%% bare_conf.tex
%% V1.3
%% 2007/01/11
%% by Michael Shell
%% See:
%% http://www.michaelshell.org/
%% for current contact information.
%%
%% This is a skeleton file demonstrating the use of IEEEtran.cls
%% (requires IEEEtran.cls version 1.7 or later) with an IEEE conference paper.
%%
%% Support sites:
%% http://www.michaelshell.org/tex/ieeetran/
%% http://www.ctan.org/tex-archive/macros/latex/contrib/IEEEtran/
%% and
%% http://www.ieee.org/

%%*************************************************************************
%% Legal Notice:
%% This code is offered as-is without any warranty either expressed or
%% implied; without even the implied warranty of MERCHANTABILITY or
%% FITNESS FOR A PARTICULAR PURPOSE! 
%% User assumes all risk.
%% In no event shall IEEE or any contributor to this code be liable for
%% any damages or losses, including, but not limited to, incidental,
%% consequential, or any other damages, resulting from the use or misuse
%% of any information contained here.
%%
%% All comments are the opinions of their respective authors and are not
%% necessarily endorsed by the IEEE.
%%
%% This work is distributed under the LaTeX Project Public License (LPPL)
%% ( http://www.latex-project.org/ ) version 1.3, and may be freely used,
%% distributed and modified. A copy of the LPPL, version 1.3, is included
%% in the base LaTeX documentation of all distributions of LaTeX released
%% 2003/12/01 or later.
%% Retain all contribution notices and credits.
%% ** Modified files should be clearly indicated as such, including  **
%% ** renaming them and changing author support contact information. **
%%
%% File list of work: IEEEtran.cls, IEEEtran_HOWTO.pdf, bare_adv.tex,
%%                    bare_conf.tex, bare_jrnl.tex, bare_jrnl_compsoc.tex
%%*************************************************************************

% *** Authors should verify (and, if needed, correct) their LaTeX system  ***
% *** with the testflow diagnostic prior to trusting their LaTeX platform ***
% *** with production work. IEEE's font choices can trigger bugs that do  ***
% *** not appear when using other class files.                            ***
% The testflow support page is at:
% http://www.michaelshell.org/tex/testflow/



% Note that the a4paper option is mainly intended so that authors in
% countries using A4 can easily print to A4 and see how their papers will
% look in print - the typesetting of the document will not typically be
% affected with changes in paper size (but the bottom and side margins will).
% Use the testflow package mentioned above to verify correct handling of
% both paper sizes by the user's LaTeX system.
%
% Also note that the "draftcls" or "draftclsnofoot", not "draft", option
% should be used if it is desired that the figures are to be displayed in
% draft mode.
%
%& --shell-escape
\documentclass[hidelinks, 10pt, conference, compsocconf]{IEEEtran}
% Add the compsocconf option for Computer Society conferences.
%
% If IEEEtran.cls has not been installed into the LaTeX system files,
% manually specify the path to it like:
% \documentclass[conference]{../sty/IEEEtran}

\usepackage{times} % assumes new font selection scheme installed
\usepackage{amsmath} % assumes amsmath package installed
\usepackage{amssymb}  % assumes amsmath package installed

\usepackage{hyperref}
\hypersetup{
    colorlinks=true,
    linkcolor=black,
    citecolor=black,
    filecolor=black,
    urlcolor=black,
}

% for the pseudocode
\usepackage{algorithm}
\usepackage{algpseudocode}

% for the figures
\usepackage{graphicx}
\usepackage[usenames,dvipsnames,svgnames,table]{xcolor}
\definecolor{yellow}{HTML}{E2C800}
\definecolor{violet}{HTML}{C136C1}
\usepackage[dot]{dot2texi}
\usepackage[edges]{forest}
%\usepackage{tikz}
\usetikzlibrary{shapes, shapes.geometric, arrows, arrows.meta, matrix, positioning}
%\usepackage{tikz-qtree}

% for the citations
\usepackage{cite}

% *** Do not adjust lengths that control margins, column widths, etc. ***
% *** Do not use packages that alter fonts (such as pslatex).         ***
% There should be no need to do such things with IEEEtran.cls V1.6 and later.
% (Unless specifically asked to do so by the journal or conference you plan
% to submit to, of course. )


% correct bad hyphenation here
\hyphenation{op-tical net-works semi-conduc-tor}


\begin{document}
%
% paper title
% can use linebreaks \\ within to get better formatting as desired
\title{Whole Genome Phylogenetic Tree Reconstruction Using Colored de Bruijn Graphs}


% author names and affiliations
% use a multiple column layout for up to two different
% affiliations

% \author{\IEEEauthorblockN{Cole A. Lyman, M. Stanley Fujimoto, \\
% Paul M. Bodily, Quinn Snell, Mark J. Clement}
% \IEEEauthorblockA{Computer Science Department\\ 
% Brigham Young University\\ 
% Provo, UT 84602 USA\\
% Email: cole@colelyman.com}
% \and
% \IEEEauthorblockN{Anton Suvorov, Michael Cormier, Justin B. Miller,\\ 
% Brandon Picket, Sage Wright, Seth M. Bybee}
% \IEEEauthorblockA{Department of Biology\\
% Brigham Young University\\ 
% Provo, UT 84602 USA}
% }

% conference papers do not typically use \thanks and this command
% is locked out in conference mode. If really needed, such as for
% the acknowledgment of grants, issue a \IEEEoverridecommandlockouts
% after \documentclass

% for over three affiliations, or if they all won't fit within the width
% of the page, use this alternative format:
% 
\author{\IEEEauthorblockN{Cole A. Lyman\IEEEauthorrefmark{1}, 
M. Stanley Fujimoto\IEEEauthorrefmark{1}, 
Anton Suvorov\IEEEauthorrefmark{2},
Paul M. Bodily\IEEEauthorrefmark{1},\\
Quinn Snell\IEEEauthorrefmark{1},
Keith A. Crandall\IEEEauthorrefmark{3},
Seth M. Bybee\IEEEauthorrefmark{2} and 
Mark J. Clement\IEEEauthorrefmark{1}}
\IEEEauthorblockA{\IEEEauthorrefmark{1}Computer Science Department\\
Brigham Young University,\\
Provo, Utah 84602 USA}
\IEEEauthorblockA{\IEEEauthorrefmark{2}Department of Biology\\
Brigham Young University,\\
Provo, Utah 84602 USA}
\IEEEauthorblockA{\IEEEauthorrefmark{3}Computational Biology Institute\\
George Washington University,\\
Washington, DC 20052 USA}
Email: \href{mailto:colelyman@byu.edu}{colelyman@byu.edu}}


% use for special paper notices
%\IEEEspecialpapernotice{(Invited Paper)}


% make the title area
\maketitle


\begin{abstract}
We present \texttt{kleuren}, a novel assembly-free method to reconstruct phylogenetic trees using the Colored de Bruijn Graph.
\texttt{kleuren} works by constructing the Colored de Bruijn Graph and then traversing it, finding bubble structures in the graph that provide phylogenetic signal.
The bubbles are then aligned and concatenated to form a supermatrix, from which a phylogenetic tree is inferred.
We introduce the algorithms that \texttt{kleuren} uses to accomplish this task, and show its performance on reconstructing the phylogenetic tree of 12 \textit{Drosophila} species.
\texttt{kleuren} reconstructed the established phylogenetic tree accurately and is a viable tool for phylogenetic tree reconstruction using whole genome sequences.
Software package available at: \href{https://github.com/Colelyman/kleuren}{https://github.com/Colelyman/kleuren}.
\end{abstract}

\begin{IEEEkeywords}
phylogenetics; algorithm; whole genome sequence; colored de Bruijn graph 
\end{IEEEkeywords}


% For peer review papers, you can put extra information on the cover
% page as needed:
% \ifCLASSOPTIONpeerreview
% \begin{center} \bfseries EDICS Category: 3-BBND \end{center}
% \fi
%
% For peerreview papers, this IEEEtran command inserts a page break and
% creates the second title. It will be ignored for other modes.
\IEEEpeerreviewmaketitle



\section{Introduction}

Whole genome sequences are readily available and affordable like never before \cite{NGS} due to the advent of high-throughput Next Generation Sequencing (NGS) which has provided researchers with vast amounts of genomic sequencing data that has transformed the landscape of understanding of genomes.
The field of phylogenetics, which discovers the evolutionary relationship between taxa, has been no exception to this transformation.
Phylogenetics has responded to the copious amounts of high throughput data with novel alignment-free and assembly-free methods \cite{AAF,CVTree} that are better suited \cite{NextGenPhylo} to handle the large amounts of data more efficiently than the traditional alignment-based phylogenetic methods.
The traditional approach to phylogenetic tree reconstruction requires a homology search throughout the genomes of the taxa, a Multiple Sequence Alignment (MSA) of the homologs, and a tree construction from the resulting matrix.
Each of these steps can be computationally expensive and may introduce many unnecessary assumptions that can be avoided by using an alignment-free and assembly-free method.

Alignment-free and assembly-free methods \cite{AlignFreeReview,AlignFreeReview2,NoMSA,Cophylog} don't come without their disadvantages, one of which being that many of these methods abstract away the source of the phylogenetic signal to a method akin to shared kmer-counting.
We propose an assembly-free whole genome phylogenetic tree reconstruction method using the Colored de Bruijn Graph (CdBG) \cite{CdBG}, a data structure that is commonly used for detecting variation and comparing genomes.

The CdBG is similar to a traditional de Bruijn Graph (dBG) in that the substrings of a certain length, referred to as kmers, of a sequence represent the vertices of the dBG and an edge exists between two vertices if the suffix of the first vertex is the prefix of the second vertex.
The CdBG differs from the traditional dBG in that each vertex is associated to an unique color (or set of colors) which could be a differing sample, species, or taxon.

We introduce the \texttt{kleuren} (Dutch for "colors" in tribute of Nicolaas Govert de Bruijn, the de Bruijn graph's namesake) software package which implements our methods.
\texttt{kleuren} works by finding \textit{bubble} regions \cite{CdBG,Bubbles} of the CdBG, which are where one or more colors diverge at a node, which act as pseudo-homologous regions between the taxa. 
The sequence for each taxon in each bubble is then extracted and a MSA is performed, then the MSA's from each bubble are concatenated to form a supermatrix in which a phylogenetic tree of evolution is constructed.

\section{Methods}

\subsection{Definitions}

Given the alphabet $\Sigma = \{A, C, G, T\}$ which are nucleotide codes, let a dBG $\mathbf{G}$, be defined as $\mathbf{G} = (V, E)$ where $V = \{v_1, v_2, \ldots, v_i, \ldots, v_s\}$ is the set of vertices and where $v_i$ is the $i^{th}$ unique sequence of length $k$ of $\mathbf{G}$ and where $E = \{e_1, e_2, \ldots, e_i, \ldots, e_t\}$ is the set of edges and where $e_i = \left(v_i, v_{i+1}\right)$ is an edge connecting two vertices such that the sequence of $v_i$ and $v_{i+1}$ overlap by $(k-1)$ characters.
Let a CdBG, $\mathbf{CG}$, be defined as $\mathbf{CG} = \{G_1, G_2, \ldots, G_i, \ldots, G_u\}$ for $u$ taxa where $G_i = \left(V_i, E_i\right)$ is the dBG of the $i^{th}$ taxon.
We refer to each $G \in \mathbf{CG}$ as a distinct color or taxon.

Furthermore, let a \textit{path}, $P = \left(v_1,\ldots, v_w\right)$ in $G_i$ be defined as a sequence of vertices from $V_i$ such that for all subsequences $\left(v_j,v_{j+1}\right)$ of $P$, the edge $\left(v_j, v_{j+1}\right) \in E_i$. 
Let a \textit{bubble}, $B$, in $\mathbf{CG}$ be defined as $B = \{P_1, \ldots, P_z\}$ such that each $P \in B$ is associated with one or more colors, that the first and last vertices of $\forall P \in B$ are identical, and that $2 \leq z \leq u$ (see Figure~\ref{fig:bubble}).

Finally, let $\mathbf{K}$ be defined as $\mathbf{K} = \{V_1 \cup V_2 \cup \ldots \cup V_i \cup \ldots \cup V_u\}$ where $V_i$ is the vertices (or the unique kmers) of the $i^{th}$ dBG, $\mathbf{G_i}$.

\begin{figure}
\centering

% \begin{dot2tex}[mathmode, options=--autosize]
% digraph G {
% 	node [style = "filled"]
% 	AGT [fillcolor = blue];
%     GGT [fillcolor = red];
%     AGG [fillcolor = red];
%     CTA [fillcolor = violet];
%     TAG [fillcolor = violet];
%     CTG [fillcolor = yellow];
%     TGT [fillcolor = yellow];
% 	ACT -> CTG;// [label = "Color\ 1"];
%     ACT -> CTA;// [label = "Colors\ 1\ \&\ 2"];
%     CTG -> TGT;// [label = "Color\ 1"];
%     CTA -> TAG;// [label = "Colors\ 1\ \&\ 2"];
%     TAG -> AGG;// [label = "Color\ 2"];
%     TAG -> AGT;// [label = "Color\ 3"];
%     AGG -> GGT;// [label = "Color\ 2"];
%     TGT -> GTG;// [label = "Color\ 2"];
%     GGT -> GTG;// [label = "Color\ 2"];
%     AGT -> GTG;// [label = "Color\ 3"];
% }
% \end{dot2tex}
\includegraphics[scale=0.55]{bubble-graph}

\begin{flushleft}
\textbf{A.} Bubble in a Colored de Bruijn Graph \\
\end{flushleft}

\begin{tabular}{cl}
{\color{yellow}Color 1} & Path: ACTGTG \\
{\color{red}Color 2} & Path: ACTAGGTG \\ 
{\color{blue}Color 3} & Path: ACTAGTG \\
\end{tabular}

\begin{flushleft}
\textbf{B.} Paths in the Bubble of Each Color \\
\end{flushleft}

\caption{\textbf{A.} An example of a bubble in a Colored de Bruijn Graph with $3$ colors (i.e. $3$ taxa), and where $k=3$.
The colors of the vertices represent the following: gray- all colors contain the vertex, purple- Color 2 and Color 3 contain the vertex, yellow- Color 1 contains the vertex, red- Color 2 contains the vertex, and blue- Color 3 contains the vertex.
In this example \textit{ACT} is the \textit{startVertex} and \textit{GTG} is the \textit{endVertex} which are both contained in all of the colors.
\textbf{B.} The extended paths of each color between the \textit{startVertex} and \textit{endVertex}.
\label{fig:bubble}}
\end{figure}

\subsection{Software Architecture}

We use the \texttt{dbgfm} software package \cite{dbgfm} to construct and represent the dBG's of the individual taxa.
\texttt{kleuren} provides an interface to interact with the individual dBG's to create a CdBG, where each taxon is considered a color.
The \texttt{dbgfm} package uses the FM-Index \cite{FM-Index}, as a space efficient representation of the dBG.

\subsection{\texttt{kleuren} Algorithms}

\subsubsection{Overall Algorithm}\label{overall}

\begin{algorithm}
\caption{kleuren Algorithm}\label{overallAlg}
\begin{algorithmic}[1]
	\Function{kleuren}{$\mathbf{K}, \mathbf{CG}$}
    	\State $bubbles \gets \left[ \ \right]$ \Comment{$bubbles$ is initialized to an empty list}
    	\ForAll{$k \in \mathbf{K}$} 
        	\If{$k$ is in $c$ or more colors of $\mathbf{CG}$}
            	\State $endVertex \gets$ \Call{findEndVertex}{$k, \mathbf{CG}$}
                \ForAll{$color \in \mathbf{CG}$}
                	\State $path \gets$ \Call{extendPath}{$k, endVertex,$ $color$}
                    \State add $path$ to $bubble$
                \EndFor
                \State append $bubble$ to $bubbles$
            \EndIf
        \EndFor
        \State $alignments \gets \left[ \ \right]$  
        \ForAll{$bubble \in bubbles$}
        	\State $alignment \gets$ multiple sequence alignment of each $path$ in $bubble$
            \State append $alignment$ to $alignments$
        \EndFor
        \State $supermatrix \gets$ concatenation of $alignments$
    \EndFunction
\end{algorithmic}
\end{algorithm}

\texttt{kleuren} works by iterating over the superset of vertices, $\mathbf{K}$, and discovering vertices that could form a \textit{bubble}.
A vertex, $s$, could form a \textit{bubble} if $s$ is present in $c$ or more colors of $\mathbf{CG}$, where $c$ is set by the user as a command line parameter.
Note that the lower that $c$ is, the more potential bubbles that may be found, but \texttt{kleuren} will take longer to run because more vertices will be considered as the starting vertex of a \textit{bubble}.
Let $s$ be considered as the starting vertex of the \textit{bubble}, $b$; then the end vertex, $e$, of $b$ is found (see Section~\ref{findEndVertex}).
After the end vertex is found, the path, $p$, between $s$ and $e$ is found for each color in $\mathbf{CG}$ (see Section~\ref{extendPath}).
This process is repeated until each vertex in $\mathbf{K}$ has been either considered as a starting vertex of a \textit{bubble}, or has been visited while extending the path between a starting and ending vertex.

\subsubsection{Finding the End Vertex}\label{findEndVertex}

\begin{algorithm}
\caption{Find End Vertex Function}\label{findEndVertexAlg}
\begin{algorithmic}[1]
	\Function{findEndVertex}{$startVertex, \mathbf{CG}$}
    	\State $endVertex \gets ``~"$ \Comment{$endVertex$ is initialized to an empty string}
        \State $neighbors \gets \Call{getNeighbors}{startVertex}$
    	\While{$!\Call{isEmpty}{neighbors}\ and\ $ $\Call{isEmpty}{endVertex}$}
        	\ForAll{$neighbor \in neighbors$}
            	\If{$k$ is in $c$ or more colors of $\mathbf{CG}$}
                	\State $endVertex \gets neighbor$
                \EndIf
            \EndFor
        \EndWhile
        \State \Return{$endVertex$}
    \EndFunction
\end{algorithmic}
\end{algorithm}

The end vertex is found by traversing the path from the $startVertex$ until a vertex is found that is in at least $c$ colors.
The $endVertex$ is then used in the function to extend the path (see Section~\ref{extendPath}). 

\subsubsection{Extending the Path}\label{extendPath}

\begin{algorithm}
\caption{Extend the Path Functions}\label{extendPathAlg}
\begin{algorithmic}[1]
	\Function{extendPath}{$startVertex, endVertex, color$, $maxDepth$}
    	\State $path \gets ``~"$
        \State $visited \gets \{\}$ \Comment{$visited$ is initialized to the empty set}
    	\If{\Call{recursivePath}{$startVertex, endVertex, path,$ $color,visited, 0, maxDepth$}}
        	\State \Return{$path$}
        \EndIf
    \EndFunction
    \\
    \Function{recursivePath}{$currentVertex, endVertex,$ $path, color, visited, depth, maxDepth$}
    	\State add $currentVertex$ to $visited$
        \If{$depth >= maxDepth$}
        	\State \Return $false$
        \EndIf
        \If{$currentKmer == endKmer$}
        	\State \Return $true$
        \EndIf
        \State $neighbors \gets \Call{neighbors}{currentVertex, color}$
        \ForAll{$neighbor \in neighbors$}
        	\If{$neighbor$ is in $visited$}
            	\State continue
            \EndIf
            \State $oldPath \gets path$ 
            \State append suffix of $currentKmer$ to $path$
            \State $depth \gets depth + 1$
            \If{!\Call{recursivePath}{$neighbor, endVertex, path,$ $color, visited, depth, maxDepth$}}
            	\State $path \gets oldPath$
            \Else
            	\State \Return{$true$}
            \EndIf
        \EndFor
    \EndFunction
\end{algorithmic}
\end{algorithm}

The main functions that discover the sequences found in a bubble are the Extend the Path Functions (see Section~\ref{extendPath}).
To extend the $path$ between the $startVertex$ and $endVertex$ we use a recursive function that traverses the dBG for a color in which every possible path between the $startVertex$ and $endVertex$ is explored up to the $maxDepth$ (provided as a command line parameter by the user).
The $maxDepth$ parameter allows the user to specify how thorough \texttt{kleuren} will search for a \textit{bubble}; the higher the $maxDepth$ the more \textit{bubbles} that \texttt{kleuren} will potentially find, but the longer \texttt{kleuren} will take because at each depth there are exponentially more potential paths to traverse.

\subsection{Data Acquisition}

To measure the effectiveness of our method we used 12 \textit{Drosophila} species, obtained from FlyBase \cite{FlyBase}.
We chose this group of species because there is a thoroughly researched and established phylogenetic tree \cite{Hahn-true-tree}.

\subsection{Tree Construction and Parameters}\label{sec:tree-construction}

We used the DSK software package \cite{DSK} to count the kmers present in all of the \textit{Drosophila} species.
To find the bubbles, we used the following parameters: $k = 17$ (kmer size of $17$) and $c = 12$ (all colors in the $\mathbf{CG}$ were required to contain a vertex in order to search for a bubble starting at that vertex) and ran $32$ instances of \texttt{kleuren} concurrently for $4$ days to find $3,277$ bubbles.
When all of the \textit{bubbles} in the CdBG had been identified, we used MAFFT \cite{MAFFT} to perform a MSA for each sequence in every \textit{bubble} that \texttt{kleuren} identified (see Figure~\ref{fig:overall} A.).
Then each MSA was concatenated to form a supermatrix (see Figure~\ref{fig:overall} B.) using Biopython \cite{Biopython}.
The phylogenetic tree was then inferred from the supermatrix by Maximum Likelihood using IQ-TREE \cite{iqtree} (see Figure~\ref{fig:overall} C.).

Once the tree was constructed, we used the ETE 3 software package \cite{ETE3} to compare the tree to the established one and Phylo.io \cite{phylo.io} to visualize the trees.

\subsection{Bubble Assumptions}

Our method is based on the assumption that \textit{bubbles} are representative of homologous regions of the taxa genomes.
We propose that this assumption is reliable because it has been shown that dBG's are a suitable method to align sequences \cite{MultipleAlignment,Sibelia,TwoPaCo}, and by identifying the \textit{bubbles} in the CdBG we find the sections of the graph that contain the most phylogenetic signal.

\begin{figure}
\centering

\begin{tabular}{cl}
{\color{yellow}Color 1} & Path: ACT\texttt{-}\texttt{-}GTG \\
{\color{red}Color 2} & Path: ACTAGGTG \\ 
{\color{blue}Color 3} & Path: ACTA\texttt{-}GTG \\
\end{tabular}

\begin{flushleft}
\textbf{A.} Multiple Sequence Alignment of the Sequences in Bubble (Figure~\ref{fig:bubble}) \\
\end{flushleft}

\medskip

\begin{tabular}{cl}
{\color{yellow}Color 1} & Path: ACT\texttt{-}\texttt{-}GTGATT\texttt{-}A... \\
{\color{red}Color 2} & Path: ACTAGGTGATTC\texttt{-}... \\ 
{\color{blue}Color 3} & Path: ACTA\texttt{-}GTGATTCA... \\
\end{tabular}

\begin{flushleft}
\textbf{B.} Supermatrix of Multiple Sequence Alignments concatenated \\
\end{flushleft}

\medskip

\begin{forest}
  forked edges,
  /tikz/every pin edge/.append style={Latex-, shorten <=2.5pt, darkgray},
  /tikz/every pin/.append style={darkgray},
  /tikz/every label/.append style={darkgray},
  before typesetting nodes={
    delay={
      where content={}{coordinate}{},
    },
    where n children=0{tier=terminus, label/.wrap pgfmath arg={right:#1}{content()}, content=}{},
  },
  for tree={
    grow'=0,
    s sep'+=10pt,
    l sep'+=15pt,
  },
  l sep'+=10pt,
  [, %!l.edge label={coordinate [pos=0, pin=-135:root] }, !1.edge label={node [pos=.65, every label, above] {branch}}, !11.edge label={coordinate [pos=0, pin={[pin distance=30pt, align=center]135:internal\\node}] }
      [
        [\color{blue} Color 3]
        [\color{red} Color 2]
      ]
      [\color{yellow} Color 1]
  ]
\end{forest}

\begin{flushleft}
\textbf{C.} Phylogenetic Tree \\
\end{flushleft}

\caption{\textbf{A.} The Multiple Sequence Alignment (MSA) of the sequences from the bubble presented in Figure~\ref{fig:bubble}.
\textbf{B.} The MSA's from each bubble are concatenated into a supermatrix, from which a phylogenetic tree is constructed.
\textbf{C.} The resulting tree from the supermatrix inferred by Maximum Likelihood.
\label{fig:overall}}
\end{figure}

\begin{figure*}
\centering
\includegraphics[scale=0.45]{Tree_mod}

\caption{The phylogenetic tree of 12 \textit{Drosophila} species constructed using \texttt{kleuren}. 
This tree resulted from using a kmer size of 17 and required all species to contain a vertex in order for the algorithm to search for a bubble starting at that vertex; and this tree is consistent with the established tree for these 12 species.
\label{fig:tree}}
\end{figure*}

\section{Results}\label{results}

\texttt{kleuren} constructed a tree (see Figure~\ref{fig:tree}) consistent with the established tree found in \cite{Hahn-true-tree} (the Robinson-Foulds distance \cite{Robinson1981} between the two trees is 0).
Even though we ran many concurrent instances of \texttt{kleuren} for multiple days (see Section~\ref{sec:tree-construction}), not all of the kmers in $\mathbf{K}$ were explored for potential bubbles; meaning that many more bubbles could be found in this CdBG which would only make the phylogeny more concrete.

Before this final successful run, there were a number of unsuccessful attempts made to construct the tree.
Initial attempts were unsuccessful because $\mathbf{K}$ (the super-set of kmers) that \texttt{kleuren} uses to find bubbles was semi-sorted (segments of the file were sorted, but all of the kmers in the file were not in lexicographic order) so the vertices that \texttt{kleuren} used to search for bubbles were skewed towards vertices that were lexicographically first.
We remedied this issue by shuffling the order of the kmer file so that there was no lexicographic bias towards the bubbles that \texttt{kleuren} finds.

A previous attempt resulted in a tree that had a $0.44$ normalized Robinson-Fould's distance from the established tree occurred because there were too few bubbles, and therefore there was not enough phylogenetic signal for the correct tree to be constructed.
To find more bubbles, we split up the kmer file into parts so that multiple instances of \texttt{kleuren} could find bubbles concurrently.
We also discovered that there was a high frequency of adenines (A) (a frequency around $40\%$ in comparison to the other nucleotides) in the final supermatrix that could skew the final tree because nucleotides have differing mutation rates.
We thought this bias towards A was due to the fact that in the $recursivePath$ function (see Algorithm~\ref{extendPathAlg}) the $neighbors$ may be sorted, so the function would traverse the $neighbor$ that started with an A before traversing the other $neighbors$ (see Algorithm~\ref{extendPathAlg}, line: 18).
Similar to the previous sorting problem, we shuffled the order of the $neighbors$ so that the first $neighbor$ that was traversed would not always be lexicographically first.
Despite this change, the final supermatrix that produced the true tree still had a bias towards A (see Section~\ref{futureWork}).
%Furthermore, we compare \texttt{kleuren} with the Assembly and Alignment-Free (AAF) \cite{AAF} software package on the same \textit{Drosophila} dataset.

\section{Conclusion}

We introduced a novel method of constructing accurate phylogenetic trees using a CdBG.
Our method, \texttt{kleuren}, uses whole genome sequences to construct a CdBG representation, then it traverses the CdBG to discover bubble structures which become the basis for phylogenetic signal between taxa and eventually produces a phylogenetic tree.

As the NGS era progresses, whole genome sequences are becoming more prevalent for more non-model organisms, in which phylogenies of these organisms have never been constructed.
\texttt{kleuren} is a viable method to relatively quickly and accurately construct the phylogenies for these newly sequenced organisms.

% conference papers do not normally have an appendix

\section{Future Work} \label{futureWork}

We plan to optimize \texttt{kleuren} so that it can find more bubbles in a shorter amount of time.
We will do this by replacing the underlying data structure for how the CdBG is represented.
\texttt{dbgfm}, the current method used to represent the dBG in \texttt{kleuren}, sacrifices time efficiency for memory efficiency by storing the FM-Index entirely on disk, thus slowing down queries into the dBG.
When \texttt{kleuren} runs faster, more bubbles will be found, and more phylogenetic signal will be present so that a more accurate tree can be constructed. 

We also plan to investigate the reasons for the high abundance of A's in the supermatrix (see Section~\ref{results}) further, and balance the frequency of nucleotides in the supermatrix.

Furthermore, we would like to look into how \texttt{kleuren} performs when the CdBG is constructed using read sequencing data rather than assembled genomes.

% use section* for acknowledgement
\section*{Acknowledgment}

This work was funded through the Utah NASA Space Grant Consortium and EPSCoR and through the BYU Graduate Research Fellowship.

The authors would like to thank Kristi Bresciano, Michael Cormier, Justin B. Miller, Brandon Pickett, Nathan Schulzke, and Sage Wright for their thoughts concerning the project.
The authors would also like to thank the Fulton Supercomputing Laboratory at Brigham Young University for their work to maintain the super-computer on which these experiments were run.

\newpage

% trigger a \newpage just before the given reference
% number - used to balance the columns on the last page
% adjust value as needed - may need to be readjusted if
% the document is modified later
%\IEEEtriggeratref{8}
% The "triggered" command can be changed if desired:
%\IEEEtriggercmd{\enlargethispage{-5in}}

% references section

% can use a bibliography generated by BibTeX as a .bbl file
% BibTeX documentation can be easily obtained at:
% http://www.ctan.org/tex-archive/biblio/bibtex/contrib/doc/
% The IEEEtran BibTeX style support page is at:
% http://www.michaelshell.org/tex/ieeetran/bibtex/
\bibliographystyle{IEEEtran}
% argument is your BibTeX string definitions and bibliography database(s)
\bibliography{biblio}
%
% <OR> manually copy in the resultant .bbl file
% set second argument of \begin to the number of references
% (used to reserve space for the reference number labels box)
% \begin{thebibliography}{1}

% \bibitem{IEEEhowto:kopka}
% H.~Kopka and P.~W. Daly, \emph{A Guide to \LaTeX}, 3rd~ed.\hskip 1em plus
%   0.5em minus 0.4em\relax Harlow, England: Addison-Wesley, 1999.

% \end{thebibliography}




% that's all folks
\end{document}


