\documentclass[twocolumn]{IEEEtran}
%\documentclass[journal,draftcls,onecolumn,12pt,twoside]{IEEEtranTCOM}
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage[version=4]{mhchem}
\usepackage{graphicx}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{color}
\usepackage{relsize}
\usepackage{colortbl}
\usepackage{kbordermatrix}
\usepackage{bm}
\usepackage{soul}
\usepackage{subfigure}
\usepackage{enumerate}
\usepackage{paralist}
\usepackage{dsfont}
\usepackage[normalem]{ulem}
\usepackage{mathtools}
\usepackage{tabularx,booktabs,caption}
\usepackage[flushleft]{threeparttable}\usepackage{tabularx,booktabs,caption}
\usepackage[flushleft]{threeparttable}
\newcommand*\mean[1]{\bar{#1}}
\newcommand{\p}{\mathrm{p}}
\newcommand{\B}{\mathrm{B}}
\newcommand{\U}{\mathrm{U}}
\newcommand{\PP}{\mathrm{P}}
\newcommand{\E}{\mathrm{E}}
\newcommand{\e}{\mathrm{e}}
\newcommand{\Var}{\mathrm{Var}}
\newcommand{\Std}{\mathrm{Std}}
\newcommand{\Exp}{\mathrm{Exp}}
\newcommand{\Cov}{\mathrm{Cov}}
\newcommand{\MSE}{\mathrm{MSE}}
\newcommand{\NMSE}{\mathrm{NMSE}}
\newcommand{\approxtext}[1]{\ensuremath{\stackrel{\text{#1}}{\approx}}}
\def\mathclap#1{\text{\hbox to 0pt{\hss$\mathsurround=0pt#1$\hss}}}
\makeatletter
\newcommand{\vastt}{\bBigg@{3}}
\newcommand{\vast}{\bBigg@{4}}
\newcommand{\Vast}{\bBigg@{5}}
\newcounter{mytempeqncnt}
\makeatother
\DeclareMathOperator*{\argmax}{arg\,max}
\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator\erfc{erfc}
\DeclarePairedDelimiter{\ceil}{\lceil}{\rceil}
\DeclarePairedDelimiter\floor{\lfloor}{\rfloor}

\begin{document}
\title{Detection in Molecular Communications with Ligand Receptors under Molecular Interference}
\author{Murat Kuscu,~\IEEEmembership{Member,~IEEE}
        and Ozgur B. Akan,~\IEEEmembership{Fellow,~IEEE}
       \thanks{An earlier version of this work was presented at IEEE SPAWC'18, Kalamata, Greece \cite{muzio2018selective}. }
       \thanks{The authors are with the Department of Electrical and Electronics Engineering, Koc University, Istanbul, 34450, Turkey  (email: \{mkuscu, akan\}@ku.edu.tr).}
       \thanks{Ozgur B. Akan is also with the Internet of Everything (IoE) Group, Electrical Engineering Division, Department of Engineering, University of Cambridge, Cambridge, CB3 0FA, UK (email: oba21@cam.ac.uk).}
\thanks{This work was supported in part by the ERC (Project MINERVA, ERC-2013-CoG \#616922) and by the AXA Research Fund (AXA Chair for Internet of Everything at Koc University).}}% <-this % stops a space

%\markboth{IEEE Transactions on Communications}%
%{Submitted paper}

\maketitle
%\thispagestyle{empty}
%\pagestyle{empty}


\begin{abstract}
	Molecular Communications (MC) is a bio-inspired communication technique that uses molecules to transfer information among bio-nano devices. In this paper, we focus on the detection problem for biological MC receivers employing ligand receptors to infer the transmitted messages encoded into the concentration of molecules, i.e., ligands. In practice, receptors are not ideally selective against target ligands, and in physiological environments, they can interact with multiple types of ligands at different reaction rates depending on their binding affinity. This molecular cross-talk can cause a substantial interference on MC. Here we consider a particular scenario, where there is non-negligible concentration of interferer molecules in the channel, which have similar receptor-binding characteristics with the information molecules, and the receiver employs single type of receptors. We investigate the performance of four different detection methods, which make use of different statistics of the ligand-receptor binding reactions: instantaneous number of bound receptors, unbound time durations of receptors, bound time durations of receptors, and combination of unbound and bound time durations of receptors within a sampling time interval. The performance of the introduced detection methods are evaluated in terms of bit error probability for varying strength of molecular interference, similarity between information and interferer molecules, number of receptors, and received concentration difference between bit-0 and bit-1 transmissions. We propose synthetic receptor designs that can convert the required receptor statistics to the concentration of intracellular molecules, and chemical reaction networks that can chemically perform the computations required for detection. 
	
\end{abstract}


\begin{IEEEkeywords}
Molecular communication, receiver, ligand receptors, interference, detection, maximum likelihood estimation, method of moments,  kinetic proofreading, synthetic receptors, chemical reaction networks.
\end{IEEEkeywords}

\IEEEpeerreviewmaketitle

\section{Introduction}
\label{sec:Introduction}
Internet of Bio-Nano Things (IoBNT) is an emerging technology built upon the artificial heterogeneous communication networks of nanomachines and biological entities, promising for novel applications such as smart drug delivery with single-molecule precision and continuous health monitoring \cite{akyildiz2015internet, kuscu2016internet, dinc2019internet, akyildiz2019microbiome}. Bio-inspired Molecular Communication (MC) has emerged as the most promising communication technique to enable IoBNT applications. MC uses molecules, instead of electromagnetic waves (EM), to transfer information, which can be encoded into the concentration or type of molecules \cite{atakan2016molecular, akan2017fundamentals, akyildiz2019moving}. Being fundamentally different from conventional EM communication techniques, MC has brought about new interdisciplinary challenges in developing communication techniques and transceiver architectures.

%MC brings about new challenges that should be tackled by interdisciplinary research. 



%requires revisiting our existing knowledge from many different aspects. 
% brings aemerge 

%Internet of Bio-Nano Things (IoBNT) refers to an emerging technology that defines the connection of biological entities and nanomachines with each other and with the Internet \cite{Akan2017, Pierobon2015, Kuscu2017}. The most promising nanoscale communication paradigm for enabling IoBNT is Molecular Communications (MC), which is a bio-inspired method, where molecules are used to encode, transmit and receive information. The great majority of studies about MC focus on the type in which the information is encoded into the concentration of molecules. The most used scheme is the passive one, where signals are transmitted through free diffusion in fluid environment \cite{Pierobon2013}.

Many efforts in MC research have been devoted to developing channel models and low-complexity communication techniques \cite{akyildiz2019moving, jamali2019channel, kuscu2011physical, kuscu2019transmitter, farsad2016comprehensive}. Of particular interest has been the detection problem. Several detection methods of varying complexity have been developed for different device architectures \cite{kuscu2019transmitter, kilinc2013receiver, li2019csi, li2015low}. Most studies focusing on MC detection, however, consider a particular receiver architecture that is capable of counting every single molecule inside its virtual reception space \cite{pierobon2011diffusion, kuscu2019transmitter}. On the other hand, an increasing research interest is being directed towards MC receivers with ligand receptors, which chemically interact with information molecules through ligand-receptor binding reaction \cite{pierobon2011noise, chou2015maximum, kuscu2018modeling, kuscu2016physical, kuscu2016modeling}. This receiver design is the most physically relevant, as the ligand-receptor interactions are prevalent in biological systems, and thus suitable for synthetic biology-enabled MC device and system architectures \cite{bialek2012biophysics, soldner2019survey, unluturk2015genetically}. This additional layer of biological interaction, while adding to the complexity of the overall MC system, yields interesting statistics that can be exploited in order to develop reliable detection methods.  


%tailoring of conventional EM techniques to this new phenomenon.  has been studied from various aspects. Different architectures .... . Detection problem is addressed mostly for absorbing or passive receivers. Ligand-receptor binding is more realistic. To this aim, many studies... ... Ligand-receptor binding is also considered... more physically-relevant.. brings additional randomness to the overall communication process. 

MC detection with ligand receptors has been widely studied; however, in existing studies, receptors are assumed to be ideally selective against the information molecules \cite{kuscu2019transmitter}. On the other hand, in practice, the selectivity of biological ligand receptors is not ideal, and receptors can bind other types of molecules that have a nonzero affinity with the receptors.
%This molecular interference is (might be) inevitable in biologically relevant environments, which are generally crowded with many different types of molecules. 
This molecular interference, also called cross-talk, is widely observed in various biological systems due to the prevalence of promiscuous ligand-receptor interactions \cite{lalanne2015chemodetection, nobeli2009protein}. A paradigmatic example is the immune recognition where T cells express promiscuous T cell receptors (TCRs) that bind both self-ligands and a large spectrum of foreign ligands \cite{sewell2012must, baker2012structural}. The detection of foreign ligands via TCR signaling evokes the immune response. Other examples include the transcriptional cross-talk due to nonspecific binding of genes and regulators in gene regulation \cite{oeckinghaus2011crosstalk}, quorum sensing (QS) where QS receptors are promiscuously activated by multiple types of ligands \cite{wellington2019quorum}, and most of the cellular communication systems, such as bone morphogenetic protein (BMP), Wnt, Notch, and fibroblast growth factor (FGF) signaling pathways \cite{su2020ligand}.

%include quorum sensing, antibody, transcriptional cross-talk,  odor recognition systems ... 

At the heart of the widespread cross-talk in biological systems lies the promiscuous proteins including cellular receptors, enzymes and antibodies, that can interact with diverse ligand structures, including small molecules, macromolecules, and ions \cite{nobeli2009protein, tawfik2010enzyme, jain2019antibody}. The prevalence of promiscuous protein interactions is mainly due to the degeneracy of the protein interface structures \cite{gao2010structural}. This can be further attributed to the structural flexibility of proteins and interacting ligands, partial recognition due to the tolerance in shape and chemical complementarity in binding, and the presence of multiple interaction sites in proteins \cite{nobeli2009protein}.


%Considering that the ... MC systems relying on synthetic biotransceivers will make use of biological molecules for signaling. 


%molecular mimicry
%
%degeneracy of protein-protein interface structures
%structural flexibility of receptors and interacting ligands, partial recognition due to the tolerance in shape and chemical complementarity in bindings, presence of multiple interaction sites,


Many biological processes, such as T cell antigen recognition and transcription/translation, are adapted to preserve the specificity in the presence of cross-talk through complex out-of-equilibrium biophysical mechanisms, such as kinetic proofreading (KPR) and adaptive sorting \cite{yousefi2019optogenetic, lalanne2013principles}. Some sensory systems, such as odor recognition system, even seem to exploit the cross-talk as an opportunity to expand the ligand search space with a limited number of receptors \cite{hallem2006coding, carballo2019receptor}. In bacteria QS, the cross-talk is considered as a mechanism that brings evolutionary advantage by facilitating the interspecies interactions in complex microbial communities \cite{wellington2019quorum}. During cell-cell communication in multicellular organisms, promiscuity of ligand-receptor interactions is shown to enable individual cells to address a larger number of cell types through combinatorial addressing \cite{su2020ligand}.   

%reduce the number of different types of receptor and signaling components in dealing with … expand the set of targets… (hedging bets as well), increase the search space. 

Considering that the synthetic biological MC systems will utilize receptors and ligands derived from biological systems, the promiscuity of the natural ligand-receptor interactions is likely to be preserved in MC devices. This could result in significant level of molecular interference in biologically relevant environments, e.g., inside human body, which are typically crowded with diverse types of proteins (e.g., transcription factors, enzymes, hormones), other macromolecules (e.g., nucleic acids), and small molecules, that can promiscuously bind the receptors synthesized on the MC receiver surface \cite{uhlen2015tissue, thul2017subcellular, luck2020reference}. Moreover, the problem can be translated into multi-user interference in crowded multi-user MC networks, where structurally similar molecules from the same family, such as isomers, are likely to be used to enable molecule-division multiple access without increasing the burden on MC transmitters and receivers \cite{dinc2017theoretical, kim2013novel}. 
% through exploiting isomers REF. 

%where express molecules from the same family. For example, isomer, ... 

% biomolecules for signaling  on the use of synthetic receptors and, biological molecules,  which are derived from ... 








%This and requires revisiting existing detection techniques and developing new ones to eliminate it in molecular communications. 


% we consider cross-talk as a problem. 

In this paper, we study the effect of molecular interference due to nonspecific ligand-receptor interactions on the reliability of synthetic MC systems. We consider an MC system encoding binary information into the concentration of molecules, i.e., utilizing binary concentration shift keying (binary CSK). The interference is resulting from a second type of molecule existing in the MC channel, whose number in the receiver's vicinity at the time of sampling follows Poisson distribution. Under these conditions, we investigate the performance of four different detection methods, which exploit different statistics of the ligand-receptor binding reaction. 

The first detection method relies on the number of bound receptors at the sampling time, which gives a measure of the total molecular concentration around the receiver. This method is the most widely studied one in MC literature concerning ligand receptors \cite{einolghozati2011capacity}.  The second method uses the maximum likelihood (ML) estimate of the total ligand concentration based on the receptors' unbound time intervals. This method has been previously introduced to overcome the saturation problem of receptors exposed to a high concentration of molecules \cite{kuscu2018maximum}. The third method relies on the estimation of the concentration ratio of information molecules, i.e., the ratio of information molecule concentration to total molecular concentration in the vicinity of the receiver, based on receptors' bound time intervals. This technique exploits the difference in the receptor binding affinities of information and interferer molecules reflected to the expected sojourn time of receptors in the bound state. The last method combines the estimates of the total ligand concentration and the concentration ratio of information molecules to obtain an estimate of the individual concentration of information molecules, which is then used for detection of molecular messages. This technique utilizes both the unbound and bound time intervals of the receptors. 

We derive the bit error probability (BEP) for each detection method, which is then used for comparing the performances of the introduced detection methods for varying strength of molecular interference, similarity between interferer and information molecules in terms of their affinity against receptors, number of receptors, and difference in received concentrations of information molecules for bit-0 and bit-1 transmissions. 
%The results reveal that the introduced detection method based on the estimate of concentration of information molecules perform quite well even in severe interference conditions. 
We also provide a comprehensive discussion on the practical implementation of these detection methods by biosynthetic devices. In particular, we propose synthetic receptor designs for the transduction of required receptor statistics, i.e., number of bound receptors, receptor bound and unbound time intervals,  into the concentration of intracellular molecules. We also propose a chemical reaction network (CRN) for each method that can perform the analog and digital computations required for detection. 


The remainder of the paper is organized as follows. In Section \ref{sec:related}, we provide a brief overview of the related work. In Section \ref{sec:statistics}, we review the fundamentals of ligand-receptor binding reactions. The considered MC system is described in Section \ref{sec:system} along with accompanying assumptions.  We introduce the detection methods in Section \ref{sec:detection}, where we also derive the BEP for each detection method. The results of the performance evaluation are discussed in Section \ref{sec:performance}. In Section \ref{sec:implementation}, we provide a discussion on the implementation, and propose synthetic receptor designs for signal transduction and CRNs for intracellular calculations. Finally, we conclude the paper in Section \ref{sec:conclusion}.

% we present the details of the MC system considered for the analysis. In Section \ref{det_meth}, we introduce the ML detection methods. Performance evaluation results are provided in Section \ref{perf_ev}. Finally, conclusions are given in Section \ref{concl}.

%interferers as well with a finite affinity.. a physically-relevant problem... Not studied before in MC literature. But in biophysics, some articles... 

%In this study, we consider the detection problem with ligand-receptors in the presence of interferer molecules. In our analysis, interference is poisson. Binary CSK modulation.  
%We propose 4 different detection techniques... based on ...
%Derive probability of error... Analysis in terms of ...

%Discuss implementation, propose synthetic receptor designs for transduction and CRNs for performing the detection. 


\section{Related Work}
\label{sec:related}
MC detection has been extensively studied for various modulation schemes, channel types, and receiver architectures. In developing received signal models for detection, simplifying assumptions are often utilized to address the intricate physical interaction between the channel and the receiver. Most of the existing MC detection studies assume passive and transparent receivers, and ignore the complex ligand-receptor interactions. Many other studies consider an absorbing architecture that actually corresponds to a receiver with infinite number of receptors, which can irreversibly react with information molecules and absorb them, i.e., remove them from the channel, at infinitely high reaction rates. Although passive and absorbing receiver architectures have little correspondence with reality, these studies provide useful theoretical performance bounds for MC detection. A recent comprehensive review of the existing MC detection methods can be found in \cite{kuscu2019transmitter}. 

The interest in MC detection with ligand receptors, on the other hand, has only recently gained momentum. In \cite{chou2013extended, chou2015markovian}, a modeling framework based on CMTPs has been introduced. Using this framework, Maximum A Posteriori (MAP) decoders have been developed based on sampling the continuous history of the receptor states, including the exact time instances of the binding events \cite{chou2015maximum, chou2019designing}. In \cite{kuscu2018maximum}, we proposed an ML detection method based on receptor unbound times to overcome the saturation problem of reactive receivers with finite number of ligand receptors. However, none of these studies consider the existence of similar types of molecules interfering with the ligand-receptor binding reaction. 

There is also a substantial body of work in biophysics literature concerning the theoretical bounds of molecular sensing with ligand receptors. Regarding the interference on the ligand-receptor reactions, in \cite{mora2015physical, singh2017simple, lalanne2015chemodetection, siggia2013decisions}, authors investigate the ML estimation of the concentrations of two different ligand types based on receptor bound times. These studies also suggest that certain types of living cells, e.g., T cells in the immune system, might be implementing similar estimation methods in discriminating against the foreign agents through a KPR mechanism, in which receptors sequentially visit a number of internal states to sample the duration of binding events. Following a similar approach with these studies, in \cite{kuscu2019channel}, we introduced a novel channel sensing method that can concurrently estimate the concentration of several different types of ligands using the receptor unbound and bound times. Lastly, in an earlier version of this study \cite{muzio2018selective}, we investigated the theoretical performance bounds of ML detection based on receptor bound times, instantaneous number of bound receptors, and receptor unbound times. 
%These studies also argue practical implementations of the ML estimators exploiting kinetic proofreading (KPR) mechanism, which is an active cellular mechanism that increases specificity, suggested to exist in T-cell receptors that can sense very low concentrations of foreign agents with extreme specificity as part of the immune system \cite{mckeithan1995kinetic}. 


Different from the conference version \cite{muzio2018selective}, this study investigates four practical detection methods that can be implemented by biological MC receivers with the use of synthetic receptors and CRNs. We derive analytical expressions for bit error probability, and propose synthetic receptor designs and CRNs for sampling the receptor states and performing the detection by biochemical means.  
%The  derive analytical expressions for? BEP. propose synthetic receptor designs and CRNs... s

%In the conference version of this work REF, we previously investigated the performance of ML detection with bound time durations in the presence of single type of interferers, and shown that it outperforms other MC detection schemes that use the samples of receptor unbound time or instantaneous receptor states.









%concentration ratio, total ligand concentration, number of bound receptors in the presence of interferer molecules... SONA .



%In biophysics literature, the literature is more abundant. Cross-talk,  


%Detection techniques are proposed for concentration shift keying (CSK) modulated MC with ligand receptors in \cite{kuscu2016modeling}, based on sampling the instantaneous number of bound receptors. Recently, the continuous history of bound and unbound states has proven to provide more information about the external ligand concentration \cite{chou2015markovian, endres2009maximum}. In this direction, maximum a posteriori (MAP) detection methods are proposed for MC in \cite{chou2015markovian, chou2015maximum}. In our previous work \cite{kuscu2018maximum}, receptor unbound time duration statistics is shown to provide a larger dynamic input range for the detector to cope with saturation at high ligand concentrations resulting from intersymbol interference (ISI). Maximum likelihood (ML) estimation of the concentration of two different ligand types based on sampling the receptor bound time durations is studied in \cite{mora2015physical, singh2017simple, lalanne2015chemodetection, siggia2013decisions}. These studies also argue practical implementations of the ML estimators exploiting kinetic proofreading (KPR) mechanism, which is an active cellular mechanism that increases specificity, suggested to exist in T-cell receptors that can sense very low concentrations of foreign agents with extreme specificity as part of the immune system \cite{mckeithan1995kinetic}. Based on a similar approach, we have previously investigated the performance of ML detection with bound time durations in the presence of single type of interferers, and shown that it outperforms other MC detection schemes that use the samples of receptor unbound time or instantaneous receptor states \cite{muzio2018selective}. However, none of the previous studies has considered the problem of sensing the concentration of more than two types of ligands at the same time. 




%Chou's work... Markov process model... in general reactive receivers with general circuits other than ligand receptors... Detection techniques... single 

%Detection problem for MC receivers with ligand receptors is elaborated in \cite{Chou2015} and \cite{Kuscu2018}. Particularly in \cite{Chou2015}, authors develop a Maximum A Posteriori (MAP) decoder based on the continuous occupation history of receptors, including the instances of binding events. In \cite{Kuscu2018}, authors propose a simpler ML detection method based on receptor unbound times, and investigate the saturation problem for more realistic receivers equipped with a finite number of receptors. %Saturation is a condition that occurs when there is high concentration of ligands and causes a decrease in sensitivity of receptors.


%Saturation problem ... maximum likelihood. 



%MC receivers must employ special detection techniques to infer the transmitted messages from the received molecules. In this direction, MC detection problem has been extensively studied in the literature \cite{Llatser2013, Kilinc2013, Meng2014}; however, the great amount of studies assume that the MC receiver is an ideal instrument capable of sensing and counting every single molecule in an arbitrarily defined reception space.
%Here, we focus on a more realistic bio-inspired architecture for the MC receiver, which employs receptor proteins, called ligand receptors, to sense information carrying molecules, i.e. ligands, through ligand-receptor binding reaction. There has been increasing interest in modeling MC receiver employing ligand receptors \cite{Pierobon2011}--\cite{Kuscu2016_SiNW}. In particular, a noise model is developed for ligand-receptor binding process in \cite{Pierobon2011}, and the impulse response and the capacity of an MC channel with ligand receptors is derived in \cite{Ahmadzadeh2016} and \cite{Aminian2015}, respectively. In \cite{Kuscu2016_physical_design, Kuscu2016_SiNW, Kuscu2018_cdr}, authors study field-effect-transistor biosensor (bioFET)-based  MC receivers with ligand receptors as bio-cyber interfaces for IoBNT. 

%Detection problem for MC receivers with ligand receptors is elaborated in \cite{Chou2015} and \cite{Kuscu2018}. Particularly in \cite{Chou2015}, authors develop a Maximum A Posteriori (MAP) decoder based on the continuous occupation history of receptors, including the instances of binding events. In \cite{Kuscu2018}, authors propose a simpler ML detection method based on receptor unbound times, and investigate the saturation problem for more realistic receivers equipped with a finite number of receptors. %Saturation is a condition that occurs when there is high concentration of ligands and causes a decrease in sensitivity of receptors.

%In practice the channel is crowded with a variety of molecules interfering with the reception process. MC detection problem considering the presence of interferer molecules of the same type as the messenger molecules is studied in \cite{Noel2014}. In this paper, we consider a channel crowded with interferer molecules of a different type, but having non-negligible affinity with the receptors. The estimation of concentration of target ligands in the presence of similar kind of interferers is studied in \cite{Mora2015}, which proposes an ML estimator for the ratio of correct ligands. Here, we extend this study for the MC detection problem, and including the one based on ratio estimation, we evaluate the performances of three different ML detection methods in the presence of interferers.
%we focus on receivers with ligand receptors and 
%considering the presence of interferer molecules in the environment. 
%in a such condition.

%The first detection method is based on the instantaneous number of bound receptors, which is informative of the total concentration of messengers and interferers \cite{Kilinc2013}, \cite{Meng2014}. The second method is also centered around the estimation of total concentration at the receiver, but in this case the observable parameter is the amount of time the receptors stay unbound within a observation time window \cite{Kuscu2018}. The last method is based on the estimation of the ratio of messenger concentration to total concentration by observing the amount of time the receptors stay bound during a  period. The three methods are numerically compared in terms of bit error probability (BEP), and results are provided under different scenarios varying in expected concentration of noise at the receiver, level of receiver saturation, affinity of the interferer molecules with the receptors, and the number of receptors on the receiver.

%The remainder of the paper is organized as follows. In Section \ref{sys_mod}, we present the details of the MC system considered for the analysis. In Section \ref{det_meth}, we introduce the ML detection methods. Performance evaluation results are provided in Section \ref{perf_ev}. Finally, conclusions are given in Section \ref{concl}.


\section{Statistics of Ligand-Receptor Binding Reactions}
\label{sec:statistics}

% monovalent
% binary switching process
% telegraph process
In ligand-receptor binding reaction, receptors randomly bind external molecules, i.e., ligands, in their vicinity. Following the canonical Berg-Purcell scheme, the stochastic ligand-receptor binding process can be abstracted as a continuous-time Markov process (CTMP) with two states; corresponding to the bound (B) and unbound (U) states of the receptors \cite{berg1977physics, ten2016fundamental}. Due to the memoryless property of the Markov processes, the dwell time at each receptor state follows exponential distribution \cite{liggett2010continuous}, with a rate parameter depending on the kinetic rate constants of the ligand-receptor binding reaction given as    
%A receptor can be either in the Bound (B) or Unbound (U) state, and the dwell time at each receptor state follows exponential distribution with a rate parameter depending on the kinetic rate constants of the reaction. 
%In the presence of single type of ligands, the state of a single receptor is governed by the following two-state stochastic process,
\begin{equation}
\ce{U  <=>[{c_L(t) k^+}][{k^-}] B},
\label{eq:bindingreaction}
\end{equation}
where $c_L(t)$ is the time-varying ligand concentration, $k^+$ and $k^-$ are the binding and unbinding rates of the ligand-receptor pair, respectively \cite{berezhkovskii2013effect}. The correlation time of this Markov process, which can be regarded as the relaxation time of the ligand-receptor binding reaction to equilibrium, is given by $\tau_B = 1/\left(c_L(t) k^+ + k^-\right)$ \cite{berezhkovskii2013effect, ten2016fundamental}. In diffusion-based MC, due to the low-pass characteristics of the diffusion channel, the bandwidth of the $c_L(t)$ is typically significantly lower than the characteristic frequency of the binding reaction \cite{pierobon2011noise}, which is given by the reciprocal of the receptor correlation time i.e., $f_\text{B} = 1/\tau_B = c_L(t) k^+ + k^-$. Therefore, the receptors are often assumed to be at equilibrium with a stationary ligand concentration, which is hereafter simply denoted by $c_L$. At equilibrium, the ligand-receptor binding reaction obeys the detailed balance, such that the rate of unbinding transitions must be equal to the rate of binding transitions, i.e., $\p_\B k^- = (1-\p_\B) c_L k^+$ \cite{endres2009maximum}. Here, $\p_\B$ is the probability of finding a receptor at the bound state at equilibrium, which can be obtained from the detailed balance condition as 
\begin{equation}
	\p_\B  = \frac{c_L k^+}{c_L k^+ + k^-} = \frac{c_L}{c_L+K_D},
	\end{equation}
where $K_D = k^-/k^+$  is the dissociation constant, which gives a measure of the \textit{affinity} between a ligand and a receptor. 
\begin{figure}[!t]
	\centering
	\includegraphics[width=9cm]{Receivernew}
	\caption{(a) A biological MC receiver with ligand receptors which can bind both information and interferer molecules. Receptors encode one or more statistics of the binding events into the concentration of intracellular molecules. (b) An example time trajectory of receptors fluctuating between the bound and unbound states. }
	\label{fig:interference}
\end{figure}
%Sampling the number of bound receptors at a given time instant previously proved effective in inferring the concentration of ligands, when the receiver is away from saturation \cite{kuscu2018maximum}, i.e., when $p_B \ll 1$. 

%When there are \textbf{\emph{multiple types of ligands}} in the channel medium, 

In the presence of two different types of molecules, e.g., information and interferer molecules, in the channel, as shown in Fig. \ref{fig:interference}(a), which can bind the same receptors with different affinities, i.e., with different dissociation constants, the bound state probability of a receptor at equilibrium becomes 
\begin{equation} \label{probBinding2}
\p_\B = \frac{c_s/K_D^s + c_{in}/K_D^{in}  }{1 + c_s/K_D^s + c_{in}/K_D^{in}},
\end{equation}
where $c_s$ and $c_{in}$ are the concentrations of information and interferer molecules, whose dissociation constants are denoted by $K_D^s$ and $K_D^{in}$, respectively (please refer to Appendix \ref{AppendixA} for the derivation). If the receiver has $N_R > 1$ independent receptors, the number of bound ones at equilibrium follows binomial distribution with the number of trials $N_R$ and the success probability $\p_\B$. 
%\begin{equation} \label{prob_M}
%p_B  = \frac{\sum_{i=1}^{M} c_i/K_{D,i}}{1 + \sum_{i=1}^{M} c_i/K_{D,i}},
%\end{equation}
%where $M$ is the number of different types of ligands co-existing in the medium. 


%The expression in \eqref{prob_M} cannot be used to infer the individual ligand concentrations $c_i$ due to the interchangeability of the summands \cite{mora2015physical}. Therefore, in the case of a mixture of different ligand types, the required insight into the individual ligand concentrations can only be acquired by examining the continuous history of binding and unbinding events over receptors, which is exemplified in Fig. \ref{fig:receiver}(b). In this case, 

On the other hand, the duration for which the receptors stay bound or unbound can reveal more information about the concentration and type of the molecules co-existing in the channel \cite{kuscu2019channel}. The likelihood of observing a set of $N$ independent binding and unbinding time intervals over any set of receptors at equilibrium can be written as
%\begin{align}\label{eq:likelihood1}
%&\p\left(\{\tau_b,\tau_u \}_{N}\right) \\ \nonumber
%&= \frac{1}{Z} \e^{-\mathlarger{\sum_{i=1}^N} \tau_{u,i} \left(\sum_{j=1}^{M} k_j^+ c_j \right)}  \mathlarger{\prod_{i=1}^{N}}  \left(\sum_{j=1}^{M} k_j^+ c_j k_j^- \e^{-k_j^- \tau_{b,i}} \right),
%\end{align}
\begin{align}\label{eq:likelihood1}
\p\left(\{\tau_b,\tau_u \}_{N}\right) &= \frac{1}{Z} \e^{-\mathlarger{\sum_{i=1}^N} \tau_{u,i} \left( k_s^+ c_s + k_{in}^+ c_{in} \right)}  \\ \nonumber
& \times \mathlarger{\prod_{i=1}^{N}}  \left(\sum_{j=\{s, in\}} k_j^+ c_j k_j^- \e^{-k_j^- \tau_{b,i}} \right),
\end{align}
where $Z$ is the probability normalization factor, $k_j^+$ and $k_j^-$ are the binding and unbinding rates for ligand $j \in \{s,in \}$, respectively, and $\tau_{u,i}$ and $\tau_{b,i}$ are the $i^\text{th}$ observed unbound and bound time durations, respectively \cite{mora2015physical}. 

%We note that an unbound or bound time duration is the duration of a time interval that a receptor continuously stays unbound or bound, respectively. Given that the receptors are independent of each other, and they are exposed to the same ligand concentration assumed to be constant during sampling, ligand-receptor binding reaction becomes a stationary ergodic process. Therefore, the likelihood function \eqref{eq:likelihood1} does not depend on the time instants the ligands bind or unbind, and the indices of bound and unbound time durations do not necessarily imply a receptor-based or chronological order. In other words, the entire set of bound and unbound duration samples $\{\tau_b,\tau_u \}_{N}$ can be obtained equivalently by observing the time trajectory of only a single receptor or multiple independent receptors. 

In the diffusion-limited case, i.e., where the reaction rates are much higher than the characteristic rate of diffusion, the binding rate can be simply written as $k^+ = 4 D a$ for circular receptors \cite{mora2015physical}, with $D$ and $a$ being the diffusion coefficient of ligands and the effective receptor size, respectively. Assuming that the ligands are of similar size, the diffusion coefficient $D$, which depends on the ligand size as well as the temperature and viscosity of the fluid medium \cite{bialek2012biophysics}, can be taken equal for all ligand types. In this case, the likelihood function \eqref{eq:likelihood1} reduces to
\begin{align}
\p\left(\{\tau_b,\tau_u \}_{N}\right) = \frac{1}{Z} e^{- T_u  k^+ c_{tot}}  (k^+ c_{tot})^N \prod_{i=1}^{N} \p\left(\tau_{b,i}\right) ,
\label{eq:likelihood}
\end{align}
where $T_u = \sum_{i = 1}^N \tau_{u,i}$ is the total unbound time of all receptors during the observation period, $c_{tot} = c_s + c_{in}$ is the total ligand concentration in the vicinity of the receptors, and $\p(\tau_{b,i})$ is the probability of observing a bound time duration, which is given as a \textit{mixture of exponential distributions}, i.e.,  
%\begin{align}
%\p\left(\tau_{b,i}\right)  =  \sum_{j=1}^M \alpha_j k_j^- \e^{-k_j^- \tau_{b,i}}.
%\end{align}
%
%remove the above. 
\begin{equation}
\p(\tau_b)=  \sum_{i \in \{s, in\}} \alpha_i k_i^- \e^{-k_i^- \tau_b} \label{eq:probboundduration}
\end{equation}
Here $\alpha_i = c_i/c_{tot}$ is the concentration ratio of a particular type of molecule.


The log-likelihood function for an observed set of unbound/bound time durations can then be written as the sum of three terms, i.e., 
\begin{align}\label{log_like}
\mathcal{L}(\{\tau_b,\tau_u \}_N) & = \ln\p(\{\tau_b,\tau_u \}_N) \\ \nonumber
& = \mathcal{L}_0  + \mathcal{L}\left(T_u | c_{tot}\right)  + \mathcal{L} \left(\{\tau_b \} | \bm{\alpha}\right), 
\end{align}
where $\mathcal{L}_0$ comprises the terms that do not depend on $c_{tot}$ or $\bm{\alpha} = [\alpha_{in}, \alpha_s]^T$, while $\mathcal{L}\left(T_u | c_{tot}\right) $ and $\mathcal{L} \left(\{\tau_b \} | \bm{\alpha}\right)$ are the functions of the total concentration $c_{tot}$ and the ligand concentration ratios $\bm{\alpha}$, respectively. For detection, we are only interested in the log-likelihoods that are functions of $c_{tot}$ and $\bm{\alpha}$, i.e.,
\begin{equation}
\mathcal{L}\left(T_u | c_{tot}\right) = N \ln(c_{tot})-k^+ c_{tot} T_u, \label{eq:likelihoodtotalligand}
\end{equation}
\begin{equation}\label{l2}
\mathcal{L} \left(\{\tau_b \} | \bm{\alpha}\right) = \sum_{i=1}^{N} \ln \p\left(\tau_{b,i}\right).
\end{equation}
Accordingly, $\mathcal{L}\left(T_u | c_{tot}\right)$ tells us that the total unbound time $T_u$ is informative of the total ligand concentration $c_{tot}$, whereas $\mathcal{L} \left(\{\tau_b \} | \bm{\alpha}\right)$ shows that the individual bound time durations $\{\tau_b \}$ are informative of the ligand concentration ratios $\bm{\alpha}$. 
%Hence, the estimators of individual ligand concentrations introduced in the next section will be based on these two likelihood functions. 

\section{MC System}
\label{sec:system}
We consider an MC system with a receiver equipped with single type of ligand receptors, attempting to decode a binary message $s \in \{0,1 \}$ encoded by a distant transmitter into the concentration of molecules, i.e., $c_s$, which propagate in the liquid MC channel through free diffusion. The following assumptions are adopted to define the system:
\begin{itemize}
	%	\item The system employs binary CSK modulation, such that the transmitter releases two different number of molecules corresponding to bit-0 and bit-1. 
	\item Following the convention in MC literature \cite{pierobon2011noise}, receiver is assumed to have a reception space of a volume $V$ around its lipid membrane, in which receptors along with incoming information and interferer molecules are uniformly distributed at any time.
	
	\item Receiver is time synchronized with the transmitter. In the absence of interferer molecules, the receiver has perfect knowledge of the channel state information (CSI) such that it exactly knows the concentration of information molecules in the reception space corresponding to $s=0$ and $s=1$ transmissions, i.e., $c_{s=0}$ and $c_{s=1}$, respectively. This is justified by the fact that molecular concentration at any point in three dimensional free diffusion channel is deterministically governed by the Fick's second law of diffusion \cite{pierobon2011diffusion}. On the other hand, in the presence of interferer molecules, receiver only knows the probability distribution of the number of interferer molecules in the reception space.  Our analysis will not explicitly take the intersymbol interference (ISI) into account for tractability of the derivations; however, we will perform analyses in Section \ref{sec:performance} for cases when the receptors approach saturation as a result of high-level ISI.  
	
	\item Sampling is performed only once for each receptor in a single signaling interval, such that the number of samples taken for each transmission is equal to the number of receptors, i.e., $N = N_R$.  Received molecular signal is taken as steady around the sampling time assuming that the MC system manifests diffusion-limited characteristics, i.e., diffusion dynamics are much slower than the binding kinetics. Receptors are assumed to be operating independently of each other. 
	
	%LOW-PASS FILTER CHARACTERISTICS OF THE MC CHANNEL ... RECEPTOR ARE ASSUMED TO BE INDEPENDENT OF EACH OTHER. 
	%	we assume that the concentration of molecules present in the reception volume is stationary for the duration of the sampling... low-past filter characteristics of the MC channel
	
	
	\item The channel and the reception space of the receiver are crowded with interferer molecules, which can bear significant affinity with the receptors. The number of interferer molecules $n_{in}$ in the reception space during a sampling period is taken as a Poisson random variable with the mean number $\mu_{n_{in}}$ \cite{bialek2012biophysics}. The binding rates of information and interferer molecules are taken as equal, i.e., $k_s^+ = k_{in}^+ = k^+$, following the assumption of diffusion-limited binding kinetics, as discussed in Section \ref{sec:statistics}.  However, the unbinding rates, determined by the affinity with the receptors, are different for information and interferer molecules, and denoted by $k_s^-$ and $k_{in}^-$, respectively. 
	
	%ARGUMENT OF DIFFUSION LIMITED REGIME, FOR SAKE OF SIMPLICITY WITHOUT LOSS OF GENERALITY.
	
	%	\item Reception volume, spherical with radius
	%	\item Same binding rate... diffusion-limited regime... for the sake of simplicity without loss of generality
	%	\item neglecting the transmitter and deterministic channel for information molecule concentration??? is this assumption... we assume that the concentration of molecules present in the reception volume is stationary for the duration of the sampling... low-past filter characteristics of the MC channel
	%	\item interferer molecules  in the reception volume follows Poisson distribution
	%	\item Receptors are independent of each other. 
\end{itemize}


\section{Detection Methods}
\label{sec:detection}
We introduce four detection methods that use different observable statistics of ligand-receptor binding reaction to decode the incoming messages in the presence of a random number of interferer molecules. These detection methods are based on the instantaneous number of bound receptors, total unbound time of receptors, receptor bound time intervals, and the combination of total unbound time time of receptors and receptor bound time intervals. 

%We obtain the probability distribution of the estimators for a given number of messenger molecules in the reception volume.   The results will be utilized in Section \ref{sec:detectionrule} to obtain the resulting BEP for a binary CSK-modulated MC system.

%At the end, the detection methods will be simplified into simple threshold testing problems assuming that the receiver is capable of computing the thresholds based on the sampled receptor states. (rephrase)


\subsection{Detection based on Number of Bound Receptors (DNBR)}

%If there are multiple receptors that are independently exposed to the same ligand concentration and not interacting with each other, the number of bound receptors becomes a binomial random variable with a success probability of $p_B$. Hence, the mean and the variance of the number of bound receptors can be written as 
%\begin{align}
%\E[n_B] &= \frac{c_L}{c_L + K_D} N_R, \\ \nonumber
%\Var[n_B] &= p_B (1-p_B) N_R,  \label{eq:Binomial}
%\end{align}
%where $N_R$ is the total number of receptors. 
The simplest detection method, which is widely studied in the MC literature, is based on sampling the number of bound receptors, exploiting the relation between ligand concentration and binding probability. As reviewed in Section \ref{sec:statistics}, the probability of finding a receptor in the bound state in the presence of interferers is given as
\begin{equation} \label{probBinding}
\p(\B|s, n_{in})  = \frac{c_s/K_D^s + c_{in}/K_D^{in}  }{1 + c_s/K_D^s + c_{in}/K_D^{in}},
\end{equation}
where $c_{in} = n_{in}/V$. Note that we condition the probability on the number of interferer molecules, $n_{in}$, instead of their concentration for mathematical convenience in dealing with the discrete Poisson distribution. 

As we discussed in Section \ref{sec:statistics}, the probability distribution of the number of bound receptors is binomial. Hence, given the number of information and interferer molecules in the reception space, the mean and variance of number of bound receptors $n_\B$ at equilibrium can be written as follows
\begin{align}
\E[n_\B|s, n_{in}] &= \p(\B|s, n_{in})  N_R, \\ \nonumber
\Var[n_\B|s, n_{in}] &= \p(\B|s, n_{in})  \bigl(1-\p(\B|s, n_{in}) \bigr) N_R.  \label{eq:Binomial}
\end{align}

%When the sensory system is composed of $N_R$ receptors, the random number of bound receptors $N$ follows Binomial distribution with success probability of $p_B$, i.e., 
%\begin{align}
%\p(n_B|&n_S, n_I) \\ \nonumber
%&= \dbinom{N_R}{n_B} \p(B|n_S, n_I)^{n_B} (1-\p(B|n_S, n_I))^{N_R-n_B}, 
%\label{binomial}
%\end{align}
%and the mean number of bound receptors becomes $\E[N] = p_B N_R$.
%
%\begin{equation}
%c_S  = n_S/V \label{eq:floor1} \nonumber
%\end{equation}
%\begin{equation}
%c_I  = n_I/V \label{eq:floor2}
%\end{equation}
The mean and variance conditioned only on the number of information molecules thus can be obtained by applying the total law of expectation and variance, i.e., 
\begin{align}
\E[n_\B|s] &=   \sum_{n=0}^\infty \E[n_\B|s,n_{in} = n] \p(n_{in} = n),  \\ \nonumber
\Var[n_\B|s] &= \E\Bigl[\Var[n_\B|s, n_{in}]\Big|s \Bigr]  + \Var\Bigl[\E[n_\B|s, n_{in}]\Big|s \Bigr],  \label{eq:numberofboundMeanAndVariance}
\end{align}
which do not lend themselves into a more tractable form, and therefore, the summations should be performed until the results converge. It is shown in Fig. \ref{fig:Gaussian_n} that the resulting probability distribution can be well approximated with a Gaussian distribution. Hence, we can write 
\begin{equation}
\p(n_\B|s) \approx \mathcal{N}\Bigl(\E[n_\B|s] , \Var[n_\B|s] \Bigr). \label{eq:DNBRpdf} 
\end{equation}
% before this, validate this through monte carlo simulation results. 


\subsection{Detection based on Receptor Unbound Time Durations  (DRUT)}

%\begin{align}
%P(T_U|c_L, c_I) = \dbinom{N_R}{n_B} P(B|c_L, c_I)^{n_B} (1-P(B|c_L, c_I))^{N_R-n_B}, 
%\label{binomial}
%\end{align}
In this method, the receiver performs the detection based on the estimation of total ligand concentration in the reception space using the total unbound time duration of receptors. From the log-likelihood function \eqref{eq:likelihoodtotalligand}, we can obtain the ML estimate of the total molecule concentration as follows: 
%Eq. (8?) tells us that 
%\begin{equation}
%\mathcal{L}\left(T_u | c_{tot}\right) = N \ln(c_{tot})-k^+ c_{tot} T_u,
%\end{equation}
\begin{align}
\frac{\partial \mathcal{L}\left(T_u | c_{tot}\right)}{\partial c_{tot}} \bigg|_{\hat{c}_{tot}^\ast} = 0,
\end{align}
which yields the ML estimator $\hat{c}_{tot}^\ast = \frac{N}{k^+ T_U}$. However, this is a biased estimator unless $N$ is very large. An unbiased version of this estimator can be obtained with a slight modification \cite{kuscu2018maximum} as follows
\begin{align}
\hat{c}_{tot} =  \frac{N-1}{k^+ T_u} \label{ctot_estimator},
\end{align}
whose mean and variance are obtained as 
\begin{align}
\E[\hat{c}_{tot}|s, n_{in}] &=  c_{tot} = c_s + n_{in}/V \label{Tu_mean},\\
\Var[\hat{c}_{tot}|s, n_{in}] &= \frac{c_{tot}^2}{N - 2}  = \frac{(c_s + n_{in}/V )^2}{N -2}\label{Tu_variance}.
\end{align}
Using the law of total expectation and variance, the mean and variance of this estimator conditioned only on the number of information molecules can be written as 
\begin{align}
\E[\hat{c}_{tot}|s] &=  \E\Bigl[\E[\hat{c}_{tot}|s, n_{in}]\Big|s\Bigr] \\ \nonumber
&= \E\bigr[c_s + n_{in}/V|s\bigr] \\ \nonumber
&= c_s + \mu_{n_{in}}/V,
\end{align}
% \sum_{n=0}^\infty \E[\hat{c}_{tot}|n_L,n_I = n] P(n_I = n),  \\ \nonumber
%&= \frac{1}{\sqrt{2 \pi \mu_I}} \int_{-\infty}^{\infty} \frac{n_L + n_I}{V} \exp \left(-\frac{(n_I-\mu_I)^2}{2 \mu_I}\right), \\ \nonumber
%& = something. \label{eq:numberofboundMeanAndVariance}
\begin{align}
&\Var[\hat{c}_{tot}|s] \\ \nonumber
&= \E\Bigl[\Var[\hat{c}_{tot}|s, n_{in}]\Big|s \Bigr] + \Var\Bigl[\E[\hat{c}_{tot}|s, n_{in}]\Big|s \Bigr]  \\ \nonumber
&= \mathlarger{\sum}_{n=0}^\infty \frac{(c_s + n/V)^2}{ (N-2)} \p(n_{in} = n) + \Var\left[c_s + n_{in}/V |s\right] \\ \nonumber
%&\stackrel{\text{(a)}}{\approx} \frac{1}{V^2 (N-2) \sqrt{2 \pi \mu_I}}  \int_{-\infty}^\infty (n_S + n)^2 \e^{ -\frac{(n-\mu_I)^2}{2\mu_I}} dn + \frac{\mu_I}{V^2}\\ \nonumber
& \approx \frac{1}{(N-2) \sqrt{2 \pi \mu_{n_{in}}}}  \int_{-\infty}^\infty \Bigl(c_s + \frac{n}{V}\Bigr)^2 \e^{ -\frac{(n-\mu_{n_{in}})^2}{2\mu_{n_{in}}}} dn + \frac{\mu_{n_{in}}}{V^2}\\ \nonumber
&\approx \frac{(c_s + \mu_{n_{in}}/V)^2} {N-2} + \frac{\mu_{n_{in}} (N-1)} {V^2 (N-2)},
\label{eq:numberofboundMeanAndVariance}
\end{align}
where the discrete Poisson distribution is approximated as a continuous Gaussian distribution. As demonstrated in Fig. \ref{fig:Gaussian_ctot}, the probability distribution of the total concentration estimator can also be approximated with a Gaussian distribution, i.e.,
\begin{equation}
\p(\hat{c}_{tot}|s) \approx \mathcal{N}\Bigl(\E[\hat{c}_{tot}|s] , \Var[\hat{c}_{tot}|s] \Bigr). \label{eq:DRUTpdf}
\end{equation}
\begin{figure}[!t]
	\centering
	\includegraphics[width=8cm]{expFigure_new}
	\caption{Probability distributions of the bound time durations corresponding to interferer molecules, information molecules, and the mixture of them. The dashed line indicates the time threshold (see \eqref{timethreshold}) that helps discriminate between information and interferer molecules by separating the longer binding events from the shorter ones. }
	\label{fig:mixture}
\end{figure}

\begin{figure*}[!t]
	\centering
	\subfigure[]{
		\includegraphics[width=7cm]{Gaussian_n}
		\label{fig:Gaussian_n}
	}
	\subfigure[]{
		\includegraphics[width=7.2cm]{Gaussian_ctot}
		\label{fig:Gaussian_ctot}
	}
	\subfigure[]{
		\includegraphics[width=7cm]{Gaussian_alpha}
		\label{fig:Gaussian_alpha}
	}
	\subfigure[]{
		\includegraphics[width=7cm]{Gaussian_c}
		\label{fig:Gaussian_c}
	}
	\caption{Gaussian approximation of decision statistics. Histograms are obtained via Monte Carlo simulations (50000 iterations) of stochastic ligand-receptor binding process under interference. Simulation parameters are set to the default values that are used in performance evaluation (see Table \ref{table:parameters}). Here we assume bit-0 is transmitted. Hence, we use $c_0 = 4 \times K_{D,S}$.  }
	\label{fig:Gaussian}
\end{figure*}

\subsection{Detection based on Receptor Bound Time Durations  (DRBT)}
The concentration ratio of information molecules in the reception space, $\alpha_s$, is also expected to be different for bit-0 and bit-1 transmissions, and thus, can be used for detection. We can obtain the ML estimation for the concentration ratio of information molecules, i.e., $\hat{\alpha}_s^{ML}$, by solving $\partial \mathcal{L} \left(\{\tau_b \} | \bm{\alpha}\right)  / \partial \alpha_s \Big|_{\hat{\alpha}_s^{ML}} = 0$, i.e., 
\begin{equation}\label{l2}
0 =\mathlarger{\mathlarger{\sum}}_{i=1}^{N} \frac{k_s^- \e^{-\left(k_s^- \tau_{b,i}\right)}} {  \hat{\alpha}_s^{ML} k_s^- \e^{-k_s^- \tau_{b,i}} + (1-\hat{\alpha}_s^{ML}) k_{in}^- \e^{-k_{in}^- \tau_{b,i}}}.
\end{equation}
However, the expression in \eqref{l2} does not have any analytical solution for ML estimate $\hat{\alpha}_s^{ML}$, and requires numerical approaches, which are not feasible for resource-limited bionanomachines \cite{kuscu2019transmitter}. Instead, in \cite{kuscu2019channel}, we proposed a feasible near-optimal estimation method for concentration ratios based on counting the number of binding events that fall in specific time intervals. In this estimation scheme, the time domain is divided into as many time intervals as the number of molecule types existing in the channel. In the presence of information molecules and only one type of interferer molecules, we need two time intervals, as demonstrated in Fig. \ref{fig:mixture}. These non-overlapping intervals are demarcated by a time threshold, which can be taken as proportional to the inverse of the unbinding rate of the molecule type with the lower binding affinity. Given that the interferer molecules bind receptors with lower affinity, we can write the time threshold as 
%\begin{equation} \label{timethreshold}
%T_i = \nu/k_i^- ~~\text{for}~~i \in \{1, \dots M-1 \}, 
%\end{equation}
\begin{equation} \label{timethreshold}
T_1 = \nu/k_{in}^-.
\end{equation}
Here, $\nu > 0$ is the proportionality constant. In this paper, we use  $\nu = 3$, which was previously shown in \cite{kuscu2019channel} to yield near-optimal performance in terms of estimation error. Note that in the same paper, we also showed that this transduction scheme is suitable for biological MC devices, as it can be implemented by active receptors based on the well-known KPR scheme. 

The probability of observing a ligand binding event with a binding duration falling in a specific time range can be obtained as
%\begin{equation}
%\p_l = \int_{T_{l-1}}^{T_l} \p(\tau_b') d\tau_b' =  \sum_{i=1}^M \alpha_i \left( \e^{-(k_i^- T_{l-1})} - \e^{-(k_i^- T_{l})} \right),
%\end{equation}


\begin{align}
\p_i = \int_{T_{i-1}}^{T_i} \p(\tau_b') d\tau_b' =&   \alpha_s \left( \e^{-k_s^- T_{i-1}} - \e^{-k_s^- T_{i}} \right) \\ \nonumber
&+ \alpha_{in} \left( \e^{-k_{in}^- T_{i-1}} - \e^{-k_{in}^- T_{i}} \right),
\end{align}
where $\p(\tau_b)$ is given in \eqref{eq:probboundduration}. Here we take $T_0 = 0$ and $T_2 = +\infty$. In matrix notation, the probabilities can be written as
\begin{equation} \label{eq:probmatrix}
\bm{\p} = \bm{Q} \bm{\alpha},
\end{equation}
where $\bm{\p} = [\p_1, \p_2]^T$,  $\bm{\alpha} = [\alpha_{in}, \alpha_s]^T$, and $\bm{Q}$ is an ($2 \times 2$) matrix with the elements 
%\begin{equation} \label{eq:Smatrix}
%s_{i,j} = \e^{-(k^-_j T_{i-1})} - \e^{-(k^-_j T_{i})}.
%\end{equation}
%
%\[
%\bm{S}=
%\begin{bmatrix}
%\e^{-(k^-_I T_{0})} - \e^{-(k^-_I T_{1})} & \e^{-(k^-_S T_{0})} - \e^{-(k^-_S T_{1})} \\
%\e^{-(k^-_I T_{1})} - \e^{-(k^-_I T_{2})} & \e^{-(k^-_S T_{1})} - \e^{-(k^-_S T_{2})} 
%\end{bmatrix}
%\]


\[
\bm{Q}=
\begin{bmatrix}
q_{1,1} & q_{1,2} \\
q_{2,1} & q_{2,2}
\end{bmatrix}
=
\begin{bmatrix}
1 - \e^{-k^-_{in} T_1} & 1- \e^{-k^-_s T_{1}} \\
\e^{-k^-_{in} T_{1}} & \e^{-k^-_s T_{1}}
\end{bmatrix}
\]



The number of binding events that fall in each time range follows a binomial distribution with the mean and the variance given by
\begin{equation}
\bm{\E[n_b}|s,n_{in}\bm{]} = \bm{\p} N,
\label{eq:meann}
\end{equation}
\begin{equation}
\bm{\Var[n_b}|s,n_{in}\bm{]} = \bigl( \bm{\p} \odot (1-\bm{\p}) \bigr) N,
\label{eq:varn}
\end{equation}
where $\bm{n_b}$ is an $(2 \times 1)$ vector with the vector elements $n_{b,i}$ being the number of binding events, whose durations are within the $i^\text{th}$ time range demarcated by $T_{i-1}$ and $T_i$, and $\odot$ denotes the Hadamard product, i.e., $(\bm{K} \odot \bm{L})_{i,j} = (\bm{K})_{i,j} (\bm{L})_{i,j}$. 

Applying the Method of Moments (MoM) with only the first moment yields a concentration ratio estimator in terms of number of binding events \cite{kuscu2019channel}, i.e., 
\begin{align} \label{Wmatrix}
\bm{\hat{\alpha}} = \left(\frac{1}{N}\right) \bm{W} \bm{n_b},  
\end{align}
where $\bm{W} = \bm{Q}^{-1}$, i.e., the inverse of $\bm{Q}$ matrix, which is also a ($2 \times 2$) matrix with elements $w_{i,j}$. 
%\[
%\bm{W}=
%\frac{1}{\e^{-(k_I^- T_1)}-\e^{-(k_S^- T_1)}}
%\begin{bmatrix}
%- \e^{-(k^-_S T_1)} & 1- \e^{-(k^-_S T_{1})} \\
%\e^{-(k^-_I T_{1})} & \e^{-(k^-_I T_{1})}-1
%\end{bmatrix}
%\]
Note that the estimated concentration ratio of information molecules becomes
%\begin{align}
%\hat{\alpha}_I &=   \left( n_1 w_{1,1} + n_2 w_{1,2} \right)/N, \\ \nonumber
%\hat{\alpha}_S &=  \left( n_1 w_{2,1} + n_2 w_{2,2} \right)/N, \label{ratio_est}
%\end{align}
\begin{align}
\hat{\alpha}_s =  \left( n_{b,1} w_{2,1} + n_{b,2} w_{2,2} \right)/N. \label{alpha_estimator}
\end{align}

%\begin{equation}
%\hat{\alpha}_l = \left(\frac{1}{N}\right)  \sum_{i=1}^M n_i w_{l,i}, \label{ratio_est}
%\end{equation}
%becomes the weighted sums of $M$ correlated binomial random variables with the weights $w_{l,i}$. 

The variance of this ratio estimator can be written as
\begin{align}\label{eq:alpha_variance}
\Var[&\hat{\alpha}_s|s, n_{in}] \\ \nonumber
&= \frac{1}{N^2}  \sum_{i=1}^2 \sum_{j=1}^2 w_{2,i} w_{2,j} ~\Cov[n_{b,i}, n_{b,j}|s, n_{in}],  
\end{align}
with the covariance function
\begin{equation}
\Cov[n_{b,i}, n_{b,j}| s, n_{in}]  = 
\begin{cases}
\Var[n_{b,i}| s, n_{in}],  & \text{if } i = j,\\
-p_i p_j N,    & \text{otherwise}.
\end{cases}
\label{eq:est_covariance}
\end{equation}
After some trivial mathematical manipulations, we can rewrite \eqref{eq:alpha_variance} in closed form as 
\begin{align}
\Var[\hat{\alpha}_s|s, n_{in}] = \frac{1}{N} \frac{\Gamma_1 (n_{in}/V)^2 + \Gamma_2 (n_{in}/V) + \Gamma_3}{(c_s + n_{in}/V)^2},  \label{eq:est_variance}
\end{align}
where, $\Gamma_1$, $\Gamma_2$ and $\Gamma_3$ are given in \eqref{eq:Gamma1}, \eqref{eq:Gamma2}, and \eqref{eq:Gamma3}, respectively.
\addtocounter{equation}{3}  
%\begin{align} 
%\Gamma_1  =& w_{2,1}^2 s_{1,1} - w_{2,1}^2 s_{1,1}^2 - 2 w_{2,1}w_{2,2}s_{1,1}s_{2,1} \\ \nonumber
%&+ w_{2,2}^2 s_{2,1} - w_{2,2}^2 s_{2,1}^2 \nonumber
%\end{align}
%\begin{align}
%\Gamma_2  =& n_S  ( w_{2,1}^2 s_{1,2} + w_{2,1}^2 s_{1,1}  - w_{2,1}^2 s_{1,1} s_{1,2}   \\ \nonumber
%&- w_{2,1}^2 s_{1,1} s_{1,2}  - 2 w_{2,1} w_{2,2} s_{1,1} s_{2,2}  - 2 w_{2,1} w_{2,2} s_{1,2} s_{2,1}  \\ \nonumber
%&+ w_{2,2}^2 s_{2,2}  + w_{2,2}^2 s_{2,1}   + w_{2,2}^2 s_{2,1} s_{2,2}  - w_{2,2}^2 s_{2,1} s_{2,2})  \\ \nonumber
%\end{align}
%\begin{align}
%\Gamma_3  =& n_S^2 (w_{2,1}^2 s_{1,2} - w_{2,1}^2 s_{1,2}^2 - 2 w_{2,1} w_{2,2} s_{1,2} s_{2,2} \\ \nonumber
%&+ w_{2,2}^2 s_{2,2} - w_{2,2}^2 s_{2,2}^2 ) 
%\end{align}


The expected value of the ratio estimator is equal to the actual value of the concentration ratio vector $\bm{\alpha}$, i.e.,
\begin{align}
\bm{\E[\hat{\alpha}}|s, n_{in} \bm{]} &= \frac{1}{N} \bm{W} \bm{\E[n_b}|s,n_{in}\bm{]}  \\ \nonumber
&= \bm{W} \bm{\p} = \bm{Q^{-1}} \bm{\p} = \bm{\alpha}.
\label{eq:meanratioest}
\end{align}

Using the law of total expectation and variance, we can write
\begin{align}
\E[\hat{\alpha}_s|s] &=  \E\Bigl[\E[\hat{\alpha}_s|S, n_{in}]\Big|s\Bigr] \\ \nonumber
&= \E\biggl[\frac{c_s}{c_s + n_{in}/V} \bigg|s\biggr] \\ \nonumber
&= \mathlarger{\sum}_{n=0}^\infty \frac{c_s}{c_s + n/V} \p(n_{in} = n),
\end{align}
\begin{align}\label{eq:boundtimeMeanAndVariance}
&\Var[\hat{\alpha}_s|s] \\ \nonumber
&= \E\Bigl[\Var[\hat{\alpha}_s|s, n_{in}] \Big|s \Bigr] + \Var\Bigl[\E[\hat{\alpha}_s|s, n_{in}] \Big|s\Bigr] \\ \nonumber
&=  \mathlarger{\sum}_{n=0}^\infty \frac{1}{N} \frac{\Gamma_1 (n/V)^2 + \Gamma_2 (n/V) + \Gamma_3 + N c_s^2}{(c_s + n/V)^2} ~\p(n_{in} = n) \\ \nonumber
& ~~~- \left(\sum_{n=0}^\infty \frac{c_s}{c_s + n/V}  \p(n_{in} = n)\right)^2.
\end{align}
It is shown in Fig. \ref{fig:Gaussian_alpha} that the distribution of the ratio estimator for information molecules can be approximated with a Gaussian distribution as follows
\begin{equation}
\p(\hat{\alpha}_s|s) \approx \mathcal{N}\Bigl(\E[\hat{\alpha}_s|s] , \Var[\hat{\alpha}_s|s] \Bigr). \label{eq:DRBTpdf}
\end{equation}
\begin{figure*}[!t]
	% ensure that we have normalsize text
	\normalsize
	% Store the current equation number.
	\setcounter{mytempeqncnt}{\value{equation}}
	% Set the equation number to one less than the one
	% desired for the first equation here.
	% The value here will have to changed if equations
	% are added or removed prior to the place these
	% equations are referenced in the main text.
	\setcounter{equation}{31}
	\begin{align} \label{eq:Gamma1}
	\Gamma_1  = w_{2,1}^2 q_{1,1} - w_{2,1}^2 q_{1,1}^2 - 2 w_{2,1}w_{2,2}q_{1,1}q_{2,1} + w_{2,2}^2 q_{2,1} - w_{2,2}^2 q_{2,1}^2 
	\end{align}
	\begin{align} \label{eq:Gamma2}
	\Gamma_2  = & c_s  \bigl( w_{2,1}^2 q_{1,2} + w_{2,1}^2 q_{1,1}  - w_{2,1}^2 q_{1,1} q_{1,2} - w_{2,1}^2 q_{1,1} q_{1,2}  - 2 w_{2,1} w_{2,2} q_{1,1} q_{2,2}  - 2 w_{2,1} w_{2,2} q_{1,2} q_{2,1}  \\ \nonumber
	&+ w_{2,2}^2 q_{2,2}  + w_{2,2}^2 q_{2,1}   + w_{2,2}^2 q_{2,1} q_{2,2}  - w_{2,2}^2 q_{2,1} q_{2,2}\bigr)  
	\end{align}
	\begin{align} \label{eq:Gamma3}
	\Gamma_3  = c_s^2 \bigl(w_{2,1}^2 q_{1,2} - w_{2,1}^2 q_{1,2}^2 - 2 w_{2,1} w_{2,2} q_{1,2} q_{2,2} + w_{2,2}^2 q_{2,2} - w_{2,2}^2 q_{2,2}^2 \bigr) 
	\end{align}
	% Restore the current equation number.
	\setcounter{equation}{\value{mytempeqncnt}}
	% IEEE uses as a separator
	\hrulefill
	% The spacer can be tweaked to stop underfull vboxes.
	\vspace*{4pt}
\end{figure*}

\subsection{Detection based on Receptor Unbound and Bound Time Durations (DRUBT)}
Combining the ratio estimator with the unbiased estimator of total ligand concentration, we can obtain an estimator for the individual concentration of information molecules as follows 
\begin{align} \label{eq:csestimator}
\hat{c}_s &= \hat{c}_{tot} \times \hat{\alpha}_s \\ \nonumber
&= \frac{N-1}{N} \frac{1}{k^+ T_u}  \left( n_{b,1} w_{2,1} + n_{b,2} w_{2,2} \right) \\ \nonumber
&\approx \frac{1}{k^+ T_u}  \left( n_{b,1} w_{2,1} + n_{b,2} w_{2,2} \right), ~~~ \text{for}~ N \gg 1.
\end{align}
The mean of this concentration estimator conditioned on the number of information and interferer molecules can be calculated as follows
\begin{align}
\E&[\hat{c}_s|s, n_{in}] = \E[\hat{c}_{tot}|s, n_{in}] \E[\hat{\alpha}_s|s, n_{in} ] = c_{tot} \alpha_s = c_s,
\end{align}
by exploiting the conditional independence of $\hat{c}_{tot}$ and $\hat{\alpha}_S$. The variance of this estimator can be obtained as follows
\begin{align} \label{eq:totalvarianceunbiased}
\Var[\hat{c}_s|s, n_{in}] &=\Var[\hat{c}_{tot}|s, n_{in}] \Var[\hat{\alpha}_s|s, n_{in}] \\ \nonumber
&+ \Var[\hat{c}_{tot}|s, n_{in}] \E[\hat{\alpha}_s|s, n_{in}]^2  \\ \nonumber
&+ \Var[\hat{\alpha}_s|s, n_{in}] \E[\hat{c}_{tot}|s, n_{in}]^2 \\ \nonumber
& \approx \frac{1}{N} \Bigl(\Gamma_1 (n_{in}/V)^2 + \Gamma_2 (n_{in}/V) + \Gamma_3 + c_s^2\Bigr) \\ \nonumber
& \text{~~~~for~} N \gg 1.  
\end{align}

Employing the law of total expectation and variance, we then obtain
\begin{align}
\E[\hat{c}_s|s] = \E\Bigl[\E[\hat{c}_s|s, n_{in}]\Big|s\Bigr] = \E[c_s|s] = c_s,
\end{align}
\begin{align} \label{eq:concentrationEstimatiorVariance}
&\Var[\hat{c}_s|s] = \E\Bigl[\Var[\hat{c}_s|s, n_{in}]\Big|s\Bigr] + \Var\Bigl[\E[\hat{c}_s|s, n_{in}]\Big|s\Bigr] \\ \nonumber
&= \frac{1}{N} \sum_{n=0}^\infty \Bigl(\Gamma_1 (n/V)^2 + \Gamma_2 (n/V) + \Gamma_3 + c_s^2 \Bigr) \p(n_{in} = n) \\ \nonumber
&\approx  \frac{1}{N \sqrt{2 \pi \mu_{n_{in}}}}  \\ \nonumber
&\times \int_{-\infty}^\infty \left(\Gamma_1 \left(\frac{n}{V}\right)^2+ \Gamma_2 \left(\frac{n}{V}\right) + \Gamma_3 + c_s^2\right) \e^{ -\frac{(n-\mu_{n_{in}})^2}{2\mu_{n_{in}}}} dn\\ \nonumber
&\approx \frac{1}{N} \left(\Gamma_1 \left(\frac{\mu_{n_{in}}}{V}\right)^2 + (\Gamma_1 + \Gamma_2) \frac{\mu_{n_{in}}}{V} + \Gamma_3 + c_s^2 \right).
\end{align}
Note that in \eqref{eq:concentrationEstimatiorVariance} we approximate the Poisson distribution of number of interferer molecules with a Gaussian distribution. 

We demonstrate in Fig. \ref{fig:Gaussian_c} that the p.d.f. of the concentration estimator can also be approximated with a Gaussian distribution, i.e.,
\begin{equation}
\p(\hat{c}_s|s) \approx \mathcal{N}\Bigl(\E[\hat{c}_s|s] , \Var[\hat{c}_s|s] \Bigr). \label{eq:DRUBTpdf} 
\end{equation}



\begin{table*}[!t]\scriptsize
	\centering
	\begin{threeparttable}
		\centering
		\caption[Comparison of detection methods under interference]{Comparison of Detection Methods}
		\label{table:comparison_interference}
		\begin{tabular}{llllll}
			\toprule	
			\textbf{Detection}   & \textbf{Sampled Receptor} & \textbf{Decision} & \textbf{Complexity}&  \textbf{Probability} \\ 
			\textbf{Method}   & \textbf{Statistics} & \textbf{Statistics} & &  \textbf{Distribution} \\
			\toprule
			DNBR  & Number of bound receptors & Number of bound receptors   & Low &  \eqref{eq:DNBRpdf}  \\ \midrule
			DRUT   & Total receptor unbound time & Total molecular concentration & Moderate & \eqref{eq:DRUTpdf} \\ \midrule
			DRBT         & Number of binding events of  & Concentration ratio of& High    &  \eqref{eq:DRBTpdf}    \\ 
			& durations in specific time intervals      & information molecules  &    &     \\ \midrule
			DRUBT &  Total receptor unbound time   & Total molecular concentration       & Very high &    \eqref{eq:DRUBTpdf} \\
			& + Number of binding events of   &  + Concentration ratio of& & \\
			&durations in specific time intervals    & information molecules & & \\  
			\bottomrule
		\end{tabular}%
	\end{threeparttable}	
\end{table*}%


\subsection{Decision Rule} 
\label{sec:detectionrule}
In previous sections, we obtain the likelihood of four different statistics given the number of information molecules in the reception space. A summary comparison of these detection methods is provided in Table \ref{table:comparison_interference}.

Considering that the system employs binary CSK, the decision rule can be simply written as
%$n_S$ corresponds to the number of information molecules received in the receiver volume for the transmitted message $S \in \{0,1\}$. 
\begin{equation}
\hat{s} = \argmax_{s \in \{0,1\}} \p(\kappa|s),
\label{gaussian_DRBT2}
\end{equation}
where $\kappa \in \{n_\B, \hat{c}_{tot}, \hat{\alpha}_s, \hat{c}_s \}$ is the received signal statistics corresponding to the introduced detection methods. The decision rule can be further simplified by defining a decision threshold $\lambda_{\kappa}$, i.e.,
\begin{equation} 
\kappa \underset{{H_0}}{\overset{H_{1}}{\gtrless}} \lambda_{\kappa}.
\label{threshold}
\end{equation}
For normally distributed statistics, the optimal decision threshold yielding the minimum error probability can be calculated as follows
%\begin{equation}
%\hat{S} = \argmax_{S \in \{0,1\}} \p(\hat{c}_S|n_S),
%\label{gaussian_DRBT2}
%\end{equation}
%which can also be simplified with a threshold, i.e.,
%\begin{equation} 
%\hat{c}_S \underset{{H_0}}{\overset{H_{1}}{\gtrless}} \lambda_{\hat{c}_S},
%\label{threshold_DRBT}
%\end{equation}
%where $ \lambda_{\hat{c}_S}$ is derived in the next section.
\begin{align} \label{threshold3}
& \lambda_{\kappa}  = \gamma_{\kappa}^{-1} \\ \nonumber
&\times  \Biggl( \Var[\kappa|s = 1] \E[\kappa|s = 0] -  \Var[\kappa|s = 0] \E[\kappa|s = 1]  \\ \nonumber
& + \Std[\kappa|s = 1]\Std[\kappa|s = 0] \\ \nonumber
& \times \sqrt{\bigl(\E[\kappa|s = 1] - \E[\kappa|s = 0] \bigr)^2+2 \gamma_{\kappa}  \ln{\frac{\Std[\kappa|s = 1]}{\Std[\kappa|s = 0]}}} ~\Biggr), 
\end{align}
%\begin{align}
%& \lambda_{\kappa}  = \gamma_{\kappa}^{-1}  \Biggl( \sigma_{\kappa|1}^2 \mu_{\kappa|0} - \sigma_{\kappa|0}^2 \mu_{\kappa|1} \\ \nonumber
%& + \sigma_{\kappa|1} \sigma_{\kappa|0} \sqrt{(\mu_{\kappa|1} - \mu_{\kappa|0})^2+2 \gamma_{\kappa}  \ln{\frac{\sigma_{\kappa|1}}{\sigma_{\kappa|0}}}} \Biggr), \label{threshold2}
%\end{align}
where $\gamma_{\kappa} = \Var[\kappa|s = 1] - \Var[\kappa|s = 0]$, and $\Std[.] = \sqrt{\Var[.]}$ denotes standard deviation. Given the decision thresholds, the BEP for each detection method can be obtained as

\begin{align} \label{eq:bepbep}
\p_\kappa(e) &=\frac{1}{2} \bigg[ \p_\kappa(\hat{s} = 1|s=0) + \p_\kappa(\hat{s} = 0|s=1) \bigg] \\ \nonumber
&=\frac{1}{4}\Biggl[\erfc \Biggl(\frac{\lambda_\kappa - \E[\kappa|s = 0] }{\sqrt{2 \Var[\kappa|s = 0] }}\Biggr) \\ \nonumber 
&~~~+  \erfc \Biggl(\frac {\E[\kappa|s = 1]  - \lambda_\kappa}{\sqrt{2 \Var[\kappa|s = 1]} }\Biggr)\Biggr]. 
\end{align}

\section{Performance Evaluation}
\label{sec:performance}
In this section, we evaluate the performance of the introduced detection methods in terms of BEP, which is calculated according to \eqref{eq:bepbep}. The default values of the system parameters used in the analyses are given in Table \ref{table:parameters}, with the reaction rates adopted from the previous literature \cite{pierobon2011noise, kuscu2018modeling, bialek2012biophysics}.

Throughout the analysis, the amount of interferer molecules in the reception space is expressed in terms of their concentration. However, the following convention is adopted to convert the concentration into the number of molecules when dealing with the discrete Poisson distribution in the computations. 
\begin{align}
\mu_{n_{in}} &= \floor*{\mu_{c_{in}} V}.
\end{align}
%\begin{align}
%\mu_{n_I} &= \floor*{\mu_{c_I} V}, \\ \nonumber
%n_S &= \floor*{c_S V}.
%\end{align}


We perform several analyses to evaluate the effect of the expected interferer concentration in the reception space, the ratio between the affinities of information and interferer molecules with the receptors, the ratio between the received information molecule concentrations corresponding to bit-$0$ ($s=0$) and bit-$1$ ($s=1$) transmissions,  and the number of receptors. In the presentation of the results, we also provide the saturation level of the receiver in terms of bound state probability $\p_\B$ of the receptors for $s=0$ and $s=1$. 



%\begin{table*}[!t]\scriptsize
%	\centering
%	\begin{threeparttable}
%		\centering
%		\caption{Comparison of Detection Methods}
%		\label{table:comparison}
%		\begin{tabular}{llllll}
%			\toprule	
%			\textbf{Detection}   & \textbf{Sampled Receptor} & \textbf{Decision} & \textbf{Complexity}&  \textbf{Probability} \\ 
%			\textbf{Method}   & \textbf{Statistics} & \textbf{Statistics} & &  \textbf{Distribution} \\
%			\toprule
%			DNBR  & Number of bound receptors & Number of bound receptors   & Low &  \eqref{eq:DNBRpdf}  \\ \midrule
%			DRUT   & Total receptor unbound time & Total molecular concentration & Moderate & \eqref{eq:DRUTpdf} \\ \midrule
%			DRBT         & Number of binding events of durations in specific time intervals      & Concentration ratio of information molecules  & High    &  \eqref{eq:DRBTpdf}    \\ \midrule
%			DRUBT &  Total receptor unbound time +   & Total molecular concentration +       & Very high &    \eqref{eq:DRUBTpdf} \\
%			\textbf{} 					 & Number of binding events of durations in specific time intervals    & Concentration ratio of information molecules & & \\
%			\bottomrule
%		\end{tabular}%
%	\end{threeparttable}	
%\end{table*}%



\begingroup
\begin{table}[!t]
	\centering
	\caption{Default Values of System Parameters}
	\renewcommand{\arraystretch}{1.2} % Default value: 1
	\begin{tabular}{ l | l }
		\hline \hline
		Binding rate for both types of molecules $(k^+)$ & $2 \times 10^{-17}$ m$^3$/s \\ \hline
		Unbinding rate for information molecules $(k^-_s)$ &$  10 $ s$^{-1}$  \\ \hline
		%Dissociation constant for correct ligands $(K_D^C)$ & $k^+ / k^-_m $ \\ \hline
		Affinity ratio $(\eta)$ & $ 0.2  $\\ \hline
		%		Dissociation constant for incorrect ligands $(K_D^I)$ & $K^M_D / \alpha$ \\ \hline
		Conc. of information molecules for $s = 0$ $(c_{s=0})$ & $4 \times K_D^s$  \\ \hline
		Conc. of information molecules for $s = 1$ $(c_{s=1})$ & $5 \times K_D^s$  \\ \hline
		%Volume of the receiver $(V_R)$ & $4/3 \pi r_R^3$ \\ \hline
		Mean concentration of interferer molecules $(\mu_{c_{in}})$ & $2 \times K_D^{in}$ \\ \hline
		Number of receptors on the receiver surface $(N_R)$ & $10000$ \\ \hline
		Volume of the reception space $(V)$ & $4000~\mu$m$^3$ \\ \hline
	\end{tabular}
	\label{table:parameters}
\end{table}
\endgroup

% write that you also give the saturation level of the receptors... which can be inferred from the equilibrium bound state probability of a receptor $\p_B$.






\subsection{Effect of Interferer Concentration}
The first analysis concerns the strength of molecular interference. We analyze the effect of expected concentration of interferer molecules in the reception space for two scenarios differing in the receiver saturation level. In the first scenario, we consider that the receiver is reasonably away from saturation by setting the received concentration of information molecules for bit-0 and bit-1 as $c_{s=0} = 4K_D^s$ and $c_{s=1} = 5K_D^s$, respectively. The results, demonstrated in Fig. \ref{fig:InterfererConcentration_NonSaturation}, show that DRUBT outperforms the other detection methods in the simulated range of interference levels. On the other hand, DRUT, while performing poorly under high-level interference, substantially outperforms DNBR and DRBT when the interference level is relatively low. In the same region, the performance improvement obtained by DRUBT is more pronounced. It is also worth noting that DRNB, which is the simplest among the investigated detection methods, performs better than DRBT at the lowest level of interference. 
\begin{figure*}[!t]
	\centering
	\subfigure[]{
		\includegraphics[width=7.5cm]{DetectionErrorAnalysis_InterfererConcentration_NonSaturation4}
		\label{fig:InterfererConcentration_NonSaturation}
	}
	\subfigure[]{
		\includegraphics[width=7.5cm]{DetectionErrorAnalysis_InterfererConcentration_Saturation4}
		\label{fig:InterfererConcentration_Saturation}
	}
	\caption{Bit error probability as a function of mean interferer concentration $\mu_{c_{in}}/K_D^{in}$ for (a) non-saturation and (b) saturation conditions of the receiver.}
	\label{fig:InterfererConcentrationMain}
\end{figure*}

%In the second case, we drive the receptors into saturation by setting the received concentrations as $c_M^0 = 19K_{D,M}$ and $c_M^1 = 20K_{D,M}$. In these conditions, the BEP for all detection methods is significantly higher than the non-saturation case as shown in Fig. \ref{fig:InterfererConcentration_Saturation}. Yet the performance improvement obtained with DRUBT is notable. This time, however, at low interference levels, DRUT, which was previously proposed for overcoming the receptor saturation problem \cite{kuscu2018maximum}, outperforms DRUBT.  
%%The relative performance of DRBT over DRNB is worth consideration. While overall DRBT performs better than DRNB, at the lowest level of interference, it performs worse in both non-saturation and saturation cases. 
%Also in the case of receptor saturation, the performance of DRBT is worsened as the mean interference level decreases below $\mu_{c_I} = 4 K_{D,I}$, in contrast to the common trend observed in other detection methods. The reason may not be obvious at first glance. In this particular range of interference, the difference between the mean of ratio estimates conditioned on bit-0 and bit-1, i.e., $\E[\hat{\alpha}_M^S|S=0]$ and $\E[\hat{\alpha}_M^S|S=1]$, respectively, decreases as the interference level is reduced. On the other hand, the corresponding variances of the ratio estimates keep increasing with decreasing interference. This hampers the receiver's ability to discriminate between two symbols. The poor performance of the ratio estimation in this range also affects the performance of DRUBT. As a result, at very low interference levels, DRUBT is outperformed by DRUT. 


In the second case, the receptors are driven into saturation by setting the received concentrations as $c_{s=0} = 19K_D^s$ and $c_{s=1} = 20K_D^s$. In these conditions, the BEP for all detection methods is significantly higher than the non-saturation case as shown in Fig. \ref{fig:InterfererConcentration_Saturation}. Yet the performance improvement obtained with DRUBT is notable. This time, however, at low interference levels, DRUT, which was previously proposed for overcoming the receiver saturation problem \cite{kuscu2018maximum}, outperforms DRUBT.  
%The relative performance of DRBT over DRNB is worth consideration. While overall DRBT performs better than DRNB, at the lowest level of interference, it performs worse in both non-saturation and saturation cases. 
Also, in the case of receiver saturation, the performance of DRBT is worsened as the mean interference level decreases below $\mu_{c_{in}} = 4 K_D^{in}$, in contrast to the common trend observed in other detection methods. This is because the difference between the mean of ratio estimates conditioned on bit-0 and bit-1, i.e., $\E[\hat{\alpha}_s|s=0]$ and $\E[\hat{\alpha}_s|s=1]$, is a concave function of mean interferer concentration, which is maximized around $\mu_{c_{in}} = 4 K_D^{in}$. On the other hand, the corresponding variances of the ratio estimates monotonically increase with the decreasing interference level. This hampers the receiver's capacity to discriminate between bit-$0$ and bit-$1$. The poor performance of the ratio estimation in this range also affects the performance of DRUBT. As a result, at very low interference levels, DRUBT is outperformed by DRUT. 


%the mean ratios corresponding to bit-1 and bit-0 decreases again, making their discrimination difficult. The reason is not obvious at first glance. The worsening performance of ratio estimation, also affects the performance of DRUBT. As a result, at very low interferer levels, DRUT outperforms DRUBT. 

%One is dealing with the ratio, the other one is nonlinear.  and the distance between the ratio, both ratios increase, increasing the variance? , increase in variance becomes more pronounced than the decrease in the distance??? does not make sense... testtest.m

\subsection{Effect of Similarity between Information and Interferer Molecules}
The affinity ratio $\eta = k_s^-/k_{in}^-$ between information and interferer molecules determines how similar they are in terms of binding affinity with the receptors. The effect of affinity ratio on the detection performance is analyzed in two parts corresponding to the scenarios when $\eta<1$ and $\eta>1$. In both analyses, we keep the unbinding rate of information molecules constant and equal to its default value $k_s^- = 10$ s$^{-1}$, and thus, the unbinding rate of interferer molecules $k_{in}^-$ changes with the varying affinity ratio. We also keep the mean concentration of interferer molecules in the reception space constant and equal to 
$\mu_{c_{in}} = 10 K_D^s = 5$ $\mu$m$^{-3}$.
\begin{figure*}[!t]
	\centering
	\subfigure[]{
		\includegraphics[width=7.5cm]{DetectionErrorAnalysis_SimilarityAlpha4}
		\label{fig:SimilarityAlpha}
	}
	\subfigure[]{
		\includegraphics[width=7.5cm]{DetectionErrorAnalysis_SimilarityAlphaReverse4}
		\label{fig:SimilarityAlphaReverse}
	}
	\caption{Bit error probability as a function of affinity ratio $\eta = k_s^-/k_{in}^-$ for the cases (a) when information molecules have more binding affinity, i.e., $\eta <1$, and (b) when interferer molecules have more binding affinity, i.e., $\eta >1$. }
	\label{fig:SimilarityAlphaMain}
\end{figure*}

In the first scenario, information molecules have higher binding affinity than the interferers, e.g., $\eta<1$. As is seen from the results provided in Fig. \ref{fig:SimilarityAlpha}, when the two types of molecules become more similar, the error probability substantially increases for all detection methods except DRUT. The performance of DRUT is not affected by the binding affinity, as the total unbound time of receptors, which DRUT solely relies on, is independent of the unbinding rate of bound ligand-receptor pairs, and only depends on the binding rate and the molecular concentration. On the other hand, DRUBT performs the best among the analyzed detection methods, when the information and interferer molecules differ greatly in terms of binding affinity. In the case of high similarity, the ratio estimator DRBT is the worst performer, as it becomes unable to discriminate between the two types of molecules based on their bound time durations. The same reasoning applies to the performance of DRUBT in this region, which also partly relies on the estimation of the concentration ratio. 

In the second analysis, we consider the case when the information molecules have lower binding affinity than the interferers. In this case, we expect that the interferer molecules occupy more receptors than the information molecules. As shown in Fig. \ref{fig:SimilarityAlphaReverse}, DNBR, the simplest detection method, performs particularly poorly for $\eta>1$.  As the affinity ratio increases, the advantage of the detection methods relying on the difference between unbinding rates becomes more pronounced. While DRUBT significantly outperforms other methods in this particular case, the performance of DRUT, which is much more practical than DRUBT, is notable. 

\begin{figure*}[!t]
	\centering
	\subfigure[]{
		\includegraphics[width=7.5cm]{DetectionErrorAnalysis_Bit0Concentration4}
		\label{fig:Bit0Concentration}
	}
	\subfigure[]{
		\includegraphics[width=7.5cm]{DetectionErrorAnalysis_Bit0ConcentrationCLOSE4}
		\label{fig:Bit0ConcentrationClose}
	}
	\caption{(a) Bit error probability with varying ratio of the received concentrations corresponding to bit-0 and bit-1 transmissions, i.e., $c_{s=0}/c_{s=1}$. A magnified view is provided in (b).}
	\label{fig:Bit0ConcentrationMain}
\end{figure*}
\subsection{Effect of Bit-0/Bit-1 Concentration Ratio}
We also analyze the effect of the distance between bit-0 and bit-1 in terms of the received concentration of information molecules. In the case of high ISI in the MC channel, it is very likely that a considerable amount of information molecules from previous transmissions remains in the reception space. As a result, the ratio between the distinct concentration levels corresponding to bit-0 and bit-1 transmissions may approach 1, obstructing the discrimination between them. Here we gradually change the ratio between the received information molecule concentrations for bit-0 and  bit-1 from 0.1 to 0.99, and provide the results in Fig. \ref{fig:Bit0Concentration} (with a magnified version in Fig. \ref{fig:Bit0ConcentrationClose}). As the concentration ratio approaches 1, all the detection methods fail to provide an acceptable error performance. On the other hand, DRUBT performs significantly better than any of the other detection methods tested. It is also to be noted that the performance of DRUT becomes the worst among all, when the concentration levels are well-separated, e.g., in the case of very low-level ISI. In this particular range, DNBR, which is the simplest detection method, performs very well. This indicates that even in the presence of interferer molecules, the instantaneous number of bound receptors can provide sufficient statistics for detection as long as the concentration levels for bit-0 and bit-1 are well-separated, and the interferer concentration is at a moderate level. 


\begin{figure}[!t]
	\centering
	\includegraphics[width=8cm]{DetectionErrorAnalysis_NumberofSamples4}
	\caption{Bit error probability as a function of number of receptors. }
	\label{fig:NumberofSamples}
\end{figure}

\subsection{Effect of Number of Receptors}
The number of receptors determines the number of independent samples taken for detection. For example, in DNBR, the instantaneous number of bound receptors is the sum of $N_R$ random variables independently following Bernoulli distribution. In DRUT and DRUBT, the total unbound time $T_u$ is the sum of $N_R$ unbound time intervals which independently follow exponential distributions. In DRBT and DRUBT, each of the independent receptors is assumed to sample only one binding event, leading to the observation of $N_R$ independent binding events. All of the investigated detection methods show similar, almost log-linear, performance trends with the varying number of receptors, as shown in Fig. \ref{fig:NumberofSamples}. Nevertheless, the most significant performance improvement is observed with DRUBT as it relies on both unbound time and bound time statistics taken independently from all receptors.
%Therefore, the number of samples in all cases is equal to the number of receptors.

%ALSO NUMBER OF SAMPLES CANNOT BE TAKEN EQUAL TO NUMBER OF RECEPTORS FOR NB AND DRUT DETECTION!. 








\section{Discussion on Implementation} 
\label{sec:implementation}
The introduced detection methods are practical in the sense that they can be implemented by biologically plausible synthetic receptors and CRNs in MC receivers. In this section, we discuss four different receptor designs that can transduce the required receptor statistics (i.e., number of bound receptors, total unbound time, number of binding events with the durations within specific time intervals) into the concentration of intracellular molecules, i.e., secondary messengers. The receptor designs incorporate an activation mechanism, which was previously introduced in \cite{kuscu2019channel}, to control the start time and the duration of the sampling process. We also discuss potential CRNs, which can chemically process the generated secondary messengers in order to perform the analog and digital computations required for the detection. Lastly, we provide a discussion of the state-of-the-art synthetic biology tools and emerging research trends that can enable the implementation of the proposed receptors and CRNs.
%
%In this section, we investigate the practical aspects of the proposed channel sensing methods. For the implementation of the method, we focus on a synthetic biology-based approach, because biosensor-based approaches for MC receiver, as overviewed in \cite{kuscu2018survey, kuscu2016physical}, do not allow inspecting the states of individual receptors, and thus, implementing the proposed estimators.
%
%The key element in the channel sensing is the biological receptors, which are the interface between the exterior and interior of a living cell, and transduce the external signals represented by the concentration of ligands into intracellular signals in the form of concentration of second messengers inside a living cell. The transduced signals need to be further processed for the estimation to be achieved. 

%The proposed estimators, both unbiased and biased, rely on two statistics, i.e., total unbound time $T_u$, and the number of binding events $n_i$ of durations within $[T_{i-1}, T_i]$ for $i \in \{1, \dots, M\}$. Our first aim is to provide a practical synthetic receptor design that can transduce both the unbound time and bound time information into the concentration of different intracellular molecules. We then investigate a chemical reaction network (CRN) that can chemically process these intracellular molecules to perform the calculations required for the proposed estimators. 

%\subsection{Acquisition of Receptor Statistics}
%The proposed estimators require the sampling of only a single pair of unbound and bound time durations from each receptor, as demonstrated in Fig. \ref{fig:kpr}(c), because the information of the exact number of independent samples is crucial for the estimation performance. To equate the number of samples and receptors, i.e., $N = N_R$, we first propose a receptor activation mechanism that can be triggered by the receiver cell when it decides to sense the channel. In this scheme, the receptors can be in one of the 6 main states, i.e., inactive unbound/bound, active unbound/bound and intermediate unbound/bound states, depending on the history of their reactions with ligands and intracellular molecules. The receptors can perform the sampling of the unbound time durations only in the active unbound state, and the bound time durations only in the active bound state through different mechanisms, which will be discussed shortly. Next, we describe the proposed activation mechanism along with the sampling of unbound time durations, and then we propose a modified kinetic proofreading (KPR) scheme for the sampling of bound time durations. 


%Proposed sampling mechanisms use intermediate states, active and inactive states.... 
%Describe the activation mechanism... Reactions, and deactivation. 

\subsection{Activation Mechanism}
In order to control the start time and the duration of the sampling, the receiver must have an activation mechanism that allows the receptors to generate secondary messengers only when they are in the active state. In \cite{kuscu2019channel}, we proposed an activation mechanism that enables the sampling of the required statistics from independent receptors only once during each sampling period. The synthetic receptor designs,  which will be introduced next, incorporate this mechanism. In this scheme, the cell generates activator molecules $A^+$, which can rapidly diffuse and interact only with the inactive receptors at the reaction rate $\omega$, and shift them to active or intermediate states depending on the adopted receptor design. The generation of the activation molecules is governed by the following reaction
\begin{equation}
\ce{ {\varnothing}  ->[{g(t)\psi^+}] {A^+}},
\end{equation}
where $g(t) \psi^+$ is the time-varying generation rate of $A^+$ molecules, with $g(t) \approx \delta(t-t_A)$ being a very short pulse signal centered around the activation time $t_A$. The generation of $A^+$ molecules is followed by 
the generation of deactivation molecules $A^-$, i.e.,
\begin{equation}
\ce{ {\varnothing}  ->[{d(t)\psi^-}] {A^-}}.
\end{equation}
The generation rate of deactivation molecules is given by $d(t) \psi^-$. Here $d(t) \approx \delta(t-t_D)$ is an impulse-like signal centered around the deactivation time $t_D$. The activation molecules are degraded by the deactivation molecules with the reaction rate $\rho$, i.e.,
\begin{equation}
\ce{ {A^+ + A^-} ->[{\rho}] {\varnothing} }.
\end{equation}
In this way, the duration of the overall sampling process is controlled by the receiver cell. Note that the reaction rates governing the activation mechanism, i.e., $\psi^+$, $\psi^-$, $\omega$, $\rho$, should be very high compared to the ligand-receptor binding/unbinding reaction rates to prevent the inactivated receptors from being re-activated in the same sampling interval. 
\begin{figure*}[!t]
	\centering
	\includegraphics[width=16cm]{Implementation2}
	\caption{(a) Receptor design for DNBR. (b) Sampling of the number of bound receptors. (c) Receptor design for DRUT. (d) Sampling of the receptor unbound time intervals. }
	\label{fig:implementation1}
\end{figure*}

%To ensure that the inactivated receptors are not re-activated during the same sampling process for the sake of obtaining only a single pair of unbound and bound duration samples from each receptor, the generation rates of activation and deactivation molecules, i.e., $\psi^+$ and $\psi^-$, respectively, as well as the rate of reaction between activation molecules and receptors, i.e., $\omega$, and the rate of deactivation reaction $\rho$ should be very high compared to the ligand-receptor binding/unbinding reaction rates.

\subsection{Implementation of DNBR}
In DNBR, we need a representation of number of bound receptors at the sampling time in terms of concentration of intracellular molecules. A potential synthetic receptor design is provided in Fig. \ref{fig:implementation1}(a). In this design, the receptor has three states, unbound state $\U$, inactive bound state $\B_I$, and active bound state $\B_A$. The receiver cell releases intracellular activation molecules at the time of sampling. The released activation molecules $A^+$ only react with $\B_I$, and convert them to $\B_A$. The active bound receptors release an intracellular molecule $M$ upon unbinding from a ligand, and returns to the unbound state. As a result, the number of $M$ molecules encodes the number of bound receptors at the sampling time. This intracellular signal can also be amplified if the receptors are designed to release multiple $M$ molecules in the active state. 

The next process is to compare the intracellular concentration of $M$ molecules to a threshold encoded by a different secondary messenger. A simple comparator can be implemented through the following reaction
\begin{equation}
\ce{ {M + X} ->[{\xi}] {\varnothing} }, 
\end{equation}
where $X$ molecules encode the threshold given in \eqref{threshold3} for DNBR. If any $M$ molecule remains inside the cell after this reaction, the receiver decides bit-1, otherwise it decides bit-0. In the case that the intracellular concentration of $M$ molecules is amplified, the threshold signal should be also amplified proportionally.  

%Activation mechanism... 
%While returning to the inactivate bound state, releases a second messenger, M, ... Therefore, the number of M molecules, will give... The mechanism can be amplified, for example, it can release multiple M molecules to amplify the intracellular signal. 

%Comparator circuit. 

\subsection{Implementation of DRUT}
DRUT requires the transduction of the total unbound time of receptors into the concentration of second messengers. We propose a synthetic receptor design with an activation mechanism demonstrated in Fig. \ref{fig:implementation1}(c), to guarantee that only one unbound time information is acquired from each independent receptor. In this design, the receptor has 5 states: inactive unbound ($\U_I$), intermediate unbound ($\U_A^\ast$), active unbound ($\U_A$), inactive bound ($\B_I$), and active bound ($\B_A$) states. At the sampling time, the receiver releases the activation molecules, which can rapidly diffuse and react only with the receptors at $\U_I$ or $\B_I$ states. If the receptor is already unbound at the time of activation, it stays idle until the next complete unbound interval, when it starts releasing intracellular molecules that encode the unbound time duration. Therefore, an already-unbound receptor first goes into the intermediate state $\U_A^\ast$ upon receiving the activation signal, which is followed by $\B_A$ and $\U_A$ states. If it is bound when activated, the next unbinding event takes it to the active unbound state $\U_A$. Receptors at $\U_A$ state release intracellular molecules $S$ at a fixed rate, and upon the first binding event, they transition to the inactive bound state $\B_I$, simultaneously releasing a single molecule of a different type $R$ in order to encode the number of independent samples taken. 


The resulting concentration of intracellular $S$ molecules encodes the total unbound time of receptors over a single period of sampling. These second messengers together with $R$ molecules can be processed by a CRN, which biochemically implements the total concentration estimator, given in Eq. \eqref{ctot_estimator}. An example CRN could be as follows: 
\begin{equation}
\ce{ {R} ->[{1}] {R} + { Y} },
\end{equation}
\begin{equation}
\ce{ S + Y ->[{k^+}] S }.
\end{equation}
In this CRN, we introduce another type of intracellular molecule $Y$, which is produced by $R$ molecules at the unit rate, while consumed by $S$ molecules at the common binding rate of ligands $k^+$. The rate equation of this CRN can be written as
\begin{align}
\frac{d\E[ n_{Y}] }{dt} = \E[n_{R}]  - k^+ \E[n_{S}]  \E[ n_{Y}], 
\end{align}
where $n_Y$, $n_R$, and $n_S$ are the number of $Y$, $R$, and $S$ molecules, respectively. Given the initial condition $\E[ n_{Y}^0] = 0$, the steady-state solution for $\E[ n_{Y}]$ can be obtained as
\begin{align}
\E[ n_{Y}^{ss}] = \frac{\E[n_{R}^{ss}]}{k^+ \E[n_{S}^{ss}]}, 
\end{align}
Given that $n_R^{ss}$ and $n_S^{ss}$ encode the number of independent receptors and the total unbound time, respectively, the resulting number of $Y$ molecules at steady-state $n_Y^{ss}$ approximates the total concentration estimator $\hat{c}_{tot} = \frac{N-1}{k^+ T_u}$.  


For decision, the comparator reaction utilized in DNBR can also be implemented here, i.e., 
\begin{equation}
\ce{ {Y + X} ->[{\xi}] {\varnothing} },
\end{equation}
where the number of $X$ molecules encodes the optimal threshold value. If any $Y$ molecule remains in the cell as a result of this reaction, the receiver decides bit-1, otherwise it decides bit-0.

%
%we have intermediate unbound state... Following the figure, describe the scenario. At the time of activation receptor can be either in inactive unbound UI or inactive bound state BI. In the inactive unbound state, it goes into the intermediate state, instead of the active state. Because the unbound time is not complete. 
%
%In the bound state, goes into active bound, which transitions into the active unbound state upon the next unbinding event. This way, a complete unbound time interval can be acquired. 
%
%In the active unbound state, the receptor generates S molecules with a specific rate, common to all receptors. WHAT ABOUT CONVERTING IT INTO THE ESTIMATOR. CHECK THAT. 
%
%Comparator circuit





\subsection{Implementation of DRBT}
DRBT requires the number of binding events with durations that fall into specific time ranges to be encoded into the concentration of intracellular molecules. Following our proposal in \cite{kuscu2019channel}, we introduce a synthetic receptor design with an activation mechanism and a modified KPR mechanism, as demonstrated in Fig. \ref{fig:implementation2}(a). The activation mechanism is similar to those introduced for DNBR and DRUT, and ensures that only one binding time information is received from each independent receptor, as shown in Fig. \ref{fig:implementation2}(b). In this design, when an active unbound receptor $\U_A$ binds to a ligand, it switches to the active bound state $\B_A$ where the KPR mechanism is activated. 


The KPR mechanism consists of two substates, $\B_A^1$ and $\B_A^2$, with a unidirectional state transition rate $\beta$, which can be set as a function of the time threshold $T_1$ as follows  
\begin{align}
\beta = \kappa_i/T_1 ,
\end{align}
where $\kappa_i$ is a tuning parameter adopted to optimize the transition rate for accurate sampling of the receptor bound time durations. Our analysis in \cite{kuscu2019channel} shows that $\kappa_i = 3/5$ provides reasonable accuracy in representing the number of binding events $n_{b,1}$ and $n_{b,2}$ with second messengers via the stochastic KPR mechanism.
\begin{figure*}[!t]
	\centering
	\includegraphics[width=16cm]{Implementation3}
	\caption{(a) Receptor design for DRBT. (b) Sampling of the number of binding events of durations within specific time intervals. (c) Receptor design for DRUBT. (d) Sampling of the receptor unbound time intervals and the number of binding events of durations within specific time intervals. }
	\label{fig:implementation2}
\end{figure*}

A receptor is allowed to return to the inactive unbound state $\U_I$ at any time by unbinding from the bound ligand. While returning to the state $\U_I$, the receptor releases a single $R$ molecule encoding the number of independent samples, and one of the intracellular molecules $D_1$ or $D_2$, depending on the last visited KPR substate. In this way, the KPR mechanism allows to discriminate long binding events, which are more likely to be resulting from the molecules with higher affinity, from short binding events through encoding the number of corresponding binding events into the number of $D_1$ and $D_2$ molecules. A steady-state analysis of a similar KPR mechanism is provided in \cite{kuscu2019channel}. 

The generated intracellular molecules $R$, $D_1$ and $D_2$ are biochemically processed by a CRN to realize the ratio estimator $\hat{\alpha}_s$, given in \eqref{alpha_estimator}. An example CRN can be as follows
\begin{equation}
\ce{ {D_1} ->[{w_{2,1}}] {D_1} + { Y} },
\end{equation}
\begin{equation}
\ce{ {D_2} ->[{w_{2,2}}] {D_2} + { Y} },
\end{equation}
\begin{equation}
\ce{ R + Y ->[{1}] R }.
\end{equation}
In this CRN, the intracellular molecules $Y$ are produced by $D_1$ and $D_2$ with the reactions rates $w_{2,1}$ and $w_{2,2}$, respectively. They are consumed by $R$ molecules at the unit rate. The rate equation of this CRN can then be written as 
\begin{align}
\frac{d\E[ n_{Y}] }{dt} = w_{2,1} \E[n_{D_1}] + w_{2,2} \E[n_{D_2}] -  \E[n_{R}]  \E[ n_{Y}].
\end{align}
Given the initial condition $\E[ n_{Y}^0] = 0$, the steady-state solution for $\E[ n_{Y}]$ can be obtained as follows
\begin{align}
\E[ n_{Y}^{ss}] = \frac{w_{2,1} \E[n_{D_1}^{ss}] + w_{2,2} \E[n_{D_2}^{ss}] }{E[n_{R}^{ss}]}. 
\end{align}
% with $\E[n_{D_1}^{ss}] = \E[n_{D_1}^{0}]$ 
%long binding events.  
If $n_{D_1}^{ss}$ and $n_{D_2}^{ss}$ encode $n_{b,1}$ and $n_{b,2}$, respectively, and $n_{R}^{ss}$ encodes the number of independent samples $N$, then the number of $Y$ molecules at steady-state $n_{Y}^{ss}$ approximates the concentration ratio estimator $\hat{\alpha}_s = \left( n_{b,1} w_{2,1} + n_{b,2} w_{2,2} \right)/N.$

The comparator reaction for decoding the transmitted bit can be realized in a similar way as DNBR and DRUT, i.e., 
\begin{equation}
\ce{ {Y + X} ->[{\xi}] {\varnothing} },
\end{equation}
where the number of $X$ molecules encodes the optimal threshold given in \eqref{threshold3}. If any $Y$ molecule remain in the cell as a result of this reaction, the receiver decides bit-1, otherwise bit-0 is decided. 

%
%Scenarios... 1 and 2. 
%
%Two-state modified KPR mechanism, which is introduced in REF. 
%
%Comparator circuit. 


\subsection{Implementation of DRUBT}

DRUBT requires the transduction of both the total unbound time, and the number of binding events of durations within specific time ranges. Hence, the combination of the synthetic receptor designs introduced for DRUT and DRBT enables the required functionality. The receptor design given in Fig. \ref{fig:implementation2}(c) ensures that only a single pair of complete unbound and bound time duration is sampled from each independent receptor. For DRUBT, receptors are not required to generate $R$ molecules, because the concentration estimator is not a function of the number of independent samples.

Given that the number of generated $S$, $D_1$ and $D_2$ molecules encodes $T_U$, $n_{b,1}$ and $n_{b,2}$, respectively, the CRN for the concentration estimator $\hat{c}_s$, given in \eqref{eq:csestimator}, can be implemented as follows 
\begin{equation}
\ce{ {D_1} ->[{w_{2,1}}] {D_1} + { Y} },
\end{equation}
\begin{equation}
\ce{ {D_2} ->[{w_{2,2}}] {D_2} + { Y} },
\end{equation}
\begin{equation}
\ce{ {S} + {Y} ->[{k^+}] S }.
\end{equation}
The rate equation of this CRN can be written as
\begin{align}
\frac{d\E[ n_{Y}] }{dt} = w_{2,1} \E[n_{D_1}] + w_{2,2} \E[n_{D_2}] -  k^+ \E[n_{S}]  \E[ n_{Y}].
\end{align}
Given the initial condition $\E[ n_{Y}^0] = 0$, the steady-state solution for $\E[ n_{Y}]$ can be obtained as
\begin{align}
\E[ n_{Y}^{ss}] = \frac{w_{2,1} \E[n_{D_1}^{ss}] + w_{2,2} \E[n_{D_2}^{ss}] }{k^+ E[n_{S}^{ss}]}. 
\end{align}
% with $\E[n_{D_1}^{ss}] = \E[n_{D_1}^{0}]$ 
%long binding events.  
As is obvious, the number of $Y$ molecules at steady-state $n_{Y}^{ss}$ approximates $\hat{c}_s =  \frac{1}{k^+ T_u}  \left( n_{b,1} w_{2,1} + n_{b,2} w_{2,2} \right)$ for  $N \gg 1$.


The following comparator can be applied here as well for decoding:  
\begin{equation}
\ce{ {Y + X} ->[{\xi}] {\varnothing} },
\end{equation}
where the number of $X$ molecules encodes the optimal threshold value given in \eqref{threshold3}. If eventually $Y$ molecules outnumber $X$ molecules, the receiver decides bit-1, otherwise it decides bit-0.


\subsection{Discussion}
	% what we need in terms of functionality... multiple states, active, inactive, intermediate states, transitions between them regulated by binding of certain molecules.
	The receptor and CRN designs proposed for the implementation of the MC detection methods are biologically plausible in the sense that similar designs are already utilized by living cells, and synthetic receptors and intracellular CRNs obtained via modification of these natural designs or via de novo designs are becoming increasingly sophisticated with the recent advances in synthetic biology. 

%modifications over these natural designs and creation of de novo designs are becoming increasingly more sophisticated with the recent advances in synthetic biology.

The introduced detection methods rely on multi-state receptors with the state transitions being realized by binding/unbinding of ligands or activation/deactivation molecules. Each of these states either determines the function of the receptors, e.g., releasing a type of intracellular molecules, or acts as an intermediate state regulating the set of next feasible state transitions. Such multi-state receptors are widely utilized in biological cells \cite{wong2020tcrbuilder, lau2013conformational}. In these natural systems, the receptor states correspond to different conformational states of the receptors, where the receptors typically manifest different binding/signaling characteristics \cite{kahsai2011multiple}. 

Our receptor designs in DRBT and DRUBT also incorporate a modified KPR mechanism to discriminate between the binding events of the information and the interferer molecules. KPR has long been speculated to be the mechanism underlying the impressive performance of T cells in discriminating between the lower-affinity self-ligands and higher-affinity antigens to evoke the immune response \cite{mckeithan1995kinetic, rabinowitz1996kinetic}, and recently a strong empirical evidence has been provided to corroborate this hypothesis \cite{yousefi2019optogenetic}. In such KPR systems, intermediate receptor states delay the activation of the receptor and consequent release of intracellular molecules as a way of exploiting the statistical difference between binding durations of self-ligands and antigens. The modified KPR scheme in our receptor designs only slightly differs from the natural KPR scheme in the sense that the unbinding of ligands during the receptors' first KPR substate (intermediate state) also results in the release of an intracellular molecule (i.e., D$_1$ molecules in Fig. \ref{fig:implementation2}). 


%At each state, receptors perform a different function, either release an intracellular molecule, or state silent? 

Various biosensing and therapeutic applications have already been developed with engineered cells through such synthetic modifications over natural receptor systems \cite{chang2020synthetic, hicks2020synthetic}. This progress has been particularly fueled by the introduction of novel design frameworks (e.g., Tango assay \cite{barnea2008genetic}, Modular Extracellular Sensor Architecture (MESA) \cite{daringer2014modular}, SynNotch \cite{morsut2016engineering}) for modular receptors that combine diverse components of natural receptors, e.g., G protein-coupled receptors (GPCRs), Notch receptors, TCRs, to enable new receptor functions. For example, binding affinity and kinetic rates of the receptors can now be tuned to favor the binding of specific types of extracellular and intracellular ligands \cite{chervin2008engineering, bowerman2009engineering}. These synthetic receptors can also be seamlessly coupled to the orthogonal intracellular signaling pathways to multiplex the biosensing \cite{chen2021programmable}. Moreover, multiple ligand inputs can be AND-gated through synthetic receptors to realize combinatorial sensing \cite{roybal2016precision}. As a practical example, the specificity of genetically modified T cells has been redirected to tumor-associated antigens with chimeric antigen receptors (CARs) to improve their therapeutic efficacy \cite{jena2010redirecting}. More importantly for the receptor designs proposed in this paper, the progress in computational de novo protein design and engineering allows the creation of arbitrary protein conformational states with tailored interaction parameters, which can be translated to multi-state synthetic receptors \cite{huang2016coming, pan2021recent, quijano2021novo, wei2020computational, dagliyan2013rational}.

% Tango, MESA, SynNotch, GEMS, 
% by the modification of interior, exterior, ... chimeric... 
%make more specific to some molecules, adjust the binding properties by combining different parts. although not relevant for the proposed designs, logic gate implementations are also realized. 
% Multiple SynNotch Receptors Can Be Used To engineer cells that combinatorially integrate multiple inputs
% integrate combinatorial environmental cues and respond only when certain dual criteria are met.


In parallel to the advances in de novo protein engineering, the research in synthetic biology of signaling networks, which are abstracted as CRNs, has been witnessing an exciting shift towards synthetic post-translational protein circuits from more conventional synthetic genetic circuits \cite{chen2021programmable}. Post-translational protein circuits are much faster than their genetic counterparts, which involve the slow transcription and translation processes \cite{chen2021programmable}. They can be more directly and rapidly linked to the protein-based receptor systems in a seamless manner, enabling receptor-integrated CRN-based computations in synthetic cells \cite{gao2018programmable}. Moreover, protein circuits reinforced with de novo protein designs offer a higher degree of orthogonality in intracellular reaction pathways, enabling the implementation of more diverse and parallel functions within synthetic cells \cite{chen2021programmable, chen2019programmable, lim2010designing}. Therefore, protein circuits are promising for the design of more sophisticated CRNs, which rely on protein interactions, e.g., binding, cleaving, and chemical modification, with the interaction rates tuned specifically to implement the desired arithmetic and logic operations. 

In summary, we believe that the ongoing progress in synthetic biology reinforced with the recently introduced receptor design frameworks and the emerging branch of protein engineering will allow the implementation of the proposed biologically-plausible multi-state synthetic receptors, receptor-integrated CRNs, and the overall MC detection systems in living cells in near future.


%
%In summary, our proposed receptor and CRN designs are although not practical yet with the current state-of-the-art synthetic biology tools, they are biologically plausible, and the ongoing progress in synthetic biology through the introduction of sophisticated receptor design frameworks, and de novo protein engineering is promising for the implementation of complex receptors and CRNs that are much more sophisticated than our proposed designs in near future.  


%Also exciting in connection with de novo protein engineering is the progress in synthetic biology of signaling networks, which witnesses a trend (shift) toward synthetic post-translational protein circuits from synthetic genetic circuits.  


%engineering of post translational protein circuits, which are faster than the genetic circuits. This enabled the diversity ... orthogonal circuitry in living cells, which would allow more sophisticated design of chemical reaction networks. 

% the great variety of synthetic intracellular transcriptional circuits that have been previously been engineered can now be linked to the outside of the cell and controlled by user-defined extracellular inputs

%- CRN design. 
%Synthetic biology of signaling networks, is gaining attention... implementation of... arithmetic operations inside living cells. 

%... gene regulation circuits... and protein circuits... Gene regulation circuits are based on slow and stochastic steps of transcription and translation, whereas protein circuits are faster without the genetic modification, therefore, attracting more interest recently. 



%---


%modular receptor designs. where different components... also important for changing the binding characteristics. and coupling to orthogonal signaling pathways, 
%protein design. 

%corresponding to different conformational states with different signaling/binding dynamics/properties. 

%multi-state receptors... 

%Once a receptor protein receives a signal, it undergoes a conformational change, which in turn launches a series of biochemical reactions within the cell. These intracellular signaling pathways, also called signal transduction cascades, typically amplify the message, producing multiple intracellular signals for every one receptor that is bound.
% Activation of receptors can trigger the synthesis of small molecules called second messengers, which initiate and coordinate intracellular signaling pathways. For example, cyclic AMP (cAMP) is a common second messenger involved in signal transduction cascades. (In fact, it was the first second messenger ever discovered.) cAMP is synthesized from ATP by the enzyme adenylyl cyclase, which resides in the cell membrane. The activation of adenylyl cyclase can result in the manufacture of hundreds or even thousands of cAMP molecules. These cAMP molecules activate the enzyme protein kinase A (PKA), which then phosphorylates multiple protein substrates by attaching phosphate groups to them. Each step in the cascade further amplifies the initial signal, and the phosphorylation reactions mediate both short- and long-term responses in the cell (Figure 2). How does cAMP stop signaling? It is degraded by the enzyme phosphodiesterase.
% https://www.nature.com/scitable/topicpage/cell-signaling-14047077/

%The proposed receptor and CRN designs are practical in the sense that they are based on biologically plausible processes and mechanisms, which are already being used and modified by the state-of-the-art synthetic biotechnology tools. 
%
%For example, the proposed receptors are multi-state receptors with the states corresponding to different conformational states with different signaling/binding dynamics/properties. 
%
%The transitions between these states are realized by the binding of certain types of molecules (either activation/deactivation molecules or ligands), just like their natural counterparts. 
%
%For all detection methods except DNBR, our designs require the definition of new intermediate states that delay the activation of the receptors. Intermediate states are common in biological receptors which employ KPR. And new stable intermediate receptor states are shown to be realized through the rational design of protein-protein interfaces. 
%
%Although state-of-the-art synthetic biology tools do not allow the design of synthetic receptors at this level of sophistication, however, the recent progress in engineering of protein circuits is promising for such advancement in near future. 


% put KPR evidence paper... optogenetics. 
% check for KPR mechanism: TCR Signaling: Mechanisms of Initiation and Propagation

% not just the amount of ligand-bound receptors, but also dynamics of binding events!!!
% In KPR the T cell does not simply measure the amount of ligand-bound TCRs (called occupancy model), '''''but monitors the dynamics of the binding events.'''' These dynamics can be described by the on-rate and the half-life of the interaction. The KPR model proposes that a long half-life of the ligand-TCR interaction, such as seen for high affinity pMHC, allows a series of biochemical reactions to be completed that eventually trigger downstream signaling. By contrast, a low affinity ligand detaches before an activatory signal is produced and the TCR then reverts quickly to the initial inactive state, thus not initiating T cell activation. Although the half-life is the decisive factor, it was recently shown that the on-rate also plays a role (Aleksic et al., 2010; Govern et al., 2010; Lin et al., 2019). If the on-rate is very fast a ligand that has detached can rapidly rebind to the same TCR before the first biochemical reactions are reverted. Again, the duration of the binding event, in this case interrupted by short dissociations, is the relevant parameter.

% The KPR model has also been extended to include feedback and feed-forward loops in the signaling network below the TCR (Altan-Bonnet and Germain, 2005; Chakraborty and Weiss, 2014; Dushek et al., 2011; Lever et al., 2016; Rabinowitz et al., 1996). Inclusion of these signaling network loops improved the mathematical description of the observed sharp ligand discrimination threshold, when relating ligand half-life to T cell activation. At the same time, the high sensitivity of the T cells towards low numbers of ligands (1–10 molecules) was retained (Irvine et al., 2002; Purbhoo et al., 2004).


% To get experimental insight into the mechanism of ligand discrimination by T cells, pMHC or TCRs have been mutated at the binding sites to generate ligand-TCR pairs of different affinities and half-lives. Although such studies are broadly consistent with KPR, other biophysical parameters, such as the free binding energy, geometry of the interaction (Adams et al., 2011), conformational changes at the TCR (Dopfer et al., 2014; Gil et al., 2002; Risueño et al., 2006) and the ability to withstand pulling (Kim et al., 2009; Liu et al., 2014), might also have been changed along with the affinity, and therefore alternative models of ligand discrimination cannot be ruled out. Unfortunately, no method to specifically modulate only the dynamics of ligand-receptor interactions is currently available. Thus, in order to disentangle the half-life from these other parameters, we engineered an optogenetic system in which the duration of ligand binding to the TCR can be remotely controlled in a reversible manner (ON-OFF switch), called the opto-ligand-TCR system


%---


% protein engineering
% Advances in protein circuit design will depend on parallel advances in protein engineering. Rational design of protein conformational changes (Langan et al., 2019; Wei et al., 2020) should allow individual proteins to play more active roles as circuit components. Proteins designed to respond to small molecules could provide additional layers of control via induced dimerization and degradation (Foight et al., 2019; Rakhit et al., 2014; Shui et al., 2021). At the supramolecular level, proteins could act as information-bearing particles to program their own self-assembly (Winfree, 1996). Orthogonal protein interaction domains will allow programmable self-assembly of proteins. This includes designed protein phase separation (Banani et al., 2016; Fisher and Elbaum-Garﬁnkle, 2020; Schuster et al., 2018), which can be rationally engineered from weak multi-valent protein interactions to provide intracellular compartments for protein circuit operations, allowing noise reduction (Klosin et al., 2020), orthogonalization (Reinkemeier et al., 2019), and more complex protein-level computation


% PROTEIN ENGINEERING... DESIGN OF MULTIPLE STATES WITH COMPARABLE FREE ENERGIES
% Allosteric regulation of protein function is widespread in biology, but is challenging for de novo protein design as it requires the explicit design of multiple states with comparable free energies. Here we explore the possibility of designing switchable protein systems de novo, through the modulation of competing inter- and intramolecular interactions. We design a static, five-helix ‘cage’ with a single interface that can interact either intramolecularly with a terminal ‘latch’ helix or intermolecularly with a peptide ‘key’. Encoded on the latch are functional motifs for binding, degradation or nuclear export that function only when the key displaces the latch from the cage. We describe orthogonal cage–key systems that function in vitro, in yeast and in mammalian cells with up to 40-fold activation of function by key. The ability to design switchable protein functions that are controlled by induced conformational change is a milestone for de novo protein design, and opens up new avenues for synthetic biology and cell engineering.

%Designing proteins that can switch conformations is more challenging because multiple states must have sufficiently low free energies relative to the unfolded state and the free-energy differences between the states must be small enough that switching can be toggled by an external input. 
%Recent advances in designing systems with multiple states include a transmembrane ion transporter7 and variants of Gβ1 that dynamically exchange between two related conformations8; however, a method for the de novo design of modular, tunable protein systems that switch conformational states in the presence of an external input has not yet been achieved

% Modularity and tunability in strategies that rely on repurposing natural proteins are limited by the evolved functions and ligands of the existing proteins, whereas altering the affinities of LOCKR components is tunable on the basis of simple design principles that are, in general, irrespective of the functional modality or application.
%Our use of a toehold for tuning helical displacement is reminiscent of DNA strand displacement technology40,41 but—unlike approaches based on nucleic acids such as genetic toggles42 or riboswitches43,44 (which have largely focused on controlling transcription)—LOCKR systems can be readily integrated with the many diverse processes controlled by proteins.
%Viewed in this light, LOCKR brings the programmability of DNA switching technology to proteins, with the added advantages of tunability and flexibility over rewired natural protein systems, and ready interfacing with biological machinery over DNA nanotechnology.
%More generally, the domain of sophisticated environmentally sensitive and switchable function no longer belongs exclusively to naturally occurring proteins.


%Computational protein design has focused primarily on the design of sequences which fold to single stable states, but in biology many proteins adopt multiple states. We used de novo protein design to generate very closely related proteins that adopt two very different states—a short state and a long state, like a viral fusion protein—and then created a single molecule that can be found in both forms. Our proteins, poised between forms, are a starting point for the design of triggered shape changes.

% Earlier this was the case: 
%Previous efforts to design new proteins for applications in research, industry, and medicine have focused on the creation of sequences that stably adopt a single target structure, ignoring the potential impact of protein dynamics on function.

% first attempts... changing the natural protein dynamics ( to reengineer natural proteins to alter their dynamics)
% Proteins are intrinsically dynamic molecules that can exchange between multiple conformational states, enabling them to carry out complex molecular processes with extreme precision and efficiency. Attempts to design novel proteins with tailored functions have mostly failed to yield efficiencies matching those found in nature because standard methods do not allow the design of exchange between necessary conformational states on a functionally relevant timescale. Here we developed a broadly applicable computational method to engineer protein dynamics that we term meta-multistate design. We used this methodology to design spontaneous exchange between two novel conformations introduced into the global fold of Streptococcal protein G domain β1. The designed proteins, named DANCERs, for dynamic and native conformational exchangers, are stably folded and switch between predicted conformational states on the millisecond timescale. The successful introduction of defined dynamics on functional timescales opens the door to new applications requiring a protein to spontaneously access multiple conformational states.

% . Given the demonstrated importance of dynamics in a range of protein functions, including enzyme catalysis9,10, allosteric regulation11, and molecular recognition12, rational design of a defined mode of dynamics into a protein fold has great potential to address this shortfall. Moreover, the ability to engineer a protein that can undergo conformational exchange between multiple states should expand the range of functionalities that can be designed, paving the way for applications that are currently inaccessible with natural proteins
% However, the introduction of functionally relevant conformational exchange into a stable protein fold is a difficult design problem, as it requires a priori knowledge of the structural features of the relevant conformational states for dynamic exchange, ''including the intermediate states that the protein must adopt as it undergoes this conformational transition'', which are often unknown.




% change of binding characteristics, orthogonal pathways, connect to intracellular transcriptional (gene regulation) or protein circuits (post-translational circuits composed of interacting proteins, protein circuits can interact directly with endogeneous 'protein-level pathways' in the cell to sense or manipulate cell function). 
% different receptor engineering frameworks have been introduced. For example, Tango, MESA, ... SynNotch.., more flexible and modularity, specific T cell targeting. 


% Nucleic acids are relatively easy to pro-gram in a predictable manner, based ontheir ability to bind, and even cleave,each other. However, they are challengingto interface with endogenous proteinpathways in the living cell. By contrast,proteins are more difficult to predictablydesign, but possess a much larger poten-tial repertoire of activities and interactionsincluding binding, cleavage, ligation, allo-steric modulation, and chemical modifica-tion. 

%Tango, MESA, SynNotch, GEMS, 
%by the modification of interior, exterior, ... chimeric... 
%make more specific to some molecules, adjust the binding properties by combining different parts. although not relevant for the proposed designs, logic gate implementations are also realized. 
% Multiple SynNotch Receptors Can Be Used To engineer cells that combinatorially integrate multiple inputs
% integrate combinatorial environmental cues and respond only when certain dual criteria are met.

% MODULARITY: We have demonstrated that it is possible to engineer highly diverse forms of synNotch receptors, for which one can change what contact ligands the cell detects and what cellular responses occur upon engagement.

% outputs are determined by what intracellular transcriptional regulator is used and what effector genes this regulator drives.


% the great variety of synthetic intracellular transcriptional circuits that have been previously been engineered can now be linked to the outside of the cell and controlled by user-defined extracellular inputs


%- CRN design. 
%Synthetic biology of signaling networks, is gaining attention... implementation of... arithmetic operations inside living cells. 

%... gene regulation circuits... and protein circuits... Gene regulation circuits are based on slow and stochastic steps of transcription and translation, whereas protein circuits are faster without the genetic modification, therefore, attracting more interest recently. 


% whether notch receptors could be used as a platform to generate synthetic signaling pathways in which both sensing and response were customized. 

% customize input sensing by swapping the extracellular recognition domain of these receptors, including the use of antibody-based domains (e.g., single-chain antibodies or nanobodies) to detect a wide range of user-specified cell-surface proteins, such as disease antigens

% imultaneously, we can link these novel inputs to customized responses, by swapping the intracellular transcription domain and providing specific downstream effector target genes.



% Initial successes in receptor engineering have come from focused efforts often based on surgical mutational changes. Recently, more modular solutions for engineering new extracellular sensing have emerged. For example, chimeric antigen receptors (CARs) have been developed that link an extracellular antigen sensing antibody domain with internal signaling domains from the T cell receptor. With the CAR system, the researchers are now able to link novel user-specified disease antigen inputs to synthetic activation of the native T cell activation program; these engineered T cells can kill tumor cells that express the cognate antigen and have proven to be clinically effective for B cell cancers (Miller and Sadelain, 2015). Despite the power of CARs, these still represent changing only the INPUT that the cell senses, the OUTPUT is still the full T cell activation program.

%In addition to these principles, protein circuits, like other computational systems, require four key circuit-level capabilities: sensing external and internal inputs, transmitting information from one component to another, processing signals, and exerting dynamic control (Figure 1B).




% Lastly, there are variations of our designs speculated to be the design behind the extraordinary selectivity of T cells etc. . pleitropic... where, or the other one, KPR mechanism with a push-pull network. shown to reach the limit of detection... 

% In  contrast,  a  pleiotropic  receptor,  which  produces  twotypes  of  downstream  signaling  molecules,  can  resolve  thisambiguity.  The  crucial  feature  of  the  model  enabling  theseproperties  is  that  the  two  output  signaling  molecules  re-flect   physically   different   features   of   the   ligand-receptorinteraction—in the case studied here, one variable is propor-tional to the bound time of the ligand, while the other reflectsthe number of distinct ligand-receptor binding events


% This idea can be implemented biologically by a cascadeof receptor conformational states triggered by binding, andproceeding irreversibly from states 1 tom, each transitionto the next state occurring with rates(Fig.4). The ligand isfree to detach from the receptor at any time, bringing thereceptor back to the unbound state 0. The receptors signalthrough the production or activation of two moleculesBandDwith opposite effects on a push-pull networkgoverning the state of a moleculeX, which provides thefinal readout forxthrough its modified stateX. If onerequires that the equilibration ofBandDare fast, and that

% The network structureproposed in this study (Fig.4) is reminiscent of kineticproofreading schemes and their generalizations, whichprovide a well-known solution to the ligand discriminationproblem[26,28,29,32,36]. An important difference is thathere signaling occurs during all steps, albeit at various, fine-tuned rates, and with potentially negative contributions,the role of which is to buffer the effect of wrong ligands.Consistent with this prediction, it was shown that a negativeinteraction through a diffusible molecule between kinetic-proofreading receptors could mitigate the effects of largenumbers of incorrect ligands in a discrimination task

% Independent Receptors. We first consider independent receptors.  McKeithan (18) proposed a KPR model explaining how a weak signal could be detected in the presence of a fixed large background of substantially dissimilar (eΔ > 10) ligands. In this  scheme, the receptor sequentially visits internal states upon  binding of a ligand. At each internal state, the receptor–ligand  complex can unbind, following which the internal state of the  receptor is assumed to quickly return to its first “unmodified”  state; see Fig. 3A corresponding to Eq. 1 with fMp =1. A


% 
%The proposed receptors are multi-state receptors and sometimes (except DNBR) require the definition of new intermediate states in addition to active and inactive states (corresponding to different conformational states with different signaling/binding dynamics/properties) with the transitions between them realized by the binding of certain types of molecules (either activation/deactivation molecules or ligands).  (we only need intermediate bound state and intermediate unbound state) (states: activated, deactivated, what other conformational states exist in bio?)
%Intermediate states are already utilized in KPR systems in biological cells, it is a part of common receptor signaling motif in these systems. These states delay the activation of the receptors. 
%Therefore, they are accessible by synthetic biology tools, through the modification of already existing receptors, through the rational design of protein-protein interactions. For example: 



%\subsubsection{\bf{Receptor Activation and Transduction of Total Unbound Time Duration}}
%We propose a receptor activation mechanism to control the start time and duration of the channel sensing, such that only one unbound/bound time duration is sampled from each receptor. In this scheme, the sampling process starts with the generation of activation molecules $A^+$, produced by the cell in an impulsive manner, when the cell decides to sample the receptor states, as demonstrated in Fig. \ref{fig:kpr}. The generation of activation signal, thus, occurs in bursts, through the following reaction
%\begin{equation}
%\ce{ {\varnothing}  ->[{s(t) \psi^+}] {A^+}},
%\end{equation}
%where the time-varying generation rate is given as $s(t) \psi^+$, with $s(t) \approx \delta(t-t_A)$ being a very short pulse signal centered around the activation time $t_A$. 
%Shortly after activation, the cell generates deactivation molecules $A^-$, through the following reaction
%\begin{equation}
%\ce{ {\varnothing}  ->[{d(t) \psi^-}] {A^-}}.
%\end{equation}
%The reaction rate is given by $d(t) \psi^-$, with $d(t) \approx \delta(t-t_D)$ being again an impulse-like signal centered around the deactivation time $t_D$. The generated deactivation molecules degrade the existing activation molecules at a rate $\rho$, i.e.,
%\begin{equation}
%\ce{ {A^+ + A^-} ->[{\rho}] {\varnothing} },
%\end{equation}
%such that the duration of the overall sampling process can be controlled.
%The inactive receptors, i.e., $U_I$ and $B_I$, transition into their intermediate states,  i.e., $U_A^\ast$ and $B_A^\ast$, upon reacting with an activation molecule $A^+$ at a rate $\omega$, i.e.,
%\begin{align}
%\ce{ {U_I + A^+} &->[{\omega}] {U_A^\ast + A^+} }\\ \nonumber
%\ce{ {B_I + A^+} &->[{\omega}] {B_A^\ast + A^+} }.
%\end{align}
%The binding of an unbound receptor in the intermediate state $U_A^\ast$, transforms it into an intermediate bound receptor $B_A^\ast$, i.e.,
%\begin{equation}
%\ce{ {U_A^\ast + L_i} ->[{k^+c_i }] {B_A^\ast} }, 
%\end{equation}
%where $L_i$ denotes a ligand molecule of $i^\text{th}$ type. Upon the first unbinding event, a bound receptor in the intermediate state $B_A^\ast$ goes into the active unbound state $U_A$, i.e., 
%\begin{equation}
%\ce{ {B_A^\ast} ->[{k_i^-}] {U_A + L_i} }.
%\end{equation}
%In the active unbound state $U_A$, the receptor produces the secondary messenger molecules $S$ at a constant rate through the following first-order reaction,
%\begin{equation} \label{eq:generationofS}
%\ce{ U_A   ->[{\mu}] {U_A + S}}.
%\end{equation}
%As a result of this reaction, the steady-state concentration of the produced $S$ molecules becomes proportional to the total unbound time $T_u$, as we will see in Section \ref{ss_analysis}.
%
%Upon binding a ligand, the active unbound receptor $U_A$ switches into the first KPR substate of the active bound state $B_A^1$, i.e., 
%\begin{equation}
%\ce{ {U_A + L_i}   ->[{k^+c_i}] {B_A^1}}.
%\end{equation}
%As a result, the modified KPR scheme, consisting of $M$ substates, $\{B_A^1, \dots, B_A^M\}$, becomes activated. 
%
%We provide some examples in Fig. \ref{fig:kpr}(c) for receptor state trajectories governed by the proposed activation mechanism. Receptor 1 is in inactive unbound state when the activation signal is sent. The reaction with activation molecules $A^+$ turns it into intermediate unbound state $U_A^\ast$. Next, with the binding of a ligand, it goes into intermediate bound state. Following the unbinding of the bound ligand, it finally gets activated in the unbound state. During the active unbound state, it generates $S$ molecules following the reaction \eqref{eq:generationofS}. When it binds a ligand again, it switches into active bound state, activating the KPR mechanism. The next unbinding event brings the receptor back into the inactive unbound state. As such one cycle of sampling of unbound and bound time duration is completed. On the other hand, Receptor 2 is in the inactive bound state $U_I$ at the time of activation. Activation reaction switches it into the intermediate bound state $B_A^\ast$, during which it is still not able to generate any second messenger. After the first unbinding event it transitions into the active bound state $U_A$, where it generates $S$ molecules. Upon the next binding, it becomes active in the bound state $B_A$, and activates the KPR mechanism. This sampling cycle is also completed with the ensuing unbinding event leading it to the inactive unbound state $U_I$.  
%
%To ensure that the inactivated receptors are not re-activated during the same sampling process for the sake of obtaining only a single pair of unbound and bound duration samples from each receptor, the generation rates of activation and deactivation molecules, i.e., $\psi^+$ and $\psi^-$, respectively, as well as the rate of reaction between activation molecules and receptors, i.e., $\omega$, and the rate of deactivation reaction $\rho$ should be very high compared to the ligand-receptor binding/unbinding reaction rates.
%
%\subsubsection{\bf{Kinetic Proofreading and Transduction of Bound Time Durations}}
%\label{kpr_scheme}
%For the sampling of the bound time durations, we propose a modified KPR scheme. In the KPR mechanism, the active bound receptor sequentially visits its $M$ substates in an irreversible manner during the bound time period by undergoing a series of conformational changes with specific transition rates, as shown in Fig. \ref{fig:kpr}(b). In each internal state, the receptor can directly return to the initial inactive unbound state $U_I$ if the bound ligand unbinds from the receptor. In our modified KPR scheme, while returning to the initial unbound state, the receptor releases an intracellular molecule $D$, type of which is specific to the last occupied KPR substate. As the unbinding rate is different for each ligand type, the last occupied substate is informative of the type of the bound ligand. This information is encoded into the number of $D_i$ molecules generated by all active bound receptors, which becomes proportional to the number of last visits made to the $B_A^i$ substate at steady-state, as discussed in Section \ref{ss_analysis}. 
%
%In order for the proposed KPR scheme to provide the required statistics for the estimation of ligand concentration ratios, the transition rates, $\beta$'s, between substates should be set in accordance with the time thresholds introduced in \eqref{timethreshold}. As such, the resulting number of second messengers, $D_i$, produced from the internal states $B_A^i$ will approximate the actual number of binding events $n_i$ of durations within corresponding time ranges. Transition rates between the KPR states can be set as a function of time thresholds $T_i$'s, i.e.,
%\begin{align}
%\beta_{i,i+1} = \kappa_i/\left(T_i^- - T_{i-1}^-\right) ~~~ \text{for}~ i \in \{1, \dots,	 M-1\} ,
%\end{align}
%where $\kappa_i$'s are tuning parameters to adjust the transition rates. In the next section, we will show that setting $\kappa_i = 3/5$ provides a good approximation for the number of binding events falling in each time interval for the unbiased estimator, where $T_i = 3/k_i^-$ for $i \in \{1, \dots,	 M-1\}$, and $T_0 = 0$. 
%
%\subsubsection{\bf{Steady-State Analysis}}
%\label{ss_analysis}
%We can now provide a steady-state analysis for the transduction of unbound and bound time durations of receptors into second messengers, i.e., $S$ and $D$ molecules. We consider the case when there are $M=3$ different types of ligands co-existing in the channel, such that each of the receptors has three KPR substates, as shown in Fig. \ref{fig:kpr}(a). For the sake of brevity of the analysis, we omit the activation mechanism, and focus only on the active receptors. The considered system for steady-state analysis is then a kinetic scheme of Markovian nature, and redrawn in Fig. \ref{fig:kinetic}, demonstrating possible states of active receptors along with the relevant transition rates.
%
%
%In order to write the chemical master equation (CME) for this kinetic scheme, at the moment, we consider the case of single type of ligands. The CME can then be given as a set of differential equations, i.e., 
%\begin{align} \label{eq:KPRode}
%\frac{dP_{U_A|i}}{dt} &= - k^+ c_i P_{U_A|i} \\ \nonumber
%\frac{dP_{B_A^1|i}}{dt} &= k^+ c_i P_{U_A|i}  - \beta_{1,2} P_{B_A^1|i} - k_i^- P_{B_A^1|i} \\ \nonumber
%\frac{dP_{B_A^2|i}}{dt} &=  \beta_{1,2} P_{B_A^1|i} - \beta_{2,3} P_{B_A^2|i}- k_i^- P_{B_A^2|i} \\ \nonumber
%\frac{dP_{B_A^3|i}}{dt} &=  \beta_{2,3} P_{B_A^2|i} - k_i^- P_{B_A^3|i}  \\ \nonumber
%\frac{dP_{D_j|i}}{dt} &=  k_i^- P_{B_A^j|i}~~~ \text{for}~ j \in \{1, \dots,	 M \},
%\end{align}
%where $P_{U_A|i}, P_{B_A^1|i}, P_{B_A^2|i}, P_{B_A^3|i}$ are the time-varying probabilities of an active receptor to be in the unbound state, and in each of the KPR substates, respectively, conditioned on the presence of only the $i^\text{th}$ type of ligand. $D_j$'s represent virtual states of absorbing nature for an active receptor, and $P_{D_j|i}$ denotes the probability of an active receptor to generate an intracellular $D_j$ molecule and return to the inactive unbound state when an $i^\text{th}$ type of ligand is bound. The escape rate from KPR substates is equal to the unbinding rate of the bound ligand, $k_i^-$.
%
%The steady-state solution of the CME in \eqref{eq:KPRode} is analytically obtained with the initial conditions  $P_{U_A|i}^0 = 1$, $P_{B_A^1|i}^0 = P_{B_A^2|i}^0 = P_{B_A^3|i}^0  = P_{D_1|i}^0 = P_{D_2|i}^0= P_{D_3|i}^0 = 0$, as follows
%\begin{align}
%P_{U_A|i}^{ss} &= P_{B_A^1|i}^{ss} = P_{B_A^2|i}^{ss} = P_{B_A^3|i}^{ss} =  0, \\ \nonumber
%P_{D_1|i}^{ss} &=  \frac{k_i^-}{\beta_{1,2} + k_i^-}, \\ \nonumber
%P_{D_2|i}^{ss} &=  \frac{\beta_{1,2} k_i^-}{\beta_{1,2} \beta_{2,3} + \beta_{1,2} k_i^- + \beta_{2,3} k_i^- + (k_i^-)^2}, \\ \nonumber
%P_{D_3|i}^{ss} &=  \frac{\beta_{1,2} \beta_{2,3}}{\beta_{1,2} \beta_{2,3} + \beta_{1,2} k_i^- + \beta_{2,3} k_i^- + (k_i^-)^2}.
%\end{align}
%In the presence of three types of ligands, the overall steady-state probabilities can be written as follows
%\begin{align}
%P_{U_A}^{ss} &= P_{B_A^1}^{ss} = P_{B_A^2}^{ss} = P_{B_A^3}^{ss} =  0, \\ \nonumber
%P_{D_j}^{ss} &= \sum_{i=1}^{M=3} \alpha_i P_{D_j|i}^{ss}.
%\end{align}
%
%Given that all active receptors independently follow the same kinetic scheme, and assuming that each receptor goes through the active state for once during a sampling process, the mean number of generated intracellular $D$ molecules at steady-state can be given as
%\begin{equation} \label{D_ss}
%\E[n_{D_j}^{ss}] = N P_{D_j}^{ss},~~ \text{for}~ j \in \{1, \dots,	 M=3 \}.
%\end{equation}
%Here we use our previous assumption that number of samples is equal to the number of receptors, i.e., $N = N_R$. Given the statistical independence of receptors, we can also write the variance of number of $D$ molecules as follows
%\begin{equation}
%\Var[n_{D_j}^{ss}] = N P_{D_j}^{ss} (1-P_{D_j}^{ss}),~~ \text{for}~ j \in \{1, \dots,	 M=3 \}.
%\end{equation}
%Assuming that the number of receptors is sufficiently high ($N_R = N = 10000$ in the considered case), we can approximate the random number of produced $D$ molecules at steady-state with a Gaussian distribution, i.e., 
%\begin{equation}
%n_{D_j}^{ss} \sim \mathcal{N}\left( \E[ n_{D_j}^{ss}], \Var[n_{D_j}^{ss}] \right),~~ \text{for}~ j \in \{1, \dots,	 M=3 \}.
%\label{eq:Ddistribution}
%\end{equation}
%
%As discussed in Section \ref{kpr_scheme}, the transition rates, $\beta$'s, between KPR substates should be optimized for obtaining the most accurate representation of actual number of binding events $n_i$'s with $D$ molecules. However, for the sake of brevity of this discussion, we leave the optimization problem as a future research. Here we provide the steady-state probability distribution of number of $D$ molecules $n_{D_j}^{ss}$ for $\kappa = 3/5$ in Fig. \ref{fig:KPR_analysis}. In the same figure, the results are compared to the histogram of the same statistic obtained through a Monte Carlo simulation of the kinetic scheme demonstrated in Fig. \ref{fig:kinetic}, and the probability distribution of number of binding events $n_i$ that fall in each time interval defined by the time thresholds $T_i$'s given according to \eqref{timethreshold} with $\nu = 3$. Here, assuming that $N$ is large enough, we also approximate the binomial distribution of $n_i$ with Gaussian distribution, i.e., $n_i \sim \mathcal{N}\left(\E[n_i], \Var[n_i] \right)$, where the mean and variance of $n_i$ are given in \eqref{eq:meann} and \eqref{eq:varn}, respectively. As is seen, the analytical results are in very good match with the simulation results. They also show that the KPR scheme with the selected transition rates can approximate the number of binding events in each time interval. However, the results also imply that the transition rates should be further optimized to obtain better approximation.
%
%The generation of intracellular $S$ molecules encoding the total unbound time duration, on the other hand, is governed by the following rate equations, 
%\begin{align} \label{rate_equation}
%\frac{d\E[n_{U_A}]}{dt} &= N \frac{dP_{U_A}}{dt} = - k^+ c_{tot} \E[n_{U_A}], \\ \nonumber
%\frac{d\E[n_S]}{dt} &= - \mu \E[n_{U_A}].
%\end{align}
%In writing \eqref{rate_equation}, for mathematical convenience, we consider the receptors as if they independently start in the active unbound state at the same time. This assumption does not degrade the accuracy of the analysis, because the generated $S$ molecules, whose generation rate is dependent on the number of active unbound receptors, are not degraded throughout the entire process (same as $D$ molecules), and we are only concerned about the steady-state statistics of the intracellular molecules, and not interested in their time-varying statistics. Hence, the steady-state solution of \eqref{rate_equation} for the initial conditions $\E[n_{U_A}^0] = N$ and $\E[n_S^0]  = 0$, is given as
%\begin{align} \label{U_ss}
%\E[n_{U_A}^{ss}] = 0.
%\end{align}
%\begin{align} \label{S_ss}
%\E[n_{S}^{ss}] =  \frac{\mu N }{k^+ c_{tot}}.
%\end{align}
%
%The produced $D$ and $S$ molecules, whose expected numbers at steady-state are given in \eqref{D_ss} and \eqref{S_ss}, respectively, will be input to the estimator CRN introduced in the next section.
%
%\subsection{Estimation with Chemical Reaction Networks}
%Once the transduction of total unbound time $T_u$ and the number of binding events $n_i$ is completed, the arithmetic operations required for the estimator can be realized through intracellular CRNs that can perform analog computations \cite{daniel2013synthetic}. Here we focus on the unbiased estimator; thus, the objective is to implement the following equation with a CRN: 
%\begin{align} \label{c_estimate}
%\hat{c_l} &= \frac{N-1}{N} \frac{1}{k^+ T_u}  \sum_{i=1}^M n_i w_{l,i}, \\ \nonumber
%&\approx \frac{1}{k^+ T_u}  \sum_{i=1}^M n_i w_{l,i}, ~~~ \text{for}~ N \gg 1.
%\end{align}
%Accordingly, we need to obtain the weighted sum of number of $M$ different types of second messengers $D_i$'s corresponding to the number of binding events that fall in each time interval, divided by the concentration of $S$ molecules encoding the total unbound time duration. This can be achieved through the following CRN, which is designed based on the methodology introduced in \cite{chou2017chemical}: 
%\begin{equation}
%\ce{ {D_i} ->[{w_{j,i}}] {D_i} + { Y_j} },~~~ \text{for}~ i,j \in \{1, \dots,	 M=3 \},
%\end{equation}
%\begin{equation}
%\ce{ S + Y_j ->[{k^+}] S },~~~ \text{for}~ j \in \{1, \dots,	 M=3 \}.
%\end{equation}
%In this CRN, while $Y$ molecules are generated by $D$ molecules with different rates set by the matrix $\bm{W} = \bm{S}^{-1}$ (see \eqref{Wmatrix}), they are consumed by the $S$ molecules that encode the total unbound time duration. The rate equation of the above CRN for the $i^\text{th}$ ligand can be written as
%\begin{align}
%\frac{d\E[ n_{Y_i}] }{dt} = \sum_{j=1}^M w_{i,j} \E[n_{D_j}]  &- k^+ \E[n_{S}]  \E[ n_{Y_i}], \\ \nonumber
%&~~~ \text{for}~ i \in \{1, \dots,	 M = 3 \}.
%\end{align}
%Given the initial condition  $\E[n_{Y_i}^0] = 0$, the steady-state solution for $\E[n_{Y_i}]$ is obtained as
%\begin{equation} \label{Y_ss}
%\E[n_{Y_i}^{ss}] = \frac{1}{k^+ \E[n_{S}^{ss}] } \sum_{j=1}^M w_{i,j} \E[n_{D_j}^{ss}].  
%\end{equation}
%Recall from \eqref{Tu} that $\E[1/T_u] = (k^+ c_{tot})/(N-1) \approx (k^+ c_{tot})/N$ for large $N$. By combining this with \eqref{S_ss}, we can see that $1/(k^+ \E[n_{S}^{ss}]) = c_{tot}/(\mu N) \approx \E[1/( \mu k^+ T_u)]$, is on average proportional to the first part of \eqref{c_estimate}, i.e., $1/(k^+ T_u)$. The proportionality constant is the rate $\mu$, which is the generation rate of $S$ molecules and can be simply set to $1\text{s}^{-1}$ for a better approximation. Given that the steady-state number of $D_i$ molecules $n_{D_i}^{ss}$ approximates the actual number of binding events $n_i$, the mean number of $Y_i$ molecules at steady-state given in \eqref{Y_ss} becomes proportional to the concentration estimate of $i^\text{th}$ type of ligand $\hat{c}_i$ given in \eqref{c_estimate}.
%
%We note that once the estimation through CRN is completed, the produced intracellular molecules, i.e., $D$, $S$ and $Y$ molecules, should be removed from the cell through a chemical degradation reaction before the cell performs the next round of channel sensing. The rate of the degradation reaction can be set according to the required frequency of channel sensing. 

\section{Conclusion}
\label{sec:conclusion}
In this paper, we investigate the performance of four different MC detection methods in the case of interference resulting from molecules having similar binding affinity as the information-carrying molecules. The detection methods are based on different statistics of the ligand-receptor binding reaction, e.g., instantaneous number of bound receptors, duration of receptors' bound and unbound time intervals, that reveal information about the concentration and binding affinity of the molecules. The methods vary in complexity; however, they are all biologically plausible, and we believe that the ongoing progress and sophistication in synthetic biology, and particularly in de novo protein engineering,  will soon enable the practical implementation of the required synthetic receptors and CRNs in biological MC devices. Our analyses show that the effect of molecular interference on the detection performance can be substantially reduced by using the combination of unbound and bound time durations of receptors instead of relying solely on the number of bound receptors, which has been the prevalent approach in the previous literature.



%We evaluate the performance of three different ML detection methods in the presence of interferer molecules, which are of similar type with the messenger molecules and can interfere with the reception process through ligand-receptor binding reaction. Results reveal that detection based on receptor bound time intervals significantly outperforms detection based on number of bound receptors and receptor unbound time intervals in most of the physiologically relevant cases. However, this initial analysis relies on simplifying assumptions, such as perfect synchronization, and thus calls for a more comprehensive approach supported by particle-based simulations. 
	\begin{figure}[!t]
	\centering
	\includegraphics[width=7cm]{Markov}
	\caption{State diagram of the CTMP model of receptor binding process.}
	\label{fig:Markov}
\end{figure}

\appendices
	\section{Bound State Probability of Receptors at Equilibrium in the Presence of Interferer Molecules}
	\label{AppendixA}
	In the presence of interferer molecules with concentration $c_{in}$, the receptor binding process can be represented by a 3-state CTMP, with the unbound state (U) accompanied by two bound states, B$_s$ and B$_{in}$, corresponding to the binding of information and interferer molecules, respectively. The state diagram of this CTMP along with the corresponding state transition rates is shown in Fig. \ref{fig:Markov} .
	The rate matrix is then constructed as follows
	%\[
	%\bm{R}=
	%\begin{bmatrix}
	%-(k_s^+ c_s + k_{in}^+ c_{in}) & k_s^+ c_s  & k_{in}^+ c_{in} \\
	%k_s^- & -k_s^- & 0 \\
	%k_{in}^- & 0 & -k_{in}^- 
	%\end{bmatrix}
	%\]
	\renewcommand{\kbldelim}{(}% Left delimiter
	\renewcommand{\kbrdelim}{)}% Right delimiter
	\[
	\bm{R} = \kbordermatrix{
		& \U & \B_{s} & \B_{in}  \\
		\U & -(k_s^+ c_s + k_{in}^+ c_{in}) & k_s^+ c_s  & k_{in}^+ c_{in} \\
		\B_{s} & k_s^- & -k_s^- & 0  \\
		\B_{in} & k_{in}^- & 0 & -k_{in}^- 
	}
	\]

The equilibrium probabilities of the receptor states $\bm{\theta} = [\p_\U ~~ \p_{\B_s} ~~ \p_{\B_{in}}]$ can be obtained by solving the following linear equations, $\bm{\theta} \bm{R} = 0$, and $\bm{\theta} \bm{e} = 1$, where $\bm{e}$ is all-ones vector \cite{whitt2006continuous}. These equations represent, respectively, the detailed balance condition at equilibrium, and the fact that the sum of all probabilities must be equal to 1. From this system of linear equations, we can obtain the unbound state probability at equilibrium as 
	\[
	\p_\U  = \frac{k_s^- k_{in}^-}{k_s^+ k_{in}^- c_s + k_s^- k_{in}^+ c_{in} + k_s^- k_{in}^-}.
	\]
	The bound state probability of the receptors at equilibrium $\p_\B = \p_{\B_s} + \p_{\B_{in}}$ is then given by
	\[
	\p_\B  = 1 - \p_\U = \frac{k_s^+ k_{in}^- c_s + k_s^- k_{in}^+ c_{in}}{k_s^+ k_{in}^- c_{s} + k_s^- k_{in}^+ c_{in} + k_s^- k_{in}^-}. 
	\]
	Substituting $K_D = k^-/k^+$, we recover Eq. \eqref{probBinding2}:
	\[
	\p_\B = \frac{c_s/K_D^s + c_{in}/K_D^{in}  }{1 + c_s/K_D^s + c_{in}/K_D^{in}}.
	\]

%\begin{equation} \label{probBinding2}
%\p_\B = \frac{c_s/K_D^s + c_{in}/K_D^{in}  }{1 + c_s/K_D^s + c_{in}/K_D^{in}},
%\end{equation}
%where, pB is equal to the sum of p1 and p2 




%\section{Introducing Unknown Ligand Types to the Channel}
%\label{AppendixA}
%In the proposed suboptimal estimators, the time thresholds, $T_i$'s, and the corresponding $S$ and $H$ matrices, given in \eqref{eq:Smatrix} and \eqref{eq:Hmatrix}, respectively, are constructed assuming that there are $M$ types of ligands with the unbinding rates known to the receiver. Here, we investigate the case when $L$ different types of additional ligands with unbinding rates unknown to the receiver are introduced to the channel. We will derive the MSE for the suboptimal unbiased concentration estimator introduced in Section \ref{sec:unbiased}. The derivation of the biased estimator, investigated in Section \ref{sec:biased}, can be done in a similar way. We will see that in the case of unknown ligands, the unbiased estimator becomes biased. 
%
%Since the receiver assumes that there are $M$ different types of ligands, the time domain is divided into $M$ different regions. When there are $L$ additional ligand types, the probability of a binding duration to fall in a specific time interval can be rewritten in vector form as follows
%\begin{equation}
%\bm{\p} =  \frac{\bm{\E[n]}}{N} =  \bm{S_r} \bm{\alpha_r}, \label{eq:appendixP}
%\end{equation}
%where $\bm{\p}$ is $(M \times 1)$ probability vector, and $\bm{\alpha_r}$ is an ($[M+L] \times 1$) vector of concentration ratios of all ligand types including the additional ones. Note that $\bm{\alpha_r}^T = [\bm{\alpha}^T  \alpha_{M+1}  \dots \alpha_{M+L} ]$, with $\bm{\alpha}$ is the $(M \times 1)$ vector of concentration ratios of known ligand types.  Here, $\bm{S_r}$ is an ($M \times [M+L]$) matrix, whose elements are given by
%\begin{align}
%\bm{S_r}(i,j) &= \e^{-(k^-_j T_{i-1})} - \e^{-(k^-_j T_{i})} \\ \nonumber
%&~~~ \text{for}~ i \in \{1, \dots,	 M \}, ~ j \in \{1, \dots,	 M+L \}.
%\end{align}
%
%Using the knowledge of only $M$ ligand types, the receiver utilizes the concentration ratio estimator given in \eqref{Wmatrix}, as follows
%\begin{align} \label{WmatrixApp}
%\bm{\hat{\alpha}} = \left(\frac{1}{N}\right) \bm{W} \bm{n},  
%\end{align}
%where $\bm{W} = \bm{S}^{-1}$ is the inverse of $\bm{S}$ matrix, that is given in \eqref{eq:Smatrix}. The mean of the ratio estimator then becomes
%\begin{equation}
%\bm{\E[\hat{\alpha}]}=  \left( \frac{1}{N}\right) \bm{W} \bm{\E[n]} = \bm{W} \bm{\p} = \bm{S^{-1}} \bm{S_r} \bm{\alpha_r}. \label{eq:appendixMean}
%\end{equation}
%The bias of the ratio estimator can then be written as: 
%\begin{align}
%\bm{\Delta[\hat{\alpha}]} &= \bm{\E[\hat{\alpha}]} - \bm{\alpha} \\ \nonumber
%&= \bm{S^{-1}} \bm{S_r} \bm{\alpha_r} - \bm{\alpha}.
%\end{align}
%
%The concentration estimator for individual ligand types is given as $\bm{\hat{c}} = \bm{\hat{\alpha}} \hat{c}_{tot}$. Note that the ML estimator of total ligand concentration, $\hat{c}_{tot}$, is unbiased. Therefore, the bias of the concentration estimator can be computed as follows
%\begin{align}
%\bm{\Delta[\hat{c}]} =  \bm{\Delta[\hat{\alpha}]} c_{tot},
%\end{align}
%where the total ligand concentration is now given as $c_{tot} = \sum_{i=1}^{M+L} c_i$. Recall from \eqref{eq:totalvarianceunbiased} that the variance of the concentration estimator is written as
%\begin{align} \label{eq:totalvarianceunbiased2}
%\bm{\Var[\hat{c}]} &= \Var[\hat{c}_{tot}] \bm{\Var[\hat{\alpha}]} + \Var[\hat{c}_{tot}] \left( \bm{\E[\hat{\alpha}]} \odot \bm{\E[\hat{\alpha}]}\right) \\ \nonumber
%&+ \bm{\Var[\hat{\alpha}]} \E[\hat{c}_{tot}]^2.
%\end{align}
%Here, $\Var[\hat{c}_{tot}] = \frac{c^2_{tot}}{N-2}~~ \text{for} ~~ N > 2$ and $\E[\hat{c}_{tot}] = c_{tot}$. The variance of the ratio estimator, $\bm{\Var[\bm{\hat{\alpha}}]}$, can be calculated by using \eqref{eq:est_variance} and \eqref{eq:est_covariance}, with the new probability vector $\bm{\p}$, given in \eqref{eq:appendixP}.
%
%Finally, the MSE of the concentration estimator in case of additional unknown ligands can be written as
%\begin{equation}
%\bm{\MSE[\hat{c}]}=  \bm{\Var[\hat{c}]} + \left( \bm{\Delta[\hat{c}]} \odot \bm{\Delta[\hat{c}]} \right).
%\end{equation}


\bibliographystyle{ieeetran}
\bibliography{DetectionUnderInterferenceArxivRevised}
\end{document}


