\documentclass[fleqn,10pt]{wlscirep}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{amsmath}
\usepackage{graphicx}% Include figure files
\usepackage{dcolumn}% Align table columns on decimal point
\usepackage{bm}% bold math
\usepackage{xcolor}
\usepackage{soul}
\usepackage{gensymb}

%%%%%%%%%%%%% 
\newcommand{\aveV}{\overline{V}}
\newcommand{\aveL}{\overline{L}}
\newcommand{\pN}{\ \mathrm{pN}}
\newcommand{\nm}{\ \mathrm{nm}}
\newcommand{\Mm}{\ \mathrm{\mu m}}
\newcommand{\bp}{\ \mathrm{bp}}
\newcommand{\bps}{\ \mathrm{bps}}
\newcommand{\persec}{\ \mathrm{s}^{-1}}
\newcommand{\steps}{\mathrm{steps}}
\newcommand{\mMol}{\mathrm{mM}}
\newcommand{\uMol}{\mathrm{\mu M}}
\newcommand{\kp}{k^+}
\newcommand{\km}{k^-}
\newcommand{\Jpm}{J^{\pm}}
\newcommand{\Jp}{J^+}
\newcommand{\Jm}{J^-}
\newcommand{\Jgamma}{J^{\gamma}}
\newcommand{\T}{\text{[T]}}
\newcommand{\DRI}{\Delta R_I}
\newcommand{\DRF}{\Delta R_F}
%\newcommand{\vR}{\mathcal{R}}
%\newcommand{\vL}{\mathcal{L}}
\newcommand{\vR}{R}
\newcommand{\vL}{L}
\newcommand{\DR}{\Delta {R}}
\newcommand{\Dl}{\Delta {L}}
\newcommand{\angstrom}{\mbox{\normalfont\AA}}
%%%%%%%%%%%%%%%%%
\title{Theory and Simulations of condensin mediated loop extrusion in DNA}

\author[1]{Ryota Takaki}
\author[2]{Atreya Dey}
\author[2]{Guang Shi}
\author[2*]{D.Thirumalai}
\affil[1]{Department of Physics, The university of Texas at Austin, Austin, 78712, USA}
\affil[2]{Department of Chemistry, The university of Texas at Austin, chemistry, Austin, 78712, USA}

\affil[*]{dave.thirumalai@gmail.com}

%\affil[+]{these authors contributed equally to this work}

%\keywords{Keyword1, Keyword2, Keyword3}

\begin{abstract}
Condensation of hundreds of mega-base-pair-long human chromosomes in a small nuclear volume is a spectacular biological phenomenon. This process is  driven  by the formation of chromosome loops. The ATP consuming motor, condensin, interacts with chromatin segments to actively extrude loops. Motivated by real-time imaging of loop extrusion (LE), we created an analytically solvable  model, predicting the LE velocity and step size distribution as a function of external load. The theory fits the available experimental data quantitatively, and suggests that condensin must undergo a large conformational change, induced by ATP binding,  bringing distant parts of the motor to proximity. Simulations using a simple model confirm that the motor transitions between an  open and a closed state in order to extrude loops by a scrunching mechanism, similar to that proposed in DNA bubble formation during bacterial transcription. Changes in the orientation of the motor domains are transmitted over $\sim$ $50\nm$, connecting the motor head and the hinge, thus providing an allosteric basis for LE.
\end{abstract}
\begin{document}

\flushbottom
\maketitle
% * <john.hammersley@gmail.com> 2015-02-09T12:07:31.197Z:
%
%  Click the title above to edit the author information and abstract
%
\thispagestyle{empty}

%\noindent \st{Please note: Abbreviations should be introduced at the first mention in the main text â€“ no abbreviations lists. Suggested structure of main text (not enforced) is provided below.}

\section*{Introduction}
How chromosomes are structurally organized in the tight space of the nucleus  is a long-standing problem in biology. Remarkably, these information-carrying polymers in humans with more than 100 million base pairs, are densely packed in the $5-10 \Mm$ cell nucleus~\cite{flemming1882zellsubstanz,alberts2013essential}. In order to accomplish this herculean feat, nature has evolved a family of SMC (Structural Maintenance of Chromosomes) complexes~\cite{hagstrom2003condensin,yatskevich2019organization} (bacterial SMC, cohesin, and condensin) to enable large scale compaction of chromosomes in both prokaryotic and eukaryotic systems. Compaction is thought to occur by an active generation of a large array of loops, which are envisioned to form by extrusion of the genomic material~\cite{nasmyth2001disseminating,fudenberg2016formation,sanborn2015chromatin} driven by ATP-consuming motors. The SMC complexes have been identified as a major component of the loop extrusion (LE) process~\cite{hagstrom2003condensin,yatskevich2019organization}. 

Of interest here is condensin, whose motor activity~\cite{terakawa2017condensin}, results in  active extrusion of loops in an ATP-dependent manner~\cite{ganji2018real}. Let us first describe the architecture of condensin, shown schematically in Fig~\ref{fig:modelfig}. Condensin is a ring-shaped dimeric motor, containing a pair of  SMC proteins (Smc2 and Smc4).  Both Smc2 and Smc4,  which have coiled-coil (CC) structures, are connected at the hinge domain. The  ATP binding domains are in the motor heads~\cite{diebold2017structure,yatskevich2019organization}.  There are kinks roughly in the middle of the CCs~\cite{diebold2017structure}. The relative flexibility in the elbow region (located near the kinks) could be the key to the conformational transitions in the CC that are powered by ATP binding and hydrolysis~\cite{buermann2019folded,yatskevich2019organization}. 

Previous studies using simulations~\cite{fudenberg2016formation,alipour2012self,goloborodko2016compaction}, which were built on the pioneering insights by Nasmyth~\cite{nasmyth2001disseminating}, suggested that multiple condensins  translocate along the chromosome extruding loops of increasing length.  In this mechanism, the two condensin heads move away from each other extruding loops in a symmetric manner. Cooperative action of many condensins~\cite{kim2020dna} might be necessary to account for the $\sim (1,000 - 10,000)$ fold compaction of human chromosomes~\cite{banigan2019limits}. The only other theoretical study that predicts LE velocity as a function of an external load~\cite{marko2019dna} is based on a four-state stochastic kinetic model, with minimally twenty parameters, for the catalytic cycle of the condensin that is coupled to loop extrusion~\cite{marko2019dna}. %by a DNA capture mechanism \textcolor{orange}{(here we seem suggest that our model has no capture mechanism but we do use word "capture" in the text several times. This may cause confusion)}\textcolor{violet}{Maybe mention the number of parameters in Marko's model here}. 

In sharp contrast, by focusing on the motor activity of condensin through ATP-driven allosteric changes in the enzyme, our theory and simulations support   "scrunching" as a plausible mechanism for  loop extrusion.  Scrunching  is reminiscent of the proposal made over a decade ago in the context of the first stage in bacterial transcription that results  in bubble formation in promoter DNA~\cite{kapanidis2006initial}, which was quantitatively affirmed using molecular simulations~\cite{chen2010promoter}. Recently, the scrunching mechanism was proposed to explain loop extrusion~\cite{ryu2020condensin}, which is fully supported by  theory and simulations presented here.

We were inspired by the real-time imaging of LE in $\lambda$-DNA by a  single condensin~\cite{ganji2018real},  which functions by extruding loops asymmetrically. To describe the experimental outcomes quantitatively, we created a simple analytically solvable theory, with two parameters, that produces excellent agreement with experiments for the LE velocity as a function of external load. We also quantitatively reproduce the distribution of LE length per cycle measured using magnetic tweezer experiments~\cite{ryu2020resolving}. The theory and simulations show that  for LE to occur there has to be an ATP-powered  allosteric transition in condensin  involving a large  conformational change that brings distant parts (head and the hinge in Fig~\ref{fig:modelfig}) of the motor to proximity. We predict that, on an average,  the distance between the head and the hinge decreases by conformational $\sim (22- 26)\nm$ per catalytic cycle. These values are in remarkably close to  the experimentally inferred values~\cite{ganji2018real}. Simulations using a simple model, with and without DNA, lend support to our findings. Our work strongly suggests that the conformational transitions are driven by a scrunching mechanism in which the motor is relatively stationary but DNA is reeled in by an allosteric mechanism.  %{\color{blue}
%This is reminiscent of the mechanism of transcription initiation by RNA polymerase, resulting in bubble formation in promoter DNA~\cite{kapanidis2006initial}, and illustrated in molecular simulations~\cite{chen2010promoter}. Recently, the scrunching mechanism was invoked in the context of loop extrusion~\cite{ryu2020condensin}, which is fully supported by our theory and simulations.
%} {\color {orange} We wrote similar sentence in previous page so we might not need this.}

% The Introduction section, of referenced text\cite{Figueredo:2009dg} expands on the background of the work (some overlap with the Abstract is acceptable). The introduction should not include subheadings.

\section*{Results}

% Up to three levels of \textbf{subheading} are permitted. Subheadings should not be numbered.

\subsection*{Model description}
In order to develop a model applicable to condensin (and cohesin), we assume that condensin is attached to  two loci ({\bf A} and {\bf B}) on the DNA (Fig.\ref{fig:modelfig}; right panel). Although we do not explicitly describe the nature of the attachment points, our model is based on the idea of scrunching motion where two distant ends of condensin move closer upon conformational change, triggered by ATP binding. For example, the green and blue sphere may be mapped onto motor heads and hinge, respectively.   
The structure of condensin-DNA complex in the LE active form is currently unavailable. However,  cryo-EM structures for the related cohesin-DNA complex~\cite{shi2020cryo} reveal that DNA is tightly gripped by the two heads of cohesin and the subunits (NIPBL and RAD21). When the results of structural studies are integrated with the observation that the hinge domain of the SMC complexes binds to DNA~\cite{chiu2004dna,griese2010structure,alt2017specialized}, we conclude that both condensin and cohesin must use a similar mechanism to engage with  DNA. The head domains in these motors interact with the DNA segment that is in proximity whereas DNA binds only  transiently to the hinge. We constructed the model in Fig.\ref{fig:modelfig} based in part on these findings.

In state 1, the spatial distance between the condensin attachment points is, $R_1$, and  the genomic length between {\bf A} and {\bf B} is $L_1$. Due to the polymeric nature of the DNA, the captured length $L_1$ could exceed $R_1$. However, $R_1$ cannot be greater than the overall dimension of the SMC motor, which is on the order of $\sim 50\nm$. Once a segment in the DNA is captured, condensin undergoes a conformational change driven most likely by ATP binding~\cite{ryu2020resolving}, shrinking the distance from $R_1$ to $R_2$ (where $R_2 < R_1$). As a result, the captured genomic length between {\bf A} and {\bf B}  reduces  to $L_2$ (state 2). Consequently, the loop grows by $L_1-L_2$ . The step size of condensin is $\DR=R_1-R_2$, and extrusion length per step is $\Dl=L_1-L_2$. After the extrusion is completed one end of condensin (blue circle in Fig.\ref{fig:modelfig}; right panel) is released from the genome segment and starts the DNA capturing process again, likely mediated by diffusion leading to the next LE cycle. 


%In our model we assume that the rate limiting step in LE is the mechanical extrusion driven by ATP binding/hydrolysis, not diffusion.  Indeed, the stepping time, which is dominated by diffusion, for well-studied molecular motors  site is considerably faster than the time scale for ATP hydrolysis~\cite{hinczewski2013design}. %For instance diffusion time for myosin V, which has comparable size to condensin, to the next binding site on actin filament is estimated to be $\sim 0.1\ \text{ms}$ compared to the time scale for hydrolysis of a few ms~\cite{hinczewski2013design}. 
%Our fit to the experimental data for LE gives $k_0 \sim 20 \persec$ (see later section) corresponding to $50\ \text{ms}$. Thus, our assumption is justified {\it a posteriori}. {\bf RT We should compare the hydrolysis rate to DNA capture time if we should keep this sentence.}% this assumption seems reasonable.  

\begin{figure}[]
\centering
\includegraphics[width=0.5\textwidth]{main_model_scale_DNA.pdf}
\caption{\label{fig:modelfig} {\bf Left panel:} Caricature of the structure of condensin, which has two heads (ATPase domains) and a hinge connected by coiled-coils, labeled Smc2 and Smc4. In the middle of the CCs, there is a flexible kink, referred to as an elbow. {\bf Right panel:} A schematic of the physical picture for one-sided loop extrusion based on the architecture of a generic SMC complex. DNA is attached to two structural regions on condensin. In state 1 (upper panel) the conformation of condensin is extended with the spatial distance between {\bf A} and {\bf B} equal to  $R_1$. The genomic length at the  start is $L_0$, which can be large or small.  After the conformational transition (state 1 to state 2) the distance between {\bf A} and {\bf B} shrinks to  $R_2$, and the length of the extrusion during the single transition is $\Dl =L_1 - L_2$, which would vary from cycle to cycle. }
\end{figure}

\subsection*{Theory for the captured length ($\vL$) of DNA}
\begin{figure}[]
\centering
\includegraphics[width=0.5\textwidth]{PLs2.pdf}
\caption{\label{fig:PLFdist} (a) Plots of $P(\vL|\vR)$ for different $\vR$ values; $\vR=40\nm$ (red), $\vR=50\nm$ (blue), and $\vR=60\nm$ (green). Inset: Peak position ($\vL_{peak}$) of $P(\vL|\vR)$, evaluated numerically by setting by $dP(\vL|\vR)/d\vL$ to zero, as a function of $\vR$. The dotted red line is a fit, $\vL_{peak}=\vR\exp(a\vR)$ with $a=0.003\ \mathrm{nm}^{-1}$.   
($b$)-($c$): The distributions of $L$ for different $\vR$ and $f$. $\vR=40\nm$ (red), $\vR=50\nm$ (blue), and $\vR=60\nm$ (green). The dots are from Eq.(\ref{eq:P(L|R,f)}) and the solid lines are the approximate probability distribution Eq.(\ref{eq:P(L|R,f)app}).  We used $l^{DNA}_p=50\nm$.
}
\end{figure}

In order to derive an expression for the loop extrusion velocity, we first estimate the loop length of DNA, $\vL$, captured by condensin when the attachment points are spatially separated by $\vR$. 
We show that on the length scale of the size of condensin ($\sim 50\nm$), it is reasonable to approximate $\vL \approx \vR $. To calculate the LE velocity it is necessary to estimate the total work done to extrude DNA with and without an external load. Based on these considerations, we derive an expression for the LE velocity, given by $k_0 \exp({-f\DR/k_BT})\DR$, where $k_0$ is the rate of mechanical step in the absence of the external load ($f$), $k_B$ is Boltzmann constant and $T$ is temperature.

\subsubsection*{Distribution of captured DNA length without tension, $\boldsymbol{P(\vL|\vR)}$ }
We examined the possibility that the loop extrusion length per step can be considerably larger than the size of condensin~\cite{terakawa2017condensin,lawrimore2017rotostep,ganji2018real,diebold2017structure} by calculating, $P(\vL|\vR)$, the conditional probability for realizing the contour length $\vL$ for a given end-to-end distance, $\vR$.  We calculated $P(\vR|\vL)$ using a mean-field theory that is an excellent approximation~\cite{hyeon2006kinetics} to the exact but complicated expression~\cite{wilhelm1996radial}.  The expression for $P(\vL|\vR)$, which has the same form as $P(\vR|\vL)$, up to a normalization constant, is given by (Sec.III in SI) %\textcolor{orange}{the equations are not centered}, 
\begin{align} 
\begin{split}
\label{eq:P(L|R)}
P(\vL|\vR) = A \frac{4\pi N\{\vL \} (\vR/\vL)^2}{\vL(1-(\vR/\vL)^2)^{9/2}} \exp\Big(-\frac{3t\{\vL\}}{4(1-(\vR/\vL)^2)}\Big),
\end{split} 
\end{align}
where $t\{\vL\}=3\vL/2l_p$, $l_p$ is the persistence length of the polymer, and  $N\{\vL\}=\frac{4\alpha^{3/2}e^{\alpha}}{\pi^{3/2}(4+12\alpha^{-1}+15\alpha^{-2})}$ with $\alpha\{\vL\}=3t/4$. In Eq.(\ref{eq:P(L|R)}), $A$ is a normalization constant that does not depend on $\vL$ where integration range for $\vL$ is from $\vR$ to $\infty$. 
The distribution $P(\vL|\vR)$, which scales as $\vL^{-3/2}$ for large $\vL$, has a heavy tail and does not have a well defined mean (see Fig.~\ref{fig:PLFdist}a for the plots of $P(\vL|\vR)$ for different $\vR$). The existence of long-tail in the distribution $P(\vL|\vR)$ already suggests that condensin, in principle, could capture DNA segment much larger than its size $\sim 50\nm$. However, this can only happen with lower probability compared to a more probable scenario where $\vR \approx \vL$ near the position of the peak in $P(\vL|\vR)$ (see Fig.\ref{fig:PLFdist}a). Therefore, we evaluated the location of the peak ($\vL_{peak}$)  in $P(\vL|\vR)$, and solved the resulting equation numerically. The dependence of $\vL_{peak}$ on $\vR$, which is almost linear (Fig.\ref{fig:PLFdist}a),  is well fit using $\vL_{peak}=\vR\exp(a\vR)$ with $a=0.003\ \mathrm{nm}^{-1}$ for $\vR < 60\nm$. Thus, with negligible corrections, we used the approximation $\vL \approx \vR$ on the length scales corresponding to the size of condensin or the DNA persistence length. 
% Note that the probability that $P(\vL|\vR)$,  for a given $l_p$ ($=50\nm$ in Fig.\ref{fig:probL}), is small for large $\vL$.  
Indeed, the location of the largest probability is at  $\vL\approx l_p \approx \vR$, which is similar to what was found for proteins~\cite{thirumalai1999time} as well. 
The presence of $f$ would stretch the DNA, in turn decrease the length of DNA that condensin captures, further justifying the assumption ($\vL\approx \vR$).  Therefore, we conclude that $R_1 \approx L_1$ and $R_2\approx L_2$ (note that $R_2 < R_1 \lesssim l^{DNA}_p$), and that LE of  DNA loop that is much larger than the size of condensin is less likely. 
Thus, we expect that the  extrusion length of DNA is nearly equal to the step size of condensin, $\DR \approx \Dl$.  

% \textcolor{orange}{(I have one question. I can understand that $R_1\approx L_1$, but how do we argue $R_2 \approx L_2$? We do provide an argument later. Maybe we can mention it here as well. )}


% The distribution of captured length of DNA by condensin is furthermore discussed using simple computational model in later section. 

% \begin{figure}[]
% \centering
% \includegraphics[width=0.5\textwidth]{main_pLR.pdf}
% \caption{\label{fig:probL} (a) Plots of $P(\vL|\vR)$ for different $\vR$ values; $\vR=40\nm$ (red), $\vR=50\nm$ (blue), and $\vR=60\nm$ (green). (b) Peak position ($\vL_{peak}$) from (a), evaluated numerically by setting by $dP(\vL|\vR)/d\vL$ to zero, as a function of $\vR$. The dotted red line is a fit, $\vL_{peak}=\vR\exp(a\vR)$ with $a=0.003\ \mathrm{nm}^{-1}$. We used $l_p^{DNA}=50\nm$ as the persistence of DNA. 
% }
% \end{figure}
%{\bf RT Please combine Figure 2 and 3 and create a 6 panel figure labeled Figure 2.}

\subsubsection*{Force-dependent distribution, $\boldsymbol{P(\vL|\vR,f)}$, of the captured DNA length}
Following the steps described above, we write the 
end-to-end distribution of semi-flexible polymer under tension $f$, $P(\vR,f|\vL)$, as $P(\vR,f|\vL) = B P(\vR|\vL)e^{f R/k_BT}$, where $B$ is normalization constant. Thus, $P(\vL|\vR,f)$ is obtained using,
\begin{align} 
\begin{split}
\label{eq:P(L|R,f)}
P(\vL|\vR,f) = C B\{\vL \} \frac{4\pi N\{\vL \} (\vR/\vL)^2}{\vL(1-(\vR/\vL)^2)^{9/2}} \exp\Big(-\frac{3t\{\vL\}}{4(1-(\vR/\vL)^2)}\Big) \exp\Big(\frac{f\vR}{k_BT}  \Big)  ,
\end{split} 
\end{align}
where $C$ is a normalization constant that does not depend on $\vL$. The constant $B\{\vL \}$ for $P(\vR,f|\vL)$, which carries the $\vL$ dependence, prevents us from deriving an analytically tractable expression for $P(\vL|\vR,f)$. We find that, for a sufficiently stiff polymer, ($\vR/l_p \lesssim 1$), $P(\vL|\vR,f)$ is well approximated by,
\begin{align} 
\begin{split}
\label{eq:P(L|R,f)app}
P_{A}(\vL|\vR,f >  0) = D \frac{4\pi N^2\{\vL \} (\vR/\vL)^2}{\vL(1-(\vR/\vL)^2)^{9/2}} \exp\Big(-\frac{3t\{\vL\}}{4(1-(\vR/\vL)^2)}\Big) \exp\Big(\frac{f\vR}{k_BT} -\big(1.0+3.3e^{-f/f_0}\big) \frac{f \vL}{k_BT} \Big) ,
\end{split} 
\end{align}
where $f_0=1/7\pN$, $t\{\vL\}=3\vL/2l_p$, $l_p$ is the persistence length of the polymer, and  $N^2\{\vL\}=\Big(\frac{4\alpha^{3/2}e^{\alpha}}{\pi^{3/2}(4+12\alpha^{-1}+15\alpha^{-2})}\Big)^2$ with $\alpha\{\vL\}=3t/4$. The constant $D$  does not depend on $\vL$ in the integration range. Note that on the scale of condensin DNA is relatively rigid ($R \lesssim l^{DNA}_p \sim 50\nm$). Probability Eq.(\ref{eq:P(L|R,f)}) and Eq.(\ref{eq:P(L|R,f)app}) for different values of $f$ in Fig.\ref{fig:PLFdist} show that the agreement between $P_{A}(\vL|\vR)$ and $P(\vL|\vR,f)$ is good. Therefore, in what follows we use  Eq.(\ref{eq:P(L|R,f)app}). The distributions for $f>0$ in Fig.\ref{fig:PLFdist} show that the position of the peak for $ P(\vL|\vR,f)$ do not change over the range of $f$ of interest. However, the height of the peak increases as $f$ increases accompanied by the shrinking of the tail for large $\vL$.

% \begin{figure}[]
% \centering
% \includegraphics[width=1.0\textwidth]{PLfdist.pdf}
% \caption{\label{fig:PLFdist} Captured length distribution for different $\vR$ and $f$; $\vR=40\nm$ (red), $\vR=50\nm$ (blue), and $\vR=60\nm$ (green). The dots are from Eq.(\ref{eq:P(L|R,f)}) and the solid lines is the approximate expression Eq.(\ref{eq:P(L|R,f)app}). (a) $f=0.1\pN$. (b) $f=0.3\pN$. (c) $f=0.5\pN$. (d) $f=0.7\pN$. We used $l^{DNA}_p=50\nm$.
% }
% \end{figure}
%{\bf RT: In the latest paper on biorxiv by Ryu (Ref 17) it is claimed even in the Abstract that "ATP-binding is the primary step-generating stage underlying loop extrusion." We say, and I agree, that hydrolysis free energy is converted to work. We need to pay attention to the preprint and understand their reasoning. {\color {red} The logic is in fact very straightforward as explained in their paper in one sentence. Section "ATP binding and hydrolysis related to two distinct mechanistic process" in their paper.}}
%Example text under a subsection. Bulleted lists may be used where appropriate, e.g.

\subsection*{Condensin converts chemical energy into mechanical work for LE}

\begin{figure}[]
\centering
\includegraphics[width=0.5\textwidth]{W.pdf}
\caption{\label{fig:delF} Energetic cost to extrude DNA, $W=W_0+W_f$, as a function of $L_\Sigma$ ($\sim$ total extruded length of DNA; $L_{\Sigma}=L_0+L_1$) for different $f$ values on DNA. 
%$L_\Sigma$ is the sum of the length of DNA  that is already extruded ($L_0$) and the length newly captured ($L_1$). % Note that $L_\Sigma \geq R_1$ by physical consideration. 
We set $R_1 = 40 \nm$ as the size of condensin in open state (State 1 in Fig.\ref{fig:modelfig}) and $R_2=14\nm$, which corresponds to $\Delta R=26 \nm$ as obtained from the fit of our theory to experiment~\cite{ganji2018real} in later section. $f=0\pN$ is in red, $f=0.2\pN$ is in blue, $f=0.4\pN$ is in green, and $f=0.6\pN$ is in brown.  We used $l^{DNA}_p=50\nm$.
}
\end{figure}

Just like other motors, condensin hydrolyzes ATP, generating $\mu \approx 20\ k_BT$ chemical energy that is converted into mechanical work, which in this case results in extrusion of a DNA loop~\cite{ganji2018real}. To derive an expression for LE velocity, we calculated the thermodynamic work required for LE. The required work $W$ modulates the rate of the mechanical process by the exponential factor $\exp(-W/k_BT)$. In our model, $W$ has two contributions. The first  is the work needed to extrude the DNA at $f=0$ ($W_{0}$). Condensin extrudes the loop by decreasing the spatial distance between the attachment points from $\vR=R_1$ to $\vR=R_2$ (Fig.\ref{fig:modelfig}). 
The associated genomic length of DNA that has to be deformed is $L_{\Sigma} = L_0 + L_1$.  
The second contribution is $W_{f}$, which comes by applying  an external load.  Condensin resists $f$ up to a threshold value~\cite{ganji2018real}, which may be thought of as the stall force. The mechanical work done when condensin takes a step, $\DR=R_1-R_2$, is $W_{f}= f \DR$. 


We calculated $W$ as the free energy change needed to bring a semi-flexible polymer with contour length $L_\Sigma$, from the end-to-end distance $R_1$ to $R_2$. It can be estimated using the relation, $W \approx -k_BT\log \big(P(R_2,f|L_\Sigma)\big)+ k_BT\log\big(P(R_1,f|L_\Sigma)\big) $, where $P(\vR,f|\vL)$ is given by $P(\vR,f|\vL) = B P(\vR|\vL)e^{f R/k_BT}$ where $B$ is a  normalization constant.   
Although $P(\vR,f|\vL)$ is a distribution, implying that there is a distribution for $W$,  for illustrative purposes, we plot $W$ in Fig.\ref{fig:delF} for a fixed $R_1=40\nm$ and $R_2=14\nm$ corresponding to $\Delta R= 26\nm$ as estimated using our theory to experiment in later section. It is evident that condensin has to overcome the highest bending penalty in the first step of extrusion, and subsequently $W$ is essentially a constant at large $L_{\Sigma}$. Note that when $f=0$ (red line in Fig.\ref{fig:delF}), $W=W_0$ because the $W_f$ term vanishes.  
If $R_1=40 \nm$, which is approximately the size of the condensin motor, we estimate that condensin pays $5\ k_BT$  to initiate the extrusion process without tension (red line in Fig.\ref{fig:delF}). %We note that initial binding site of condensin on DNA is reported to be a A/T rich domain~\cite{terakawa2017condensin}, which is more flexible than sequences with high G/C content~\cite{sarai1989sequence,mitchell2017sequence}. Thus, it is possible  that condensin starts the extrusion from A/T rich domain to prevent the failure of the initiation process. 

Once the energetic costs for LE are known, we can calculate the LE velocity as a function of an external load applied to condensin. From energy conservation, we obtain the equality, $n \mu=W_{0} + W_{f} + Q$, where $n$ is the number of ATP molecules consumed per mechanical step, $\mu$ is the energy released by ATP hydrolysis, and $Q$ is the heat dissipated during the extrusion process.  The maximum  force is obtained at equilibrium when the equality $n \mu = W_{0} +  W_{f}\{ f_  {max} \}$ holds.  If we denote the rate of mechanical transition as $k^+$ and reverse rate as $k^-$, fluctuation theorem~\cite{seifert2005fluctuation,seifert2012stochastic,mugnai2020theoretical} together with conservation of energy gives the following relation:  

\begin{align} 
\begin{split}
\label{eq:thermo}
k^{+}/k^{-} = e^{(n \mu-W_{0} - W_{f})/k_BT}. 
\end{split} 
\end{align}
Note that condensin operates in non-equilibrium by transitioning from state 1 to 2, which requires input of energy. The load dependent form for the above equation may be written as,
\begin{align} 
\begin{split}
\label{}
k^{+}= k_0 e^{-W_{f}/k_BT},
\end{split} 
\end{align}
where $k_0=k^{-} e^{(n \mu-W_{0})/k_BT}$ is the rate of the mechanical transition at $0$ load. 
Thus, with  $\DR$ being the extruded length per reaction cycle , the velocity  of LE, $\Omega$, may be written as, 
\begin{align} 
\begin{split}
\label{eq:omega}
\Omega\{f\}= k_0 e^{-f \DR / k_BT}\DR.
\end{split} 
\end{align}
It is worth stating that $k_0$ is the chemical energy dependent term, which includes $\mu$, depends on the nucleotide concentration. In order to obtain the ATP dependence, we assume the Michaelis-Menten form for $k_0$, $\frac{\hat{k}_0[\text{T}]}{[\text{T}]+K_M}$, where [\text{T}] is the concentration of ATP, $\hat{k}_0$ is the maximum rate at saturating ATP concentration, and $K_M$ is the Michaelis-Menten constant (see Fig.\ref{fig:fit_model}b). In principle, $K_M$ should be determined from the measurement of LE velocity as a function of  [\text{T}]. Here, we use $K_M=0.4\ \mMol$ obtained as Michaelis-Menten constant for ATP hydrolysis rate~\cite{terakawa2017condensin}, assuming that ATP hydrolysis rate is the rate-limiting step in the loop extrusion process.

%Both $\hat{k}_0$ and $K_M$ can be obtained by measuring ATPase activity as a function of [T] (see Fig.\ref{fig:fig_model}b).  

In order to calculate $\Omega$ as a function of the relative DNA extension, $x$, we use the expression ~\cite{marko1995stretching,rubinstein2003polymer},
%we follow  Ganji et al.~\cite{ganji2018real} $f$ is related to the relative DNA extension by using  the expression~\cite{marko1995stretching,rubinstein2003polymer},
\begin{align} 
\begin{split}
\label{eq:f}
f = \frac{k_BT}{2l_p}\Big[2x+\frac{1}{2}\Big(\frac{1}{1-x}\Big)^2-\frac{1}{2}\Big].
\end{split} 
\end{align}
The dimensionless variable, $x$, is the $f$-dependent relative extension. In the Ganji et al.~\cite{ganji2018real}  experiment $x$ is measured, and the $f$-dependence of $\Omega$ is obtained by expressing  $x$ in terms of $f$ using a numerical procedure. 

% $x=R/L$, where $R$ is end-to-end distance of the whole DNA and $L$ is the contour length corresponding unextruded part of DNA~\cite{ganji2018real}. 
%It is unclear if the above equation obtained for semi-flexible polymer in isolation holds for the condensin construct. {\color {red} This equation is used for DNA not condensin.} \textcolor{violet}{(In definition of extension is $L = 20\mu m - L_\Sigma?$ That is the definition in the Ganji paper. If so, maybe you should write L is the contour length of unextruded part of DNA. Also I think the notation of R and L are repeated because initially they are used to describe a single loop and a single step. In the experiment end-to-end distance is fixed, so maybe if your definition is same as experiment you can write $x = 9.1\mu m/(20\mu m - L_\Sigma)$}
%{\bf RT I know the SI has LE velocity as a force. Did you fit that using Eq. 6? Did you get the same values for the two parameters?  } 
%{\color {red} LE velocity as a function of $f$ comes from the simple conversion of $\chi$ to $f$ using Eq.\ref{eq:f} so they are completely same graph but different unit. I did not do any fit for the $\Omega$ vs $f$ graph for that reason.  }

\subsection*{Analysis of experimental data}
\begin{figure}[]
\centering
\includegraphics[width=\textwidth]{LE_vel_dists.pdf}
\caption{\label{fig:fit_model}  (a) {\bf Left panel:} The LE velocity  as a function of the relative extension of the DNA ($x$). Red dots are experimental data~\cite{ganji2018real}, and the solid blue line is the fit obtained using Eq. (\ref{eq:omega}). We used $l_p^{DNA}= 50\nm$ for the persistence length of DNA. {\bf Right panel:} Extrusion velocity as a function of the external load acting on DNA ($f$). The blue line is $\Omega$ from Eq. (\ref{eq:omega}) and the line in black is reproduced from Fig. 6C in~\cite{marko2019dna}. The unit of LE velocity, nm/s, is converted to kbp/s using the conversion $1\bp=0.34\nm$~\cite{alberts2013essential}. (b)  The dependence of LE velocity on ATP concentration  for different relative extension of DNA. The parameters used are $\hat{k}_0=20\persec$, $\DR=26 \nm$, and $K_M=0.4\ \mMol$, where $K_M=0.4\ \mMol$ is Michaelis-menten constant for ATP hydrolysis rate~\cite{terakawa2017condensin}. We used $l_p^{DNA}= 50\nm$ for the persistence length of DNA.
(c)
Distribution of the DNA extrusion length per step ($\Delta L$) using $l^{DNA}_p=42\nm$, which is the value reported in the experiment~\cite{ryu2020resolving}. The histograms are the experimental data taken from Ryu et al.~\cite{ryu2020resolving}(blue) and Strick et al.~\cite{strick2004real}(inset;green). The distributions in red are the theoretical calculations. The distributions for theory and the experiments are both in $f=0.4\pN$.
(d) LE length distributions for various external load on DNA using $l_p^{DNA}= 42\nm$~\cite{ryu2020resolving}. $f=0\pN$ is in magenta, $f=0.2\pN$ is in blue, and $f=1.0\pN$ is in green. The inset compares the results for $f=0.2\pN$ and $f=0.4\pN$.
}
\end{figure}


\subsubsection*{Loop extrusion velocity}
We used Eq.(\ref{eq:omega}) to fit the experimentally measured LE velocity as a function of DNA extension~\cite{ganji2018real}. The two fitting parameters are $\DR$, and $k_0$, the average step size for condensin, and the extrusion rate at $f=0$, respectively. Excellent  fit of theory to experiments, especially considering the  dispersion in the data, gives  $k_0=20 \persec$ and $\DR=26 \nm$.  This indicates that condensin  undergoes a conformational change that brings the head and the hinge to within $\Delta R \sim 26 \nm$ ($\sim$ 76 bps), during each extrusion cycle. This prediction is remarkably close to the value measured in the recent AFM experiment $\sim 22\nm$~\cite{ryu2020condensin}, and is further supported by our simulations (see below).
We note that $k_0=20 \persec$ is roughly ten times greater than the bulk hydrolysis rate estimated from ensemble experiments~\cite{ganji2018real,terakawa2017condensin}. 
A plausible reason for the apparent discrepancy, already provided in the experimental studies~\cite{ganji2018real,terakawa2017condensin}, is that  bulk hydrolysis rate could underestimate the true hydrolysis rate due to the presence of inactive condensins. Thus, the estimated rate of $k_0 = 2 \persec$ should be viewed as a lower bound~\cite{ganji2018real}. Another possible reason for the discrepancy may be due to methods used to estimate $k_0$ in previous experiments~\cite{ganji2018real,terakawa2017condensin}. It is clear that additional experiments are needed to obtain better estimates of the hydrolysis rate, which is almost all theories is seldom estimated.%Additional remarks related to extraction of parameters made in section VII in the SI, lead us to conclude that the $k_ 0$ value obtained by fitting theory to experiments is not unreasonable. 


In Fig.\ref{fig:fit_model}a right panel we compare the dependence of $\Omega$ on $f$ obtained in a previous kinetic model that has in excess of 20 parameters~\cite{marko2019dna}. In contrast to our theory, even the shape of the LE velocity for $\Omega$ versus $f$ does not agree with experiment. In addition, there is a major discrepancy (factor of 2-3) between the predicted and the measured values of $\Omega$ at low force.  



\subsubsection*{LE length distribution}
Recently Ryu et al.~\cite{ryu2020resolving} measured the distribution of LE length per step using magnetic tweezers. We calculated the LE distribution using Eq.(\ref{eq:P(L|R)}) and Eq.(\ref{eq:P(L|R,f)app}).  Of interest here  is the distribution for $L_1-L_2$ where $L_1$ is the captured length of DNA in open shape (see Fig.\ref{fig:modelfig}; right panel). We use the length of condensin in the open state as $R_1=40\nm$ (roughly the peak in the distance between the head and the hinge in the O shape in the wild type condensin), and assume that  $\Delta R=26\nm$ during a single catalytic cycle, as theoretically calculated in the previous section. This gives the length of the closed state, $R_2=14\nm$. Previous experiments~\cite{ryu2020condensin} reported significant fluctuations in the size of the open and closed states, leading to the standard deviation for the change between these two states to be $\Delta=13\nm$. For simplicity, we include the standard deviation for the conformational change in the open state $R_1\pm \Delta$ and keep $R_2$ fixed. This is justified because as $R$ decreases ($R_2 < R_1$), not only does the peak in 
 $P(\vL|\vR)$ and $P_A(\vL|\vR,f)$ moves to small $\vL$ but also there is a decrease in the fluctuation (width of the distribution is smaller) (see Fig.\ref{fig:PLFdist}). This suggests that the variance of the extrusion length owing to $R_2$ is negligible. We assume that the distribution of $R_1$ is a Gaussian centered at $40\nm$ with a standard deviation of $13\nm$. It is reasonable to approximate $L_2 \approx R_2$ since $R_2=14\nm < l^{DNA}_p=50\nm$. With these  parameters in hand, the captured DNA length, Eq.(\ref{eq:P(L|R)}) and Eq.(\ref{eq:P(L|R,f)app}), directly lead to an expression for the distributions for LE length per step. The probability for the LE length per step ($\Delta \vL$) is, $P(\Delta \vL) =  P(\vL+R_2|R_1)$ for $f=0$ and $ P(\Delta \vL | f) = P_{A}(\vL+R_2|R_1,f)$ for $f>0$. %Note that we use Gaussian random variable for $R_1$ in this section in order to take into account the fluctuation for condensin.  

We compare in Fig.\ref{fig:fit_model} the theoretically calculated distribution for $f=0\pN$ from Eq.(\ref{eq:P(L|R)}) and $f=0.4\pN$ from Eq.(\ref{eq:P(L|R,f)app}) with experiments. The distribution for $f=0 \pN$ cannot be measured using magnetic tweezers but provides insights into the range of LE length that condensin could take at $f=0$.   Remarkably, the calculated distribution is in excellent agreement with the experimental data~\cite{ryu2020resolving}. 


%{\bf RT: Fig 5 (a) LE velocity as a function of $f$ experiments, our theory and the one from Marko's paper. (b) Like (a) except add Marko result. (c) ATP Depenence.} {\color {red} Fig4.(a) is LE velocity vs extension and Marko's result is LE velocity vs f so I added our result for LE velocity vs f and Marko's result in the inset.}

% \begin{figure}[]
% \centering
% \includegraphics[width=0.5\textwidth]{dist_experiment_lp=42.pdf}
% \caption{\label{fig:dist_exp}  Distribution for the DNA extrusion length per step ($\Delta \vL$)  of condensin. We used $l^{DNA}_p=42\nm$, which is specifically measured in the experiment~\cite{ryu2020resolving}. (a) Prediction for the distribution derived from our theory in the case $f=0$, $P(\Delta \vL)$.   (b) Red squares are the distribution obtained from theory, $P(\Delta \vL|f)$. Empty circles are the distribution for forward steps obtained in the experiment~\cite{ryu2020resolving}. $f=0.4\pN$.  
% }
% \end{figure}

% {\bf RT Besides improving the theoretical data (as per Guang Shi suggestion) report results for 0.1, 0.2, 0.3 and 0.5 also. Total 6 panels.
% {\color {red} I put together multiple distributions for different loads so that it is easy to compare. The choice of $f$ values is simply visibility. Other loads are shown in SI section VII. }
% Fig. 7 top not needed. But may AD could superimpose the experimental with simulations in Fig. 7b. Also Fig. 8 and 7 should be combined.
% {\color {red} I made two versions where Fig. 7 top is not deleted and Fig. 7 top is deleted with structures is inserted instead. I like the former version but please let me know what you think. }

% Fig. 2 and 3 into one Fig with 6 panel. {\color {red} Did it.}}

\subsection*{Plausible conformational change of condensin in LE process}
\begin{figure}[]
\centering
\includegraphics[width=\textwidth]{simulation_new2.pdf}
\caption{\label{fig:delR} Simulations for the transition between O$\rightarrow$B transition. (a) {\bf Left panel:} Representative pictures from the simulation. Red, blue, light blue, orange, and green spheres are the heads, hinge, CC, elbows, and DNA, respectively. {\bf Right panel:} The trajectory for the change in head-hinge distance ($\vR$). $\Delta t_L$ is the time step of simulation (SI Sec.I). (b) Predicted distributions, $P(\vR)$s, of the head-hinge distance during one catalytic cycle of the motor. The peak positions for open state and closed state ($R_1=38\nm$ and $R_2=16\nm$, respectively). The large width and the overlap between the two distributions implies a great deal of conformational heterogeneity that contributes also to broad step size distribution. (c) The distribution for $\vR$ taken from Ryu et al.~\cite{ryu2020condensin}.  The agreement between the simulations and experiments is remarkable, especially considering that the model has {\it no fitting parameters}.
%The distributions were calculated from 50 trajectories (100,000 time points). 
(d) Histograms in orange are the distribution of $\vL$ in the open shape, and the blue histograms are the distribution in the closed shape. The inset shows the distributions for $L>90\nm$. We performed 50 simulations from which 100,000 sample points were used to create the histograms. 
}
\end{figure}

% \begin{figure}[]
% \centering
% \includegraphics[width=\textwidth]{simulations_structures.pdf}
% \caption{\label{fig:delR} Simulations for the transition between O$\rightarrow$B transition. The top figure is the trajectory for head-hinge distance adapted from the experiment~\cite{ryu2020condensin} (a) Example of a trajectory showing the change in the head-hinge distance $R$. The trajectory is for $l_p^{CC} \sim 70 \nm$. $\Delta t_L$ is the time step of simulation (SI) (b) Distributions of $P(R_1$) and $P(R_2)$ with $l_p^{CC} \sim 70 \nm$. The histogram for open state and closed state are in blue and red, respectively. The mean values of open state and closed state ($R_1=41\nm$ and $R_2=20\nm$, respectively) are indicated in the figure. The inset is the distribution for head-hinge distance adapted from Ryu et al.~\cite{ryu2020condensin}.
% The distributions were calculated from 50 trajectories (40,000 time points). 
% (c)-(d) Histogram in orange shows the distribution of head-hinge distance, and the blue gives the distribution for the captured length of DNA by condensin.  Connected beads depicted in pink shows DNA.  (c) is the distributions for closed state (B state) and (d) is the distributions for open state (O state).  Parameters for condensin in this simulation are identical to the simulations without DNA (see Table I in SI). We performed 50 simulations of 10,000 sample points to create the histograms.
% }
% \end{figure}

%\subsubsection*{Simulation without DNA}

% \begin{figure}[]
% \centering
% %\includegraphics[width=0.5\textwidth]{main_dist.pdf}
% \includegraphics[width=0.5\textwidth]{simulation_with_experiment.pdf}
% \caption{\label{fig:delR} Simulations for the transition between O$\rightarrow$B transition. The top figure is the trajectory for head-hinge distance adapted from the experiment~\cite{ryu2020condensin} (a) Example of a trajectory showing the change in the head-hinge distance $R$. The trajectory is for $l_p^{CC} \sim 70 \nm$. $\Delta t_L$ is the time step of simulation (SI) (b) Distributions of $P(R_1$) and $P(R_2)$ with $l_p^{CC} \sim 70 \nm$. The histogram for open state and closed state are in blue and red, respectively. The mean values of open state and closed state ($R_1=41\nm$ and $R_2=20\nm$, respectively) are indicated in the figure. The inset is the distribution for head-hinge distance adapted from Ryu et al.~\cite{ryu2020condensin}.
% The distributions were calculated from 50 trajectories (40,000 time points). 
% }
% \end{figure}


{\bf Simulations without DNA:}
Next we tested whether the predicted value of $\DR\sim 26\nm$, which is in  fair agreement with the experiment, is reasonable using simulations of a simple model.
Because the ATPase domains are located at the heads of condensin, it is natural to assume that the head domain undergoes conformational transitions upon ATP binding and/or hydrolysis. Images of the CCs of the yeast condensin (Smc2-Smc4) using liquid atomic force microscopy (AFM) show they could adopt a few distinct shapes~\cite{eeftens2016condensin,ryu2020condensin}.  Based on these experiments, we hypothesize that the conformational changes initiated at the head domain result in changes in the angle at the junction connecting the motor head to the CC that  propagates through the whole condensin via the CC by an allosteric mechanism~\cite{muir2020structure}. The open (O-shaped in~\cite{ryu2020condensin}), with the hinge that is $\approx 40 \nm$ away from the motor domain, and the closed (B-shape~\cite{ryu2020condensin}) in which the hinge domain is in proximity to the motor domain are the two relevant allosteric states  for LE~\cite{ryu2020condensin,eeftens2016condensin}. To capture the reaction cycle (O $\rightarrow$ B $\rightarrow$ O), we model the CCs as kinked semi-flexible polymers (two moderately stiff segments connected by a flexible elbow), generalizing a similar description of stepping of Myosin V on actin~\cite{hinczewski2013design}. By altering the angle between the two heads the allosteric transition between the open (O-shaped) and closed (B-shaped) states could be simulated (SI contains the details). 

We tracked the head-hinge distance ($\vR$) changes during the transition from the open ($\vR=R_1$) to the closed state ($\vR=R_2$) in order to calculate the distribution of  $\Delta R_s =R_1-R_2$. The sample trajectory in Fig.\ref{fig:delR}a, monitoring the conformational transition between the two states, shows that $\DR_s$ changes by $\sim 22\nm$ for the persistence length of condensin ($l^{CC}_p$; see Sec.II in SI for the detail) $l^{CC}_p = 24 \nm$, which roughly coincides with the value extracted by fitting the theory  to the experimental data.  Higher (smaller) values of $\Delta R_s$ may be obtained using larger (smaller) values of $l_p^{CC}$ (Fig.S3 in SI).
The distributions, $P(\vR)$, calculated from multiple trajectories (Fig.\ref{fig:delR}b) are broad, suggestive of high degree of conformational heterogeneity in the structural transition between the open and closed states. The large dispersions found in the simulations is in surprisingly excellent agreement with experiments~\cite{ryu2020condensin}, which report that the distance between the peaks is $22 \pm  13\nm$.  We find that the corresponding value is $22 \pm 9 \nm$. The uncertainty is calculated using standard deviation in the distributions.  Overall the simulations not only provide insight into the physical basis of the theory but also lend support to recent single molecule experiments~\cite{ryu2020condensin}.
%{\color {red} Although our simulation here without DNA suffices for our purpose to probe the change of head-hinge distance, we conducted simulations including DNA in order to give insight for DNA capture process by condensin (see section IX in SI). } 


{\bf Simulations with DNA:}
The purpose of the simulations discussed in the previous section was to assess whether the allosteric mechanism produces a structural rationale for the value of $\DR\sim 26\nm$ extracted from the theory.  We also created a simple model of condensin with DNA to give additional insights into the DNA-capture mechanism, which is directly related to the extrusion length of DNA per step. We assume that the capture length of DNA by condensin, $\vL$, is governed by  diffusion of the hinge domain, and that the $\vL$ is solely determined by the semi-flexible polymer nature of DNA.
We attached one end of the DNA to the heads of condensin. The other end of DNA diffuses freely during the simulations. We define the DNA "capture" event by the distance between a DNA segment and the condensin hinge, with a cut-off length of $4\nm$: if the distance is less than $4\nm$ we assume that condensin captures the DNA segment. Captured DNA length is the contour length of DNA held between the heads and the hinge. 
% This working mechanism using hinge domain is motivated by the recent AFM experiment~\cite{ryu2019afm}. 
We used a coarse-grained bead-spring model for DNA~\cite{hyeon2006kinetics,dey2017toroidal}. Each bead represents 10 base-pairs, which implies that the bead size is $\sigma_{DNA}=3.4 \nm$. The chain has $N=100$ beads or 1,000 base-pairs. The simulation model for DNA is in the SI (Sec.I).
The simulations with DNA explain an interesting aspect of the DNA capture process. In contrast to well-studied molecular motors that take a step that is nearly constant (conventional kinesin and myosin V) and walks on rigid linear track (microtubule and actin filament, respectively), condensin could in principle capture variable length of DNA during each catalytic cycle because it is a flexible polymer unlike microtubule. 
%In Fig.~\ref{fig:delR}c the peak of the distribution for $\vL$ at closed state nearly coincides with the peak for $\vR$, reflecting that the head-hinge distance is below $l^{DNA}_p\sim 50\nm$. 
%In open state (Fig.~\ref{fig:delR}d) the peak position slightly shifts towards a longer distance. The peak position for $\vR$ is $\sim 38\nm$ and the peak for $\vL$ is $\sim40\nm$. This accords well with the theoretical estimate of the peak position (see Fig.~\ref{fig:PLFdist}b). Note that the estimate of the peak position in Fig.~\ref{fig:PLFdist}b is for fixed $\vR$, on the contrary, in this simulation head-hinge distance is fluctuating quantity. 
 Fig.~\ref{fig:delR}d shows that there is a finite probability that $\vL$ exceeds the position of the peak by a considerable amount, as predicted in Fig.~\ref{fig:PLFdist}a. This implies that the $L_1$ can be as large as (60-100) nm ($\sim (180 - 290)  \mathrm{bps}$), which allows for condensin to extrude substantial length of DNA in each catalytic cycle (see the distributions $P(\Delta \vL)$ in Fig.\ref{fig:fit_model}). 
%Because our theory to derive LE velocity, $\Omega$, is built on the peak position of $P(\vL|\vR)$ we neglect the long tail of the distribution as a higher order correction. Nevertheless, 

Our results show that theory and simulations for the LE velocity [Eq.(\ref{eq:omega})] predicts the extent of the conformational change of condensin during the LE process fairly accurately ($\sim 26\nm$ in theory and $\sim 22\nm$ in the experiment~\cite{ryu2020condensin} and simulations) and gives the distributions of LE length per cycle in good agreement with experiment. 
% Future experiments will access the range of applicability of the first order approximation $\vR \approx \vL$ in $\Omega$.   
Although our theory is in good agreement with experiments for load-dependent LE velocity and distribution of step sizes, the calculated persistence length (24 nm) of the SMC coils is much higher than the value ($\sim$ 3.8 nm) estimated from analyses of AFM images in combination with simulations using the worm-like chain model~\cite{eeftens2016condensin}. It is possible that due to interaction between condensin and other proteins (Brn1 for example) could constrain the head movement, and thus stiffen the coiled coil. However, the discrepancy between theoretical predictions and simulations is  too large to be explained by such effects.  Despite performing many simulations using a variety of polymer models, the origin of this discrepancy is unclear. Simulations with flexible coiled-coil do not  reproduce the measurements of quantities such as $\Delta R$, that are directly monitored in single molecule experiments~\cite{ryu2020condensin} (see Fig.S3). We believe that additional experiments and simulations based on higher resolution structures in different nucleotide states are required to close the gap.


\section*{Discussion}
{\bf Connection to experiments:}
Two of the most insightful experimental studies~\cite{ryu2020condensin,ryu2020resolving} have reported the mean LE velocity and step size distribution as a function of $f$.  Because these are direct single molecule measurements that have caught the motor in the act of LE, the results are unambiguous, requiring little or no interpretation.  Minimally any viable theory must account for these measurements as quantitatively as possible,  using only a small number of physically meaningful parameters. To our knowledge, our theory is currently the only one that reproduces the experimental observations accurately with just two parameters. The only other theory~\cite{marko2019dna} that reported LE velocity as a function of force, not only has a large number of parameters but it cannot be used to calculate the step size distribution.

We first showed that the calculated LE velocity could be fit to experimental data in order to extract the hydrolysis rate and the mean step size. To provide a physical interpretation of the theoretically predicted mean step size, we performed polymer based simulations using a simple model of the CCs. The model for the simulations was based on the AFM images~\cite{ryu2020condensin}, which  showed that, during loop extrusion, there is a transition between the O shape (head and the hinge are far apart) to the B shape where they are closer. Remarkably, our simulations capture the distributions of the head-hinge distances in the O and B states well {\it without any fitting parameters}. The  mean distance between the peaks ($\Delta R_s \approx 22\nm$) in the simulations is in excellent agreement with measured value (see Fig. 2c in~\cite{ryu2020condensin}).  It is worth emphasizing that our theoretical fit to experiment yielded  $\Delta R \sim 26 \nm$, which is also in very good agreement with $\sim22\nm$ measured in the high speed AFM imaging experiment~\cite{ryu2020condensin}. Thus, both experiments and simulations support the mechanism that repeated O$\rightarrow$B shape transitions result in extrusion of the DNA loops. 
% Finally, the calculated distribution of extruded DNA length at $f=0.4\pN$ is in excellent agreement with the most recent measurements using single molecule magnetic tweezer experiments. Taken together, the theory (2 free parameters) and simulations (no adjustable  parameter) based on a very simple model explains in near quantitative manner the experimental data. In addition, we have made two testable predictions: (i)  LE velocity as a function of ATP concentration and $f$, which is needed to obtain insights details of the workings of condensin and by implication cohesin. (ii) The most recent magnetic tweezer experiment reported the distribution of the length of the extruded loop only at $f = 0.4\pN$. 
%{\color {red} (There are experimental plots for other $f$ values in the SI in Ryu et.al. but they are poor statistics $N\sim 100$. Our theoretical distributions fit well for lower load $f<0.5$ but does not fit for higher $f$. Please check section VII in SI. I think by the time they publish the manuscript the shape of the distributions will change as happened in the case Ryu et.al. 2020, AFM paper.) } We have calculated this quantity for other values of $f$ as well (see Fig...). Both these predictions await future experiments.

{\bf Relation to a previous study:} 
Recently, a four state chemical kinetic model~\cite{marko2019dna}, similar to the ones used to  interpret experiments on stepping of myosin and kinesin motors on polar tracks (actin and microtubule)~\cite{mugnai2020theoretical,kolomeisky2007molecular}, was introduced in order to calculate the $f$-dependent LE velocity and the loop size. The agreement between the predicted dependence on LE velocity and loop size as a function of $f$ is not satisfactory (see Fig. \ref{fig:fit_model}(a)).  Apart from the very large number of parameters (about twenty one in the simplified version of the theory~\cite{marko2019dna}), our two parameter theory differs from the previous study in other important ways. (1) The model~\cite{marko2019dna} is apparently based on the rod-like (or I shape) X-ray structure of the prokaryotic CC of the SMC dimer~\cite{diebold2017structure}, which was pieced together by joining several segments of the CC.  However, the theory itself does not incorporate any structural information but is based on a number of rates connecting the four assumed states in the reaction cycle of the motor, and energetics associated with the isolated DNA.  (2) Because the previous purely kinetic model~\cite{marko2019dna} does not explicitly consider the structure of condensin, it implies that an allosteric communication between the hinge and the head - an integral part of our theory and observed in AFM experiments~\cite{eeftens2016condensin}, is not even considered for the LE mechanism~\cite{marko2019dna}.  The lack of conformational changes in response to ATP-binding implies that the substantial decrease in the head-hinge distance by about $\sim 22\nm$ observed in AFM imaging experiments cannot be explained, as was noted previously~\cite{ryu2020condensin}.  (3) In the picture underlying the DNA capture model~\cite{marko2019dna} (referred to as the DNA pumping model elsewhere~\cite{ryu2020condensin}), the distance between the head and the hinge changes very little, if at all. Such a scenario is explicitly ruled out in an experimental study by Ryu et.  al.,~\cite{ryu2020condensin} in part because they seldom observe the I shape in the holocomplex by itself or in association with DNA. For this reason, we believe that the mechanism proposed in the recent simulation study~\cite{nomidis2021dna} is unlikely to be viable. Rather, it is the O$\leftrightarrow$B transition that drives the loop extrusion process, as found in experiments~\cite{ryu2020condensin}, and affirmed here using our simulations. 


{\bf Structural basis for LE:}
The paucity of structures for condensin and cohesin in distinct nucleotide bound states makes it difficult to interpret experiments, theory and simulations in molecular terms.  The situation is further exacerbated because even  the biochemical reaction cycle of condensin (or the related motor cohesin) has not been determined. A recent $8.1\angstrom$ structure of the yeast condensin holocomplex in both the {\it apo} non-engaged state (the one  in which the regulatory element, YCS4, bring the heads in proximity), and the  {\it apo}-bridged state in which the heads interact with each other show a sharp turn in the elbow region, resembling an inverted letter J in the representation  in Fig.1 in a recent study~\cite{lee2020cryo}. The functional importance of the inverted J  state is unclear because when condensin is active (extruding loops in an ATP-dependent manner) only the O- and B- shaped structures are apparently observed~\cite{ryu2020condensin}. Furthermore, the structure of the related motor, cohesin, with DNA shows that the heads are bound to DNA with the hinge is in proximity~\cite{shi2020cryo}, which is inconceivable if the CCs adopt only the I shape.  For this reason, we compared our simulations directly with experiments that have measured the distance changes between the head and the hinge during the {\it active}  LE process~\cite{ryu2020condensin}.  

The partially resolved structure of the ATP-bound state shows a large opening of the CC near the heads~\cite{lee2020cryo}, suggestive of an allosterically driven conformational change.  In contrast, the structure of only the prokaryotic SMC coiled-coil at 3.2 $\angstrom$  resolution, which was created by piecing together several fragments in the CC, showed that it adopts the I shape. As noted elsewhere~\cite{ryu2020condensin}, the I-shaped structure is almost never observed in yeast condensin during its function. The paucity of structures prevents any meaningful inclusion of structural details in theory and simulations. It is for this reason, we resorted to comparisons to AFM imaging data and results from single molecule magnetic tweezer experiments, which have caught the yeast condensin as it executes its function, in validating our theory.  After all it is the function that matters. In the SI (Sec. VI), we performed structural alignment and normal mode analysis using the partially available cryo-EM structures~\cite{lee2020cryo} in order to capture the possible conformational transition in the {\it apo} state that is poised to transition to the LE active state. Even using only partially resolved structures, the normal mode analyses show that ATP binding induces a substantial opening  in the CC region that interacts with the head domains. This preliminary analysis does suggest that for loop extrusion to occur there has to be an allosteric mechanism that brings the head and hinge of the motor close to each other spatially.

{\bf Scrunching versus Translocation:} Using a combination of simulations based on a simple model and theory, we have proposed that LE occurs by a scrunching mechanism. The crux of the scrunching mechanism is that the motor heads, once bound to the DNA, are relatively stationary. Extrusion of the loop occurs by the change in the distance between the head and the hinge by about $\sim 22\nm$. This conformational change is likely driven by ATP binding to condensin~\cite{ryu2020resolving}.  As a result, the head reels in the DNA, with the mean length that could be as large as $\sim 100 \nm$, although the most probable value is $\approx (25-40) \nm$ depending on the external load (Fg. \ref{fig:fit_model}(c) and (d)). Recent experiments have suggested that LE occurs by a scrunching mechanism~\cite{ryu2020resolving}, although it was (as stated earlier) proposed in the context of DNA bubble formation~\cite{kapanidis2006initial}, which is the initial stage in bacterial transcription. The near quantitative agreement with experiments for load-dependent LE velocity and step size distribution shows that the theory and the mechanism are self-consistent. 

In contrast, the other mechanism is based on the picture that condensin must translocate along the DNA in order to extrude loops~\cite{banigan2019limits}. Elsewhere~\cite{ganji2018real} it is argued that the translocation observed in the experiment is an artifact due to salt and buffer conditions, thus casting doubt on the motor translocating along DNA.%If multiple motors are bound to the chromosome, as is surely needed to extrude loops on genomic length scale, in a biologically reasonable  timescale, it is envisioned that a motor translocates bidirectionally until it encounters another motor, which could result in pausing or stoppage of LE~\cite{goloborodko2016compaction}.  It could be argued that  single molecule experiments by Terakawa~\cite{terakawa2017condensin} have provided evidence that condensin motor translocates on DNA without extruding loops. However, in these experiments DNA is fully stretched and serves as a stiff track, much like microtubules for kinesin. {\color{blue}
%Therefore, the motor is forced to walk or translocate on DNA. Elsewhere it is argued that the translocation observed in the experiment is an artifact due to salt and buffer conditions~\cite{ganji2018real}. It is important to incorporate the characteristics found in the scrunching mechanism into future studies especially computational work for genome folding since the microscopic model could affect the resulting 3d structure.   }


% therefore, the motor is forced to walk or translocate on DNA.  Thus, to distinguish between the scrunching and translocation mechanisms, experiments involving both force and single molecule FRET, with dyes on specific locations on DNA and the motor, would be needed as was done sometime ago in a different context~\cite{kapanidis2006initial}. 

{\bf Directionality of LE:}
Directionality is a vital characteristic of biological molecular machines with SMC proteins being no exception. In our model (Fig.~\ref{fig:modelfig}), the unidirectional LE arises during two stages. One is the DNA capture process and the other is the actual active loop extrusion  (State1 to State2 in Fig.~\ref{fig:modelfig}). Directional loop extrusion  emerges as a result of binding  of the SMCs and the associated subunits to DNA via anisotropic interactions. Indeed, the structure of cohesin~\cite{shi2020cryo} suggests that the subunit (STAG1) interacts with the hinge of cohesin by interactions that are anisotropic. This implies that the directional LE could be set by the very act of binding of the condensin motor heads to DNA, which poises the hinge to preferentially interact with  DNA downstream of the motor head, we show schematically as point A in Fig.~\ref{fig:modelfig}.  This in turn ensures  that  the probability of capture of DNA resulting in $\Delta L <0$ is minimized. 

In addition, in our theory there is an asymmetry in the expressions for $k^+$ and $k^-$ in Eq.(4) arising from the free energy, $\mu$, due to the ATP hydrolysis. This process leads to a decrease in the head-hinge distance, as seem in the liquid AFM  images of condensin in action.   However, we believe that, with small probability, $\Delta L<0$ could arise because of the slippage of DNA from the extruded DNA by the strong resistive force on the motor~\cite{ryu2020resolving}. This situation is reminiscent of the slippage of kinesin on microtubule under resistive load~\cite{sudhakar2021germanium}. In the absence of any opposing force, the probability that $\Delta L<0$ is likely to be small.
%{\bf On the origin of directionality in LE:}
%Directionality is a vital characteristic of biological machines with  SMC proteins being no exception. In our model (Fig.~\ref{fig:modelfig}), the directionality has to be met during the two processes. One is the DNA capture process mediated by diffusion, and the other is the extrusion process (State1 to State2). The capture process, predominantly downstream of DNA (site B in Fig.~\ref{fig:modelfig}),  could be accomplished by SMCs and the associated subunits binding to DNA. In other words, binding alone could introduce directionality, as in kinesin. Evidence that this could be the case comes from the structure for cohesin~\cite{shi2020cryo}, which reveals that the subunit (STAG1) interacts with the hinge of cohesin. This implies that the directionality of the subunits interacting with  DNA may dictate the directionality of the hinge,   which in turn, leads to biased diffusion even during the DNA capture process.  In this way, the probability of capture of  DNA upstream (extruded DNA; $L_0$) captured, which would result in $\Delta L <0$,  could  be minimized. 


%The second step, extrusion of DNA, is more subtle. In our theory we incorporated the directionality by the asymmetry for the term $k^+$ and $k^-$ in Eq.~\ref{eq:thermo}, coming from the energy supply by the ATP hydrolysis $\mu$. We surmised this process is realized by the decrease of head-hinge distance, however, the detailed molecular mechanism for the process has yet to be investigated. Especially whether this process utilizes "power stroke" or "Brownian ratchet" is in debate. We extend the discussion of this issue using an example of well-studied molecular machine kinesin-1 in the next section. If negative loop extrusion $\Delta L <0$ were to occur as seen in Ref.~\cite{ryu2020resolving},  the negative extrusion would be related to the extrusion process, not diffusive DNA capturing. However, we speculate that $\Delta L<0$ arises because of the slippage of DNA from the extruded DNA ($L_0$) by the strong resistive force rather than condensin actively reeling away DNA by consuming ATP, reminiscent of the slippage of kinesin on microtubule under resistive load~\cite{sudhakar2021germanium}. 


{\bf Brownian ratchet and power-stroke:}
Whether SMC proteins employ Brownian ratchet mechanism~\cite{astumian1997thermodynamics} or power stroke~\cite{sindelar2002two} is an important question that has to be answered to fully elucidate the molecular mechanism of LE.  A recent study proposed Brownian ratchet model for cohesin~\cite{higashi2021brownian}. An analogy to the well-studied motor conventional kinesin-1 (Kin1), which walks towards the plus end of the stiff microtubule,  is useful. It could be argued that Kin1 makes use of both power stroke and biased diffusion.  Previous studies~\cite{zhang2017parsing,zhang2012dissecting} showed that power stoke (neck-linker docking) propels the trailing head of Kin1 only by $\approx$ 5-6 nm forward, which creates  a strong bias to the next binding site. The rest of the step ($\approx$ 6-8 nm) is completed by diffusion. Therefore, it is possible that both power stroke (sets the directionality)  and Brownian ratchet  are not mutually exclusive. Both the mechanisms could play a role in the LE process as well. Further structural, experimental  and computational studies are  required to resolve fully the interplay between these mechanisms in LE. 



%{\bf Brownian ratchet and power-stroke:}
%Whether SMC proteins employ Brownian ratchet mechanism~\cite{astumian1997thermodynamics} or power stroke~\cite{sindelar2002two} is an important question that has to be answered to fully elucidate the molecular mechanism of LE. A recent study proposed Brownian ratchet model for the cohesin motor based on biochemical study~\cite{higashi2021brownian}. An analogy to the well-studied motor kinesin-1 (Kin1), which walks with substantial probability towards the plus end of the stiff microtubule,  is useful. It could be argued that Kin1 makes use of both power stroke and biased diffusion.  Computational studies~\cite{zhang2017parsing,zhang2012dissecting} showed that power stoke (neck-linker docking) propels the trailing head of Kin1 by $\approx$ 5-6 nm forward and create a strong bias to the next binding site. The rest of the step ($\approx$ 6-8 nm) is completed by diffusion. Therefore, it is possible that both power stroke and power-stroke are not mutually exclusive, and both the mechanisms could play a role in the LE process as well. Further structural, experimental  and computational studies are  required to resolve fully the interplay between these mechanisms in LE. }


\subsection*{Conclusions}
We conclude with a few additional remarks. 
(1) We focused only on one-sided loop extrusion (asymmetric process)  scenario for a single condensin, as demonstrated in the {\it in vitro} experiment~\cite{ganji2018real}.  Whether symmetric LE could occur when more than one condensin loads onto DNA producing Z-loop structures~\cite{kim2020dna}, and if the LE mechanism depends on the species~\cite{kong2020human} is yet to be settled. Similar issues likely exist in loop extrusion mediated by cohesins~\cite{kim2019human,davidson2019dna}. We believe that our work, which only relies on the polymer characteristics of DNA and on an allosteric (action at a distance) mechanism for loop extrusion,  provides a framework for theoretical investigation of LE, accounting for different scenarios. (2) The $f$ dependence of  LE velocity allows us to estimate the time scale for compacting the whole genome.  In particular, if the loop extrusion velocity at $f=0$ is taken to be $\sim$1 kbp/s, we can calculate the LE  time using the following assumptions. The number of condensin I and condensin II that are likely bound to DNA is $\sim$ 3,000 and $\sim$ 500, respectively~\cite{walther2018quantitative}.  Therefore, the loops in the entire chromosome1 ($\sim$ 250 Mbps) could be extruded in a few minutes with the motors operating independently. The assumption that the motors operate independently is reasonable because the linear density (number of motors per genomic base pair) of the bound motors is low. A similar estimate has been made for loop extrusion time by cohesin in the G1 phase of HeLa cells~\cite{davidson2019dna}. These times are faster than the time needed to complete mitosis ($\sim$ an hour)~\cite{gibcus2018pathway}. %With $f$ dependence of $\Omega$ during mitosis $\sim$ an hour may give reasonable upper bound.      
(3) Finally, if LE occurs by scrunching, as gleaned from simulations, and advocated through experimental studies~\cite{ryu2020condensin}, it would imply that the location of the motor is relatively fixed on the DNA and the loop is extruded by ATP-driven shape transitions in the coiled-coils.



% The Discussion should be succinct and must not contain subheadings.

%\section*{Methods}

%Topical subheadings are allowed. Authors must ensure that their Methods section includes adequate experimental and characterization data necessary for others in the field to reproduce their work.

%\bibliography{mybib}
% \begin{thebibliography}{10}
% \urlstyle{rm}
% \expandafter\ifx\csname url\endcsname\relax
%   \def\url#1{\texttt{#1}}\fi
% \expandafter\ifx\csname urlprefix\endcsname\relax\def\urlprefix{URL }\fi
% \expandafter\ifx\csname doiprefix\endcsname\relax\def\doiprefix{DOI: }\fi
% \providecommand{\bibinfo}[2]{#2}
% \providecommand{\eprint}[2][]{\url{#2}}

% \bibitem{flemming1882zellsubstanz}
% \bibinfo{author}{Flemming, W.}
% \newblock \emph{\bibinfo{title}{Zellsubstanz, kern und zelltheilung}}
%   (\bibinfo{publisher}{Vogel}, \bibinfo{year}{1882}).

% \bibitem{alberts2013essential}
% \bibinfo{author}{Alberts, B.} \emph{et~al.}
% \newblock \emph{\bibinfo{title}{Essential cell biology}}
%   (\bibinfo{publisher}{Garland Science}, \bibinfo{year}{2013}).

% \bibitem{hagstrom2003condensin}
% \bibinfo{author}{Hagstrom, K.~A.} \& \bibinfo{author}{Meyer, B.~J.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Condensin and cohesin: more than
%   chromosome compactor and glue}}.
% \newblock {\emph{\JournalTitle{Nature Reviews Genetics}}}
%   \textbf{\bibinfo{volume}{4}}, \bibinfo{pages}{520--534}
%   (\bibinfo{year}{2003}).

% \bibitem{yatskevich2019organization}
% \bibinfo{author}{Yatskevich, S.}, \bibinfo{author}{Rhodes, J.} \&
%   \bibinfo{author}{Nasmyth, K.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Organization of chromosomal {DNA}
%   by {SMC} complexes}}.
% \newblock {\emph{\JournalTitle{Annual Review of Genetics}}}
%   \textbf{\bibinfo{volume}{53}}, \bibinfo{pages}{445--482}
%   (\bibinfo{year}{2019}).

% \bibitem{nasmyth2001disseminating}
% \bibinfo{author}{Nasmyth, K.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Disseminating the genome: joining,
%   resolving, and separating sister chromatids during mitosis and meiosis}}.
% \newblock {\emph{\JournalTitle{Annual Review of Genetics}}}
%   \textbf{\bibinfo{volume}{35}}, \bibinfo{pages}{673--745}
%   (\bibinfo{year}{2001}).

% \bibitem{fudenberg2016formation}
% \bibinfo{author}{Fudenberg, G.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Formation of chromosomal domains by
%   loop extrusion}}.
% \newblock {\emph{\JournalTitle{Cell reports}}} \textbf{\bibinfo{volume}{15}},
%   \bibinfo{pages}{2038--2049} (\bibinfo{year}{2016}).

% \bibitem{sanborn2015chromatin}
% \bibinfo{author}{Sanborn, A.~L.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Chromatin extrusion explains key
%   features of loop and domain formation in wild-type and engineered genomes}}.
% \newblock {\emph{\JournalTitle{Proceedings of the National Academy of
%   Sciences}}} \textbf{\bibinfo{volume}{112}}, \bibinfo{pages}{E6456--E6465}
%   (\bibinfo{year}{2015}).

% \bibitem{terakawa2017condensin}
% \bibinfo{author}{Terakawa, T.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{The condensin complex is a
%   mechanochemical motor that translocates along {DNA}}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{358}},
%   \bibinfo{pages}{672--676} (\bibinfo{year}{2017}).

% \bibitem{ganji2018real}
% \bibinfo{author}{Ganji, M.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Real-time imaging of {DNA} loop
%   extrusion by condensin}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{360}},
%   \bibinfo{pages}{102--105} (\bibinfo{year}{2018}).

% \bibitem{diebold2017structure}
% \bibinfo{author}{Diebold-Durand, M.-L.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Structure of full-length {SMC} and
%   rearrangements required for chromosome organization}}.
% \newblock {\emph{\JournalTitle{Molecular cell}}} \textbf{\bibinfo{volume}{67}},
%   \bibinfo{pages}{334--347} (\bibinfo{year}{2017}).

% \bibitem{buermann2019folded}
% \bibinfo{author}{Buermann, F.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{A folded conformation of {MukBEF}
%   and cohesin}}.
% \newblock {\emph{\JournalTitle{Nature structural \& molecular biology}}}
%   \textbf{\bibinfo{volume}{26}}, \bibinfo{pages}{227} (\bibinfo{year}{2019}).

% \bibitem{alipour2012self}
% \bibinfo{author}{Alipour, E.} \& \bibinfo{author}{Marko, J.~F.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Self-organization of domain
%   structures by {DNA}-loop-extruding enzymes}}.
% \newblock {\emph{\JournalTitle{Nucleic acids research}}}
%   \textbf{\bibinfo{volume}{40}}, \bibinfo{pages}{11202--11212}
%   (\bibinfo{year}{2012}).

% \bibitem{goloborodko2016compaction}
% \bibinfo{author}{Goloborodko, A.}, \bibinfo{author}{Imakaev, M.~V.},
%   \bibinfo{author}{Marko, J.~F.} \& \bibinfo{author}{Mirny, L.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Compaction and segregation of
%   sister chromatids via active loop extrusion}}.
% \newblock {\emph{\JournalTitle{Elife}}} \textbf{\bibinfo{volume}{5}},
%   \bibinfo{pages}{e14864} (\bibinfo{year}{2016}).

% \bibitem{kim2020dna}
% \bibinfo{author}{Kim, E.}, \bibinfo{author}{Kerssemakers, J.},
%   \bibinfo{author}{Shaltiel, I.}, \bibinfo{author}{Haering, C.} \&
%   \bibinfo{author}{Dekker, C.}
% \newblock \bibinfo{journal}{\bibinfo{title}{{DNA}-loop extruding condensin
%   complexes can traverse one another}}.
% \newblock {\emph{\JournalTitle{Biophysical Journal}}}
%   \textbf{\bibinfo{volume}{118}}, \bibinfo{pages}{380a} (\bibinfo{year}{2020}).

% \bibitem{banigan2019limits}
% \bibinfo{author}{Banigan, E.~J.} \& \bibinfo{author}{Mirny, L.~A.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Limits of chromosome compaction by
%   loop-extruding motors}}.
% \newblock {\emph{\JournalTitle{Physical Review X}}}
%   \textbf{\bibinfo{volume}{9}}, \bibinfo{pages}{031007} (\bibinfo{year}{2019}).

% \bibitem{marko2019dna}
% \bibinfo{author}{Marko, J.~F.}, \bibinfo{author}{De~Los~Rios, P.},
%   \bibinfo{author}{Barducci, A.} \& \bibinfo{author}{Gruber, S.}
% \newblock \bibinfo{journal}{\bibinfo{title}{{DNA}-segment-capture model for
%   loop extrusion by structural maintenance of chromosome {(SMC)} protein
%   complexes}}.
% \newblock {\emph{\JournalTitle{Nucleic acids research}}}
%   \textbf{\bibinfo{volume}{47}}, \bibinfo{pages}{6956--6972}
%   (\bibinfo{year}{2019}).

% \bibitem{kapanidis2006initial}
% \bibinfo{author}{Kapanidis, A.~N.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Initial transcription by {RNA}
%   polymerase proceeds through a {DNA}-scrunching mechanism}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{314}},
%   \bibinfo{pages}{1144--1147} (\bibinfo{year}{2006}).

% \bibitem{chen2010promoter}
% \bibinfo{author}{Chen, J.}, \bibinfo{author}{Darst, S.~A.} \&
%   \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Promoter melting triggered by
%   bacterial {RNA} polymerase occurs in three steps}}.
% \newblock {\emph{\JournalTitle{Proceedings of the National Academy of
%   Sciences}}} \textbf{\bibinfo{volume}{107}}, \bibinfo{pages}{12523--12528}
%   (\bibinfo{year}{2010}).

% \bibitem{ryu2020condensin}
% \bibinfo{author}{Ryu, J.-K.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{The condensin holocomplex cycles
%   dynamically between open and collapsed states}}.
% \newblock {\emph{\JournalTitle{Nature structural \& molecular biology}}}
%   \bibinfo{pages}{1--8} (\bibinfo{year}{2020}).

% \bibitem{ryu2020resolving}
% \bibinfo{author}{Ryu, J.-K.}, \bibinfo{author}{Rah, S.-H.},
%   \bibinfo{author}{Janissen, R.}, \bibinfo{author}{Kerssemakers, J.~W.} \&
%   \bibinfo{author}{Dekker, C.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Resolving the step size in
%   condensin-driven {DNA} loop extrusion identifies {ATP} binding as the
%   step-generating process}}.
% \newblock {\emph{\JournalTitle{Available at SSRN 3728949}}}
%   (\bibinfo{year}{2020}).

% \bibitem{shi2020cryo}
% \bibinfo{author}{Shi, Z.}, \bibinfo{author}{Gao, H.}, \bibinfo{author}{Bai,
%   X.-c.} \& \bibinfo{author}{Yu, H.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Cryo-em structure of the human
%   cohesin-{NIPBL}-{DNA} complex}}.
% \newblock {\emph{\JournalTitle{Science}}}  (\bibinfo{year}{2020}).

% \bibitem{chiu2004dna}
% \bibinfo{author}{Chiu, A.}, \bibinfo{author}{Revenkova, E.} \&
%   \bibinfo{author}{Jessberger, R.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Dna interaction and dimerization of
%   eukaryotic smc hinge domains}}.
% \newblock {\emph{\JournalTitle{Journal of Biological Chemistry}}}
%   \textbf{\bibinfo{volume}{279}}, \bibinfo{pages}{26233--26242}
%   (\bibinfo{year}{2004}).

% \bibitem{griese2010structure}
% \bibinfo{author}{Griese, J.~J.}, \bibinfo{author}{Witte, G.} \&
%   \bibinfo{author}{Hopfner, K.-P.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Structure and dna binding activity
%   of the mouse condensin hinge domain highlight common and diverse features of
%   smc proteins}}.
% \newblock {\emph{\JournalTitle{Nucleic acids research}}}
%   \textbf{\bibinfo{volume}{38}}, \bibinfo{pages}{3454--3465}
%   (\bibinfo{year}{2010}).

% \bibitem{alt2017specialized}
% \bibinfo{author}{Alt, A.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Specialized interfaces of smc5/6
%   control hinge stability and dna association}}.
% \newblock {\emph{\JournalTitle{Nature communications}}}
%   \textbf{\bibinfo{volume}{8}}, \bibinfo{pages}{1--14} (\bibinfo{year}{2017}).

% \bibitem{lawrimore2017rotostep}
% \bibinfo{author}{Lawrimore, J.}, \bibinfo{author}{Friedman, B.},
%   \bibinfo{author}{Doshi, A.} \& \bibinfo{author}{Bloom, K.}
% \newblock \bibinfo{title}{Rotostep: a chromosome dynamics simulator reveals
%   mechanisms of loop extrusion}.
% \newblock In \emph{\bibinfo{booktitle}{Cold Spring Harbor symposia on
%   quantitative biology}}, vol.~\bibinfo{volume}{82}, \bibinfo{pages}{101--109}
%   (\bibinfo{organization}{Cold Spring Harbor Laboratory Press},
%   \bibinfo{year}{2017}).

% \bibitem{hyeon2006kinetics}
% \bibinfo{author}{Hyeon, C.} \& \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Kinetics of interior loop formation
%   in semiflexible chains}}.
% \newblock {\emph{\JournalTitle{The Journal of Chemical Physics}}}
%   \textbf{\bibinfo{volume}{124}}, \bibinfo{pages}{104905}
%   (\bibinfo{year}{2006}).

% \bibitem{wilhelm1996radial}
% \bibinfo{author}{Wilhelm, J.} \& \bibinfo{author}{Frey, E.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Radial distribution function of
%   semiflexible polymers}}.
% \newblock {\emph{\JournalTitle{Physical review letters}}}
%   \textbf{\bibinfo{volume}{77}}, \bibinfo{pages}{2581} (\bibinfo{year}{1996}).

% \bibitem{thirumalai1999time}
% \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Time scales for the formation of
%   the most probable tertiary contacts in proteins with applications to
%   cytochrome c}}.
% \newblock {\emph{\JournalTitle{The Journal of Physical Chemistry B}}}
%   \textbf{\bibinfo{volume}{103}}, \bibinfo{pages}{608--610}
%   (\bibinfo{year}{1999}).

% \bibitem{seifert2005fluctuation}
% \bibinfo{author}{Seifert, U.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Fluctuation theorem for a single
%   enzym or molecular motor}}.
% \newblock {\emph{\JournalTitle{EPL {(Europhysics Letters)}}}}
%   \textbf{\bibinfo{volume}{70}}, \bibinfo{pages}{36} (\bibinfo{year}{2005}).

% \bibitem{seifert2012stochastic}
% \bibinfo{author}{Seifert, U.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Stochastic thermodynamics,
%   fluctuation theorems and molecular machines}}.
% \newblock {\emph{\JournalTitle{Reports on progress in physics}}}
%   \textbf{\bibinfo{volume}{75}}, \bibinfo{pages}{126001}
%   (\bibinfo{year}{2012}).

% \bibitem{mugnai2020theoretical}
% \bibinfo{author}{Mugnai, M.~L.}, \bibinfo{author}{Hyeon, C.},
%   \bibinfo{author}{Hinczewski, M.} \& \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Theoretical perspectives on
%   biological machines}}.
% \newblock {\emph{\JournalTitle{Reviews of Modern Physics}}}
%   \textbf{\bibinfo{volume}{92}}, \bibinfo{pages}{025001}
%   (\bibinfo{year}{2020}).

% \bibitem{marko1995stretching}
% \bibinfo{author}{Marko, J.~F.} \& \bibinfo{author}{Siggia, E.~D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Stretching {DNA}}}.
% \newblock {\emph{\JournalTitle{Macromolecules}}} \textbf{\bibinfo{volume}{28}},
%   \bibinfo{pages}{8759--8770} (\bibinfo{year}{1995}).

% \bibitem{rubinstein2003polymer}
% \bibinfo{author}{Rubinstein, M.}, \bibinfo{author}{Colby, R.~H.} \emph{et~al.}
% \newblock \emph{\bibinfo{title}{Polymer physics}}, vol.~\bibinfo{volume}{23}
%   (\bibinfo{publisher}{Oxford university press New York},
%   \bibinfo{year}{2003}).

% \bibitem{strick2004real}
% \bibinfo{author}{Strick, T.~R.}, \bibinfo{author}{Kawaguchi, T.} \&
%   \bibinfo{author}{Hirano, T.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Real-time detection of
%   single-molecule {DNA} compaction by condensin {I}}}.
% \newblock {\emph{\JournalTitle{Current biology}}}
%   \textbf{\bibinfo{volume}{14}}, \bibinfo{pages}{874--880}
%   (\bibinfo{year}{2004}).

% \bibitem{eeftens2016condensin}
% \bibinfo{author}{Eeftens, J.~M.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Condensin {Smc2-Smc4} dimers are
%   flexible and dynamic}}.
% \newblock {\emph{\JournalTitle{Cell reports}}} \textbf{\bibinfo{volume}{14}},
%   \bibinfo{pages}{1813--1818} (\bibinfo{year}{2016}).

% \bibitem{muir2020structure}
% \bibinfo{author}{Muir, K.~W.}, \bibinfo{author}{Li, Y.}, \bibinfo{author}{Weis,
%   F.} \& \bibinfo{author}{Panne, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{The structure of the cohesin
%   {ATP}ase elucidates the mechanism of {SMC}--{kleisin} ring opening}}.
% \newblock {\emph{\JournalTitle{Nature Structural \& Molecular Biology}}}
%   \textbf{\bibinfo{volume}{27}}, \bibinfo{pages}{233--239}
%   (\bibinfo{year}{2020}).

% \bibitem{hinczewski2013design}
% \bibinfo{author}{Hinczewski, M.}, \bibinfo{author}{Tehver, R.} \&
%   \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Design principles governing the
%   motility of myosin {V}}}.
% \newblock {\emph{\JournalTitle{Proceedings of the National Academy of
%   Sciences}}} \textbf{\bibinfo{volume}{110}}, \bibinfo{pages}{E4059--E4068}
%   (\bibinfo{year}{2013}).

% \bibitem{dey2017toroidal}
% \bibinfo{author}{Dey, A.} \& \bibinfo{author}{Reddy, G.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Toroidal condensates by
%   semiflexible polymer chains: Insights into nucleation, growth and packing
%   defects}}.
% \newblock {\emph{\JournalTitle{The Journal of Physical Chemistry B}}}
%   \textbf{\bibinfo{volume}{121}}, \bibinfo{pages}{9291--9301}
%   (\bibinfo{year}{2017}).

% \bibitem{kolomeisky2007molecular}
% \bibinfo{author}{Kolomeisky, A.~B.} \& \bibinfo{author}{Fisher, M.~E.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Molecular motors: a theorist's
%   perspective}}.
% \newblock {\emph{\JournalTitle{Annu. Rev. Phys. Chem.}}}
%   \textbf{\bibinfo{volume}{58}}, \bibinfo{pages}{675--695}
%   (\bibinfo{year}{2007}).

% \bibitem{lee2020cryo}
% \bibinfo{author}{Lee, B.-G.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Cryo-em structures of holo
%   condensin reveal a subunit flip-flop mechanism}}.
% \newblock {\emph{\JournalTitle{Nature Structural \& Molecular Biology}}}
%   \textbf{\bibinfo{volume}{27}}, \bibinfo{pages}{743--751}
%   (\bibinfo{year}{2020}).

% \bibitem{sudhakar2021germanium}
% \bibinfo{author}{Sudhakar, S.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Germanium nanospheres for
%   ultraresolution picotensiometry of kinesin motors}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{371}}
%   (\bibinfo{year}{2021}).

% \bibitem{astumian1997thermodynamics}
% \bibinfo{author}{Astumian, R.~D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Thermodynamics and kinetics of a
%   brownian motor}}.
% \newblock {\emph{\JournalTitle{science}}} \textbf{\bibinfo{volume}{276}},
%   \bibinfo{pages}{917--922} (\bibinfo{year}{1997}).

% \bibitem{sindelar2002two}
% \bibinfo{author}{Sindelar, C.~V.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Two conformations in the human
%   kinesin power stroke defined by x-ray crystallography and epr spectroscopy}}.
% \newblock {\emph{\JournalTitle{Nature structural biology}}}
%   \textbf{\bibinfo{volume}{9}}, \bibinfo{pages}{844--848}
%   (\bibinfo{year}{2002}).

% \bibitem{higashi2021brownian}
% \bibinfo{author}{Higashi, T.~L.}, \bibinfo{author}{Tang, M.},
%   \bibinfo{author}{Pobegalov, G.}, \bibinfo{author}{Uhlmann, F.} \&
%   \bibinfo{author}{Molodtsov, M.}
% \newblock \bibinfo{journal}{\bibinfo{title}{A brownian ratchet model for dna
%   loop extrusion by the cohesin complex}}.
% \newblock {\emph{\JournalTitle{bioRxiv}}}  (\bibinfo{year}{2021}).

% \bibitem{zhang2017parsing}
% \bibinfo{author}{Zhang, Z.}, \bibinfo{author}{Goldtzvik, Y.} \&
%   \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Parsing the roles of neck-linker
%   docking and tethered head diffusion in the stepping dynamics of kinesin}}.
% \newblock {\emph{\JournalTitle{Proceedings of the National Academy of
%   Sciences}}} \textbf{\bibinfo{volume}{114}}, \bibinfo{pages}{E9838--E9845}
%   (\bibinfo{year}{2017}).

% \bibitem{zhang2012dissecting}
% \bibinfo{author}{Zhang, Z.} \& \bibinfo{author}{Thirumalai, D.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Dissecting the kinematics of the
%   kinesin step}}.
% \newblock {\emph{\JournalTitle{Structure}}} \textbf{\bibinfo{volume}{20}},
%   \bibinfo{pages}{628--640} (\bibinfo{year}{2012}).

% \bibitem{kong2020human}
% \bibinfo{author}{Kong, M.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Human condensin {I} and {II} drive
%   extensive {ATP}-dependent compaction of nucleosome-bound {DNA}}}.
% \newblock {\emph{\JournalTitle{Molecular cell}}} \textbf{\bibinfo{volume}{79}},
%   \bibinfo{pages}{99--114} (\bibinfo{year}{2020}).

% \bibitem{kim2019human}
% \bibinfo{author}{Kim, Y.}, \bibinfo{author}{Shi, Z.}, \bibinfo{author}{Zhang,
%   H.}, \bibinfo{author}{Finkelstein, I.~J.} \& \bibinfo{author}{Yu, H.}
% \newblock \bibinfo{journal}{\bibinfo{title}{Human cohesin compacts {DNA} by
%   loop extrusion}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{366}},
%   \bibinfo{pages}{1345--1349} (\bibinfo{year}{2019}).

% \bibitem{davidson2019dna}
% \bibinfo{author}{Davidson, I.~F.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{{DNA} loop extrusion by human
%   cohesin}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{366}},
%   \bibinfo{pages}{1338--1345} (\bibinfo{year}{2019}).

% \bibitem{walther2018quantitative}
% \bibinfo{author}{Walther, N.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{A quantitative map of human
%   condensins provides new insights into mitotic chromosome architecture}}.
% \newblock {\emph{\JournalTitle{Journal of Cell Biology}}}
%   \textbf{\bibinfo{volume}{217}}, \bibinfo{pages}{2309--2328}
%   (\bibinfo{year}{2018}).

% \bibitem{gibcus2018pathway}
% \bibinfo{author}{Gibcus, J.~H.} \emph{et~al.}
% \newblock \bibinfo{journal}{\bibinfo{title}{A pathway for mitotic chromosome
%   formation}}.
% \newblock {\emph{\JournalTitle{Science}}} \textbf{\bibinfo{volume}{359}},
%   \bibinfo{pages}{eaao6135} (\bibinfo{year}{2018}).

% \end{thebibliography}
\bibliography{mybib.bib}

%\noindent LaTeX formats citations and references automatically using the bibliography records in your .bib file, which you can edit via the project menu. Use the cite command for an inline citation, e.g.  \cite{Hao:gidmaps:2014}.

%For data citations of datasets uploaded to e.g. \emph{figshare}, please use the \verb|howpublished| option in the bib entry to specify the platform and the link, as in the \verb|Hao:gidmaps:2014| example in the sample bibliography file.

\section*{Acknowledgements}
We thank Rasika Harshey, Changbong Hyeon, Mauro Mugnai, and Johannes Stigler for useful comments and discussions. We are grateful to John Marko for clarifying some aspects of his model.  This work was supported by NSF (CHE 19-00093), NIH (GM - 107703) and the Welch Foundation Grant F-0019 through the Collie-Welch chair. 

\section*{Author contributions statement}


R.T and D.T. conceived the theory,  R.T.,  A.D., G.S., and D.T performed the calculations and simulations, the experiment(s), R.T., A.D., FG. S., and D. T.  analyzed the results.  All authors reviewed the manuscript. 

%\section*{Additional information}

% To include, in this order: \textbf{Accession codes} (where applicable); \textbf{Competing interests} (mandatory statement). 

% The corresponding author is responsible for submitting a \href{http://www.nature.com/srep/policies/index.html#competing}{competing interests statement} on behalf of all authors of the paper. This statement must be included in the submitted article file.

% \begin{figure}[ht]
% \centering
% \includegraphics[width=\linewidth]{stream}
% \caption{Legend (350 words max). Example legend text.}
% \label{fig:stream}
% \end{figure}

% \begin{table}[ht]
% \centering
% \begin{tabular}{|l|l|l|}
% \hline
% Condition & n & p \\
% \hline
% A & 5 & 0.1 \\
% \hline
% B & 10 & 0.01 \\
% \hline
% \end{tabular}
% \caption{\label{tab:example}Legend (350 words max). Example legend text.}
% \end{table}

% Figures and tables can be referenced in LaTeX using the ref command, e.g. Figure \ref{fig:stream} and Table \ref{tab:example}.

\end{document}