%% ****** Start of file apstemplate.tex ****** %
%%
%%
%%   This file is part of the APS files in the REVTeX 4 distribution.
%%   Version 4.1r of REVTeX, August 2010
%%
%%
%%   Copyright (c) 2001, 2009, 2010 The American Physical Society.
%%
%%   See the REVTeX 4 README file for restrictions and more information.
%%
%
% This is a template for producing manuscripts for use with REVTEX 4.0
% Copy this file to another name and then work on that file.
% That way, you always have this original template file to use.
%
% Group addresses by affiliation; use superscriptaddress for long
% author lists, or if there are many overlapping affiliations.
% For Phys. Rev. appearance, change preprint to twocolumn.
% Choose pra, prb, prc, prd, pre, prl, prstab, prstper, or rmp for journal
%  Add 'draft' option to mark overfull boxes with black boxes
%  Add 'showpacs' option to make PACS codes appear
%  Add 'showkeys' option to make keywords appear
%\documentclass[aps,prl,preprint,unsortedaddress]{revtex4-1}
\documentclass[aps,preprint]{revtex4-1}
%\documentclass[journal=jacsat,manuscript=article]{achemso}
%\documentclass[onecolumn,showpacs,superscriptaddress,groupedaddress]{revtex4}
%\documentclass[aps,prl,preprint,superscriptaddress]{revtex4-1}
%\documentclass[aps,prl,reprint,groupedaddress]{revtex4-1}
\usepackage[dvipdfmx]{graphicx}
%\usepackage{graphicx}
\usepackage{color,hyperref,lineno,setspace,amsmath,epstopdf}
\usepackage{amsmath,amsfonts,amsthm} % Math packages
\usepackage{amssymb}
% You should use BibTeX and apsrev.bst for references
% Choosing a journal automatically selects the correct APS
% BibTeX style file (bst file), so only uncomment the line
% below if necessary.
%\bibliographystyle{apsrev4-1}

\newcommand{\angstrom}{\textup{\AA}}
\newcommand{\aveV}{\overline{V}}
\newcommand{\aveL}{\overline{L}}
\newcommand{\pN}{\ \mathrm{pN}}
\newcommand{\nm}{\ \mathrm{nm}}
\newcommand{\bp}{\ \mathrm{bp}}
\newcommand{\bps}{\ \mathrm{bps}}
\newcommand{\persec}{\ \mathrm{s}^{-1}}
\newcommand{\steps}{\mathrm{steps}}
\newcommand{\mMol}{\mathrm{mM}}
\newcommand{\uMol}{\mathrm{\mu M}}
\newcommand{\kp}{k^+}
\newcommand{\km}{k^-}
\newcommand{\Jpm}{J^{\pm}}
\newcommand{\Jp}{J^+}
\newcommand{\Jm}{J^-}
\newcommand{\Jgamma}{J^{\gamma}}
\newcommand{\T}{\text{[T]}}
\newcommand{\DRI}{\Delta R_I}
\newcommand{\DRF}{\Delta R_F}
%\newcommand{\vR}{\mathcal{R}}
%\newcommand{\vL}{\mathcal{L}}
%\newcommand{\DR}{\Delta {R}}
%\newcommand{\Dl}{\Delta {l}}

\newcommand{\vR}{R}
\newcommand{\vL}{L}
\newcommand{\DR}{\Delta R}
\newcommand{\Dl}{\Delta l}
%\setcounter{tocdepth}{1} % Show sections
%\setcounter{tocdepth}{2} % + subsections
%\setcounter{tocdepth}{3} % + subsubsections
%\setcounter{tocdepth}{4} % + paragraphs
%\setcounter{tocdepth}{5} % + subparagraphs
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Link to SI
\usepackage{xcite}

\usepackage{xr}
\makeatletter
\newcommand*{\addFileDependency}[1]{% argument=file name and extension
  \typeout{(#1)}% latexmk will find this if $recorder=0 (however, in that case, it will ignore #1 if it is a .aux or .pdf file etc and it exists! if it doesn't exist, it will appear in the list of dependents regardless)
  \@addtofilelist{#1}% if you want it to appear in \listfiles, not really necessary and latexmk doesn't use this
  \IfFileExists{#1}{}{\typeout{No file #1.}}% latexmk will find this message if #1 doesn't exist (yet)
}
\makeatother

\newcommand*{\myexternaldocument}[1]{%
    \externaldocument{#1}%
    \addFileDependency{#1.tex}%
    \addFileDependency{#1.aux}%
}
%%% END HELPER CODE

% put all the external documents here!
\myexternaldocument{main_v2}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\externaldocument{main_v2}
\usepackage{here}
% S before figure/table/equation
\renewcommand{\theequation}{S\arabic{equation}}
\renewcommand{\thefigure}{S\arabic{figure}}
\renewcommand{\thetable}{S\arabic{table}}
 \newcommand{\rchem}{\bar{r}_{C}}

\begin{document}

% Use the \preprint command to place your local institutional report
% number in the upper righthand corner of the title page in preprint mode.
% Multiple \preprint commands are allowed.
% Use the 'preprintnumbers' class option to override journal defaults
% to display numbers if necessary
%\preprint{}

%Title of paper
\title{Supplementary Information \\Theory and Simulations of condensin mediated loop extrusion in DNA}

% repeat the \author .. \affiliation  etc. as needed
% \email, \thanks, \homepage, \altaffiliation all apply to the current
% author. Explanatory text should go in the []'s, actual e-mail
% address or url should go in the {}'s for \email and \homepage.
% Please use the appropriate macro foreach each type of information

% \affiliation command applies to all authors since the last
% \affiliation command. The \affiliation command should follow the
% other information
% \affiliation can be followed by \email, \homepage, \thanks as well.




\author{Ryota Takaki}
\affiliation{Department of Physics, The university of Texas at Austin}%Lines break automatically or can be forced with \\
\author{Atreya Dey}
\affiliation{Department of Chemistry, The university of Texas at Austin}
\author{Guang Shi}
\affiliation{Department of Chemistry, The university of Texas at Austin}
\author{D. Thirumalai}

\affiliation{Department of Chemistry, The university of Texas at Austin}
%Collaboration name if desired (requires use of superscriptaddress
%option in \documentclass). \noaffiliation is required (may also be
%used with the \author command).
%\collaboration can be followed by \email, \homepage, \thanks as well.
%\collaboration{}
%\noaffiliation

\date{\today}

%\begin{abstract}
% insert abstract here
%\end{abstract}

% insert suggested PACS numbers in braces on next line
%\pacs{}
% insert suggested keywords - APS authors don't need to do this
%\keywords{}

%\maketitle must follow title, authors, abstract, \pacs, and \keywords
\maketitle

% body of paper here - Use proper section commands
% References should be done using the \cite, \ref, and \label commands
\tableofcontents
\newpage
% Because we cover a number of topics in the Supplementary Information, we provide a road map. We begin by describing the two different waiting state for ATP binding. Expression for the forward, backward, and detachment rate as a function of [T] are derived. These result are used to derive the run length and velocity distribution function for both the kinetic scheme. Analytic expression for the mechanical and chemical randomness parameter, whose [T] dependence we propose could be used to distinguish the two model, are then provided. The procedure to obtain the parameters and additional results are given in the final section.


\section{Simulations}
\begin{figure}[]
\centering
\includegraphics[width=1\textwidth]{SI_simulation.pdf}
\caption{\label{fig:simulation} Cartoon representation of the condensin motor based in part on partially resolved structures and FM images.  The simulation model is based on this representation. The two heads are shown as red spheres. A magnified image of the angle between the motor heads at the junctions to the two arms of the SMC is shown in the lower box. The angle at the elbow is depicted in the upper box. The coiled coils (CCs) connecting the motor to the hinge (purple sphere) are treated as a semi-flexible polymers that are kinked  at the flexible elbow region. We envision that the allosteric transition between the open and the closed states (shown as by the green arrow) is driven by ATP binding to the motor domains. 
}
\end{figure}
The purpose of the simulations using a simplified model is to show that the extracted parameter values, obtained by fitting the theoretical extrusion rate or equivalently the velocity of loop extrusion (LE) to experiments, are reasonable. In particular, we use simulations to argue that the value of $\Delta R \approx$ 26 nm or equivalently $\sim$ 76 bps (see the main text for details) is consistent with experiments that have imaged the shape transitions during the LE process. To this end, we imagine that during the ATPase cycle, the SMC motor undergoes a conformational change from an "open" (top structure in Fig. \ref{fig:simulation}) to a "closed"  (bottom structure) state, thus decreasing  the distance between the motor domains and the hinge region. We theorize that this process is allosterically driven, in a manner similar to other cargo-carrying  motors (myosins, kinesins and dynein), by binding ATP. We envision that in the SMC the allosteric transitions are effectuated through the movement of the flexible elbow region. Such a picture is consistent with high speed AFM imaging~\cite{eeftens2016condensin} and measurements of head-hinge distance as the motor transitions between the two active (open and closed) states~\cite{ryu2020condensin}. 

\subsection{Model for Condensin \label{sec:model}}
We modeled the two heads of condensin as spheres that are connected by finitely-extensible nonlinear elastic (FENE) potential~\cite{kremer1991erratum} to the coiled coils (CCs) that connect the motor domains to the hinge (Fig.\ref{fig:simulation}). The CC in the SMCs are reminiscent of the lever arm in Myosin V. The angle $\theta_1$ and $\theta_2$ are formed at the junctions connecting the motor heads to the first bead on the CCs (Fig.\ref{fig:simulation}).  As in the well-studied molecular motors, a  change in the conformation  initiated in the head domain, is  amplified to the rest of the motor through the CCs.  We envision this process as the principle mechanism by which a spool (roughly $\frac{\Delta R}{0.34}=$ 76 base pairs (bps)  in a single step) of DNA could be extruded. 

We used 19 and 18 beads for upper CC and lower CC, respectively. The diameter of each bead is $1 \nm$ diameter. We used 3 beads (diameter $0.4\nm$ each) in the middle of the CCs for the elbow region. The strength of the angle potential in the elbow region, marking the break in an otherwise stiff CC, is chosen to facilitate the allosteric propagation of conformational changes in the motor head. For the hinge and two motor heads we used $4\nm$ diameter beads. 

All the lengths are measured in units of $\sigma=1\nm$, corresponding to the diameter of the beads in the CC. We express energy in  $k_BT$ units, where $k_B$ is the Boltzmann constant and $T$ is the temperature. The mass of all the particles were set to $m=1$. We performed low-friction Langevin dynamics simulations using OpenMM \cite{eastman2017openmm} software using a  time-step of $\Delta t_L=0.01\tau_L$, where $\tau_L=0.4\sqrt{m\sigma^2/k_BT}$. The value of the friction coefficient is $0.01/\tau_L$. %The friction coefficient was chosen to be as low as possible for a stable simulation. 

\subsection{Energy function} Because our goal is to merely illustrate that the proposed allosteric mechanism for SMC-mediated LE is plausible, we chose a simple energy function to monitor the conformational changes in condensin. The explicit form of the energy function is,
\begin{align} 
\begin{split}
\label{eq:Hamil}
%H(\Vec{r}_1,\Vec{r}_2,...,\Vec{r}_N)
E(\Vec{r}_1,\Vec{r}_2,...,\Vec{r}_N, \vec{\phi},\vec{\theta})  &= \sum_{i=1}^{N-1}U_{FENE}(r_{i,i+1}) + \sum_{i\neq j}^{N}U_{N}(r_{i,j}) +\Big( \sum_{i\in CC\neq El}U^{CC }_{ANG}(\phi_{i}) + \sum_{i\in El}U^{El}_{ANG}(\phi_{i}) \Big) \\
& + \sum_{i\in Head}^{}U_{CNF}^{}(\theta_{i}) .
\end{split}  
\end{align}
The first term in Eq.(\ref{eq:Hamil}) enforces the connectivity of the beads and is given by,
\begin{align} 
\begin{split}
\label{}
U_{FENE}(r_{i,i+1}) = -\frac{1}{2}k_F R_F^2 \log\Big[1-\frac{(r_{i,i+1}-r^0_{i,i+1})^2}{R_F^2}\Big],
\end{split}  
\end{align}
where $k_F$ is the stiffness of the potential, $R_F$ is the upper bound for the displacement, and $r^0_{i,i+1}$ is the equilibrium distance between the beads, $i$ and $i+1$.
The second term in Eq.(\ref{eq:Hamil}), accounting for excluded volume interactions, is given by,
\begin{equation} 
%\begin{split}
\label{}
U_{N}(r_{i,j})  =\epsilon_{N}\Big (\frac{\sigma}{r_{i,j}}\Big )^{12},
%\end{split}  
\end{equation}
where $\epsilon_{N}$ and $\sigma$  are the strength and range of the interaction, respectively. We used additive interactions, which means that $\sigma$ is  the sum of the radii of the two interacting beads. %{\bf NOTE I USE phi for one angle potential to avoid confusion}.
The third and the fourth terms in Eq.(\ref{eq:Hamil}) are the two angle potentials that control the bending stiffness of the CCs. The potential, $U^{}_{ANG}(\phi_{i})$, is taken to be, 
\begin{equation} 
%\begin{split}
\label{}
U^{}_{ANG}(\phi_{i})  =\epsilon _b(1+\cos \phi_i),
%\end{split}  
\end{equation}
where $\epsilon_b$,  the energy scale for bending, is related to the persistence length of the semi-flexible CC.
We used a smaller value of $\epsilon_b$ for $U^{El}_{ANG}(\epsilon_b^{El})$ to model the difference in the stiffness between the elbow region, and the rest of the CC. %Because the persistence of the CC, $l_p^{CC}$, is not known we varied $\epsilon_b^{CC}$ to cover a range of plausible values of $l_p^{CC}$ (see Table~\ref{Table:parameters}).
%{\bf RT what are the two different $\epsilon$ values? They should be listed in the Table as two rows. Variable is not enough}

The last term in Eq.(\ref{eq:Hamil}) models the conformation change in the motor head  due to ATP binding, and is taken as, 
\begin{equation} 
%\begin{split}
\label{}
U_{CNF}^{}(\theta_{i})= \frac{1}{2}k_C\big(\theta^0_i-\theta_i\big)^2,
%\end{split}  
\end{equation}
where $k_C$ is the spring constant for the potential, and $\theta^0_i$ is the equilibrium angle for the angle potential. Before the conformational change, we set $\theta_i^0=2.4$ (radian) in the open state, which is roughly the angle calculated by ATP engaged state of  prokaryotic SMC~\cite{diebold2017structure}. Because the structure for the closed state is unavailable, we chose $\theta_i^0=4.0$ (radian) for the closed state, which leads to $\theta_i\sim \pi$ (radian) in equilibrium. The transition between the open and closed states results in the scrunching of the DNA (the nearly stationary motor reels in the DNA), and extrusion of the loop.
The parameter values in the energy function used in the simulations are in Table~\ref{Table:parameters}. 
%Because of the large difference between $\epsilon_N$ and $k_C$, small changes in the motor head are greatly exaggerated across the SMC, which would represent a power stroke that is characteristic of most motors.


\subsection{Simulation with DNA}
We used a coarse-grained bead-spring model for DNA~\cite{hyeon2006kinetics,dey2017toroidal}. Each bead represents 10 base-pairs, which implies that the bead size is $\sigma_{DNA}=3.4 \nm$. The chain has $N=100$ beads or 1,000 base-pairs. Consecutive beads along the chain are  connected by a FENE potential (Eq.(S2)) with $r^0_{i,i+1}=\sigma_{DNA}$. The DNA stiffness is modeled using a harmonic bending potential given by,
\begin{equation}
    U_{BEND} = \frac{1}{2}\sum_{i=1}^{N_{ang}}\alpha\cdot\theta^2,
\end{equation}
where $N_{ang}=98$ is the number of bond angles, $\theta_i$ is the deviation of the $i^{th}$ bond-angle in the chain from $180\deg$, and $\alpha = 15.55$ $(k_B T /\text{rad}^2)$ is a constant~\cite{hyeon2006kinetics}. We chose  $\alpha$ so that the persistence length is $\approx 50\nm$, which is the canonical value for DNA in monovalent salts.   %We leave the detailed structural analysis leading to the extrusion process to future work.  Nevertheless, the physical basis of DNA capture and the distribution of capture length are in accord with the currently available structures of condensin.

\begin{table}[t]
\begin{center}
%\begin{ruledtabular}
  \begin{tabular}{|c | c|}
    \hline
    \text{Parameter} &\text{Value}\\
    \hline \hline
    $k_F$      &$50 (k_BT/\mathrm{nm}^2)$\\ 
    $R_F$    &$1.5 (\mathrm{nm})$\\
    $\epsilon_b^{CC}$ &$150 (k_BT)$\\
    %$ $ &$150 (k_BT)^*$\\
    $\epsilon_b^{El}$ &$2 (k_BT)^*$\\
    $\epsilon_{N}$    &$5(k_BT)$\\
    $k_C$&$200(k_BT/{\text{rad}}^2)$\\
    \hline
  \end{tabular}
%\end{ruledtabular}
\end{center} 
\caption{\label{Table:parameters}Parameters for the molecular dynamics simulation. $^*$ The dependence of the distance distribution between the head and hinge during one cycle on   $\epsilon_b^{El}$ is given in Fig. \ref{fig:dists_lp}.}
\end{table}

% \section{Persistence length of the Coiled-coil\label{sec:persistence}}
% \begin{figure}[]
% \centering
% \includegraphics[width=0.7\textwidth]{lp_Toan.pdf}
% \caption{\label{fig:persistence}Persistence length ($l_p^{CC}$) as a function of bending stiffness ($\epsilon_b^{CC}$). Red circles; $l_p^{CC}$ calculated using end-to-end distance. Blue circles; $l_p^{CC}$ calculated using the correlation of tangent angle used in Toan and Thirumalai~\cite{toan2012origin}. Green squares are for a semi-flexible polymer {\it without} flexible elbow. %$l_b\epsilon_b^{CC}/k_BT$, where $l_b$ is equilibrium bond length. 
% }
% \end{figure}

% % \begin{figure}[H]
% % \centering
% % \includegraphics[width=1\textwidth]{DelR_with:wo_elbow.pdf}
% % \caption{\label{fig:DelR_comparison}(a) $\Delta R_s$ as a function of $l_p^{CC}$. ($l_p^{CC}$ is calculated for a semi-flexible polymer {\it without} flexible elbow.) (b) $\Delta R_s$ as a function of effective $l_p^{CC}$. ($l_p^{CC}$ is calculated for a semi-flexible polymer {\it with} flexible elbow.)
% % }
% % \end{figure}

% We calculated the persistence length ($l_p^{CC}$) for coiled-coil that suffices using simulations for a single isolated semi-flexible polymer with the elbow. The single polymer has 40 beads in total with 3 flexible beads, which provides a coarse-grained  description of the CC for condensin (Sec.\ref{sec:model}). We set $\epsilon_b^{El}=4(k_BT)$ and varied $\epsilon_b^{CC}$ (see Table.\ref{Table:parameters}).
% Since the CC in our model has flexible kink at the elbow, we computed the effective persistence length using end-to-end distance of the semi-flexible polymer. Namely, we obtained contour length ($L$) and end-to-end distance ($R$) from simulations then numerically solved $<R^2>=2l_pL\big(1-\frac{l_p}{L}(1-e^{-l/l_p})\big)$~\cite{kratky1949rontgenuntersuchung} for $l_p$. We also computed the persistence length from the decay of tangent angle correlation used previously following a previous study on DNA~\cite{toan2012origin}. We found that the two methods gives consistent values for $l_p$ (see Fig.\ref{fig:persistence}). 

% Fig.\ref{fig:persistence} shows the persistence length of the CC as a function of $\epsilon_b^{CC}$. Without the flexible elbow, the persistence length is well approximated by $l_p=l_b\epsilon_b/(k_BT)$, where $l_b\sim1.3\nm$ is mean bond length, as described elsewhere~\cite{benkova2017structural,li2015modeling,midya2019phase}. In the presence of the kink at the elbow the effective persistence length is smaller.
   
% \section{Distribution of head-hinge distance}
% \begin{figure}[]
% \centering
% \includegraphics[width=\textwidth]{SI_distributions.pdf}
% \caption{\label{fig:distributions} Distributions of $R$ for various $l_p^{CC}$. $R_1$ and $R_2$ are mean value of head-hinge distance for open state and closed state, respectively. The distributions are calculated from 50 trajectories, 40,000 time points. (a) For $l_p^{CC} \sim 5 \nm$, $\Delta R_s=1\pm8 \nm$, where error is calculated using standard deviation of the distributions. (b) For $l_p^{CC}\sim 40 \nm$, $\Delta R_s=11\pm9 \nm$. (c) For $l_p^{CC}\sim 60 \nm$, $\Delta R_s=16\pm8 \nm$. (d) For $l_p^{CC}\sim 90 \nm$ $(\epsilon_b^{CC}=300 k_BT)$, $\Delta R_s=25\pm 6 \nm$.
% }
% \end{figure}
% \begin{figure}[]
% \centering
% \includegraphics[width=0.6\textwidth]{SI_DeltaR.pdf}
% \caption{\label{fig:DeltaR} Difference of head-hinge distance between the open and closed states ($\Delta R_s=R_1-R_2$) as a function of $l_p^{CC}$.
% Each point is calculated from 5 trajectories (4000 time points). Error bars are mostly smaller than the plot marker.}
% \end{figure}
% In Fig.5 in the main text we show the distribution of head-hinge distance for $l_p^{CC} \sim 70\nm$. Here we list distributions for different values of $l_p^{CC}$ (Fig.\ref{fig:distributions}). We also plot the mean value of the change of head-hinge distance between open and closed state ($\Delta R_s=R_1-R_2$) as a function of $l_p^{CC}$  (Fig.\ref{fig:DeltaR}). For $l_p^{CC}\sim 5 \nm$ (Fig.\ref{fig:distributions}(a)), we find the two distributions for open state and closed state overlaps almost completely. This is because the persistence length of CC is too small to propagate the conformational change initiated at the head domain. 
% As $l_p^{CC}$ increases the separation of the distributions become distinct. We note that for $l_p\sim 90 \nm$, $\Delta R_s$ reaches $25\nm$, which is comparable to the conformation change estimated in our theory. 


% Fig.\ref{fig:DeltaR} shows $\DR_S$ as a function of $l_p^{CC}$. $\DR_s$ linearly increases with $l_p^{CC}$ reaching $\sim 25\nm$ at $l_p^{CC}\sim 90\nm$. 


% \section{Condensin Power Stroke}
% \begin{figure}[]
% \centering
% \includegraphics[width=0.5\textwidth]{angle_powerstroke.jpg}
% \caption{\label{fig:angle_dist}Density distribution for the angles for head ($\theta_1$) and for elbow ($\phi_{El}$) from simulation. We plot the two dimensional distribution of angle from 50 trajectories, 40,000 sample points, for $\epsilon_b^{CC}=150(k_BT)$ corresponding $\l_p^{CC}\sim 70\nm$.
% The two maximum are $(\theta_1,\phi_{El})=(2.4,2.1)$ and $(\theta_1,\phi_{El})=(3.1,1.4)$. 
% On top and the right side we show the distribution of $\theta_1$ and $\phi_{El}$, respectively.
% }
% \end{figure}
% A salient feature of molecular motors is the conversion of the chemical energy released due to ATP hydrolysis to mechanical work, which is often accompanied by a power stroke involving conformational changes. Processive motors, such as kinesin, myosin or dynein, undergo dynamic allosteric transitions initiated by ATP binding to the motor head. We posit that a similar power stroke mechanism that produces allosteric transitions must also be operative in the SMC class of motors to translocate along the DNA, thus extruding loops. Based on AFM experiments~\cite{eeftens2016condensin,ryu2019afm} and structural studies~\cite{diebold2017structure} we envision that open to close transition (scrunching process) corresponds to condensin power stroke. The scrunching is assumed to be initiated at heads (ATPase domain), which is amplified through the CC. 

% In the main text, we obtained head-hinge distance as a measure of the power stroke. We illustrate in Fig.\ref{fig:angle_dist} the distribution for the angle for the head, $\theta_1$, and for the elbow, $\phi_{El}$, (see Fig.\ref{fig:simulation} for the definition of angles) calculated from the simulation trajectories. There is a clear separation in the two dimensional distribution of the angles $\theta_1 $ and $\phi_{El}$ (Fig.\ref{fig:simulation}). The distribution of $\theta_1$ in both the open (O shape displayed in top structure in Fig.\ref{fig:simulation}) state and the closed state (B shape - a terminology used in~\cite{eeftens2016condensin} to describe the structure in the bottom of Fig.\ref{fig:simulation}) is narrower than the fluctuations of $\phi_{El}$. Thus, even using the simple model we find that conformations changes in the head is transmitted through the elbow leading to the open to closed transition (Fig. \ref{fig:simulation}). 
% % More importantly, a small change in $\theta_1$ results in a large change in $\phi_{El}$, thus manifesting the powe stroke in condension. 

\section{Persistence length of condensin}
\begin{figure}[]
\centering
\includegraphics[width=\textwidth]{persistence_condensin.pdf}
\caption{\label{fig:persistence} Extracting $l_p^{CC}$ from the simulations. Histograms are for the head-hinge distance in the O shape from the  simulations. Solid lines are from the theoretical expression for $P(R)$. The value of $\epsilon^{CC}_b=150 (k_BT)$ in all the panels except in the inset of (a). (a) $l_p^{CC}\sim13\nm$ for $\epsilon^{El}_b=0 (k_BT)$.  Inset: $l_p^{CC}\sim4\nm$ for $\epsilon^{CC}_b=4 (k_BT)$ and $\epsilon^{El}_b=4 (k_BT)$. (b) $l_p^{CC}\sim18\nm$ and $\epsilon^{El}_b=1 (k_BT)$. (c) $l_p^{CC}\sim24\nm$  and $\epsilon^{El}_b=2 (k_BT)$. (d) $l_p^{CC}\sim34\nm$  and $\epsilon^{El}_b=3 (k_BT)$. 
}
\end{figure}

\begin{figure}[]
\centering
\includegraphics[width=\textwidth]{SI_dists_lp.pdf}
\caption{\label{fig:dists_lp} $P(R)$ for different values of $l_p^{CC}$ in the simulations. (a) $l_p^{CC}\sim13\nm$ [$\epsilon^{El}_b=0 (k_BT)$]. The position of the peaks are $R_2 \sim 13 \nm$ and $R_1 \sim 28 \nm$.  Inset: $l_p^{CC}\sim4\nm$ [$\epsilon^{CC}_b=\epsilon^{El}_b=4 (k_BT)$].  The position of the peaks are $R_2 \sim 15 \nm$ and $R_1 \sim 16 \nm$. (b)  $l_p^{CC}\sim 18\nm$ [$\epsilon^{El}_b=1 (k_BT)$]. The position of the peaks are $R_2 \sim 14 \nm$ and $R_1 \sim 32 \nm$.  (c) $l_p^{CC}\sim24\nm$ [$\epsilon^{El}_b=2 (k_BT)$]. The position of the peaks are $R_2 \sim 16 \nm$ and $R_1\sim 38 \nm$.  (d)  $l_p^{CC}\sim34\nm$ [$\epsilon^{El}_b=4 (k_BT)$]. The peak positions are $R_2 \sim 18 \nm$ and $R_1\sim 40 \nm$. 
}
\end{figure}


In order for the allosteric transition that brings the head-hinge distance to within $\Delta R\sim 26\nm$ ($\approx 22\nm$ in experiments and simulations), condensin has to be sufficiently rigid but not overly so. %Previous experiment~\cite{eeftens2016condensin} using AFM image reported the persistence length of condenins $\approx 4\nm$. 
We estimate the persistence length of condensin ($l^{CC}_p$) by fitting the theoretical expression for end-to-end distance ($R$) with contour length $L$~\cite{bhattacharjee1997distribution} to the simulation results. Here, $R$ is the head-hinge distance in the O shape and $L=51\nm$ is the length of the CC in the simulation. %For $l^{CC}_p \sim 4 \nm$ (the inset in Fig.\ref{fig:persistence}a for the fit) we did not see perceptible difference for O shape and B shape, suggesting the lack of conformational transition~(see the overlapped distributions in the inset of Fig.\ref{fig:dists_lp}a).   
As we varied $\epsilon_b^{El}$, $l_p^{CC}$ changes, resulting in different $\Delta R_s$ values (Fig.~\ref{fig:dists_lp}). 
For $l_p^{CC} \approx 24 \nm $, we find that  $\Delta R_s=22\nm$, which coincides with the experimental value~\cite{ryu2020condensin}.  
The value of $l_p^{CC}$ is roughly six times larger than the experimental data. This may be due to the differences in the fitting used in the simulations or the errors in accurate measurements or in the method used to extract $l_p^{CC}$ from the experimental data. It should be emphasized that for comparison between theory and experiment $l_p^{CC}$ does not play an important rule. Nevertheless, it would be most interesting to design experiments to obtain precise estimates of $l_p^{CC}$.

The results in the inset in Fig.S3(a) are calculated using $l_p^{CC} \sim 4 \nm$. In this case, the elbow effect  is eliminated by setting $\epsilon^{CC}_b  = \epsilon^{El}_b =4 k_BT$, making  the entire CCs flexible. The simulations show that $\Delta R \approx 15\nm $, which disagrees with the experimental value ($\Delta R \approx 22\nm $).   We infer that  the existence of rigid portion of CC with flexibility in the elbow region  is essential for the scrunching mechanism. 


\section{Derivation of $\boldsymbol{P(\vL|\vR)}$}
\begin{figure}[]
\centering
\includegraphics[width=0.5\textwidth]{SI_deriv.pdf}
\caption{\label{fig:derivation}A picture of a conformation of condensin bound to two loci separated by a genomic distance $s$ (extruded loop length). The  spatial distance between the attachment points in the DNA is $r$. For LE to occur condensin has to engage with at least two loci on the DNA.
}
\end{figure}
A major ingredient in the theory (see Eq.(1) in the main text) is the calculation of the contour length of the extruded loop as condensin is powered by ATP binding to the motor head, followed by hydrolysis, and subsequently resetting to complete the catalytic cycle. To obtain Eq.(1) in the main text, let us consider condensin separated by the spatial distance $r$ that pinches a loop whose genomic length is $s$ (Fig.\ref{fig:derivation}). Given the distribution of the spatial distance $r$ between two loci separated by a linear genomic distance $s$, $P(r|s)$, we would like to derive the distribution of $s$, $P(s|r)$. Indeed, $P(s|r)$ is the probability density of the extruded length of DNA, $s$, by condensin whose DNA binding domains are separated by the distance, $r$.  According to the Bayes theorem, we have $P(r|s)P(s) = P(s|r)P(r)$. The normalization dictates that $P(r) = \int_0^{L}P(r|s)P(s)\mathrm{d}s$. These two equations lead to,
\begin{align*}
P(s|r)=\frac{P(r|s)P(s)}{\int_0^{L}P(r|s)P(s)\mathrm{d}s}.
\end{align*}

We assume that there is no preference for picking a specific genomic distance $s$ on the DNA to which the motor attaches to initiate the extrusion process. Consequently,  we take $P(s)=1/L$. Therefore, 
\begin{align*}
P(s|r)&=\frac{(1/L)P(r|s)}{(1/L)\int_0^{L} P(r|s)\mathrm{d}s}\\
&=\frac{P(r|s)}{\int_0^{L}P(r|s)\mathrm{d}s}.
\end{align*}
Thus, $P(s|r)$ and $P(r|s)$ differ only by a constant, $\int_0^{L}P(r|s)\mathrm{d}s$, if we consider a fixed $r$.
It is clear that $P(r|s)$ is the radial probability density for the interior segments separated by a distance $r$ for a semi-flexible polymer, which is derived elsewhere \cite{hyeon2006kinetics}. For the case $s=L$, $P(r|s)$  is the result for the distribution of end-to-end distance for semi-flexible chains, $P(\vR|\vL)$~\cite{bhattacharjee1997distribution}. It is known that the simple analytic result for $P(\vR|\vL)$~\cite{bhattacharjee1997distribution} is accurate when compared to the exact result~\cite{Wilhelm96PRL} or numerical simulations. Thus, we employ the simpler expression $P(\vR|\vL)$ and assume that $P(\vL|\vR)$ is equivalent to $P(\vR|\vL)$ up to a normalization constant when expressed in terms of $L$ with fixed $R$.   Calculation of the distribution of loop sizes requires knowing, $P(\vL |\vR,f)$, which can also be derived from $P(\vR,f |\vL)$ in a similar manner. 


\section{Effect of varying DNA persistence length}
\begin{figure}[]
\centering
\includegraphics[width=1.0\textwidth]{SI_pLR.pdf}
\caption{\label{fig:rate_lp}Effect of variable persistence length for DNA. (a) Plot of $P(\vL|R=50\nm)$. $l_p=50\nm$ (red), $l_p=40\nm$ (blue), $l_p=30\nm$ (green), and $l_p=20\nm$ (brown). (b) Extrusion rate of DNA for different $l_p$. $l_p=50\nm$ (red), $l_p=40\nm$ (blue), $l_p=30\nm$ (green), and $l_p=20\nm$ (brown). $\DR$ is fixed to be $26\nm \sim 76\bps$. 
}
\end{figure}


In the main text, we used $l_p=50 \nm$ ($\sim$147 bps) as the persistence length of DNA, which is widely accepted value for DNA~\cite{rubinstein2003polymer}.
It could be interesting to explore the consequences of varying $l_p$, which can be drastically altered in the presence of divalent cations, as a variable in our theory. In Fig.\ref{fig:rate_lp}(a) we plotted $P(\vL|R=50\nm)$ using Eq.(1) in the main text for different $l_p$. As DNA becomes flexible the distribution of $P(\vL|R=50\nm)$ becomes wider, suggesting that most probable value of the captured length of DNA by condensin would be larger with a large dispersion. Thus, in this situation our approximation, $\vL \approx \vR$, would become less accurate. Nevertheless, we can explore the velocity of extrusion for different $l_p$ shown in Fig.\ref{fig:rate_lp}(b) for a fixed $\DR=26\nm$. As $l_p$ decreases, the velocity of extrusion becomes linear and slower because the load acting on DNA is higher for smaller $l_p$ at the same extension.The decrease of $\Omega$ as $l_p$ decreases can be deduced from the linear (small $x$) expansion of Eq.(7) in the main text. At small forces we find that  $f=\frac{3k_BT}{2 l_p}$. Substituting this linear expansion in the expression for $\Omega$ [Eq.(6)] confirms that as $l_p$ decreases $\Omega$ becomes smaller. Note  that this holds  for fixed extrusion length per step ($\sim$26 nm obtained by fitting Eq. (6) in the main text to the measured  LE velocity).  


% \section{DNA Capture}
% The purpose of the simulations, reported in the main text,  was  to assess whether the scrunching mechanism produces a structural rationale for the value of $\DR\sim 26\nm$ extracted from our theory.  In this section, we created a simple model of condensin with DNA to give some insight into the DNA-capture mechanism. We attached one end of the DNA to the heads of condensin. The other end of DNA diffuses freely during the simulations. We define the DNA "capture" event by the distance between a DNA segment and hinge of condensin with a cut-off length of $4\nm$: if the distance is less than $4\nm$ we assume that condensin captures the DNA segment. Captured DNA length is the contour length of DNA held between the heads and the hinge (see the inset of Fig.\ref{fig:hist_capture}). This working mechanism using hinge domain is motivated by the recent AFM experiment~\cite{ryu2019afm}. 

% We used a coarse-grained bead-spring model for DNA~\cite{hyeon2006kinetics,dey2017toroidal}. Each bead represents 10 base-pairs, which implies that the bead size is $\sigma_{DNA}=3.4 \nm$. The chain has $N=100$ beads or 1,000 base-pairs. Consequitive beads along the chain are  connected by a FENE potential (Eq.(S2)) using $r^0_{i,i+1}=\sigma_{DNA}$. The DNA stiffness is modeled using a harmonic bending potential given by,
% \begin{equation}
%     U_{BEND} = \sum_{i=1}^{N_{ang}}\alpha\cdot\theta^2
% \end{equation},
% where $N_{ang}=98$ is the number of bond angles, $\theta_i$ is the deviation of the $i^{th}$ bond-angle in the chain from $180\deg$, and $\alpha = 7.775$ $(k_B T /\text{rad}^2)$ is a constant~\cite{hyeon2006kinetics}. We chose  $\alpha$ so that the persistence length is $\approx 50\nm$, which is the canonical value for DNA in monovalent salts.   We leave the detailed structural analysis leading to the extrusion process to future work.  Nevertheless, the physical basis of DNA capture and the distribution of capture length are in accord with the currently available structures of condensin.


% \begin{figure}[]
% \centering
% \includegraphics[width=0.5\textwidth]{captured_histgram.pdf}
% \caption{\label{fig:hist_capture} Red histogram shows the distribution of head-hinge distance, and the blue gives the distribution of the captured length of DNA by condensin. Insets display  the snap shots from the simulations. Connected beads depicted in pink correspond to DNA. (a) The distributions for open state (O state).  (b) The distributions for closed state (B state). Parameters for condensin in this simulation are identical to the simulations in the main text (see Table.\ref{Table:parameters}). We performed  50 simulations of 10,000 sample points to create the histograms.  
% }
% \end{figure}



% This simulations confirm that our approximation, $\DR \approx \Dl$, used in theory (see Fig. 2 in the main text) is reasonable. In particular,  the simulations show the long-tail distribution of the captured length of DNA, as predicted by the theory (see Fig.\ref{fig:hist_capture}). Fig.\ref{fig:hist_capture}(a) shows that in the closed state, the captured length of DNA is similar to the head-hinge distance. In the open state, the peak position of the captured length slightly shifts toward longer distance. However, there is a substantial probability that $\Dl$ can exceed the mean (Fig.\ref{fig:hist_capture}(b)). This implies that the capture length can be as large as (60-70) nm ($\sim (180 - 200)  \mathrm{bps}$). It is clear from the simulations that $\DR$ cannot greatly exceed $\Dl$, which is physically reasonable because the capture length should not greatly exceed the motor dimensions (head-hinge distance $\sim$ 50 nm).

\section{Distribution of LE length per cycle}
\begin{figure}[]
\centering
\includegraphics[width=\textwidth]{dists_SI.pdf}
\caption{\label{fig:dist_SI} Distributions for LE length per step for various external loads on DNA. The points in blue are from the  experiment~\cite{ryu2020resolving} and the distributions in red are from the theory (main text). The distributions in blue for high external loads are for the different standard deviation ($\Delta=5\nm$) for $R_1$.  
}
\end{figure}


%\begin{figure}[]
%\centering
%\includegraphics[width=0.5\textwidth]{median_L.pdf}
%\caption{\label{fig:median_L} Median for the distribution $P(\Delta L)$ as a function of $f$. The line in magenta (circle) is from the experiment~\cite{ryu2020condensin} and the line in cyan (square) is from the distributions in Fig.\ref{fig:dist_SI}.
%}
%\end{figure}
% \begin{figure}[]
% \centering
% \includegraphics[width=0.5\textwidth]{peak_position_f.pdf}
% \caption{\label{fig:peak_SI} LE length, calculated from the peak position of the probability $P(\vL|\vR,f)$, as a function of $f$. Red, blue, and green colors are for $\vR=50\nm,40\nm,$ and $30\nm$, respectively. Solid lines are for the peak position of $P(\vL|\vR,f)$ and dashed lines are for the step size (peak position - 26nm).    
% }
% \end{figure}
In the main text, we showed that the theoretically derived distribution of LE length agrees well with the one obtained in the experiment~\cite{ryu2020resolving} for $f=0.4\pN$, {\it without adjusting any parameters}. The agreements persist for different values of load unless $0.5\pN \leq f$ (Fig.\ref{fig:dist_SI}). The discrepancies for high loads can be eased by using smaller $\Delta$ as shown in the blue distributions. It is worth mentioning that the sample sizes for obtaining the experimental results in Fig.~\ref{fig:dist_SI} are much smaller than for  $f=0.4\pN$.  They are $N=153,131,102,118,155,140$ for $f=0.2\pN,0.3\pN,0.5\pN,0.6\pN,0.7\pN,1.0\pN$ respectively whereas $N=1727$ for $f=0.4\pN$ in the main text. The smaller sample sizes surely affects the accuracy of the measured distribution. Considering the low stall load for condensin, likely to be $f \approx 0.8 \pN$ (see the inset of Fig.4a in the main text),  we believe that our theory predicts, with reasonable accuracy, the LE length by condensin during DNA compaction. 

%In Fig.\ref{fig:median_L} we plot median for $\Delta L$ as a function of $f$. Median value decreases as a function of $f$, reflecting the shortening of the long-tail in $P(\Delta L)$. We find the discrepancy for the median values between the experiment!\cite{ryu2020condensin} and the predicted from our theory, especially for large $f$. Because the sampling in the experiment is limited experiment may not be able to access the long-tail in the distributions, which could result in the smaller median values for large $f$ comparing to the theory.  




%\section{Detailed comparison with a previous study:}

%{\bf Compilation of parameters needed to evaluate loop extrusion velocity Eq. (3) in the MRBG Paper}
 
%A recently published theory~\cite{marko2019dna} produces an expression for LE  velocity and step size as a function of load. %We tabulate the parameters needed to calculate the LE velocity using the MRBG (Marko-Rios-Barducci-Gruber) model. We extracted the parameters starting from the final form for the LE velocity contained in Eq.(3) in the MRBG study. 
%The full expression is far too complicated~\cite{marko2019dna}, and therefore MRBG used a simpler equation based on a four state model for the ATPase cycle for the enzyme along with a DNA model to calculate the LE velocity ($\Omega$ in our notation) as a function of force.  Even the simpler expression has over twenty parameters, making it difficult to analyze experiments.% Rather than restate the equations in the MRBG model, we list the parameters needed to calculate $\Omega$ in Table.\ref{Table:parameters_marko}. 
%The numerical values of $\eta$ and $k_{syn}$, which are not contained in~\cite{marko2019dna}, were kindly provided by Marko.   

%We first calculated $\Omega$ as a function of load using the parameters in Table.\ref{Table:parameters_marko}, which were extracted from~\cite{marko2019dna}. 
%The solid black line in Fig.\ref{fig:marko} is in near quantitative agreement with the result in Fig. 6C in~\cite{marko2019dna}. 
%The prediction  does deviate from the experimental data (red dots).  
%More importantly, the shape of the curve is different from measurements (compare the dots and the solid black line in Fig.4a).   
%Informed by  the fit of our theory to experiments, we varied $k_{eng}$ and $f_0$ (see Table.\ref{Table:parameters_marko}) to better fit the data. The resulting dashed blue curve using $k_{eng}=30\persec$ and $f_0=0.3 \pN$, with the remaining parameters fixed at the values in Table.\ref{Table:parameters_marko}, gives a very good fit to the data. %  shape of the curve differs Using the parameters we calculated the LE velocity and compared it to experiments in Fig.\ref{fig:marko}.

%\end{ruledtabular}

%\end{table}

%We first calculated $\Omega$ using the parameters in Table.\ref{Table:parameters_marko}. The solid line in F
% We cannot find the values for $\eta$ and $k_{sys}$.



% \begin{figure}[]
% \centering
% \includegraphics[width=0.5\textwidth]{Marko_LE.pdf}
% \caption{\label{fig:marko} Reproduction of Fig.6C in \cite{marko2019dna}. Blue solid curve is the theoretical LE velocity originally plotted in \cite{marko2019dna}. Because the shape of the solid blue curve does not mach with experiments, we adjusted two parameters to obtain the dashed curve. We used $k_{eng}=30\persec$ and $f_0=0.3\pN$ instead of $k_{eng}=2\persec$ and $f_0=0.1\pN$ \cite{marko2019dna} (the rest of the parameters are the same as in Table.\ref{Table:parameters_marko} modified). Red points are the data from the experiment~\cite{ganji2018real}. We used the standard conversion between base pair and distance, $1\bp=0.34\nm$~\cite{alberts2018molecular}.
% }
% \end{figure}
% \newpage
% \begin{figure}[]
% \centering
% \includegraphics[width=1\textwidth]{persistence_angle_end.pdf}
% \caption{\label{}(a) Persistence length calculated using angle. $l_p=l_b/(1+<\cos\theta>)$, where $l_b$ is bond length and $\theta$ is angle between consecutive bonds. Red circles are with flexible elbow, blue circles are without flexible elbow. Green squares are reference values calculated from $l_p=l_b \epsilon/k_BT$. (b) Persistence length calculated using end-to-end distance. $<R^2>=2l_p L (1-\frac{l_p}{L}(1-e^{-L/l_p}))$. Red circles are with flexible elbow, blue circles are without flexible elbow. Green squares are reference values calculated from $l_p=l_b \epsilon/k_BT$. 
% }
% \end{figure}

%\section{Critique of the reported values of $k_0$, $\Delta R$, and length of base pair.}
%Two experiments on yeast condensin that used the same length (48.5 kbp $\lambda$) DNA estimated two widely differing values for the step size ($\Delta R$) as $\sim$ 30 bps~\cite{terakawa2017condensin} and $\sim$ 300 bps~\cite{ganji2018real}. The inconsistency between the vastly different estimates for the step size deserves further scrutiny.  (i) It appears that the step sizes were obtained using the relation, $\Delta R = \Omega/k_{hyd}$, where $k_{hyd}$ is hydrolysis rate. Because the LE velocity, $\Omega$ is load dependent, it implies that $\Delta R$ changes with the external load. This formula cannot be correct because it would predict that the step size of the well-studied kinesin-1 (Kin1), would be $\sim$ 8.2 nm (the correct and constant value) at zero external load ($\Omega \sim 800 \nm/s$ and $k_{hyd} \sim 100\persec$~\cite{cross2004kinetic}). However, at the stall force $\Delta R$ would be predicted to be zero, which is incorrect, and at intermediate values of the resistive force the step size of Kin1 would change.  However, many careful single molecule experiments (see for example~\cite{hua1997coupling}) have shown that step size of kinesin is a constant. (ii) Terakawa et al. obtained the average value of $\Omega  \sim 60 \bps/s$ (see Fig. 3E in~\cite{terakawa2017condensin}) and $k_{hyd}=2\persec$. The small  $\Omega$ value implies that DNA is under tension ($x \sim 0.72$) in the experiment. In contrast, it is unclear why the value of $\Omega \sim 600 \bps/s$ was used in~\cite{ganji2018real}. We concluded that it must be the mean value of the LE velocity versus relative extension of DNA shown in Fig. 3I in~\cite{ganji2018real}. However, combining such a mean value with $k_{hyd}$ cannot give meaningful estimate of step size, as illustrated by  the example of Kin1. The point is that the current experimental estimates of condensin step sizes are mere guesses. The only important conclusion one can draw, which is fully supported by our theory ($\Delta R = 76\ \mathrm{bps}$),  is that the step sizes are considerably larger than what one finds in DNA motors (at most few bps). 

% In order to assess how the LE velocity changes as a function of load, we calculated $\Omega$ by varying $k_0$~(Fig.\ref{fig:diff_k0}) using our theory and the one by MRBG ~\cite{marko2019dna}. For a given $k_0$, we obtained $\Delta R$ as the best fit of our theory (Eq.4 in the main text) to experimental data. We find that the best agreement with measurements are obtained for $k_0 = 20\persec$ although predictions using $k_0 = 10\persec$ with a step size of 54 nm ($\sim$ 160 bps) are not unreasonable.  We also calculated $\Omega$ as a function of load at various $k_{hyd}$ using the MRBG theoty~\cite{marko2019dna} using the parameters in  Table.\ref{Table:parameters_marko} except for $k_{eng}$ and $f_0$, which we increased to fit the experimental data. The figures show that reasonable fits require that $k_0 (k_{hyd})$ has to exceed $10 \persec$. Of course, one could argue that the theories are simply wrong but given that the measured $k_0$ is stated to be a lower bound~\cite{ganji2018real} such arguments are likely to be tenuous. 

%Finally, one last technical but important point should be made.  This pertains to the length associated with a single base pair. Terakawa et al.~\cite{terakawa2017condensin} used $1 \bp \sim 0.25 \nm$. They obtained this value appears by dividing the distance with  fixed end of DNA (12 $\mu$m) by the total number of base pairs (48.5 kbs). The use of this procedure is very approximate. It is  doubtful if it provides a reliable estimate of the step size since the value obtained in this conversion is not the genomic length of the DNA. From the numbers ($600 \bps \sim 110 \nm$) given in Ganji et al.~\cite{ganji2018real} we get $1 \bp \sim 0.18 \nm$. It is likely the reported numbers were calculated using the same formula as Terakawa et al.~\cite{terakawa2017condensin}.  The DNA end-to-end distance (9.1 $\mu$m) divided by the total number of base pairs (48.5 kbs) yields $\sim 0.18 \nm$. To the best of our knowledge, these unusually small numbers are never used in the nucleic acid literature. Indeed, these values would be totally inconsistent with well-established canonical value of the persistence length ($150 \bps \sim 50 \nm$), and force extension curves for $\lambda$-DNA~\cite{smith1992direct}.

%Had we used  $1 \bp = 0.18\nm$~\cite{ganji2018real}, we obtain $k_0=11\persec$  by fitting our theory for $\Omega$ to the experimental data. Concomitantly, the step size in base pair would increase as well. The $k_0$ value, setting the scale of the curve, depends on the bp to nm conversion because we fit the data obtained in bp unit while Eq.(4) in the main text is in nm. It is best to measure LE velocity directly using conventional length scale (nm), otherwise the relation between $k_0$ and hydrolysis rate of condensin would be less stringent.




%\begin{figure}[]
%\centering
%\includegraphics[width=1.0\textwidth]{diff_k0.pdf}
%\caption{\label{fig:diff_k0} LE velocity for various $k_0$ and $k_{hyd}$. (a) From our theory (Eq.4 in the main text). Red solid line: $k_0=20\persec$ and $\Delta R=26\nm$, blue solid line: $k_0=10\persec$ and $\Delta R=54\nm$, green solid line: $k_0=5\persec$ and $\Delta R=83\nm$, and brown solid line: $k_0=2\persec$ and $\Delta R=93\nm$. (b) From MRBG theory~\cite{marko2019dna}. Red solid line: $k_{hyd}=20\persec$, blue solid line: $k_{hyd}=10\persec$, green solid line: $k_{hyd}=5\persec$, and brown solid line: $k_{hyd}=2\persec$. We used that the length of a bp is $1\bp=0.34 \nm$.
%}
%\end{figure}


\section{Conformational transition to the LE active state}
\begin{figure}[]
\centering
\includegraphics[width=1.0\textwidth]{alighnment.pdf}
\caption{\label{fig:alighn} (a) Structural alignment obtained by minimizing the root mean squared displacement for the head domain. Structure in green is the {\it apo} state (PDB:6YVU) where the heads are aligned to the ATP bound state, and the structure in magenta is the state with ATP (PDB:6YVD). 
%Structure colored magenta is ATP bound state (PDB:6YVD) and the structure for {\it apo} state (PDB:6YVU) aligned to magenta is colored in green. 
Alignment is constructed only for the heads. (b) Normal mode analysis using the crystal structure for the {\it apo} state and ATP bound state (6YVU and 6YVD, respectively). Blurred regime shows the possible movements of the residues.  
}
\end{figure}
The lack of structures in various nucleotide states of condensin in the presence of DNA makes it virtually impossible to provide a molecular basis of loop extrusion. In the absence of viable structures, and prompted by the theoretical prediction that during a single turnover the distance between the hinge and motor should come within $\Delta R \approx (22-26) \nm$, we envisioned that the SMCs transition between the open and closed states in order to extrude loops. This picture is fully consistent with experiments~\cite{ryu2020condensin}. The shape of condensin from the partial structures of condensin in the inactive (in the absence of nucleotides or DNA) obtained by cryo-EM at $\sim 8.1 \angstrom$ resolution~\cite{lee2020cryo} cannot account for the O shape, which likely represents the functionally active state. Because the cryo-EM structures~\cite{lee2020cryo} do not contain DNA, they correspond to an inactive state.  Nevertheless, even these inactive structures~\cite{lee2020cryo} reveal that binding of ATP,  sandwiched between the two heads, leads to a substantial conformational change ($\sim 50 \degree$ rotation and $\sim 20 \angstrom$ translation) near the junction between the CC and the heads.
%, which should affect the conformation of the unresolved parts of the CC. {\bf I DO NOT UNDERSTAND WHAT THE PART THAT FOLLOWS IS?}unless the movement of CC is structurally constraint by folded conformation frequently observed in the study for both condensin~\cite{lee2020cryo} and cohesin~\cite{buermann2019folded} in the absence of DNA. %Another possible constraint is braiding topology of CC suggested by computational study~\cite{krepel2020braiding}. 


To provide insights into the ATP-induced conformational changes, we first aligned the structures in the head-domains of the {\it apo} state (PDB:6YVU) and the head-domains of the ATP-bound state (PDB:6YVD). Using VMD's multiseq tool~\cite{roberts2006multiseq}, we aligned residues: 1-148, 1036-1170 for Chain A and 150-311, 1285-1415 for Chain B using STAMP structural alignment.
The structural alignment shows that there is a large change in the  CC orientation between the {\it apo} state and the ATP bound state. %{\bf What does this mean?} unless CC is structurally constrained. 
Fig.~\ref{fig:alighn}a  shows the structural alignment obtained by minimizing the root mean squared displacement between the head domains in the {\it apo} state (PDB:6YVU), and the head domain in the ATP bound state (PDB:6YVD). The alignment suggests that the CC could undergo a wide opening motion upon ATP binding. %Possible reason that CC-opened structure in green is not observed in \cite{lee2020cryo} maybe be attributed to the fact that CC has a bend at the elbow, which creates structural constraint for condensin. We speculate that in order to transient to LE active state the bend at the elbow has to be liberated. Future structural study will access the transition from LE inacrtive state to LE active state for condensin.  

% {\bf AGAIN STRUCTURALLY CONSTRAINED BY WHAT?} unless CC is restrained as in original ATP engaged state colored magenta. %The opened structure (in green) rather resemble to the head domain of ATP$\gamma$S state obtained for bacterial SMC~\cite{diebold2017structure}. 
% %This alignment also insinuates that the open state or O-state may correspond to ATP bound state. 
% {\bf RT - the following paragraph - speculative - do we need?}
% Association of condensin to DNA triggered by ATP binding may alter the stability of folded domain on the CC, possibly by hinge-DNA interaction or hinge-subunit (STAG1) interaction observed for cohesin~\cite{shi2020cryo}, transforming into the structures where the conformational change at the head domain can readily transmit to CC as in open and closed state discussed in this manuscript.  


{\bf Normal mode analysis:}
To understand the correlation between the head and hinge movement, we performed Normal Mode Analysis on the condensin structure in the {\it apo} state (PDB:6YVU). We created a variant of an Elastic Network Model using a second-order Taylor series expansion of the Self-organized Polymer model with Side-chains (SOP-SC) for proteins~\cite{mugnai2020role}. Since the {\it apo}-aligned to ATP bound state does not have the full condensin structure, we calculated a displacement vector ($\textbf{D}$) between 6YVU (aligned to 6YVD) with the original coordinate of 6YVU using only the beads that were present in both the full condensin structure, and the {\it apo}-aligned structure. Using the normal modes, we determined the eigenvector that best approximates this structural transition. $\textbf{D}$ is a $3N$ dimensional vector where N is the total number of beads  (PDB: 6YVD).  $\textbf{D}_i = [(\textbf{D})_{1,x},...,(\textbf{D})_{M,z}]$,
where $(\textbf{D})_{j,\alpha}$ is the entry associated with bead j, and direction $\alpha\in {x, y, z}$. The beads which are not present in the ATP-aligned structure have $(\textbf{D})_{j,\alpha}=0$.  The overlap between the eigenvector $v_n$ (corresponding to normal mode $n$), and displacement $\textbf{D}$ is given by,
\begin{equation}
    I_n(\textbf{D}) = \frac{\sum_{i=1}^{3N} v_{n,i}\cdot D_i }{\sqrt{\sum_{i=1}^{3N} v_{n,i}^2}\sqrt{\sum_{i=1}^{3N} \textbf{D}_i^2}}.
\end{equation}
We found that mode 7 has the highest overlap value.

Fig.~\ref{fig:alighn}b shows the motion of the residues calculated from the largest eigenvector that produces the maximum overlap. In the blurred region residues experience correlated movement. Our analysis shows that the normal modes mostly overlap at the head and hinge. In other words, head and hinge are likely to undergo the largest conformational change during the transition between the two states. If this finding holds when the structures of condensin in various nucleotide binding states are determined, it would provide a molecular basis for the O and B shapes that the condensin clearly samples during the process of loop extrusion~\cite{ryu2020condensin}. %This would also rule out the apparent suggestion that the hinge does not undergo significant amount of movement to extrude DNA~\cite{diebold2017structure,marko2019dna}.               



\bibliography{mybib}
\end{document}
%
% ****** End of file apstemplate.tex ******

