% \vspace{-1em}
\section{Introduction}
% \vspace{-0.2em}
\iffalse
Advancements in multimodal single-cell technologies provide the capability to simultaneously profile multiple data types in the same cell, including chromatin accessibility~\cite{cao2018joint, chen2019high}, DNA methylation~\cite{gaiti2019epigenetic}, nucleosome occupancy~\cite{pott2017simultaneous}. For example, CITE-seq~\cite{stoeckius2017simultaneous} utilizing oligonucleotide-conjugated antibodies can quantify RNA and surface protein abundance in the same cells. Here, protein abundance is measured via the antibody-derived tags (ADTs) read counts. The Single Cell Multiome ATAC $+$ Gene Expression technology~\cite{belhocine2021single} %\footnote{https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression} 
concurrently profiles assay of transposase-accessible chromatin (ATAC-seq) ~\cite{buenrostro2013transposition} and RNA expression from the same cell. These technologies offer an exciting opportunity to characterize cell identity and state at an unprecedented resolution, enabling a better understanding of gene regulatory networks in multicellular organisms and tissues~\cite{zhu2020single}. 
%overcome the limitations of single-cell RNA sequencing (scRNA-seq) and to examine the impact of multiple cellular modalities among cellular state and function~\cite{zhu2020single}.
\fi 

Advancements in multimodal single-cell technologies provide the capability to simultaneously profile multiple data types in the same cell, including chromatin accessibility~\cite{cao2018joint, chen2019high}, DNA methylation~\cite{gaiti2019epigenetic}, nucleosome occupancy~\cite{pott2017simultaneous}. These technologies offer an exciting opportunity to characterize cell identity and state at an unprecedented resolution, enabling a better understanding of gene regulatory networks in multicellular organisms and tissues~\cite{zhu2020single}.  Despite the rapid accumulation of multimodal single-cell data, the analyses of such data are still faced with numerous challenges. First, single-cell measurements often exhibit high sparsity and noise levels, making it difficult to draw meaningful insights from the data~\cite{eraslan2019single}. 
% The measurement process is impacted by various environmental factors, such as amplification bias, cell cycle effects, variations in library size, and RNA capture rate, all of which contribute to substantial noise in single-cell data~\cite{eraslan2019single}.
Furthermore, samples are often measured under different conditions, including batches, times, locations, or using different instruments, which leads to systematic variations in the measured values, which further complicates the interpretation of single-cell data. These imperfections can result in biased estimates of cell-cell interactions and poses tremendous challenges for computational models to exploit such interactions. 

%Innovations in multimodal single-cell technologies enable the simultaneous profiling of the transcriptome alongside other cellular modalities such as chromatin accessibility~\cite{cao2018joint, chen2019high}, DNA methylation~\cite{gaiti2019epigenetic}, nucleosome occupancy~\cite{pott2017simultaneous}, or spatial location~\cite{rodriques2019slide}. Specifically, CITE-seq~\cite{stoeckius2017simultaneous} utilizes oligonucleotide-conjugated antibodies to quantify both RNA and surface protein abundance in single cells through the sequencing of antibody-derived tags (ADTs). Single Cell Multiome ATAC $+$ Gene Expression technologies\footnote{https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression} simultaneously profile assay of transposase-accessible chromatin (ATAC-seq)~\cite{buenrostro2013transposition} and RNA measurements from the target cells. These approaches offer exciting solutions to overcome the limitations of single-cell RNA sequencing (scRNA-seq) and to examine the impact of multiple cellular modalities among cellular state and function~\cite{zhu2020single}.

\iffalse
Despite the rapid accumulation of multimodal single-cell data, the analyses of such data are still faced with numerous challenges. First, single-cell measurements frequently exhibit high sparsity levels, making it difficult to draw meaningful insights from the data. The measurement process is impacted by various environmental factors, such as amplification bias, cell cycle effects, variations in library size, and RNA capture rate, all of which contribute to substantial noise in single-cell data~\cite{eraslan2019single}. Furthermore, samples are often measured under different conditions, including batches, times, locations, or using different instruments, which leads to systematic variations in the measured values, which further complicates the interpretation of single-cell data. These imperfections can result in biased estimates of cell-cell interactions and poses tremendous challenges for computational models to exploit such interactions. 
\fi
% The availability of multimodal single-cell data is increasing, but the challenges of analyzing existing data are still emerging. Single-cell measurements often show extreme sparsity, making extracting meaningful information from the data challenging. In the measurement process, various technical factors, such as  amplification bias, cell cycle effects, library size differences, and especially low RNA capture rate, result in substantial noise in single-cell data~\cite{eraslan2019single}. In addition, samples in the data are frequently measured in different batches, at different times, or using different instruments, which can lead to systematic variations in the measured values and further complicate the interpretation of single-cell data. The imperfections of sequencing data may lead to biased estimation of the underlying molecule interactions and cell-cell associations. 
% \hongzhi{Can we use cell-cell 'correlation' instead of 'interaction'? For "molecule interactions", we can refer to MLP. For "cell-cell interactions", we should connect to kNN graph-based methods. Basically, we are demonstrating the drawback of MLPs and kNN-based GNNs}

% \wei{you can first talk about the existing methods; limitations of existing methods; why need a new approach (motivation); why the introduced new approach can work...} 

% \jt{we mentioned challenges in the above paragraph?  I think for this paragraph, we should mention what are existing solutions? why they are not sufficient??? I can tell you somehow follow this way, but we need a better logic, which is consistent with the aforementioned challenges.}

To effectively capture the intricate interactions within cells and genes, current research focuses on constructing static graphs based on heuristic criteria and then employing graph neural networks (GNNs)~\cite{kipf2016semi, velickovic2017graph, ma2021deep} to extract information from the built graph. 
% For example, Hao et al.~\cite{hao2021integrated} and Van et al.~\cite{van2018recovering} have built the k-nearest neighbor (k-NN) graph for cells, which assesses the connections between cells by measuring the similarity in gene expression of cells. 
% However, the quality of k-NN graphs is contingent upon the selected heuristic similarity measure and it does not incorporate downstream task information.
However, the quality of built graphs is contingent upon the selected heuristic similarity measure and it does not incorporate downstream task information.
%, which can introduce noise for GNNs to perform well on the downstream task. 
An alternative way for building the interaction graph is to construct the graph based on domain knowledge, e.g., utilizing publicly accessible databases~\cite{wen2022graph}. % For instance, Wen et al.~\cite{wen2022graph} construct the graph based on the pathway data~\cite{liberzon2015molecular} extracted from the Molecular Signatures Database~\cite{subramanian2005gene}, which describes the correlations between gene features. 
%which is able to mitigate the influence of noisy representations.  
However, similar to k-NN graphs, such graph construction process does not leverage downstream task information and the knowledge base may not include all relevant genes/proteins. Furthermore, GNNs have some inherent limitations that hinder their success in applications: the over-smoothing~\cite{kreuzer2021rethinking} and over-squashing~\cite{alon2021on} issues where GNNs produce poor results when we deeply stack GNN layers.
Hence, one question naturally arises: \textit{can we have a better approach to construct interaction graphs (among cells, genes, and proteins) that utilize downstream information while avoiding the aforementioned issues?}
%of GNNs result in indistinguishable embeddings of connected nodes within the prior knowledge graph, while the rest of the nodes are isolated to the aggregation process. In view of this, one question naturally arises: \textit{is there a better way to explore both the molecule interactions and cell-cell association without the aforementioned issues?}

% \wei{how do they utilize the KNN graph?}  In addition,
% graph neural networks have been employed  to model the biological interactions \wei{what graph they have?}. 

% Incorporating prior domain knowledge into GNNs may mitigate the influence of noisy representations~\cite{}, but it also introduces new challenges. 


% The majority of existing works capture these interactions by building a k-nearest neighbor (k-NN) graph~\cite{hao2021integrated, van2018recovering}. The k-NN graphs quantify the associations between cells by calculating the distances between them, which significantly overlook the heterogeneity and batch effects of the data.\jt{is GNN also based on the k-NN graphs??? } Recently Graph Neural Networks (GNNs)~\cite{rao2021imputing, reau2023deeprank} have been introduced to model the biological interactions, which could lead to inconsistent results when the input is of poor quality. Incorporating prior domain knowledge into GNNs may mitigate the influence of noisy representations \jt{add references}, but it also introduces new challenges. Although there are numerous publicly accessible databases, the use of GNNs may be hindered if the prior information does not encompass all the nodes of interest in the data. In addition, the over-smoothing~\cite{kreuzer2021rethinking} and over-squashing~\cite{alon2021on} issues of GNNs result in indistinguishable embeddings of connected nodes within the prior knowledge graph, while the rest of the nodes are isolated to the aggregation process. Thus, alternative methods are needed to further explore both the molecule interactions and cell-cell associations.

% \jt{I do not think we should name different types of transformer like call, omics, .... I think we can think about it is a multimodal transformers. Also, before we introduce the multimodal transformers, we need to first motivate why transformer can handle the shortcomings of existing methods. After that, we need to introduce what are the challenges we face when we try to  adopt transformers to our case: maybe data is multimodal or some other challenges we have.  }

%\jt{I think in the above paragraph, we need to mention the challenges, first, second, third. Then in this paragraph, we discuss how transformer has the potential to tackle these challenges. Now this pragraph is not well organized. we first discuss transformer and then graph transformer? why? }

In light of the recent advances of transformers~\cite{devlin2018bert, kitaev2020reformer, liu2021swin, liu2022swin} in capturing pairwise relations among objects, we seek to utilize transformers for learning the interaction graph for cells, genes, and proteins in an end-to-end manner. Transformers are well-suited to address the limitations of static graphs: they learn the interaction between objects through the self-attention mechanism, where all objects are attended to each other with learnable attention scores indicating their interaction strength. Thus, the attention matrix provides an advanced approach to characterize the interaction between objects in a data-driven way and has demonstrated success in reducing unwanted variance and noise across batches~\cite{yang2022scbert}. 
% Moreover, multi-head and multi-layer transformers have the capability of capturing more complex and nuanced relationships during the training process~\cite{huang2021moltrans, ieremie2022transformergo}. 
However, these traditional transformers do not account for the available graph structure and are therefore unable to leverage prior information present in graph data, such as biological knowledge graphs. In this context, graph transformers~\cite{rampasek2022recipe, ying2021transformers, dwivedi2020generalization}  offer a solution by combining the strengths of GNNs and transformers to make use of graph data. These approaches allow for the incorporation of prior insights from structural information learned from GNNs, while still allowing for data-specific interactions to be learned through the attention mechanism.
%In other words, graph transformers can leverage prior insights by incorporating structural information learned from GNNs, while still being able to learn data-specific interactions through the attention mechanism. 
On top of that, graph transformers also alleviate the over-smoothing and over-squashing problems in GNNs by enabling individual objects to attend to unconnected objects~\cite{chen2022structure, dwivedi2021graph, kreuzer2021rethinking, rampavsek2022recipe}. 
Therefore, it is of great importance to investigate the potential of (graph) transformers in single-cell analysis.


%\jt{we need to state what are the challenges to adopt transformer into multi-modal data? what we have done. Please merge the following two paragraphs. For our methods, we do not need to give too many details now. The high-level is a multi-modal transformer: for each modality, we adopt the corresponding transformer and we use xxx to bridge all modalities}
In this work, we aim to design a transformer framework for multimodal single-cell  data. In essence, to utilize the strengths of (graph) transformers, we are faced with two non-trivial challenges.  \textbf{First}, since multimodal data contains diverse information from various sources, e.g., genes, proteins, and cells, it can be difficult for a single transformer to capture all aspects. \textbf{Second}, traditional transformers suffer a quadratic computation complexity w.r.t. the number of objects, which poses a challenge for single-cell analysis where the number of cells can be large. 
To address the first challenge, we introduce the \underline{S}ingle-\underline{C}ell Multi\underline{mo}dal Trans\underline{former} \method{}, which employs multiple transformers to model the multimodal data, allowing each transformer to deal with a specific data modality. The core of \method{} is the cross-modality aggregation component which builds a bridge between these transformers and aggregates the necessary information from individual ones.
% For genes and proteins, we apply modality-specific transformers to learn the high-order interactions within each modality.  To incorporate existing domain knowledge of protein-protein interactions and gene-gene networks, we adapt GraphGPS~\cite{rampasek2022recipe} on genes and proteins respectively, where the graph is constructed among prior information. We also integrate Laplacian positional encoding~\cite{dwivedi2020generalization} and random walk positional encoding~\cite{dwivedi2022graph} to further improve the expressiveness of the graph transformer modules. 
For the second challenge, \method employs linearized transformers~\cite{choromanski2021rethinking} to the cells which greatly reduces the computational complexity.
%\hongzhi{We had some experiments to show that pre-computed kNN graphs from scRNAseq are not consistent with protein similarity}
% \hongzhi{Please only mention scBERT. scBERT also claimed that they are the first ones to apply a transformer. You can mention the difference between our method and scBERT. scFormer is already withdrawn from ICLR.} 
% In this paper, we investigate the application of transformer models in single-cell analysis, which is a pioneering research topic. To the best of our knowledge, only a few works~\cite{yang2022scbert} implemented transformer modules among scRNA-seq data, and we are the first to introduce transformers to multimodal analysis. Here, we propose \method{} for multimodal integration with a focus on modality prediction tasks. 
% As a novel extension of our previous work ScMoGNN~\cite{wen2022graph}, \method{} incorporates GNNs with transformer and achieves remarkable results on the benchmark datasets compared to existing models.
To the best of our knowledge, we are the first to employ transformers to advance the analysis of multimodal single-cell  data. Our proposed framework achieves promising results on the benchmark datasets, providing a very strong baseline for follow-up research. Our contributions can be summarized as follows: 
\vspace{-1em}
\begin{itemize}[leftmargin=*]
    \item We study the problem of multimodal single-cell data analysis and propose a transformer framework \method{} to capture the intricate relations within modalities and between modalities. %integrate multimodal data and predict the target modality.
    \item The proposed \method{} is versatile and can flexibly incorporate domain knowledge regarding genes and proteins. %by GNNs and positional encoding. of biological databases
    \item The proposed \method{} achieves superior performance on various benchmark datasets. Remarkably, we are one of the top winners in a NeurIPS 2022 competition. 
    % \jt{since we also do experiment in last year competition, if it performs better than our last year's model, we can mention it beats last year's winner performance?} \wenzhuo{We change last year's model from transductive to inductive for fair comparison, which may not be adequate to state like that.}
\end{itemize}

% TODO: add hyperlink of each section
% The rest of the paper is structured as follows. In Section 2, we review some of the related works. Section 3 contains the notations and a formal statement of the problem. We specify our proposed framework in Section 4. The experimental results are organized in Section 5. Finally, we wrap up the work and outline potential future directions in Section 6.