 \documentclass[11pt,notitlepage]{article}

\usepackage{booktabs} % For formal tables


% Copyright
%\setcopyright{none}
%\setcopyright{acmcopyright}
%\setcopyright{acmlicensed}
%\setcopyright{rightsretained}
%\setcopyright{usgov}
%\setcopyright{usgovmixed}
%\setcopyright{cagov}
%\setcopyright{cagovmixed}

\usepackage{fullpage}
\usepackage{theorem}
\usepackage{graphicx}
\usepackage{subfig}
\usepackage{float}
\usepackage{afterpage}
%\usepackage{multirow}
%\usepackage{needspace}
\usepackage{amsmath,amssymb}
%\usepackage{dsfont}
%\usepackage{caption, subcaption}
%\DeclareCaptionType{copyrightbox}
%\usepackage{framed,url}
%\usepackage{citesort}
%\usepackage{epsfig}
\usepackage{color}
\usepackage{url}
%\usepackage{subcaption}
\usepackage{bbm}
%\usepackage{tikz}
%\usepackage{wrapfig}
%\usepackage[margin=1in]{geometry}
\usepackage{epsfig}
\usepackage{epstopdf}
%\usepackage[linesnumbered,ruled,vlined, algo2e]{algorithm2e}
%\allowdisplaybreaks

\newtheorem{claim}{Claim}
\newtheorem{property}{Property}
\newtheorem{observation}{Observation}
\newtheorem{heuristic}{Heuristic}
\newtheorem{solution}{Solution}
\newtheorem{scheme}{Scheme}
\newtheorem{remark}{Remark}
\newtheorem{fact}{Fact}
\newtheorem{definition}{Definition}
\newtheorem{theorem}{Theorem}

\newcommand{\eps}{\epsilon}
\newcommand{\abs}[1]{\left| #1 \right|}
\newcommand{\norm}[1]{\left\lVert #1 \right\rVert}
\newcommand{\E}{\mathcal{E}}
\newcommand{\var}{\mathbf{Var}}
\newcommand{\err}[1]{\text{Err}_{#1}^k}


\newcommand{\bx}{\mathbf{x}}
\newcommand{\bz}{\mathbf{z}}
\newcommand{\by}{\mathbf{y}}
\newcommand{\bw}{\mathbf{w}}
\newcommand{\bv}{\mathbf{v}}
\newcommand{\bpi}{\boldsymbol{\pi}}

\newcommand{\infnorm}[1]{\|#1\|_{\infty}}

\renewcommand{\Pr}{\mathbf{Pr}}

\newcommand{\A}{\mathcal{A}}
\newcommand{\X}{\mathcal{X}}
\newcommand{\D}{\mathcal{D}}
\renewcommand{\O}{\mathcal{O}}
\newcommand{\T}{\mathcal{T}}
\renewcommand{\H}{\mathcal{H}}
\newcommand{\F}{\mathcal{F}}
\renewcommand{\S}{\mathcal{S}}

\newcommand{\genoa}{{\tt GEN50kS}}
\newcommand{\genoaa}{{\tt GEN50kS-aligned}}
\newcommand{\genob}{{\tt GEN20kS}}
\newcommand{\genoc}{{\tt GEN20kM}}
\newcommand{\genod}{{\tt GEN20kL}}
\newcommand{\genoe}{{\tt GEN80kS}}
\newcommand{\genof}{{\tt GEN320kS}}
\newcommand{\trec}{{\tt TREC}}
\newcommand{\uniref}{{\tt UNIREF}}

\newcommand{\ebdjoin}{{\tt EmbedJoin}}
\newcommand{\ebdpara}{{\tt EmbedJoinPara}}
\newcommand{\pass}{{\tt PassJoin}}
\newcommand{\edjoin}{{\tt EDJoin}}
\newcommand{\adpjoin}{{\tt AdaptJoin}}
\newcommand{\qchunk}{{\tt QChunk}}
\newcommand{\sft}{{\tt sft}}



\newcommand{\median}{\operatornamewithlimits{median}}
\newcommand{\mean}{\operatornamewithlimits{mean}}
\newcommand{\argmin}{\operatornamewithlimits{argmin}}
\newcommand{\argmax}{\operatornamewithlimits{argmax}}

\newcommand{\qinsays}[2][]{\textcolor{blue} {\textsc{Qin #1:} \emph{#2}}}
\newcommand{\haoyusays}[2][]{\textcolor{red} {\textsc{Haoyu #1:} \emph{#2}}}

\renewcommand{\paragraph}[1]{\medskip \noindent {\bf #1.}}

%\renewenvironment{proof}{\trivlist\item[]\emph{Proof}:}%
%{\unskip\nobreak\hskip 1em plus 1fil\nobreak$\Box$
%\parfillskip=0pt%
%\endtrivlist}

\usepackage{algpseudocode}
\usepackage{algorithmicx}
\usepackage{algorithm}
\renewcommand{\algorithmicrequire}{ \textbf{Input:}} %Use Input in the format of Algorithm  
\renewcommand{\algorithmicensure}{ \textbf{Output:}} %UseOutput in the format of Algorithm  

\algnewcommand\algorithmicforeach{\textbf{for each}}
\algdef{S}[FOR]{ForEach}[1]{\algorithmicforeach\ #1\ \algorithmicdo}

\usepackage{multirow}
\usepackage{array}
\usepackage[mathscr]{eucal}
\usepackage{chngpage}



%\copyrightyear{2017} 
%\acmYear{2017} 
%\setcopyright{acmlicensed}
%\acmConference{KDD'17}{}{August 13--17, 2017, Halifax, NS, Canada.}
%\acmPrice{15.00}
%\acmDOI{http://dx.doi.org/10.1145/XXXXXXX.XXXXXXX}
%\acmISBN{978-1-4503-4887-4/17/08} 
%Authors, replace the red X's with your assigned DOI string. See pdf attached to ACM rightsreview confirmation email.

%\fancyhead{}
%\settopmatter{printacmref=false}

\usepackage{authblk}
\usepackage{blindtext}


\begin{document}
\title{EmbedJoin: Efficient Edit Similarity Joins via Embeddings}
%\titlenote{Both authors are supported in part by NSF CCF-1525024 and IIS-1633215.}

\author{Haoyu Zhang\thanks{Email: hz30@umail.iu.edu} }
\author{Qin Zhang\thanks{Email: qzhangcs@indiana.edu} }
\affil{Department of Computer Science, Indiana University}
\date{}
\maketitle



\begin{abstract}
We study the problem of edit similarity joins, where given a set of strings and a threshold value $K$, we want to output all pairs of strings whose edit distances are at most $K$.  Edit similarity join is a fundamental problem in data cleaning/integration, bioinformatics, collaborative filtering and natural language processing, and has been identified as a primitive operator for database systems.  This problem has been studied extensively in the literature. However, we  have observed that all the existing algorithms fall short on long strings and large distance thresholds.  

In this paper we propose an algorithm named \ebdjoin+ which scales very well with string length and distance threshold.  Our algorithm is built on the recent advance of metric embeddings for edit distance, and is very different from all of the previous approaches.  We demonstrate via an extensive set of experiments that \ebdjoin+ significantly outperforms the previous best algorithms on long strings and large distance thresholds. 
%For example, on a collection of 20,000 real-world DNA sequences each of length 20,000 and a distance threshold that is 1\% of the string length (1\% errors), the previous best  algorithms that we have tested cannot finish in 10 hours, while \ebdjoin\ finished in less than 6 minutes. Moreover, \ebdjoin\ scales very well up to 20\% errors which is critical in applications such as bioinformatics, and is far beyond the reach of existing algorithms.
\end{abstract}



\maketitle

\input{intro}

\input{related}

\input{tools}

\input{algo}

\input{improved}

\input{exp}

\input{conclude}

\section{ACKNOWLEDGMENT}
The authors would like to thank Djamal Belazzougui and Michal Kouck\'y for many helpful discussions, and Haixu Tang and Diyue Bu for their help on preparing the human genome datasets.  The authors would also like to thank Michal Kouck\'y for introducing us the efficient implementation of the algorithm for computing exact edit distance by Ukkonen~\cite{Ukkonen85}.

\bibliographystyle{acm}
\bibliography{paper} 




%\appendix
%\input{appendix}

\end{document}
