%
%
%
%
%
%
%
%
%
%
%
%
%
%
%
%
%
%
%
%
\documentclass[%
 reprint,
superscriptaddress,
%
%
%
%
%
%
%
%
%
 amsmath,amssymb,
 aps,
%
%
%
%
%
%
onecolumn]{revtex4-2}
\usepackage{floatrow}
\usepackage[table,xcdraw,dvipsnames]{xcolor}
\usepackage{graphicx}%
\usepackage{dcolumn}%
\usepackage{bm}%
\usepackage{filecontents}
\usepackage{xr}
\makeatletter
\newcommand*{\addFileDependency}[1]{% argument=file name and extension
  \typeout{(#1)}
  \@addtofilelist{#1}
  \IfFileExists{#1}{}{\typeout{No file #1.}}
}
\makeatother

\newcommand*{\myexternaldocument}[1]{%
    \externaldocument{#1}%
    \addFileDependency{#1.tex}%
    \addFileDependency{#1.aux}%
}
\myexternaldocument{PRE-manuscript-final-upload-cleaned}
%
%
%
%
%
%
%
%
%
%
\usepackage{graphicx}
\usepackage{nccmath}
\usepackage{float}
%
%
\usepackage{mathrsfs}\newcommand{\rmd}{\mathrm{d}}
\usepackage[english]{babel}
\newcommand{\lb}{\langle}
\newcommand{\rb}{\rangle}
\newcommand{\comm}[1]{\textcolor{Brown}{#1}}
\global\long\def\E{\mathrm{E}}
\global\long\def\Cov{\mathrm{Cov}}
\global\long\def\CV{\mathrm{CV}}
 \global\long\def\Exp{\mathrm{Exp}}
 \global\long\def\sgn{\mathrm{sgn}}
 \global\long\def\Var{\mathrm{Var}}
\pdfpageattr {/Group << /S /Transparency /I true /CS /DeviceRGB>>} 
\usepackage{mathtools}
\newlength{\arrow}
\settowidth{\arrow}{\scriptsize$1000000000$}
\newcommand*{\myrightarrow}[1]{\xrightarrow{\mathmakebox[\arrow]{#1}}}
\newcommand*{\myxrightleftharpoons}[2]{\xrightleftharpoons[{#1}]{\mathmakebox[\arrow]{#2}}}
\newlength{\extralongarrow}
\settowidth{\extralongarrow}{\scriptsize$100000000$}
\newcommand*{\myextralongrightarrow}[1]{\xrightarrow{\mathmakebox[\extralongarrow]{#1}}}

\usepackage{hyperref}
\hypersetup{
    colorlinks=true,
    linkcolor=blue,
    filecolor=blue,      
    urlcolor=blue,
    citecolor=blue,
}

%
 
\renewcommand{\thesection}{\arabic{section}}
\renewcommand\thefigure{S\arabic{figure}}
\begin{document}

\preprint{APS/123-QED}



\title{\large{Inferring gene regulation dynamics from static snapshots of gene expression variability} \\ $\quad$ \\
\large{\emph{Supplementary Material}}}%
%



\author{Euan Joly-Smith}
\affiliation{Department of Physics, University of Toronto, 60 St.~George St., Ontario M5S 1A7, Canada}
%
\author{Zitong Jerry Wang}
\affiliation{California Institute of Technology, Division of Biology and Biological Engineering, Pasadena CA 91125, USA}
\author{Andreas Hilfinger}
\affiliation{Department of Physics, University of Toronto, 60 St.~George St., Ontario M5S 1A7, Canada}
%
\affiliation{Department of Mathematics, University of Toronto, 40 St.~George St., Toronto, Ontario M5S 2E4}
\affiliation{Department of Cell \& Systems Biology, University of Toronto, 25 Harbord St, Toronto, Ontario M5S 3G5}


%
             %


\maketitle
\numberwithin{equation}{section}




%
\vspace{-0.8cm}
\tableofcontents{}

\makeatletter\@input{xx.tex}\makeatother

\setcounter{page}{1}





\section{Inferring the timescale of upstream variability --- numerical bound}

Using Eq.~\eqref{EQ: no-oscillation 1-step eta bar autocorrelation integral}, we have 
\begin{align*}
 \eta_{\bar{x}\bar{x}} - a \eta_{\bar{y}\bar{y}} = 
 \int_{0}^{\infty} \eta_{RR}A_{R}(t) \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt \quad \text{where} \quad T \leq a \leq 1 \quad .
\end{align*}
 
The expression in parentheses is negative for $t \leq t^{*}$ and positive for $t \geq t^{*}$, where $t^{*} = \ln(a/T)\tau_{x}\tau_{y}/(\tau_{x} - \tau_{y})$. We thus have 
\begin{align}
 \eta_{\bar{x}\bar{x}} - a \eta_{\bar{y}\bar{y}} &= 
 \int_{0}^{t^{*}} \eta_{RR}A_{R}(t) \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt  + \int_{t^{*}}^{\infty} \eta_{RR}A_{R}(t) \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt \nonumber\\
 &\geq \int_{0}^{t^{*}} \eta_{RR} \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt  +  \int_{t^{*}}^{\infty} \eta_{RR}A_{R}(t) \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt \nonumber\\
 &\geq \int_{0}^{t^{*}} \eta_{RR} \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt  +  \int_{t^{*}}^{\infty} \eta_{RR}e^{-t/\tau_{R}} \left( \frac{e^{-t/\tau_{x}}}{\tau_{x}} - a \frac{e^{-t/\tau_{y}}}{\tau_{y}}\right)  dt \quad , \label{EQ: integral}
\end{align}
where in the second step we use the fact that the expression in parantheses is negative and $A_{R}(t) \leq 1$, and the third step comes from the fact that $A_{R}(t) \geq e^{-t/\tau_{R}}$. 
Without loss of generality we work in units where $\tau_{x} = 1$ and $\tau_{y} = T$, in which case we solve the above integral to obtain 
\begin{align}
  &\eta_{\bar{x}\bar{x}} - a \eta_{\bar{y}\bar{y}} \geq \eta_{RR} h(a,T,\tau_{R}) \\
  \text{where} \quad h(a,T,\tau_{R}) &= \left(1 - a  +   a\left(\frac{a}{T}\right)^{-\frac{1}{1 - T}} - \left(\frac{a}{T}\right)^{-\frac{T}{1 - T}} +  \frac{\tau_{R}\left(\frac{a}{T}\right)^{-\frac{(1+\tau_{R})T}{\tau_{R}(1-T)}}}{1+\tau_{R}} - \frac{a\tau_{R}\left(\frac{a}{T}\right)^{-\frac{(r+T)}{\tau_{R}(1-T)}}}{T+\tau_{R}} \right) \quad .\nonumber
\label{EQ: no-oscillation 1-step timescale a equation}
\end{align}
Now, we can infer that $h(a, T, \tau_{R})$ is zero for some $a^{*} \in [T,1]$. In particular, for $a = T$ we have
\begin{align*}
    h(T,T,\tau_{R}) = \frac{\tau_{R}}{1+\tau_{R}} - \frac{T\tau_{R}}{T + \tau_{R}} = \frac{\tau_{R}^{2}(1  - T)}{(1+\tau_{R})(T + \tau_{R})} > 0 \quad .
\end{align*}
Moreover, we note that the $\tau_{R}$ dependence of $h$ comes from the integral on the right in Eq.~\eqref{EQ: integral}, and since the expression in parenthesis in this integral is always positive, the integral will increase as $\tau_{R}$ increases. As a result, $h(a,T,\tau_{R})$ is a monotonically increasing function of $\tau_{R}$ and so $h(1,T,\tau_{R}) < h(1,T, \tau_{R} \to \infty) = 0$. Thus, we've shown that $h(a,T,\tau_{R})$ is positive for $a = T$ and negative for $a = 1$, and so by the mean value theorem there must exist an $a^{*}\in [T,1]$ s.t. $h(a^{*},T,\tau_{R}) = 0$. This $a^{*}$ can be found numerically for a given $\tau_{R}$ and $T$, and plugging this $a^{*}$ into Eq.~\eqref{EQ: no-oscillation 1-step timescale a equation} gives us the strictness bound on systems that satisfy $e^{-t/\tau_{R}} \leq A_{R}(t)$
\begin{align}
    \eta_{\bar{x}\bar{x}} - a^{*}\eta_{\bar{y}\bar{y}} \geq 0 \quad \text{where} \quad h(a^{*},T,\tau_{R}/\tau_{x}) = 0\quad ,
\end{align}
where the $\tau_{x}$ factor is added for systems with arbitrary units. We were unable to solve for an analytical solution to $a^{*}$ given $T$ and $\tau_{R}$. However, through trial and error, and by looking at the functional form of the solution of systems with exponential auto-correlations, we proposed the ansatz that $h(a^{*}_{ansatz},T,\tau_{R}/\tau_{x}) \geq 0$, where $a^{*}_{ansatz} =  (0.9 + T\tau_{x}/\tau_{R})/(0.9 + \tau_{x}/\tau_{R})$. By numerically plotting $h(a^{*}_{ansatz},T,\tau_{R}/\tau_{x})$ over different $T \in [0,1]$ and $\tau_{R} \in [0,100]$ (see Fig.~\ref{FIG: Timescales ansatz}), we find that $h(a^{*}_{ansatz},T,\tau_{R}/\tau_{x}) \geq 0$ in this regime of parameters, and so through Eq.~\eqref{EQ: no-oscillation 1-step timescale a equation} we have the slightly weaker bound
\begin{align}
    \eta_{\bar{x}\bar{x}} - a^{*}_{ansatz}\eta_{\bar{y}\bar{y}} \geq 0 \quad \text{where} \quad a^{*}_{ansatz} = \left( \frac{0.9 + \frac{T\tau_{x}}{\tau_{R}}}{0.9 + \frac{\tau_{x}}{\tau_{R}}}\right) \quad .
\end{align}

\begin{figure*}[htb!]
\vspace{-0.4cm}
\centering
  \includegraphics[width=0.45\columnwidth]{Figure-SI-1.pdf}
   \caption{Plot of $h(a^{*}_{ansatz},T,\tau_{R}/\tau_{x})$ over different $T$ and $\tau_{R}/\tau_{x}$ values, where $a^{*}_{ansatz} = (0.9+ T\frac{\tau_{x}}{\tau_{R}})/(0.9 + \frac{\tau_{x}}{\tau_{R}})$. Note that $h$ here is always positive and for most values of the domain it is approximately zero (the peak is $h_{max} \approx 0.02$).}
    \label{FIG: Timescales ansatz}
\end{figure*}

We recall from Eq.~\eqref{EQ: no-feedback 1-step barbar equals xR} that $\eta_{\bar{x}\bar{x}} = \eta_{xR}$ and $\eta_{\bar{y}\bar{y}} = \eta_{yR}$, and we can solve 
for $\eta_{xR}$ and $\eta_{yR}$ in terms of measurable (co)variances through Eq.~\eqref{EQ: 1-step eta bar in terms of covariances}. Substituting these into the above inequality 
gives us
\begin{align}
 (a - T)\rho_{xy} \leq \left( \frac{a + T}{1+T} \right) \left( \frac{CV_{x}}{CV_{y}} - T\frac{CV_{y}}{CV_{x}}\right) \quad \text{where} \quad  a = \left(\frac{0.9 + \frac{T\tau_{x}}{\tau_{R}}}{0.9 + \frac{\tau_{x}}{\tau_{R}}}\right) \quad .
 \label{EQ: stricter bound}
\end{align}
Systems that break the above inequality cannot satisfy $e^{-t/\tau_{R}} \leq A_{R}(t)$, and so the upstream variability must 
fluctuate with timescale smaller than $\tau_{R}$, see Fig.~\ref{FIG: Timescales}. 


For fluorescent protein reporters of the class of systems in Fig.~\ref{FIG: 3 step class of systems with maturation (manuscript)}A of the paper, we can follow the same analysis as above. In particular, 
we consider systems in which the following holds
\begin{align*}
 e^{-t/\tau_{R}} \leq A_{\bar{F}}(t) \quad \text{where} \quad \bar{F}(t) = \mathrm{E}\big[ F(\mathbf{u}(t), \mathbf{u}_{x}) | \mathbf{u}[-\infty,t] \big] \quad .
\end{align*}
That is, $\bar{F}(t)$ is the translation rate when we average out all the intrinsic fluctuations originating from the mRNA intrinsic system, and so 
$A_{\bar{F}}(t)$ characterizes the timescale of the upstream fluctuations originating from the cloud of components $\mathbf{u}(t)$. 
Similarly to the above, we have 
\begin{align*}
  \eta_{\bar{x}''\bar{x}''} - a \eta_{\bar{y}''\bar{y}''} 
  \geq \int_{g(t) < 0} \eta_{RR} g(t)  dt 
  +  \int_{g(t) \geq 0} \eta_{RR} e^{-t/\tau_{u}}g(t) dt \quad \\
  \text{where} \quad g(t) = \left( \frac{\tau_{mat,x}e^{-t/\tau_{mat,x}}-\tau''e^{-t/\tau''}}{\tau_{mat,x}^{2} - \tau''^{2}} - a \frac{\tau_{mat,y}e^{-t/\tau_{mat,y}}-\tau''e^{-t/\tau''}}{\tau_{mat,y}^{2} - \tau''^{2}}\right) \quad ,
\end{align*}
In order to find the strongest bound on these systems, the largest $T_{m} \leq a^* \leq 1$ value that would set the right hand side of the above inequality to zero would need to be found. This can be done 
numerically, where this $a^*$ value would depend on $\tau_{R}$, $T$, and $\tau''$. If $\tau'$ is not known, then the largest $a$ that would set the right hand side of the above inequality to be positive for all $\tau'$ can be used. In both of these cases we would end up with 
\begin{align*}
 a \eta_{\bar{y}''\bar{y}''} \leq \eta_{\bar{x}''\bar{x}''}  \quad \text{where} \quad a = a^{*} \text{ (see above paragraph)}\quad . 
\end{align*}
When there is no feedback, recall that Eq.~\eqref{EQ: no-feedback 3-step fluctuation-balance in terms of bared variables} and Eq.~\eqref{EQ: no-feedback 3-step intrinsic noise bound} hold. 
Using these equations with the above inequality we find 
\begin{align}
 (a - T_{m})\rho_{x''y''} \leq \left( \frac{a + T_{m}}{1+T_{m}} \right) \left( \frac{CV_{x''}}{CV_{y''}} - T_{m}\frac{CV_{y''}}{CV_{x''}}\right) \quad  .
\end{align}
Systems that break the above inequality must break the inequality given by $e^{-t/\tau_{R}} \leq A_{F}(t)$, and so the upstream variability must 
fluctuate with timescales smaller than $\tau_{R}$. 




{\section{Examples of non-ergodic systems}}
{
Non-ergodic systems are systems where time averages do not correspond to population averages. Though the cloud of components $\mathbf{u}(t)$ in Fig.~\ref{FIG: Class 1 space of solutions -- feedback (manuscript)}A is dynamic and can vary as a function of time, it is left unspecified and we do not need to assume ergodicity in $R(\mathbf{u}(t))$. As an example, consider the case where $\mathbf{u}(t)$ is such that the rate $R(\mathbf{u}(t))$ evolves from some initial value and eventually reaches one of two values, $\lambda_{1}$ or $\lambda_{2}$, from which it stays constant, see Fig.~\ref{FIG: Ergodicity}A. Here, let's say that the choice of states $\lambda_{1}$ or $\lambda_{2}$ is random with a probability of 1/2. If we follow any single cell over time from some fixed initial state, the transcription rate will evolve to either $R = \lambda_{1}$ or $R = \lambda_{2}$ with equal probability, and then stay constant. If we wait for stationarity, in which each system has left the transient state, then we have two sub-populations. Such a system, though not ergodic, will satisfy the constraints presented in the main text. In particular, this exact system will lie along the right boundary of the open-loop constraint given by Eq.~\eqref{EQ: No feedback constraint (manuscript)} of the main text. In previous sections we've conditioned on the history of the upstream variables $\mathbf{u}(t)$ for systems without feedback. For the system in Fig.~\ref{FIG: Ergodicity}A this corresponds to conditioning on $R = \lambda_{1}$ or $R = \lambda_{2}$. When we take the average over all histories we effectively average over the two $R$ states. 
}

\begin{figure*}[h]
\centering
  \includegraphics[width=0.89\columnwidth]{Figure-SI-2.pdf}
   \vspace{-.5em}
   \caption{
    {\textbf{Irreversible cell states are non-ergodic processes.} A) Here are possible realizations of the transcription rate $R(\mathbf{u}(t))$ in a particular system within the class of Fig.~\ref{FIG: Class 1 space of solutions -- feedback (manuscript)}A in which the rate R either remains at the constant state of $R=\lambda_{1}$ or $R=\lambda_{2}$ after the initial transient dynamics have decayed. In each realization, the rate has a probability of $1/2$ of growing into the $R = \lambda_{1}$ state or into the $R = \lambda_{2}$ state. This could, for example, model a population of cells in which each cell must irreversibly switch to one of two states. This system is not ergodic, as following an individual cell over time can only provide information on the sub-population that shares that cell's state, and not the whole population. B) A similar system as A, with one of the states having an oscillating transcription rate and the other a stochastic transcription rate. }}
    \label{FIG: Ergodicity}
\end{figure*}


{
We can consider a more general non-ergodic system in which the cells move into one of two states with equal probability, defined by either having a transcription rate $R_{1}(t)$ or $R_{2}(t)$. For example, $R_{1}$ could be a stochastic signal and $R_{2}$ could be an oscillation, see Fig.~\ref{FIG: Ergodicity}B. Here when we condition on the history we are also conditioning on the state of the cell by specifying whether the history is an oscillation or not. When we take the average over all histories we in effect take the average of all histories in each state and then average over each state. }

{
It is worth noting that for such non-ergodic systems with different transcription states, if there is any feedback between the downstream variables $X$ and $Y$ and the cloud of components $\mathbf{u}(t)$ in any of the different cell states, then that constitutes feedback in our framework and could break the open-loop constraint given by Eq.~\eqref{EQ: No feedback constraint (manuscript)}.}

{
Moreover, the bound on stochastic systems given by Eq.~\eqref{EQ: No oscillation constraint (manuscript)} bounds systems with non-negative autocorrelation. The autocorrelation here is the autocorrelation obtained by ensemble averages, defined by 
\begin{equation*}
A(s) : = \frac{\lb R(t+s) R(t) \rb - \lb R(t+s)\rb \lb R(t) \rb}{\Var[R(t)]} ,
\end{equation*}
where outer brackets are averages over the population, and the variance at the bottom is the variance over the population. For a non-ergodic system of the form in Fig.~\ref{FIG: Ergodicity}B, this autocorrelation would be an average between the autocorrelations of the two possible transcription rates. It's thus possible that a system will break the constraint given by Eq.~\eqref{EQ: No oscillation constraint (manuscript)} if only one of the sub-populations has an oscillating transcription rate. }


\iffalse
Just like the previous example, it follows from the laws of total expectation and total variance that 
\begin{align*}
    \langle x \rangle &= \frac{1}{2}\langle x_{1} \rangle + \frac{1}{2} \langle x_{2} \rangle \\
    \text{Var}(X) &= \left[\frac{1}{2}\text{Var}(X_{1}) + \frac{1}{2}\text{Var}(X_{2})\right] + \left[\frac{1}{2}(\langle x_{1} \rangle - \langle x_{2} \rangle )^{2} \right] \\
    \text{Cov}(X,Y) &= \left[\frac{1}{2}\text{Cov}(X_{1}, Y_{1}) + \frac{1}{2}\text{Cov}(X_{2}, Y_{2})\right] + \left[\frac{T}{2}(\langle x_{1} \rangle - \langle x_{2} \rangle )^{2} \right] \quad ,
\end{align*}
where the numbered subscripts represent averages over the respective sup-population (i.e. $\langle x_{1} \rangle = E[X|R = R_{1}]$ and $\text{Var}(X_{1}) = \text{Var}(X|R = R_{1})$). Let's consider the system where $R_{1}$ is a stochastic signal and $R_{2}$ is an oscillation. More precisely, we let 
\begin{align*}
    R_{1} = \lambda z \quad \text{ where } \quad z \myrightarrow{\lambda_{z}} z + 1\quad  & \quad z \myrightarrow{z/\tau_{z}}  z - 1 \quad , \\
    R_{2} = A\sin(\omega t + \gamma) + 4A 
\end{align*}
where $\gamma$ is a random variable that serves to de-synchronize the cells so that the ensemble is at stationarity. Each of these sub-systems are analytically tractable. In particular, using the flux-balance and the fluctuation balance relations in Eqs.~\eqref{EQ: 1-step flux-balance relations} and \eqref{EQ: 1-step fluctuation-balance relations} on the $R_{1}$ system we find 
\begin{align*}
    \langle x_{1} \rangle = \tau_{x}\lambda \langle z \rangle \quad \quad  \langle y_{1} \rangle = \tau_{y}\lambda \langle z \rangle \\
    \text{Var}(X_{1}) = \langle x_{1} \rangle + \frac{\tau_{x}^{2}\lambda^{2}\langle z \rangle }{1 + \frac{\tau_{x}}{\tau_{z}}} \quad \quad  \text{Var}(Y_{1}) = \langle y_{1} \rangle + \frac{\tau_{y}^{2}\lambda^{2}\langle z \rangle }{1 + \frac{\tau_{y}}{\tau_{z}}} \\ \text{Cov}(X_{1}, Y_{1}) = \left[ T\frac{\tau_{x}^{2}\lambda^{2}\langle z \rangle}{1 + \frac{\tau_{x}}{\tau_{z}}} + \frac{\tau_{y}^{2}\lambda^{2}\langle z \rangle}{1 + \frac{\tau_{y}}{\tau_{z}}}\right]\frac{1}{1+T} \quad .
\end{align*}
Similarly, for the $R_{2}$ system we have 
\begin{align*}
    \langle x_{2} \rangle = \langle y_{1} \rangle /T = 4\tau_{x}A \quad \text{Var}(X_{2}) = \langle x_{2} \rangle + \frac{A^{2}\tau_{x}^{2}}{2(1+\omega^{2}\tau_{x}^{2})} \quad \text{Var}(Y_{2}) = \langle y_{2} \rangle + \frac{A^{2}\tau_{y}^{2}}{2(1+\omega^{2}\tau_{y}^{2})} \quad \text{Cov}(X_{2}, Y_{2}) = \frac{A^{2}\tau_{x}\tau_{y}(1+\omega^{2}\tau_{x}\tau_{y})}{2(1+\omega^{2}\tau_{x}^{2})(1 + \omega^{2}\tau_{y}^{2})}
\end{align*}
Putting it all together we can solve for $CV_{x}/CV_{y}$ and $\rho_{xy}$ as messy expressions that depend on the parameters in the toy model. 
\fi


{\section{Flow cytometry data analysis}}


\begin{figure}[hbt!]
\includegraphics[width=0.8\columnwidth]{Figure-SI-3.pdf}
\caption{ 
{
\textbf{Filtering cells from debris based on scattering and fluorescence profiles.} Here we plot a figure illustrating the filtering strategy that was used to filter out debris from the raw flow cytometry datasets. The side (SSC) and forward (FSC) scattering signals were used to separate the cluster of cell-like objects from debris measurements that lie outside the cluster. Additionally, the green (FITC) and cyan (mCFP) signals were used with the scattering signals to separate the cluster of fluorescent objects. On the bottom right is the resulting green fluorescence distribution for mEGFP, with statistics showing the abundance of post-sorted events (with pre-sorted at 10,000), the mean, and the coefficient of variation.}}
\label{FIG: flow software}
\end{figure} 

{
For each fluorescent protein, coefficients of variation (CVs) were obtained for three independent cell cultures according to Fig.~\ref{FIG: flow software}, and the average was taken with an uncertainty given by the standard error. The CVs that were obtained this way correspond to abundance CVs. In order to obtain the concentration CVs, we used Eq.~\eqref{EQ: Appendix concentration CVs from abundance CVs (manuscript)} as described in Appendix \ref{SEC: Appendix Experimental data analysis}. Note that $CV_{V}$ can be estimated from first principles for a given cellular growth dynamics. For example, if we assume that the volume is growing exponentially between divisions with symmetric division times, then $CV_{V} \approx 0.2$ (similarly for a volume growing linearly). This is independent of the cell division time and can be used as a lower bound for $CV_{V}$.  In general, $CV_{V}$ will be bigger than 0.2 due to stochastic effects like asymmetric divisions and division times. We thus estimated $CV_{V}$ using separate data that explicitly quantifies \emph{E.~coli} growth dynamics through time-lapse microscopy \cite{wang2010robust}. In particular, in \cite{wang2010robust} individual \emph{E.~coli} cells are followed and their lengths are measured as a function of time for about 100 divisions, see Fig.~\ref{FIG: cell size time traces}. From the publicly available data we analyzed the length time-traces of the mother cells from one experiment with \emph{E.~coli} MG1655 (CGSC 6300) grown in LB media at 37$^\circ$C. This includes over 100 length time-traces like the one in Fig.~\ref{FIG: cell size time traces}, thus constituting approximately $10^4$ divisions. We noticed that many time-traces exhibited large fluctuations near the end (Fig.~\ref{FIG: cell size time traces}), which suggests a systematic error. Therefor, we computed the length CV from the first half of each time-trace, and then took the average, resulting in $CV_{V} = 0.261\pm0.005$. }

\begin{figure}[]
\includegraphics[width=0.68\columnwidth]{Figure-SI-4.pdf}
\caption{{ \textbf{Example time-trace of growing and dividing \emph{E.~coli} cells.} Data taken from \cite{wang2010robust} where individual \emph{E.~coli} cells were followed using time-lapse microscopy and their lengths measured over time. One pixel corresponds to $0.0645\mu m$. Many of these time-traces exhibited large fluctuations near the end, suggesting a possible systematic error. We therefor computed the CVs of the first half of each time-trace.}}
\label{FIG: cell size time traces}
\end{figure} 

\begin{table}[b]
\includegraphics[width=0.75\columnwidth]{Table-SI-1.pdf}
\caption{{\textbf{Maturation times and concentration CVs for the selected fluorescent proteins.} In the top row we present the 10 selected fluorescent proteins from the publicly available data that are most closely modelled by first-order maturation kinetics, along with their maturation half-life (related to the maturation time through $\tau_{mat} = \tau_{mat}^{50}/\ln(2)$), and the concentration CVs computed from the flow cytometry datasets. Numbers in parentheses are the uncertainties, taken from \cite{Balleza2018} for the maturation times and taken as the standard error of three replicate flow cytometry data sets along with error propagation for the CVs. In the white cells we report the maturation time ratios and the ratio of CVs. Our constraints are unique for a given pair of CVs and maturation times, thus we show only one ratio for a given fluorescent protein pair. Each panel corresponds to a set of reporter pairs where the $Y$ reporter is held fixed and the $X$ reporter is varied (equivalent to varying $\tau_{mat,x}$ while keeping $\tau_{may,y}$ fixed). Each column corresponds to a panel in Fig.~\ref{FIG: All data}.  } }
\label{TAB: Data}
\end{table}

{
Of the 50 fluorescent proteins that were studied in \cite{Balleza2018}, varying maturation kinetics were observed, including first-order exponential kinetics as assumed in Fig.~\ref{FIG: 3 step class of systems with maturation (manuscript)}, second-order kinetics, and other more complex maturation kinetics. As a result, two effective maturation times are reported: $\tau_{mat}^{50}$ and $\tau_{mat}^{90}$, corresponding to the time it takes for 50\% or 90\% of an ensemble of fluorescent proteins to become mature, respectively. The former, $\tau_{mat}^{50}$, is the maturation half-life, which for first-order maturation kinetics as assumed in Fig.~\ref{FIG: 3 step class of systems with maturation (manuscript)} is related to the average maturation time through $\tau_{mat}^{50}/\tau_{mat} = \ln(2)$, and to $\tau_{mat}^{90}$ through $\tau_{mat}^{50}/\tau_{mat}^{90} = \ln(2)/\ln(10)$. In order to pick a subset of fluorescent proteins that are the most closely modelled by a first-order maturation step, we picked the 10 fluorescent proteins for which the experimental ratio $\tau_{mat}^{50}/\tau_{mat}^{90}$ lies closest to $\ln(2)/\ln(10)$. However, we found that this algorithm would capture a few fluorescent proteins whose maturation curves in \cite{Balleza2018} were noisy and could possibly also be described by a second-order process. Therefore, we picked the 10 fluorescent proteins for which the ratio $\tau_{mat}^{50}/\tau_{mat}^{90}$ lies closest to $\ln(2)/\ln(10)$ and that also display a clear first-order trend in the maturation kinetic time-traces reported in \cite{Balleza2018}. The chosen fluorescent proteins with their $\tau_{mat}^{50}$ maturation times and their concentration CVs are presented in Table \ref{TAB: Data}. }








\section{Simulations}
Though the constraints presented in the main text are analytically proven, we performed gillespie simulations to demonstrate {that the entire bounded regions are achievable}. Here we summarize what systems were simulated.

To demonstrate the open-loop constraint we performed gillespie simulations of certain systems that are part of the classes of Fig.~\ref{FIG: Class 1 space of solutions -- feedback (manuscript)}A and Fig.~\ref{FIG: 3 step class of systems with maturation (manuscript)}A of the paper. The systems that were simulated include those driven by a sinusoidal, those driven by an oscillating step function, those driven by a poisson process $z$, and those driven by the square of a poisson process $z^2$. In all these cases many simulations were performed with varying reporter lifetimes and reporter birthrate parameters (like the sinusoidal amplitude and frequency, or the half-life of the upstream poisson process, etc.). The time (co)variances for these particular system realizations was then computed, and these correspond to the blue dots in the paper figures. For closed-loop systems, we performed gillespie simulations of systems driven by rates with feedback. In particular, to fill most of the region in the figures, rates of the form $R(x) \propto (K + x^{n})^{-1}$, $R(y) \propto (K + y^{n})^{-1}$, and $R(y) \propto (K + y^{n} + x^{m})^{-1}$ were done over different $n$, $m$, and $K$. The region outside of the orange lines in Fig.~\ref{FIG: Class 1 space of solutions -- feedback (manuscript)}B and Fig.~\ref{FIG: 3 step class of systems with maturation (manuscript)}B near the $\rho = 1$ line was only accessible by rates of the form $R(x,y) \propto U(x - ky)$, where $U$ is a step function and $k$ is a parameter that was varied. 

To demonstrate the constraint on stochastic systems we performed gillespie simulations of open-loop systems that are known to be stochastic and periodic. These include those driven by a poisson process and those driven by a sinusoidal. We performed many simulations where the parameters of these specified birthrates were varied.

To simulate concentrations of growing and dividing cells, we performed gillespie simulations of molecular numbers that undergo a binomial split at particular times separated by the same period $\ln(2)\tau_{c}$. The volume was then modelled as an exponentially growing volume $V \propto e^{t/\tau_{c}}$ that is reduced by a factor of 1/2 at the same moments that the molecular numbers undergo a binomial split. The reporter numbers and the cell volume were kept track of, and from these the concentration (co)variances were computed from the time averages of $x(t)/V(t)$ and $y(t)/V(t)$. The concentration production rates $R_{c} := R/V$ that we simulated include poisson processes, sinusoidals, rates proportional and inversely proportional to the cell volume, and oscillating step functions.  


%

%
%
%
%
%
%
%
%
%
\begin{thebibliography}{2}%
\makeatletter
\providecommand \@ifxundefined [1]{%
 \@ifx{#1\undefined}
}%
\providecommand \@ifnum [1]{%
 \ifnum #1\expandafter \@firstoftwo
 \else \expandafter \@secondoftwo
 \fi
}%
\providecommand \@ifx [1]{%
 \ifx #1\expandafter \@firstoftwo
 \else \expandafter \@secondoftwo
 \fi
}%
\providecommand \natexlab [1]{#1}%
\providecommand \enquote  [1]{``#1''}%
\providecommand \bibnamefont  [1]{#1}%
\providecommand \bibfnamefont [1]{#1}%
\providecommand \citenamefont [1]{#1}%
\providecommand \href@noop [0]{\@secondoftwo}%
\providecommand \href [0]{\begingroup \@sanitize@url \@href}%
\providecommand \@href[1]{\@@startlink{#1}\@@href}%
\providecommand \@@href[1]{\endgroup#1\@@endlink}%
\providecommand \@sanitize@url [0]{\catcode `\\12\catcode `\$12\catcode
  `\&12\catcode `\#12\catcode `\^12\catcode `\_12\catcode `\%12\relax}%
\providecommand \@@startlink[1]{}%
\providecommand \@@endlink[0]{}%
\providecommand \url  [0]{\begingroup\@sanitize@url \@url }%
\providecommand \@url [1]{\endgroup\@href {#1}{\urlprefix }}%
\providecommand \urlprefix  [0]{URL }%
\providecommand \Eprint [0]{\href }%
\providecommand \doibase [0]{https://doi.org/}%
\providecommand \selectlanguage [0]{\@gobble}%
\providecommand \bibinfo  [0]{\@secondoftwo}%
\providecommand \bibfield  [0]{\@secondoftwo}%
\providecommand \translation [1]{[#1]}%
\providecommand \BibitemOpen [0]{}%
\providecommand \bibitemStop [0]{}%
\providecommand \bibitemNoStop [0]{.\EOS\space}%
\providecommand \EOS [0]{\spacefactor3000\relax}%
\providecommand \BibitemShut  [1]{\csname bibitem#1\endcsname}%
\let\auto@bib@innerbib\@empty
%
\bibitem [{\citenamefont {Wang}\ \emph {et~al.}(2010)\citenamefont {Wang},
  \citenamefont {Robert}, \citenamefont {Pelletier}, \citenamefont {Dang},
  \citenamefont {Taddei}, \citenamefont {Wright},\ and\ \citenamefont
  {Jun}}]{wang2010robust}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {P.}~\bibnamefont
  {Wang}}, \bibinfo {author} {\bibfnamefont {L.}~\bibnamefont {Robert}},
  \bibinfo {author} {\bibfnamefont {J.}~\bibnamefont {Pelletier}}, \bibinfo
  {author} {\bibfnamefont {W.~L.}\ \bibnamefont {Dang}}, \bibinfo {author}
  {\bibfnamefont {F.}~\bibnamefont {Taddei}}, \bibinfo {author} {\bibfnamefont
  {A.}~\bibnamefont {Wright}},\ and\ \bibinfo {author} {\bibfnamefont
  {S.}~\bibnamefont {Jun}},\ }\bibfield  {title} {\bibinfo {title} {Robust
  growth of escherichia coli},\ }\href@noop {} {\bibfield  {journal} {\bibinfo
  {journal} {Current biology}\ }\textbf {\bibinfo {volume} {20}},\ \bibinfo
  {pages} {1099} (\bibinfo {year} {2010})}\BibitemShut {NoStop}%
\bibitem [{\citenamefont {Balleza}\ \emph {et~al.}(2018)\citenamefont
  {Balleza}, \citenamefont {Mark~Kim},\ and\ \citenamefont
  {Cluzel}}]{Balleza2018}%
  \BibitemOpen
  \bibfield  {author} {\bibinfo {author} {\bibfnamefont {E.}~\bibnamefont
  {Balleza}}, \bibinfo {author} {\bibfnamefont {J.}~\bibnamefont {Mark~Kim}},\
  and\ \bibinfo {author} {\bibfnamefont {P.}~\bibnamefont {Cluzel}},\
  }\bibfield  {title} {\bibinfo {title} {Systematic characterization of
  maturation time of fluorescent proteins in living cells},\ }\href@noop {}
  {\bibfield  {journal} {\bibinfo  {journal} {Nature Methods}\ }\textbf
  {\bibinfo {volume} {15}},\ \bibinfo {pages} {47} (\bibinfo {year}
  {2018})}\BibitemShut {NoStop}%
\end{thebibliography}%



\end{document}