\begin{filecontents}{PRE-manuscript-final-upload-cleaned.aux}
\relax 
\bbl@cs{beforestart}
\citation{Thattai2001,Ozbudak2002,Blake2003,Bar-Even2006,Newman2006,Eldar2010,elf2018,kaern2005}
\citation{hilfinger2015b}
\citation{Taniguchi2010}
\citation{Cai2014}
\citation{Elowitz2002}
\citation{Elowitz2002,Raser2004,maamar2007}
\citation{Balleza2018}
\newlabel{FirstPage}{{}{1}{}{}{}}
\@writefile{toc}{\contentsline {title}{Inferring gene regulation dynamics from static snapshots of gene expression variability}{1}{}}
\@writefile{toc}{\contentsline {abstract}{Abstract}{1}{}}
\@writefile{toc}{\contentsline {section}{\numberline {I}Introduction}{1}{}}
\citation{shen2002network}
\citation{Raser2004,Bar-Even2006,Elowitz2002,baudrimont2019contribution}
\citation{hilfinger2015b}
\citation{hilfinger2015a}
\citation{baudrimont2019contribution,raj2006stochastic}
\@writefile{toc}{\contentsline {section}{\numberline {II}Detecting gene regulation feedback from static snapshots of population heterogeneity}{2}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {A}Mathematical correlation constraints for open-loop ``dual reporters''}{2}{}}
\newlabel{EQ: Definition of 1-step Dual-Reporter System}{{1}{2}{}{}{}}
\newlabel{EQ: No feedback constraint (manuscript)}{{2}{2}{}{}{}}
\citation{golding2013}
\citation{Bar-Even2006}
\citation{Balleza2018}
\citation{Balleza2018}
\@writefile{lof}{\contentsline {figure}{\numberline {1}{\ignorespaces  \textbf  {Feedback in gene regulation affects the space of possible mRNA covariability.} A) We consider all stochastic processes in which two components, X and Y, are made with an identical, but unspecified, rate. This rate can depend in any way on a cloud of unknown components $\mathbf  {u}(t)$, which in turn can depend in arbitrary ways on the number of X and Y molecules. The shared production rate of X and Y, together with first-order degradation of X and Y with respective lifetimes $\tau _x$ and $\tau _y$, are the only specified parts within this arbitrarily large network, as defined in Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Definition of 1-step Dual-Reporter System}\unskip \@@italiccorr )}}. B) Space of possible covariability for different values of the life-time ratio $T:=\tau _y / \tau _x$. All systems must satisfy \mbox  {$\rho _{xy} (1+T) \leqslant CV_{x}/CV_{y} + TCV_{y}/CV_{x}$} corresponding to the area below the dashed orange line. Allowing for feedback (grey dots), the entire space below the dashed orange line is accessible. In the absence of feedback (blue dots), $\rho _{xy}$ is additionally constrained by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No feedback constraint (manuscript)}\unskip \@@italiccorr )}}, corresponding to the region between the solid orange lines. Dots are stochastic simulation data for specific models of co-regulated genes within the class defined in Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Definition of 1-step Dual-Reporter System}\unskip \@@italiccorr )}}. {Plotted are a subset of simulations with arbitrary density to demonstrate the accessibility of the constrained regions. Blue and grey curves are exemplary toy models (Appendix \ref  {SEC: Appendix particular systems}) that illustrate the effect of decreasing transcription rate variability (blue) or increasing feedback strength (grey)}. C) Experimental set-up to detect feedback in the regulation of a native gene of interest \emph  {geneX}. The native and reporter genes are regulated by identical promoters in the same cell. If the covariability of the transcripts X and Y of such genes falls outside the open-loop constraint of Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No feedback constraint (manuscript)}\unskip \@@italiccorr )}}, we can conclude that the gene of interest \emph  {geneX} regulates its own transcription. {To maximize the discriminatory power of the approach the reporter \emph  {geneY} should be engineered to be a passive read-out of transcription without significantly affecting gene expression.}}}{3}{}}
\newlabel{FIG: Class 1 space of solutions -- feedback (manuscript)}{{1}{3}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {B}Experimentally exploiting mRNA correlations to detect feedback}{3}{}}
\@writefile{lof}{\contentsline {figure}{\numberline {2}{\ignorespaces  \textbf  {Feedback in gene regulation affects the possible covariability of fluorescent protein measurements.} A) As before X and Y correspond to co-regulated mRNA, but we now explicitly include the dynamics of immature fluorescent proteins denoted as X' and Y', as well as mature fluorescent proteins X'' and Y''. Because fluorescent proteins are typically stable and are thus effectively diluted with a common ``degradation'' time set by the cell-cycle, we focus on gene expression dynamics that is symmetric apart from the maturation step. The asymmetry between the co-regulated genes is then entirely characterized by the ratio of average fluorescent maturation times $T_\mathrm  {m} := \tau _{\mathrm  {mat},y} / \tau _{\mathrm  {mat},x}$. B) Without feedback control (blue dots), fluorescence correlations are constrained to the region between the dashed and solid orange lines, the latter corresponding to the bound of Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no-feedback constraint (manuscript)}\unskip \@@italiccorr )}}. Allowing for feedback (grey dots), the entire region becomes available. Correlations in fluorescence levels can thus be used to detect causal feedback in gene regulation from static population snapshots. {Dots are selected numerical simulations of specific fluorescent reporter systems within this class that illustrate the full accessibility of the region constrained by the analytically proven bounds.}}}{4}{}}
\newlabel{FIG: 3 step class of systems with maturation (manuscript)}{{2}{4}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {C}Experimentally exploiting fluorescent protein correlations to detect feedback}{4}{}}
\newlabel{EQ: two-step no-feedback constraint (manuscript)}{{3}{4}{}{}{}}
\citation{Raser2004,Bar-Even2006,Elowitz2002,baudrimont2019contribution}
\citation{eling2019challenges}
\citation{baudrimont2019contribution,raj2006stochastic}
\citation{emory19925,cheneval2010review}
\@writefile{toc}{\contentsline {section}{\numberline {III}Distinguishing stochastic from deterministic transcription rate variability}{5}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {A}Mathematical correlation constraints for stochastic transcriptional noise}{5}{}}
\newlabel{EQ: No oscillation constraint (manuscript)}{{4}{5}{}{}{}}
\citation{Raj2013}
\@writefile{lof}{\contentsline {figure}{\numberline {3}{\ignorespaces  \textbf  { Periodic transcriptional variability can be distinguished from stochastic variability {without experimental access to transcription rates and without following individual cells over time.}} A) In a noisy cellular milieu, the auto-correlation of a periodic signal is not perfectly periodic but decays. We thus operationally define a signal as periodic when its periodicity is strong enough such that its auto-correlation function becomes negative at some point. Conversely, we classify signals with non-negative auto-correlations as stochastic. B) The left-side of the ``open-loop'' region defined by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No feedback constraint (manuscript)}\unskip \@@italiccorr )}} is only accessible by mRNA reporters that are driven by oscillatory production rates rather than purely stochastic upstream variability. The boundary (dashed black) line is defined by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No oscillation constraint (manuscript)}\unskip \@@italiccorr )}}. C) Sequential measurements of mRNA reporters X and Y, or fluorescent proteins X'' and Y'', can be used to discriminate between stochastic and oscillatory transcription rates if a system violates the respective bounds of Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No oscillation constraint (manuscript)}\unskip \@@italiccorr )}} or Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no oscillation constraint (manuscript)}\unskip \@@italiccorr )}} (indicated by dashed black line). {Selected numerical simulation (dots) illustrate the achievability of the constrained regions, and the arrowed curves indicate how specific models (Appendix \ref  {SEC: Appendix particular systems}) behave as the downstream response becomes slower. For these systems, we find that oscillatory systems cross the black dashed line when $2\pi \sqrt  {\tau _{x}\tau _{y}}$ or $2\pi \sqrt  {\tau _{\mathrm  {mat},x}\tau _{\mathrm  {mat},y}}$ become slower than the period of the driving oscillation. To detect oscillations it is thus advantageous to choose slow reporters. } D) Periodic transcription rates of a gene of interest \textit  {geneZ} can be experimentally detected either utilizing mRNA reporters, X and Y (left panel), or fluorescent protein reporters, X'' and Y'' (right panel), driven by the promoter of \textit  {geneZ}. If experimental reporters violate Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No oscillation constraint (manuscript)}\unskip \@@italiccorr )}} or \textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no oscillation constraint (manuscript)}\unskip \@@italiccorr )}}, they land to the left of the dashed black line (panel C), and \emph  {geneZ} must be driven by a periodically varying transcription rate.}}{6}{}}
\newlabel{FIG: Class 1 space of solutions -- oscillations (manuscript)}{{3}{6}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {B}Experimentally exploiting mRNA correlations to detect periodic transcription rates}{6}{}}
\citation{baudrimont2019contribution,raj2006stochastic}
\citation{Raser2004,Bar-Even2006,Elowitz2002}
\citation{Balleza2018}
\citation{Balleza2018}
\citation{Balleza2018}
\citation{Balleza2018}
\@writefile{toc}{\contentsline {subsection}{\numberline {C}Experimentally exploiting fluorescent protein correlations to detect periodic transcription rates}{7}{}}
\newlabel{EQ: two-step no oscillation constraint (manuscript)}{{5}{7}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {IV}Applying theoretical bounds to experimental data}{7}{}}
\newlabel{SEC: Practical Applicability}{{IV}{7}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {A}Unknown life-time ratios}{7}{}}
\newlabel{EQ: T (manuscript)}{{6}{7}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {B}Proportional transcription rates}{7}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {C}Systematic undercounting}{7}{}}
\@writefile{lof}{\contentsline {figure}{\numberline {4}{\ignorespaces  \textbf  {Utilizing bounds to analyze concentrations in growing and dividing cells.} A) We want to distinguish stochastically varying transcription rates whose average remains constant throughout the cell-cycle (red line) from transcription rates that change periodically during the cell-cycle (exemplified by blue line). Numbers of molecules oscillate in both types of systems but only systems with cell-cycle dependent driving oscillate in \emph  {concentrations}. {Our constraints can identify such periodic gene expression from static snapshots of population variability without access to time-series data.} B) Solid orange lines denote open-loop constraints proven for systems in which the reporter concentrations are independent of the cell volume, under the assumption that the cell volume grows exponentially between division events. For mRNA reporters (left panel) concentration correlations are then bounded by the same constraints derived for absolute numbers in the absence of feedback. When transcription rates are periodic, the open-loop constraint of Eq.~\ref  {EQ: No feedback constraint (manuscript)} for absolute numbers does not strictly apply to concentrations. Numerical simulations of example systems (blue dots) suggest that such systems only fall fall marginally outside the orange bounds. For fluorescent reporters (right panel), the feedback bounds (orange lines) for concentrations depend on the additional parameter \mbox  {$T_{m}^{c} := (1/\tau _{\mathrm  {mat},x} + \qopname  \relax o{ln}(2)/\tau _c)/(1/\tau _{\mathrm  {mat},y} + \qopname  \relax o{ln}(2)/\tau _c)$} where $\tau _c$ is the average cell-cycle time, see Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no-feedback constraint concentrations (manuscript)}\unskip \@@italiccorr )}}. However, for both mRNA and fluorescence levels, Eqs.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No oscillation constraint (manuscript)}\unskip \@@italiccorr )}} and \textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no oscillation constraint (manuscript)}\unskip \@@italiccorr )}} (dashed black lines) can be used to detect cell cycle dependent genes from asynchronous static snapshots of dual reporter concentrations. { C) Previously reported gene expression variability data for constitutively expressed fluorescent proteins that exhibit first order maturation dynamics in \emph  {E.~coli} \cite  {Balleza2018}. Plotted are variability ratios and reporter asymmetries with respect to the observed dynamics of mEYFP. As expected for constitutively expressed fluorescent proteins, the experimental data is consistent with the class of gene expression models that do not exhibit feedback and are not periodically driven (pink region) bounded by Eq.\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no oscillation constraint (manuscript)}\unskip \@@italiccorr )}} and Eq.\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step sequential no-feedback constraint (manuscript)}\unskip \@@italiccorr )}}. With the exception of the indicated mEGFP outlier, all data fall along the right hand boundary that is only accessible for systems whose fluorescence variability is dominated by a translation, maturation, or machine read-out step. The yellow corridor indicates the estimated uncertainty in reported cell-cycle time $\tau _c =(28.5\pm 2)$~min and the reference mEYFP maturation half-life $\tau _{mat,y}\qopname  \relax o{ln}(2) = (9.0\pm 0.7)$~min. Variability ratios with respect to all other reference fluorescence proteins confirm the above picture with the exception of the observed mEGFP variability which violates the expected behaviour, see Appendix \ref  {SEC: Appendix Experimental data analysis}. } }}{8}{}}
\newlabel{FIG: Concentrations (manuscript)}{{4}{8}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {D}Concentration measurements instead of absolute numbers}{8}{}}
\citation{Balleza2018}
\citation{Balleza2018}
\citation{Balleza2018}
\citation{Elowitz2002,raj2006stochastic}
\citation{Balleza2018}
\citation{galbusera2020using,grun2014validation}
\citation{galbusera2020using}
\citation{grun2014validation}
\citation{raj2008imaging}
\citation{eng2019transcriptome}
\newlabel{EQ: two-step no-feedback constraint concentrations (manuscript)}{{7}{9}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {E}Data from constitutively expressed fluorescent proteins fall within the expected bounds}{9}{}}
\newlabel{EQ: two-step sequential no-feedback constraint (manuscript)}{{8}{9}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {F}Measurement noise \& technological limitations}{9}{}}
\@writefile{toc}{\contentsline {section}{\numberline {V}Discussion}{9}{}}
\citation{Rhee2014}
\citation{vank1992,Lestas2008}
\citation{Lestas2008}
\citation{hilfinger2015a}
\@writefile{toc}{\contentsline {section}{\numberline {}Acknowledgments}{10}{}}
\@writefile{toc}{\appendix }
\@writefile{toc}{\contentsline {section}{\numberline {}Appendix}{10}{}}
\@writefile{toc}{\contentsline {section}{\numberline {A}Fluctuation balance relations for systems as defined in Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Definition of 1-step Dual-Reporter System}\unskip \@@italiccorr )}}}{10}{}}
\newlabel{SEC: Appendix fluctuation balance relations}{{A}{10}{}{}{}}
\newlabel{EQ: Appendix general flux balance equation}{{A1}{10}{}{}{}}
\newlabel{EQ: Appendix general covariance balance equation}{{A2}{10}{}{}{}}
\newlabel{EQ: 1-step flux-balance relations}{{A3}{10}{}{}{}}
\newlabel{EQ: 1-step fluctuation-balance relations}{{A4}{10}{}{}{}}
\citation{Hilfinger2011,Hilfinger2012}
\citation{Hilfinger2011}
\citation{Hilfinger2011}
\@writefile{toc}{\contentsline {section}{\numberline {B}Constraints on open-loop systems for systems as define in Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Definition of 1-step Dual-Reporter System}\unskip \@@italiccorr )}}}{11}{}}
\newlabel{Appendix section on open-loop constraint}{{B}{11}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1}Derivation of the open-loop constraint Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No feedback constraint (manuscript)}\unskip \@@italiccorr )}}}{11}{}}
\newlabel{EQ: no-feedback 1-step differential equations}{{B1}{11}{}{}{}}
\newlabel{EQ: no-feedback 1-step barbar equals xR}{{B2}{11}{}{}{}}
\newlabel{EQ: 1-step no-feedback fluctuation-balance}{{B3}{11}{}{}{}}
\newlabel{EQ: 1-step no-feedback conditioned averages}{{B4}{11}{}{}{}}
\newlabel{EQ: 1-step eta bar in terms of covariances}{{B5}{11}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2}Discriminating types of feedback}{11}{}}
\newlabel{SEC: Appendix types of feedback}{{B\tmspace  +\thinmuskip {.1667em}2}{11}{}{}{}}
\newlabel{EQ: no-feedback 1-step no feedback in X}{{B6}{12}{}{}{}}
\newlabel{EQ: no-feedback 1-step no feedback in Y}{{B7}{12}{}{}{}}
\@writefile{lof}{\contentsline {figure}{\numberline {5}{\ignorespaces  \textbf  {The space of possible covariability between mRNA-levels of co-regulated genes depends on the type of feedback} A) Systems that have no feedback in $X$ must lie in the region bounded by the light-red solid lines which corresponds to the region bounded by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: no-feedback 1-step no feedback in X}\unskip \@@italiccorr )}}. Systems that lie outside of this region must have feedback in X. Similarly, systems that have no feedback in $Y$ must lie in the region bounded by the light-blue solid lines which corresponds to the region bounded by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: no-feedback 1-step no feedback in Y}\unskip \@@italiccorr )}}. B) Here we define feedback to be negative when the mRNA transcription rate is negatively correlated with the mRNA levels: $\eta _{xR} < 0$. This would correspond to highly regulated systems that exhibit noise suppression. Using co-regulated reporters, $\eta _{xR}$ and $\eta _{yR}$ can be measured from static (co)variance measurements of $X$ and $Y$ using Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: 1-step eta bar in terms of covariances}\unskip \@@italiccorr )}}.}}{12}{}}
\newlabel{FIG: Types of feedback}{{5}{12}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {C}Dynamics from static transcript variability}{12}{}}
\newlabel{Appendix section on time-scales}{{C}{12}{}{}{}}
\newlabel{EQ: no-oscillation 1-step eta bar autocorrelation integral}{{C1}{12}{}{}{}}
\newlabel{EQ: Appendix stochastic Laplace}{{C2}{12}{}{}{}}
\citation{SI}
\citation{SI}
\citation{SI}
\citation{davenport1958introduction}
\newlabel{EQ: Appendix eta time-scale result}{{C3}{13}{}{}{}}
\newlabel{EQ: Fluctuation timescale bound (manuscript)}{{C4}{13}{}{}{}}
\newlabel{EQ: no-feedback 1-step single bar variances from Fourier}{{C5}{13}{}{}{}}
\@writefile{lof}{\contentsline {figure}{\numberline {6}{\ignorespaces  \textbf  {The space of possible correlations between co-regulated reporters depends on the timescale of the upstream fluctuations.} A) The timescale of stochastic fluctuations is characterized by how quickly the auto-correlation of the signal (red) decays to zero. By bounding the auto-correlation from below by $e^{-t/\tau _{R}}$ (black dashed curve) we set a lower bound on how quickly the signal auto-correlation can decay to zero. This thus sets a lower bound on the "speed" of the signal fluctuations. B) Open-loop systems in which the upstream auto-correlation is bounded below by $e^{-t/\tau _{R}}$ are bounded further towards the right. The solid dark red lines correspond to the loose bound given by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Fluctuation timescale bound (manuscript)}\unskip \@@italiccorr )}}, whereas the black dashed lines are given by the stricter nuerical bound presented in the supplement \cite  {SI}. These constraints can be used to gain information on the timescale of upstream fluctuations with only access to static snapshots of dual reporter transcript concentrations obtained, e.g., from single-cell sequencing methods that report snapshots of transcript abundances rather than temporal information.}}{13}{}}
\newlabel{FIG: Timescales}{{6}{13}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {}Setting bounds on the spectral density of upstream influences using co-regulated reporters}{14}{}}
\newlabel{EQ: spec 1}{{C6}{14}{}{}{}}
\newlabel{EQ: spec 2}{{C7}{14}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {D}The general class of co-regulated fluorescent proteins}{14}{}}
\newlabel{SEC: General class of fluorescent proteins}{{D}{14}{}{}{}}
\citation{hilfinger2015a}
\@writefile{lof}{\contentsline {figure}{\numberline {7}{\ignorespaces  \textbf  {General class of dual reporter systems with co-regulated fluorescent proteins} We consider a class of systems identical to Fig.~\ref  {FIG: 3 step class of systems with maturation (manuscript)}A with the exception that we now include multiple unspecified steps in the intrinsic dynamics of gene expression. Here $\mathbf  {u}_{x}$ and $\mathbf  {u}_{y}$ are identical (though independent) systems of components that are affected by the upstream cloud of components $\mathbf  {u}(t)$ in the same way. These two smaller clouds of components model the intrinsic steps in gene expression and allow for a wide range of possible mRNA dynamics {and post-translational modifications that occur before the maturation step. They are left unspecified except for the fact that we assume they do not form circuits of components that can create oscillations, so that any oscillatory variability is caused by the shared environment $\mathbf  {u}(t)$.} We model the dynamics of immature fluorescent proteins denoted as X' and Y', as well as mature fluorescent proteins X'' and Y''. The birthrate of X' and Y' can now depend on the components in $\mathbf  {u}_{x}$ and $\mathbf  {u}_{y}$ respectively, in an arbitrary way. The asymmetry between the co-regulated genes is characterized by the ratio of average fluorescent maturation times $T_\mathrm  {m} := \tau _{\mathrm  {mat},y} / \tau _{\mathrm  {mat},x}$. }}{15}{}}
\newlabel{FIG: Class of systems for FPs}{{7}{15}{}{}{}}
\newlabel{EQ: 3-step flux-balance}{{D1}{15}{}{}{}}
\newlabel{EQ: 3-step fluctuation-balance}{{D2}{15}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {E}Constraints on open-loop systems for systems as define in Fig.~\ref  {FIG: 3 step class of systems with maturation (manuscript)}A }{15}{}}
\newlabel{Appendix section on open-loop constraint for fluorescent reporters}{{E}{15}{}{}{}}
\citation{davenport1958introduction}
\citation{davenport1958introduction}
\citation{Hilfinger2012}
\newlabel{EQ: no-feedback 3-step single bar variances from Fourier}{{E1}{16}{}{}{}}
\newlabel{EQ: no-feedback 3-step etaxy covariance equation}{{E2}{16}{}{}{}}
\newlabel{EQ: no-feedback 3-step double bar differential equations}{{E3}{17}{}{}{}}
\newlabel{EQ: 3-step no-feedback double bar extrinsic}{{E4}{17}{}{}{}}
\newlabel{EQ: no-feedback 3-step double bar variances from Fourier}{{E5}{17}{}{}{}}
\newlabel{EQ: no-feedback 3-step fluctuation-balance in terms of bared variables}{{E6}{17}{}{}{}}
\newlabel{EQ: no-feedback 3-step intuitive no-feedback bound}{{E7}{18}{}{}{}}
\newlabel{EQ: no-feedback 3-step intrinsic noise bound}{{E8}{18}{}{}{}}
\newlabel{EQ: no-feedback-3step-extrinsic-noise-bounds}{{E9}{19}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {F}Dynamics from static fluorescent protein variability}{19}{}}
\newlabel{Appendix section on time-scales for fluorescent reporters}{{F}{19}{}{}{}}
\newlabel{SEC: Appendix dynamics fluorescent proteins}{{F}{19}{}{}{}}
\newlabel{EQ: fp strong constraint}{{F1}{19}{}{}{}}
\citation{Balleza2018}
\@writefile{toc}{\contentsline {section}{\numberline {G}Behaviour of specific example systems}{20}{}}
\newlabel{SEC: Appendix particular systems}{{G}{20}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {H}The effects of stochastic undercounting on dual reporter correlations}{20}{}}
\newlabel{SEC: Undercounting}{{H}{20}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1}Undercounting mRNA}{20}{}}
\newlabel{EQ: Undercounting-1step-nofeedback-bound}{{H1}{20}{}{}{}}
\newlabel{EQ: Undercounting-1step-no-oscillation-bound}{{H2}{20}{}{}{}}
\citation{hilfinger2015a}
\newlabel{EQ: Undercounting-1step-covariance-relations}{{H3}{21}{}{}{}}
\newlabel{EQ: Undercounting-1step-intrinsic-ratio}{{H4}{21}{}{}{}}
\newlabel{EQ: Undercounting-1step-extrinsic-noise}{{H5}{21}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2}Undercounting fluorescent proteins}{21}{}}
\newlabel{SEC: Undercounting fluorescent proteins}{{H\tmspace  +\thinmuskip {.1667em}2}{21}{}{}{}}
\newlabel{EQ: no-feedback 3-step intuitive no-feedback bound-final}{{H6}{22}{}{}{}}
\newlabel{EQ: Undercounting-3step-no-oscillation-bound}{{H7}{22}{}{}{}}
\newlabel{EQ: Undercounting-3step-covariance-relations-1}{{H8}{22}{}{}{}}
\newlabel{EQ: Undercounting-3step-eta-xr-to-eta-x}{{H9}{22}{}{}{}}
\newlabel{EQ: Undercounting-3step-covariance-relations}{{H10}{22}{}{}{}}
\newlabel{EQ: no-feedback 3-step intuitive no-feedback bound-2}{{H11}{22}{}{}{}}
\citation{hilfinger2015a}
\citation{hilfinger2015a}
\newlabel{EQ: Undercounting-3step-extrinsic-noise}{{H12}{23}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {I}Genes with proportional transcription rates}{23}{}}
\newlabel{SEC: Proportional rates}{{I}{23}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1}mRNA with proportional transcription rates}{23}{}}
\newlabel{EQ: Proportional-rates-1step-reactions}{{I1}{23}{}{}{}}
\newlabel{EQ: Proportional-1step-nofeedback-bound}{{I2}{23}{}{}{}}
\newlabel{EQ: Proportional-1step-no-oscillation-bound}{{I3}{23}{}{}{}}
\citation{Hilfinger2011}
\newlabel{EQ: Proportional-1step-intrinsic-ratio}{{I4}{24}{}{}{}}
\newlabel{EQ: Proportional-rates-1step-extrinsic-noise}{{I5}{24}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2}Fluorescent proteins with proportional translation rates}{24}{}}
\newlabel{EQ: Proportional-rates-3step-reactions}{{I6}{24}{}{}{}}
\newlabel{EQ: Proportional-3step-nofeedback-bound-final}{{I7}{24}{}{}{}}
\newlabel{EQ: Proportional-3step-no-oscillation-bound}{{I8}{24}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {3}Systematic undercounting of systems with proportional transcription rates}{24}{}}
\newlabel{SEC: Proportional both}{{I\tmspace  +\thinmuskip {.1667em}3}{24}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {J}Additional reporters}{25}{}}
\newlabel{SEC: Additional reporters}{{J}{25}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1}Measuring unknown life-time ratios}{25}{}}
\newlabel{SEC: Appendix inferring unknown lifetime ratios}{{J\tmspace  +\thinmuskip {.1667em}1}{25}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2}Measuring birthrate proportionality constants and reporter detection probabilities}{25}{}}
\newlabel{SEC: Measuring alpha and detection probabilities}{{J\tmspace  +\thinmuskip {.1667em}2}{25}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {3}Stronger constraints on fluorescent reporters using a third reporter}{26}{}}
\newlabel{SEC: Stronger constraints}{{J\tmspace  +\thinmuskip {.1667em}3}{26}{}{}{}}
\newlabel{EQ: FP three reporters mrna eqns}{{J3}{26}{}{}{}}
\newlabel{EQ: FP three reporters FP eqns}{{J4}{26}{}{}{}}
\newlabel{EQ: FP three reporters no-feedback constraints}{{J5}{26}{}{}{}}
\citation{Hilfinger2012}
\newlabel{EQ: FP three reporters no-oscillation constraint}{{J8}{27}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {K}Concentrations of growing and dividing cells}{27}{}}
\newlabel{SEC: Appendix concentrations}{{K}{27}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {1}Derivation of Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No feedback constraint (manuscript)}\unskip \@@italiccorr )}} and Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: No oscillation constraint (manuscript)}\unskip \@@italiccorr )}} for mRNA concentrations}{27}{}}
\newlabel{EQ: Concentrations-1step-differential-equation-numbers}{{K1}{28}{}{}{}}
\newlabel{EQ: Concentrations-1step-differential-equation-concentrations-general}{{K2}{28}{}{}{}}
\newlabel{EQ: Concentrations-1step-differential-equation-concentrations-exponential}{{K3}{28}{}{}{}}
\newlabel{EQ: Appendix concentrations (co)variance relations (manuscript)}{{K4}{28}{}{}{}}
\newlabel{EQ: Appendix concentrations flux balance (manuscript)}{{K5}{28}{}{}{}}
\newlabel{EQ: Concentrations-1step-flux-balance}{{K6}{28}{}{}{}}
\newlabel{EQ: Concentrations-1step-3-equations}{{K7}{28}{}{}{}}
\newlabel{EQ: Consentrations-1step-xtotal-law-of-total-variance}{{K8}{29}{}{}{}}
\newlabel{EQ: Concentrations-1step-xtotal-law-of-total-variance-2}{{K9}{29}{}{}{}}
\newlabel{EQ: Concentrations intrinsic terms general}{{K10}{29}{}{}{}}
\newlabel{EQ: Concentrations intrinsic terms volume independent}{{K11}{29}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {2}Derivation of Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no oscillation constraint (manuscript)}\unskip \@@italiccorr )}} and Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step no-feedback constraint concentrations (manuscript)}\unskip \@@italiccorr )}} for volume independent fluorescent protein concentrations}{29}{}}
\newlabel{EQ: Concentrations-3step-differential-equations-1}{{K12}{30}{}{}{}}
\newlabel{EQ: Concentrations-3step-differential-equations-2}{{K13}{30}{}{}{}}
\newlabel{EQ: FP concentration flux balance}{{K14}{30}{}{}{}}
\newlabel{EQ: Concentrations-3step-fluctuation-balance}{{K15}{30}{}{}{}}
\newlabel{EQ: FP concentration eta}{{K16}{30}{}{}{}}
\newlabel{EQ: Concentrations-3step-intrinsic-relation}{{K17}{30}{}{}{}}
\@writefile{toc}{\contentsline {subsection}{\numberline {3}Volume dependent genes --- exact constraints using a third reporter}{30}{}}
\newlabel{SEC: Appendix exact constraints on volume dependent genes}{{K\tmspace  +\thinmuskip {.1667em}3}{30}{}{}{}}
\citation{SI}
\citation{SI}
\citation{Balleza2018}
\citation{online-data}
\citation{SI}
\citation{galbusera2020using}
\citation{wang2010robust}
\citation{SI}
\@writefile{toc}{\contentsline {section}{\numberline {L}Experimental data analysis}{31}{}}
\newlabel{SEC: Appendix Experimental data analysis}{{L}{31}{}{}{}}
\newlabel{EQ: Appendix concentration CVs from abundance CVs (manuscript)}{{L1}{31}{}{}{}}
\newlabel{EQ: CV concentrations general}{{L2}{31}{}{}{}}
\newlabel{EQ: Negligible transcription noise model}{{L3}{31}{}{}{}}
\citation{Balleza2018}
\citation{galbusera2020using}
\citation{galbusera2020using}
\@writefile{lof}{\contentsline {figure}{\numberline {8}{\ignorespaces  {\textbf  {Utilizing constraints on flow-cytometry data from constitutively expressed fluorescent proteins.} Here we plot all of the possible pairs from Table 1 in the Supplemental Material \cite  {SI}. We plot 9 separate plots instead of combining the data into a single plot because for a given $T_{m}^{c}$ the right bound given by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: two-step sequential no-feedback constraint (manuscript)}\unskip \@@italiccorr )}} is only determined if we specify one of the maturation times. Thus for fixed Y reporter, the right bound is well defined for any given X reporter, with the yellow corridor indicating the estimated uncertainty in $\tau _c$ and $\tau _{mat,y}$. Because all fluorescent proteins were constitutively expressed and were not fused to cellular proteins we expect the experimental data to be consistent with the class of gene expression models that do not exhibit feedback and are not periodically driven (pink region). All variability ratios with respect to all reference fluorescence proteins confirm the above picture with the exception for mEGFP (with ratios indicated in red) for which the data violate the expected behaviour. With the exception of the indicated mEGFP outlier all data fall along the right hand boundary.}}}{32}{}}
\newlabel{FIG: All data}{{8}{32}{}{}{}}
\newlabel{EQ: Negligible translation noise model}{{L4}{32}{}{}{}}
\citation{SI}
\citation{Balleza2018}
\citation{SI}
\citation{Balleza2018}
\bibcite{Thattai2001}{{1}{2001}{{Thattai\ and\ van Oudenaarden}}{{}}}
\bibcite{Ozbudak2002}{{2}{2002}{{Ozbudak\ \emph  {et~al.}}}{{Ozbudak, Thattai, Kurtser, Grossman,\ and\ van Oudenaarden}}}
\bibcite{Blake2003}{{3}{2003}{{Blake\ \emph  {et~al.}}}{{Blake, K{\ae }rn, Cantor,\ and\ Collins}}}
\bibcite{Bar-Even2006}{{4}{2006}{{Bar-Even\ \emph  {et~al.}}}{{Bar-Even, Paulsson, Maheshri, Carmi, O'Shea, Pilpel,\ and\ Barkai}}}
\bibcite{Newman2006}{{5}{2006}{{Newman\ \emph  {et~al.}}}{{Newman, Ghaemmaghami, Ihmels, Breslow, Noble, DeRisi,\ and\ Weissman}}}
\bibcite{Eldar2010}{{6}{2010}{{Eldar\ and\ Elowitz}}{{}}}
\bibcite{elf2018}{{7}{2018}{{Jones\ and\ Elf}}{{}}}
\bibcite{kaern2005}{{8}{2005}{{K{\ae }rn\ \emph  {et~al.}}}{{K{\ae }rn, Elston, Blake,\ and\ Collins}}}
\bibcite{hilfinger2015b}{{9}{2016{}}{{Hilfinger\ \emph  {et~al.}}}{{Hilfinger, Norman,\ and\ Paulsson}}}
\bibcite{Taniguchi2010}{{10}{2010}{{Taniguchi\ \emph  {et~al.}}}{{Taniguchi, Choi, Li, Chen, Babu, Hearn, Emili,\ and\ Xie}}}
\bibcite{Cai2014}{{11}{2014}{{Lubeck\ \emph  {et~al.}}}{{Lubeck, Coskun, Zhiyentayev, Ahmad,\ and\ Cai}}}
\bibcite{Elowitz2002}{{12}{2002}{{Elowitz\ \emph  {et~al.}}}{{Elowitz, Levine, Siggia,\ and\ Swain}}}
\bibcite{Raser2004}{{13}{2004}{{Raser\ and\ O'Shea}}{{}}}
\bibcite{maamar2007}{{14}{2007}{{Maamar\ \emph  {et~al.}}}{{Maamar, Raj,\ and\ Dubnau}}}
\bibcite{Balleza2018}{{15}{2018}{{Balleza\ \emph  {et~al.}}}{{Balleza, Mark~Kim,\ and\ Cluzel}}}
\bibcite{shen2002network}{{16}{2002}{{Shen-Orr\ \emph  {et~al.}}}{{Shen-Orr, Milo, Mangan,\ and\ Alon}}}
\bibcite{baudrimont2019contribution}{{17}{2019}{{Baudrimont\ \emph  {et~al.}}}{{Baudrimont, Jaquet, Wallerich, Voegeli,\ and\ Becskei}}}
\@writefile{lof}{\contentsline {figure}{\numberline {9}{\ignorespaces  {\textbf  {The data is best described by a model with negligible transcription noise.} Plotted are the concentration CVs from Table 1 of the Supplemental Material \cite  {SI} as a function of their respective fluorescent protein maturation times. The pink curve corresponds to the model given by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Negligible transcription noise model}\unskip \@@italiccorr )}} with $A = 1.3$ and $\tau _{c}$ as reported in \cite  {Balleza2018}: $28.5\pm 2$ min. The pink corridor indicates the result of the uncertainty in $\tau _{c}$. The blue curve corresponds to the model given by Eq.~\textup  {\hbox {\mathsurround \z@ \normalfont  (\ignorespaces \ref  {EQ: Negligible translation noise model}\unskip \@@italiccorr )}} in which translation noise is negligible, with $B = 1.2$, $\tau _{F} = 28.5/10$ min, and $\tau _{c} = 28.5\pm 2$ min. We find that the data follows the pink curve and is thus best described by the negligible transcription noise model. The data point with the smallest CV corresponds to mEGFP, and the dashed red curve corresponds to the same model as the pink curve but with $\tau _{c}$ increased by a factor of 1.25. A possible explanation for this relatively low CV observed for mEGFP is that the cell cultures expressing mEGFP had slightly slower growth rates.}}}{33}{}}
\newlabel{FIG: CV plot}{{9}{33}{}{}{}}
\@writefile{toc}{\contentsline {section}{\numberline {}References}{33}{}}
\bibcite{hilfinger2015a}{{18}{2016{}}{{Hilfinger\ \emph  {et~al.}}}{{Hilfinger, Norman, Vinnicombe,\ and\ Paulsson}}}
\bibcite{raj2006stochastic}{{19}{2006}{{Raj\ \emph  {et~al.}}}{{Raj, Peskin, Tranchina, Vargas,\ and\ Tyagi}}}
\bibcite{golding2013}{{20}{2013}{{Skinner\ \emph  {et~al.}}}{{Skinner, Sep{\'u}lveda, Xu,\ and\ Golding}}}
\bibcite{eling2019challenges}{{21}{2019}{{Eling\ \emph  {et~al.}}}{{Eling, Morgan,\ and\ Marioni}}}
\bibcite{emory19925}{{22}{1992}{{Emory\ \emph  {et~al.}}}{{Emory, Bouvet,\ and\ Belasco}}}
\bibcite{cheneval2010review}{{23}{2010}{{Cheneval\ \emph  {et~al.}}}{{Cheneval, Kastelic, Fuerst,\ and\ Parker}}}
\bibcite{Raj2013}{{24}{2013}{{Shaffer\ \emph  {et~al.}}}{{Shaffer, Wu, Levesque,\ and\ Raj}}}
\bibcite{galbusera2020using}{{25}{2020}{{Galbusera\ \emph  {et~al.}}}{{Galbusera, Bellement-Theroue, Urchueguia, Julou,\ and\ van Nimwegen}}}
\bibcite{grun2014validation}{{26}{2014}{{Gr{\"u}n\ \emph  {et~al.}}}{{Gr{\"u}n, Kester,\ and\ Van~Oudenaarden}}}
\bibcite{raj2008imaging}{{27}{2008}{{Raj\ \emph  {et~al.}}}{{Raj, Van Den~Bogaard, Rifkin, Van~Oudenaarden,\ and\ Tyagi}}}
\bibcite{eng2019transcriptome}{{28}{2019}{{Eng\ \emph  {et~al.}}}{{Eng, Lawson, Zhu, Dries, Koulena, Takei, Yun, Cronin, Karp, Yuan \emph  {et~al.}}}}
\bibcite{Rhee2014}{{29}{2014}{{Rhee\ \emph  {et~al.}}}{{Rhee, Cheong,\ and\ Levchenko}}}
\bibcite{vank1992}{{30}{1992}{{Van~Kampen}}{{}}}
\bibcite{Lestas2008}{{31}{2008}{{Lestas\ \emph  {et~al.}}}{{Lestas, Paulsson, Ross,\ and\ Vinnicombe}}}
\bibcite{Hilfinger2011}{{32}{2011}{{Hilfinger\ and\ Paulsson}}{{}}}
\bibcite{Hilfinger2012}{{33}{2012}{{Hilfinger\ \emph  {et~al.}}}{{Hilfinger, Chen,\ and\ Paulsson}}}
\bibcite{SI}{{34}{}{{SI}}{{}}}
\bibcite{davenport1958introduction}{{35}{1958}{{Davenport\ \emph  {et~al.}}}{{Davenport, Root \emph  {et~al.}}}}
\bibcite{online-data}{{36}{2017}{{Balleza}}{{}}}
\bibcite{wang2010robust}{{37}{2010}{{Wang\ \emph  {et~al.}}}{{Wang, Robert, Pelletier, Dang, Taddei, Wright,\ and\ Jun}}}
\bibstyle{apsrev4-2}
\citation{REVTEX42Control}
\citation{apsrev42Control}
\newlabel{LastBibItem}{{37}{34}{}{}{}}
\newlabel{LastPage}{{}{34}{}{}{}}

\end{filecontents}
