\documentclass[aps,prx,noshowpacs,twocolumn,nofootinbib]{revtex4-2}
\usepackage[utf8]{inputenc}
\usepackage{xcolor}
\usepackage{graphics,graphicx,epsfig,multirow}
\usepackage{amssymb,amsfonts,amsmath,bm}
\usepackage{algorithm,algpseudocode}
\usepackage{ifthen}



\usepackage{hyperref}
\renewcommand*{\bibfont}{\footnotesize}
\usepackage{dsfont}
\renewcommand{\paragraph}[2][.]{\noindent {\bf #2#1}}		

\newcommand{\EQ}{\begin{equation}}
\newcommand{\EE}{\end{equation}}
\newcommand{\EQA}{\begin{eqnarray}}
\newcommand{\EEA}{\end{eqnarray}}
\newcommand{\x}{{\bf{x}}}
\renewcommand{\d}{{\text d}}
\newcommand{\ext}{{\text{ext}}}
\newcommand{\esc}{{\text{esc}}}
\newcommand{\eq}{{\text{eq}}}
\newcommand{\ts}{{\text{ts}}}
\newcommand{\tv}{{\text{tv}}}
\newcommand{\B}{{\text B}}
\newcommand{\Binom}{{\text {Binom}}}
\renewcommand{\L}{{\mathcal{L}}}
\newcommand{\seemethods}{{\text {see Methods}}}
\newcommand{\expect}[1]{\underset{#1}{\mathbb{E}}\,}
\newcommand{\E}{{\mathbb{E}}}
\newcommand{\av}[1]{\langle{#1}\rangle{}}
\newcommand{\abs}[1]{|#1|}
\newcommand{\degree}{{}^{\rm o}}
\newcommand{\bs}{ \mbox{\boldmath$\sigma$}}
\renewcommand{\baselinestretch}{1}
\newcommand{\comment}[1]{{\color{blue} #1}}
\newcommand{\commentR}[1]{{\color{red} #1}}
\newcommand{\commentF}[1]{{\color{orange} #1}}


\begin{document}

\title{Design of an optimal combination therapy with broadly neutralizing antibodies to suppress HIV-1}
\author{Colin LaMont}
\affiliation{Max Planck Institute for Dynamics and Self-organization, Am Fa\ss berg 17, 37077 G\"ottingen, Germany}
\author{Jakub Otwinowski}
\altaffiliation{Current address: Dyno Therapeutics, 1 Kendall Square, Cambridge, MA 02139}
\affiliation{Max Planck Institute for Dynamics and Self-organization, Am Fa\ss berg 17, 37077 G\"ottingen, Germany}
\author{Kanika Vanshylla}
\affiliation{Laboratory of Experimental Immunology, Institute of Virology, Faculty of Medicine and University Hospital Cologne, University of Cologne, 50931 Cologne, Germany}
\author{Henning Gruell}
\affiliation{Laboratory of Experimental Immunology, Institute of Virology, Faculty of Medicine and University Hospital Cologne, University of Cologne, 50931 Cologne, Germany}
\author{Florian Klein}
\affiliation{Laboratory of Experimental Immunology, Institute of Virology, Faculty of Medicine and University Hospital Cologne, University of Cologne, 50931 Cologne, Germany}
\affiliation{Partner Site Bonn-Cologne, German Center for Infection Research, 50931 Cologne, Germany}
\author{Armita Nourmohammad}
\email{Correspondence should be addressed to Armita Nourmohammad: armita@uw.edu}
\affiliation{Department of Physics, University of Washington, 3910 15th Ave Northeast, Seattle, WA 98195, USA}
\affiliation{Max Planck Institute for Dynamics and Self-organization, Am Fa\ss berg 17, 37077 G\"ottingen, Germany}
\affiliation{Fred Hutchinson Cancer Research Center, 1100 Fairview ave N, Seattle, WA 98109, USA}


\date{\today} 
\begin{abstract}
\noindent 
Broadly neutralizing antibodies (bNAbs) are promising targets for vaccination and therapy against HIV. Passive infusions of bNAbs have shown promise in clinical trials as a potential alternative for anti-retroviral therapy. A key challenge for the potential clinical application of bnAbs is the suppression of viral escape, which is more effectively achieved with a combination of bNAbs. However, identifying an optimal bNAb cocktail is combinatorially complex. Here, we propose a computational approach to predict the efficacy of a bNAb therapy trial based on the population genetics of HIV escape, which we parametrize using high-throughput HIV sequence data from a cohort of untreated bNAb-naive patients. By quantifying the mutational target size and the fitness cost of HIV-1 escape from bNAbs, we reliably predict the distribution of rebound times in three clinical trials. Importantly, we show that early rebounds are dominated by the pre-treatment standing variation of HIV-1 populations, rather than spontaneous mutations during treatment. Lastly, we show that a cocktail of three bNAbs is necessary to suppress the chances of viral escape below 1\%, and we predict the optimal composition of such a bNAb cocktail. Our results offer a rational design for bNAb therapy against HIV-1, and more generally show how genetic data could be used to predict treatment outcomes and design new approaches to pathogenic control.
\end{abstract}
\keywords{Population genetics, Stochastic evolution, HIV therapy, Broadly neutralizing antibodies, HIV control}
\maketitle


\section{Introduction}
Recent discoveries of highly potent broadly neutralizing antibodies (bNAbs) provide new opportunities to successfully prevent, treat, and potentially cure infections from evolving viruses such as HIV-1~\cite{Walker:2009cd,Walker:2011ew,Liao:2013gs,Mouquet:2013he,Klein:2013eb,Kwong:2013ia,Caskey:2015hm,Caskey:2017el,bar-onSafetyAntiviralActivity2018,sokRecentProgressBroadly2018,zwickBroadlyNeutralizingAntibodies2001,burtonBroadlyNeutralizingAntibodies2012}, influenza~\cite{Sparrow:2016ew}, and the Dengue virus~\cite{Ekiert:2012gr,Durham:2019fq}. 
bNAbs target vulnerable regions of a virus, such as the CD4 binding site of HIV {\em env} protein,  where escape mutations {can be} costly for the virus~\cite{Walker:2009cd,Chen:2009ig,Zhou:2010gx,Walker:2011ew,Liao:2013gs,West:2014jq,Burton:2016ek}. As a result, eliciting bNAbs is  the goal of a universal vaccine design against the otherwise rapidly evolving HIV-1. Apart from vaccination, bNAbs  can also offer significant advances in  therapy against both HIV and influenza~\cite{West:2014jq,Caskey:2016kk,Gruell:2018ixa,Durham:2019fq}. Specifically, augmenting current anti-retroviral therapy (ART) drugs with bNAbs may provide the next generation of HIV therapies~\cite{Horwitz:2013gb,Gruell:2018ixa}. 

Recent studies have used bNAb therapies to curb infections by the Simian immunodeficiency virus (SHIV)  in non-human primates~\cite{Shingai:2013kn,Barouch:2013gn,Julg:2017kg}, and  HIV-1 infections in human clinical trials~\cite{Caskey:2015hm,Bar:2016hg,Caskey:2017el,bar-onSafetyAntiviralActivity2018}. Monotherapy  trials with  potent bNAbs, including  3BNC117~\cite{Caskey:2015hm}, VRC01~\cite{Bar:2016hg}, and 10-1074~\cite{Caskey:2017el}  indicate that administering bNAbs is safe and can suppress viral load in patients.  Nonetheless, in each trial, escape mutants emerge resulting in a viral rebound after about 20 days past infusion of  the bNAb. However, in trials that administered a combination of 10-1074 and 3BNC117, viral rebound was substantially suppressed~\cite{Shingai:2013kn,bar-onSafetyAntiviralActivity2018}. Combination therapy has been repeatedly used against many infectious agents,  including current HIV ART cocktails and combination antibiotic treatments against Tuberculosis~\cite{Lienhardt:2012fl}. The principle behind combination therapy is clear: It is harder for a pathogen population to acquire resistance  against multiple treatment targets simultaneously than to acquiring resistance against each target separately. 


{Prior work has focused on optimizing combination therapy of bNAbs for breadth and potency \cite{yuPredictingBroadlyNeutralizing,waghOptimalCombinationsBroadly2016a} without considering viral dynamics.}   Theoretical approaches have been used to model the dynamics of viremia in patients following passive infusion of bNAbs~\cite{Lu:2016id,Reeves:2020ca,Saha:2020fd,meijersPredictingVivoEscape2021}. However the predictive power of models relying on trial data is limited by the small number of individuals enrolled in these trials, and increasing the size of a trial may be impractical. One can view the outcome of these trials in the context of an evolutionary competition among susceptible and resistant strains with different probabilities to emerge and establish in an HIV population within a patient, due to their different differential fitness effects  in the presence or absence of a bNAb. Studies of population genetics of HIV have found rapid intra-patient evolution and turnover of the virus~\cite{Lemey:2006wb,Zanini:2015gg} and have indicated that the efficacy of drugs in anti-retroviral therapy can severely impact the mode of viral evolution and escape~\cite{Feder:2016bc}. 
Despite the  complex evolutionary dynamics of HIV-1 within patients due to individualized immune pressure~\cite{Nourmohammad:2019ij}, genetic linkage~\cite{Zanini:2015gg}, recombination~\cite{Neher:2010dw,Zanini:2015gg}, and epistasis between loci~\cite{Bonhoeffer:2004cf,Zhang:2020bs}, the genetic composition of a population can still provide valuable information about the evolutionary significance of specific mutations, especially  in highly vulnerable regions of the virus. For example, analysis of genomic covariation in the Gag protein  of HIV-1 has been successful in predicting fitness effect of mutations in relatively conserved regions of the virus, which could inform the design of rational T-cell therapies that  target these vulnerable regions~\cite{Ferguson:2013kb}.   

Here, we present a statistical inference framework that uses the high throughput longitudinal survey of genetic data collected from 11 ART-naive patients over about 10 years of infection~\cite{Zanini:2015gg} to characterize the evolutionary fate of escape mutations and to predict patient outcomes in recent mono- and combination therapy  trials with 10-1074 and 3BNC117 bNAbs~\cite{Caskey:2015hm,Caskey:2017el,bar-onSafetyAntiviralActivity2018}. Using the accumulated intra-patient genetic variation from deep sequencing of HIV-1 populations in ART--naive patients~\cite{Zanini:2015gg}, we can estimate the diversity and the fitness effects of mutations at sites mediating escape. These variables parametrize our  individual-based model  for viral dynamics to characterize the expected path for a potential escape of HIV-1 populations in response to bNAb therapies in patients enrolled in the clinical trials. Our analysis accurately predicts the distribution of viral rebound times in response to passive  bNAb  infusions~\cite{Caskey:2015hm,Caskey:2017el,bar-onSafetyAntiviralActivity2018}, measuring the efficacy of these clinical trials. 

Our prediction for the viral rebound time in response to a bNAb is done based on the inferred genetics parameters from the deep sequencing of HIV-1 populations in a separate cohort of ART-naive patients. Therefore, we use our approach to  assess a broader panel of nine bNAbs, for which escape sites can be identified from prior deep mutational scanning experiments~\cite{Dingens:2019fd}, to characterize the therapeutic efficacy of each of these bNAbs and to propose optimal combination therapies that can efficiently curb an HIV infection. Our results showcase how the wealth of genetic data can be leveraged to guide rational therapy approaches against HIV. Importantly, this approach is potentially applicable to therapy designs against other evolving pathogens, such as resistant bacteria or cancer. 

\section{Model}
\begin{figure*} [t!]
    \centering
	\includegraphics[width=0.8\textwidth]{Fig1}
    \caption{
    {\bf Schematics for the evolutionary dynamics of viral rebound.}
{\bf (A) }
	%Deterministic dynamics and rebound time
	The viral dynamics after the initiation of a treatment with bNAb infusion ($t = 0$) is determined by two competing processes. 
	Susceptible strains (sus) undergo exponential decay (red line) with decay rate given by $r$, 
		while the resistant mutants (mut) undergo logistic growth back up to the carrying capacity ($N_k$) of the patient.
	In the deterministic limit (eq.~\ref{eq:logisticpiecewise}), the rebound time is linearly related to the log-frequency of the mutant fraction.
{\bf (B) }
	The schematic shows the four stochastic processes of birth, death, mutation, and neutralization with their respective rates for susceptible (purple) and resistant (blue) variants. These processes define the evolution of a  viral population. Note that both the susceptible and the resistant variants are subject to birth and death with their respective rates.
{\bf (C) }
	%
	The birth and death rates can be visualized as a region of size $\lambda = \beta+\delta$ which is partitioned into birth and death events.
	In the absence of antibodies, the susceptible population has balanced birth and death rates, $\beta_s=\delta_s$, while
		the resistant population has a negative net birth rate equal to the fitness difference $\Delta =\delta_m-\beta_m$.
	After introduction of the antibody, the susceptible population decays at rate $r$, and without competition from the susceptible population, the resistant population grows at the free growth-rate $\gamma$.
{\bf (D) }
	Mutational target size is inferred {\it a priori} from the genotype-phenotype mapping, 
		which can be visualized as a bipartite graph.
	The nodes correspond to codons, while the
		edges are the mutations which link one codon to another, weighted according to the respective mutation rates.
	The average edge weight from codons of susceptible variants to the escape  mutants determines the rate of escape mutations $\mu$. 
	Mutations can be divided into two types: transitions (black) are within-class, and transversions (red) are out of class nucleotide changes.
	Transitions occur at about  $8$ times higher rate than transversions (Fig.~\ref{Fig2}).
	 {\bf (E) }
	A coarse grained fitness and mutation model for two of the escape sites (281 and 282) against antibody 3BNC117 are shown.
	Left: At each escape mediating site, amino acids fall into one of three groups: (i) susceptible (wild-type), (ii) escape mutant, and (iii) fatal.
	For an escape-class amino acid at site $i$ the virus incurs a fitness cost $\Delta_i$, and these costs are additive across sites.
	Right: Mutations at a given site $i$ occur with (independent) forward $\mu_i$ and backward $\mu^\dagger_i$  rates which govern the substitution events between amino acid classes.}
   \label{Fig1}
\end{figure*}
\subsection*{HIV response to therapy}
After infusion of bNAbs in a patient, the antibodies bind and neutralize the susceptible strains of HIV. The neutralized subpopulation of HIV no longer infects T-cells, 
	and the plasma RNA copy-number associated with this neutralized population decays. The dynamics of viremia in HIV patients off ART following a bNAb therapy with 3BNC117~\cite{Caskey:2015hm}, 10-1074~\cite{Caskey:2017el}, and their combination~\cite{bar-onSafetyAntiviralActivity2018} are shown in Figs.~S1-S3.
With competition of the neutralized strains removed,	the resistant subpopulation grows until the viral load typically recovers to a level close to the pretreatment state (i.e., the carrying capacity); see Fig.~\ref{Fig1}A. The time it takes for the viral load to recover is the {\em rebound time}-- a key quantity that characterizes treatment efficacy within a patient. Although the details of the viremia dynamics, especially at beginning and at the end of the therapy,  may be complex~\cite{Lu:2016id,Reeves:2020ca,Saha:2020fd,meijersPredictingVivoEscape2021}, the rebound time  can be approximately modeled using a logistic growth after bNAb infusion ($t>0$),
\begin{align}
\label{eq:logisticpiecewise}
N(t) = 
\begin{cases}
N_k 
	& t \leq 0 \\
(1-x) N_k e^{-r t} + \frac{N_k}{1+ \frac{1-x}{x}e^{- \gamma t}}
	& t>0
\end{cases}
\end{align}
with the initial condition set for pre-treatment fraction of resistant subpopulation $x=  N_r(0)/ (N_r(0) + N_s (0))$, where $N_r(0)$ and $N_s(0)$ denote the size of resistance and susceptible subpopulations at time $t=0$, respectively. Here, $\gamma$ is the growth rate of the resistant population, $r$ is the neutralization rate impacting the susceptible subpopulation, and $N_k$ is the carrying capacity (Fig.~\ref{Fig1}A, Methods).  In our analysis, we set $\gamma = 1/3 \text{ days}^{-1}$ or a doubling time of $\sim 2$ days, which is the characteristic of HIV growth in patients~\cite{Perelson:1996hv}. We infer the neutralization rate $r$ as a global parameter for each trial, since it depends on the neutralization efficacy of a bNAb at the concentration used in the trial. We will infer the patient-specific pre-treatment fraction  of resistant subpopulation $x$, using a population genetics based approach based on which we characterize the mutational target size and selection cost of escape in the absence of a bNAb (see below).  The resulting viremia fits in Figs.~S1-S3 specify the rebound time $T$ in each patient, which in this simple model, is given by $T =-\gamma^{-1} \log x$ (Methods).


The rebound time following passive infusion of  3BNC117~\cite{Caskey:2015hm} and 10-1077~\cite{Caskey:2017el} bNAbs range from 1 to 4 weeks, with a small fraction of patients exceeding the monitoring time window in the studies (late rebounds past 56 days);  see Figs.~S1-S3. The distribution of rebound times summarizes the escape response of the virus to a therapy and directly relates to the distribution for the pre-treatment fraction of resistant variants $P(x)$ across patients $P(T) \sim x^{-1} P(x)$. 


\subsection*{Stochastic evolutionary dynamics of HIV subject to bNAb therapy}
The fate of an HIV population subject to bNAb therapy depends on the composition of  the pre-treatment population with resistant and susceptible variants, and the establishment of resistant variants following the treatment. To capture these effects, we construct an individual-based stochastic model for viral rebound (Fig.~\ref{Fig1}B). We specify a coarse-grained phenotypic model, where a viral strain of type $a$ is defined by a binary state vector $\vec \rho^a = [\rho^a_1,\dots,\rho^a_\ell]$, with $\ell$ entries for potentially escape-mediating epitope sites; the binary entry of the state vector at the epitope site $i$ represents the presence ($\rho^a_i=1$) or absence ($\rho^a_i=0$)  of a escape mediating mutation against a specified bNAb at this site of  variant $a$. We assume that a variant is resistant to a given antibody if at least one of the entries of its corresponding state vector is non-zero.

At each generation, a phenotypic variant $a$ can undergo one of three processes: birth, death and mutation to another type $b$, with rates $\beta_a$, $\delta_a$, and $\mu_{a \to b}$, respectively (Fig.~\ref{Fig1}B).  The net growth rate of  variant $a$  is its birth rate minus the death rate, $\gamma_a= \beta_a - \delta_a$ (Fig.~\ref{Fig1}C).  The total rate of events (birth and death) $\lambda = \beta_i + \delta_i$ modulates the amount of stochasticity in this birth-death process (Methods), which we assume to be constant across phenotypic variants.  The continuous limit for this birth-death process results in a stochastic evolutionary dynamics  for the sub-population of size $N_a$,
\EQA
{\small \frac{dN_a}{dt}= \begin{cases}
\text{absence of bNAb {\bf or} if $a$ is resistant:} \\
 N_a \left(f_a - \phi \right) +   \sum_{b}\left( N_b \mu_{{}_{b\to a}} - N_a \mu_{{}_{a\to b}} \right)\\ \quad +   \sqrt{N_{a} \lambda } \, \eta(t)     \\\\
\text{presence of bNAb {\bf and} if $a$ is susceptible:} \\
-r N_a +  \sum_{b}\left( N_b \mu_{{}_{b\to a}}- N_a \mu_{{}_{a\to b}} \right) +   \sqrt{N_{a} \lambda } \, \eta(t)  
\end{cases} }
\label{Langevin}
\EEA 
where  $\eta(t) $ is a Gaussian random variable with mean $\langle \eta(t)\rangle =0$ and correlation $\langle \eta(t) \eta(t')\rangle =\delta(t-t')$ (Methods). Here, $f_a$ denotes  the intrinsic  fitness of variant $a$ and its net growth rate $\gamma$ is mediated by a competitive pressure ${\small \phi =  \frac{1}{N_k} {\sum_{b} N_b f_b}}$ with the rest of the  population constrained by the carrying capacity $N_k$, such that $\gamma_a = f_a-\phi$.  In the presence of a bNAb, birth is effectively halted for susceptible variants and their death rate is set by the neutralization rate of the antibody, resulting in a net growth rate, $\gamma_\text{sus.} = - r$. At the carrying capacity, the competitive pressure is the mean population fitness  $\phi=\overline{f}$, making the net growth rate of the whole population zero (Fig.~\ref{Fig1}C). When susceptible variants are neutralized by a bNAb, the competitive pressure $\phi$ drops, and as a result, the resistant variants can rebound to carrying capacity, at growth rates near their intrinsic fitness.

To connect the birth-death model (eq.~\ref{Langevin}) with data, we should relate the simulation parameters of a birth-death process to  molecular observables. We have already made a connection between the birth and death rates of a variant  and its   intrinsic fitness in eq.~\ref{Langevin}.  In addition, the neutral diversity $\theta$ of a population at steady-state can  be expressed as $ \theta = {2 N_k \mu}/{ \lambda}$, where $\mu\approx 10^{-5} / \text{ day} $ is the per-nucleotide mutation rate, which we infer from intra-patient longitudinal  HIV sequence data~\cite{Zanini:2015gg}  (Methods). For consistency,  we set the  total rate of events $\lambda$ to be at least as large as the fastest process in the dynamics, which in this case is the growth rate of susceptible viruses $\gamma  \approx (3 \text{ days})^{-1}$, we choose $\lambda  = (0.5 \text{ days})^{-1}$ (Methods). Therefore, the key parameters of the birth-death model, i.e., $ \beta,\, \delta,\, \text{ and } N_k$ can be expressed in terms of the intrinsic fitness of the variants $f_a$ and the neutral diversity $\theta$, which we will infer from data. 


\section{Results}
\subsection*{Population genetics of HIV escape from bNAbs} 
HIV escape from different bNAbs has been a subject of interest for vaccine and therapy design, and a number of escape variants against different bNAbs have been identified  in clinical trials or in infected individuals~\cite{lynchHIV1FitnessCost2015,Caskey:2015hm,scheidHIV1Antibody3BNC1172016,Caskey:2017el,bar-onSafetyAntiviralActivity2018}. This {\em in-vivo} data is often complemented with information from co-crystallized structures of bNAbs with the HIV envelope protein~\cite{Pancera:2017dw}, and {\em in-vitro} deep mutational scanning (DMS) experiments, in which the relative change in the growth rate of tens of thousands of viral mutants are measured in the presence of different bNAbs~\cite{Dingens:2017bk,Dingens:2019fd,Schommers:2020er}.  We identify escape mutations against each of the bNAbs in this study by using information from clinal trials,  the characterized binding sites, and the DMS assays (Methods); the list of escape mutations against each bNAb is given in Table~S1. 

The rise and establishment of an escape variant against a specific bNAb depend on three key factors, (i) {neutral} genetic diversity of the viral population, (ii) the mutational target size for escape from the bNAb and (iii) the intrinsic fitness associated with such mutations. Although viremia traces in clinical trials can be used to model the escape dynamics~\cite{Lu:2016id,Reeves:2020ca,Saha:2020fd,meijersPredictingVivoEscape2021}, they do not offer a comprehensive statistical description for HIV escape as they are limited by the number of  enrolled individuals. Alternatively, mutation and fitness characteristics of such escape-mediating variants can be inferred from a broader cohort of untreated and bNAb-naive patients. We  will infer these quantities from the large amount of high-throughput HIV sequence data from ref.~\cite{Zanini:2015gg} (see Fig.~\ref{Fig2}A,~B for details) and use them to parameterize the birth-death model (Fig.~\ref{Fig1}B).

	\begin{figure*} [t!]
    \centering
	\includegraphics[width=0.85\textwidth]{Fig2_barchart}
    \caption{
    {\bf Statistics of the genetic data from bNAb-naive HIV patients.} 
{\bf (A)} 
	Statistics of the high throughput longitudinal data collected from  HIV populations in 11 ART-naive patients from ref.~\cite{Zanini:2015gg}, is shown. Some of the have low diversity (vertical red lines) and were not usable for our study. 
	Usable samples (vertical black lines) amount to 4-10 samples per patient, collected over 5 -10 years of infection.
{\bf (B)} 
	Lower panels show the relative frequencies (cube-root transformed for legibility) of different amino acids in four patients at the 3 escape sites against the 10-1074 bNAb, estimated from the polymorphism data  at the nucleotide level  in each patient over time. Despite 10-1074 being a broadly neutralizing antibody, mutations associated with escape (indicated by a $*$) are commonly observed in untreated patients. The upper right panel shows the number of individuals (out of the cohort of 11 bNAb-naive HIV patients) that carry mutations associated with escape against the indicated bNAbs. {\bf (C)}
	The nucleotide diversity associated with transversion $\theta_\tv= 2N_e \mu_\tv$ is shown against the transition diversity $\theta_\ts= 2N_e \mu_\ts$  for all patients (colors) and all time points.
	The covariance of these two diversity measurements yields an estimate for the transition/transversion ratio $\theta_\ts/\theta_\tv=7.6$.
{\bf (D)}
	Left: The transition diversity is shown to grow as a function of time since infection in all the 11 patients (colors according to (A)).
	%We find that transition diversity begins low due to the bottleneck of the initial infection event. 
	%The diversity increases with infection time as the population size grows.
	Right: The neutral diversity of viral populations in patients (points) from the three different clinical trials~\cite{Caskey:2015hm,Caskey:2017el,bar-onSafetyAntiviralActivity2018} analyzed in  this study resemble the larger diversities of long-established viral populations in untreated patients.
{\bf (E)} The inferred forward and backward mutation rates ($\mu,\,\mu^\dagger$), relative to the transition rate, and the median selection strength $\sigma$ at each escape site against the two bNAbs (10-1074, and 3BNC117) from the trial data used in this study are shown. Compared to the 10-1074 bNAb, escape from the  3BNC117 bNAb appears to be less costly, and is associated with a smaller mutational target.}
   \label{Fig2}
\end{figure*}


\noindent{\bf Diversity of the viral population.} The neutral genetic diversity $\theta = 2N_e\mu/\lambda$ (i.e., the number of segregating alleles) determines the chance to observe a rare (e.g. resistant) mutation in a patient prior to treatment, and it modulates the strength of selection in an HIV population to escape a bNAb; here $N_e$ is the effective population size, $\mu$ is the per-nucleotide mutation rate, and $\lambda$ is the total number of events per virus in the birth-death process which determines the noise amplitude (Methods).  We use synonymous changes as a proxy for diversity  associated with the neutral variation in an HIV population at a given time point within a patient.  By developing a maximum-likelihood  approach based on the multiplicities of different synonymous variants, we can  accurately infer the neutral diversity of a population from the large survey of synonymous sites in the HIV genome (Methods and Fig.~S4).
Importantly, we infer the neutral diversity of transition $\theta_\text{ts}$ and transversion $\theta_\text{tv}$ mutations separately, and consistent with previous work~\cite{Feder:2016bc,Zanini:2017in}, find that transitions occur with a rate of about 8 times larger than transversions (Fig.~\ref{Fig2}C,~Fig.~S4B-E).

Our inference indicates that the neutral diversity grows over the course of an infection in untreated HIV patients from ref.~\cite{Zanini:2015gg} (Fig.~\ref{Fig2}D). The patients enrolled in the three bNAb trials~\cite{Caskey:2015hm,Caskey:2017el,bar-onSafetyAntiviralActivity2018} show a broad range of neutral diversity prior to bNAb therapy (Fig.~\ref{Fig2}D). In addition to the circulating viruses in a patient's sera, the viral reservoir, which consists of replication-competent HIV in latently infected cells or un-sampled tissue,  can also contribute to a bNAb escape in a patient. Evidence for the effect of reservoir is directly visible in trials as the failure of pre-trial sequencing to exclude
	patients who do not harbor escape variants.
We model the effect of the reservoir as augmenting the neutral diversity by a constant multiplicative factor $r_\text{resv.}$,
	so that patients with more diverse sera, representing usually longer infections, 
	are also expected to have correspondingly more diverse reservoir populations. By fitting the observed rebound data, we infer the reservoir factor $r_\text{resv.}\simeq 2.07$ (Methods, Fig.~S4). We use the  augmented genetic diversity  of HIV prior to the bNAb therapy in each trial  to generate the rebound time and the probability of HIV escape in patients. \\
	

\noindent{\bf Mutational target size for escape.} We define the mutational target size for escape from a bNAb as the number of trajectories that connect the susceptible codon to codons associated with escape variants, weighed by their probability of occurrences (Methods).  The connecting paths with only single nucleotide transitions or transversions dominate the escape and can be represented as connected graphs shown in Fig.~\ref{Fig1}D. To characterize the  target size of escape for each bNAb, we determine the forward mutation rate {$\mu \equiv \mu_{\text{sus.} \to \text{res.}}$ from the susceptible codons to the resistant (escape) codons, and the reverse mutation rate  $\mu^\dagger \equiv \mu_{\text{sus.} \leftarrow \text{res.}}$} back to the susceptible variant (Fig.~\ref{Fig1}D, Methods). The  mutational target sizes vary across  bNAbs,  with HIV escape being most restricted from 10E8 ($\mu/\mu_{\ts} = 1.8$) and most accessible in the presence of 10-1074 ($\mu/\mu_{\ts} = 4.9$); see Fig.~\ref{Fig4}C and Table~S1 for the list of mutational target size for escape against all bNAbs in this study.\\


\begin{figure*} [t!]
    \centering
	\includegraphics[width=0.82\textwidth]{Fig3}
    \caption{
    {\bf Statistics of  viral rebound  in clinical trials with bNAbs.}
    {\bf (A)} 
Panels show viremia  of three patients from  the 3BNC117 trial over time (black circles)  and the fitted model of the viral decay and rebound processes from eq.~\ref{eq:logisticpiecewise} (orange line). The viral rebound time $T$ and the fitted carrying capacity $N_k$ is shown in each panel. Shown are examples of a non-responder (NR; left),	 a rebound occurring during the trial window ($0<T<56$ days; center), and a late rebound ($T>56$ days; right).  Viremia traces from all patients and all trials are shown in Figs.~S1-S3.
    {\bf (B)} 
	We compare the distribution rebound times in patients from the three clinical trials with 10-1074~\cite{Caskey:2017el}, 3BNC117~\cite{Caskey:2015hm}, and the combination of the two bNAbs~\cite{bar-onSafetyAntiviralActivity2018} to the predictions from the simulations based on our evolutionary model (Fig.~\ref{Fig1}, and Methods). 
	The error bars show the inter decile range (0.1-0.9 quantiles) 
		generated by the simulations for the corresponding trial.
    {\bf (C)} The summary table shows the  number of patients for whom the infecting HIV population shows  no response (NR), rebound during the trial window $0<T<56$, and a late rebound  ($T>56$ days) in each trial.  Note that 3 patients were excluded from the 3BNC117 trial (*) because of insufficient dosage leading to weak viral response: $1\text{mg}/\text{kg}$ compared to the $3-30\text{mg}/\text{kg}$ in the other treatment groups.
    {\bf (D)} 
    	Plotted are $1,200$ trajectories of the mutant viral population simulated using our individual based model. 
    	Due to the individual birth-death events, fluctuations are larger when the population size is smaller.
	At a critical threshold, $x_\text{ext}$, fluctuations are large enough to lead to almost certain extinction in the existing viral population.
	The critical threshold (yellow line) is an order of magnitude larger than the post-treatment spontaneously-generated mutant fraction (red line).
    {\bf (E)}
    	The predicted fraction of escape events associated with post-treatment spontaneous mutations (red) and the pre-treatment standing variation (yellow) are shown for the three trials. The fraction of events associated with late rebound is indicated in blue. 	Because the spontaneously-generated mutant fraction is smaller than the extinction threshold, these mutations  contribute to less than 4\% of escape events (red), and  escape is likely primarily driven driven by standing variation (yellow), i.e., pre-existing escape variants.    }
   \label{Fig3}
\end{figure*}

\noindent{\bf Fitness effect of escape mutations.}  Since bNAbs target highly conserved regions of the virus, we expect HIV escape mutations to be intrinsically deleterious for the virus~\cite{Ferguson:2013kb,meijersPredictingVivoEscape2021}, and incur a fitness cost relative to pre-treatment baseline $f_0$. We assume that fitness cost associated with escape mutations are additive and background-independent so the fitness of a variant $a$ in the absence of bNAb follows, $f_a  =  f_0 - \sum_{i} \Delta_{i} \sigma_{i}^{a} $, where $ \Delta_{i}$ is the cost associated with the presence of a escape mutation at site $i$ (i.e., for $\sigma_{i}^{a}  =1$); see Fig.~\ref{Fig1}D. 

Interestingly, we observe the escape variants against different bNAbs to be circulating in the HIV populations from the cohort of ART- and bNAb-naive patients~\cite{Zanini:2015gg} (Fig.~\ref{Fig2}B). We use this data~\cite{Zanini:2015gg} and extract the multiplicity of susceptible and escape variants in HIV populations at each sampled time point  from a given patient. We use a single locus approximation under strong selection to represent the stationary distribution of the  underlying frequency of escape alleles $x$ in each patient from ref.~\cite{Zanini:2015gg}, 
	$P(x;\sigma, \theta,\theta^\dagger)\sim x^{-1+\theta} (1-x)^{-1+ \theta^\dagger} \exp[-\sigma x]$, 
	given the (scaled) fitness difference between the susceptible and the escape variants
	$\sigma= 2N_e (f_\text{sus}-f_\text{mut})$; see Methods.

Based on the statistics of escape and susceptible variants in all patients, we define a likelihood function that determines a Bayesian posterior for  selection $\sigma$ associated with escape at each site (Methods). We found that it is statistically more robust to infer the strength of selection relative to a reference diversity measure $\sigma/ \theta_\text{ts} = (f_\text{sus.}-f_\text{res.}) / \mu_\text{ts}$, for which we choose the transition rate (Methods). This approach generates unbiased selection estimates in simulations and is robust to effects of linkage and recombination (Methods and Figs.~S5,~S6). The inferred values of the scaled fitness costs $\sigma/ \theta_\text{ts}$ are shown for the   escape-mediating sites of the  trial bNAbs in Fig.~\ref{Fig2}E, and are reported in Table~S1. 

	
\begin{figure*} [t!]
    \centering
	\includegraphics[width=0.9\textwidth]{Fig4}
    \caption{ {\bf Statistics of viral escape for optimal combination therapy with bNAbs.}
    {\bf  (A) } 
The posterior distribution for inferred selection strength on the escape-mediating sites associated with each of the 9 bNAbs in this study is shown (right); white line: median, box: 50\% around the median, bar: 80\%
around median.
Each escape site is color coded  by its location on the {\em env} gene (left) and each antibody by its associated epitope location.     {\bf (B)} The harmonic mean of the selection strength $\sigma$ associated with cost of escape (scaled by transition diversity $\theta_\ts$) is shown against the mutational target size for each bNAb; error bars indicate 50\% around the median.
For antibodies to be broadly-neutralizing, it is sufficient that viral escape from  them to be associated with a small mutational target size or a large fitness cost. The mutational target size is found to be weakly correlated with the average  cost of escape from a given bNAb. We identify two distinct strategies for antibody breadth---selection limited and mutational-target-size limited escape pathways each highlighted in gray.
    {\bf (C)}   
  bNAb therapies with 1, 2, and 3 antibodies are ranked based on the predicted probability of early  viral rebound, and in each case,  six therapies with  highest efficacies are shown; best ranked  therapy is associated with the lowest probability of early rebound; indicate 50\% around the median.
   Also for reference, the probability of early viral rebound two therapies from the trials in this study (10-1074 and 10-1074+3BNC117) are shown.
    }
   \label{Fig4}
\end{figure*}



\subsection*{Predicting the efficacy of bNAb therapy in clinical trials}
Monotherapy trials with 10-1074~\cite{Caskey:2017el} and with 3BNC117~\cite{Caskey:2015hm}, and the combination therapy with both of these antibodies in ~\cite{bar-onSafetyAntiviralActivity2018} have shown variable outcomes. In some patients bNAb therapy did not suppress the viral load, whereas in others suppression was efficient and no rebound was observed up to 56 days after infusion (end of  surveillance in these trials); see Fig.~\ref{Fig3}A for  examples of patients with different rebound times, Figs.~S1-S3 
 for the viremia traces in all patients, and Figs.~\ref{Fig3}B,~C for the distributions and the summary statistics of the rebound times in patients  in different trials. 

Although we infer a large intrinsic fitness cost for a virus to harbor an escape allele (Fig.~\ref{Fig2}E), these variants can emerge or already be present due to the large intra-patient diversity of HIV populations (Fig.~\ref{Fig2}C), or a larger mutational target size for these escape variants. Deep sequencing data in untreated (likely bNAb-naive) patients shows circulation of resistant variants against a panel of bNAbs in the majority of patients (Fig. \ref{Fig2}B). Our goal is to predict the efficacy of a bNAb trial, using the fitness effect and  the mutational target size for escape from a given bNAb, both of which we infer from the high-throughput HIV sequence data collected from bNAb-naive patients in ref.~\cite{Zanini:2015gg} (Figs.~\ref{Fig2}E, Table~S1). In addition, we modulate these measures with the patient-specific neutral diversity $\theta$ inferred from whole genome sequencing of HIV populations in each patient prior to bNAb therapy (Fig.~\ref{Fig2}D). These quantities parametrize the birth-death process for viral escape in a bNAb therapy (Fig.~\ref{Fig1}A), which we use to characterize the  distribution of rebound times in a given trial (Methods).


For both the  the 3BNC117 and the 10-1074 trials~\cite{Caskey:2015hm,Caskey:2017el}, we see an excellent agreement between our predictions of the rebound time distribution and data; see Fig.~\ref{Fig3}B, and Methods and Fig.~S7 for statistical accuracy of this comparison. By assuming an additive fitness effect for escape from 10-1074 and  3BNC117, we also accurately predict the distribution of rebound times  in the combination therapy~\cite{bar-onSafetyAntiviralActivity2018} (Fig.~\ref{Fig3}B). The agreement of our results with  data for combination therapy is consistent with the fact that the escape mediating sites from 10-1074 and 3BNC117 are spaced farther apart on the genome than 100bp, beyond which linkage disequilibrium diminishes due to  
frequent recombination in HIV~\cite{Zanini:2015gg}. Importantly, in all the trials, our evolutionary model accurately predicts the fraction of participants for whom  we should expect a late viral rebound (more than 56 days passed bNAb infusion)---the quantity that determines the efficacy of a treatment. 

Apart from the overall statistics of the rebound times, our stochastic model also enables us to characterize the relative contributions of the pre-treatment standing variation of the HIV population versus the spontaneous  mutations emerging during a trial to viral escape from a given bNAb. Given the large population size of HIV  and a high mutation rate ($\mu =10^{-5} \text{nt}$ per generation), spontaneous mutations generate a fraction $x^{(\mu)}$  of resistant variants during a trial, which we can express as, 
\begin{align}
x^{(\mu)}&=  \int_0^{56 \text{ wks}} (1-x(0))  e^{-r t } \mu \gamma dt \end{align}
In the best case scenario, there are no resistant virions prior to treatment i.e., $x(0)=0$. Since the neutralization rate $r$ and the growth rate $\gamma$ are comparable, this deterministic approach predicts that mutations can generate a resistant fraction of $x^{(\mu)} \approx 10^{-5}$ during a trial. 
However, stochastic effects from random birth and death events play an important role in the fate  and establishment of these resistant variants. The probability of extinction for a variant at frequency $x$ can be approximated as  $p(\text{extinct}) \approx 1- e^{- x/ x_\text{ext}}$ (Methods); here $x_\text{ext}= \frac{\mu_{ts}}{\gamma \theta_{ts}}\approx 10^{-4}$ and a variant with fraction $x$ that falls below this critical value is likely to go extinct (Fig.~\ref{Fig3}D). Since the total integrated mutational flux fraction during a trial is $x^{(\mu)} \sim 10^{-5}$, mutational flux rarely decides the outcome of patient treatment. Indeed we infer that spontaneous mutations contribute to less than 4\% of escape events in all the three trials and  escape is primarily attributed to the standing variation prior from the serum or the reservoirs prior to therapy (Fig.~\ref{Fig3}E). A similar conclusion was previously drawn based on a mechanistic model of escape in VRC01 therapy trials~\cite{Saha:2020fd}.  


\section{Devising optimal bNAb therapy cocktails}
Clinical trials with bNAbs have been instrumental in demonstrating the potential role of bNAbs  as therapy agents and in measuring the efficacy of each bNAb to suppress HIV. Still, these clinical trials {can only test a small fraction of the potential therapies that can be devised. It is therefore important that trials test therapies that have been optimized based on surrogate estimates of treatment efficacy.}  The accuracy of our predictions  for the rebound time of a HIV population subject to  bNAb therapy suggests a promising approach to the rational design of therapies based on genetic data of HIV populations collected from bNAb-naive patients. 
	
	Here, we use genetic data to infer the efficacy of therapies with bNAbs, for which clinal trials  are not yet performed. To do so, we first need to identify the routes of HIV escape from these  bNAbs. We  use deep mutational scanning data on HIV subject to 9 different bNAbs from ref.~\cite{Dingens:2019fd} together with information from literature  to identify the escape mediating variants from each of these bNAbs (Methods and Table~S1). We then determine the mutational target size and the fitness effect of these escape variants using high-throughput sequences of HIV in bNAb-naive patients from ref.~\cite{Zanini:2015gg}; these inferred values are reported in Fig.~\ref{Fig4}A,~B and Table~S1. Using the inferred fitness and mutational parameters and by setting the pre-treatment neutral diversity $\theta$ to  be comparable to that of the patients in the previous three trials with 10-1074 and 3BNC117 (Fig.~\ref{Fig2}C), we simulate treatment outcomes for these 9 bNAbs and their 2-fold and 3-fold combinations. 
	
	Interestingly, we infer that escape from mono-therapies is almost certain and a  combination of at least 3 antibodies is necessary to limit the probability of early rebound to below 1\% (Fig.~\ref{Fig4}C).  When considering all nine antibodies together, interesting  patterns emerge. 
	We find that the mutational target size and the fitness cost of escape, estimated as the harmonic-mean selection cost of individual sites, obey a roughly linear relationship (Fig.~\ref{Fig4}B). As all these bNAbs have similar overall breadth (i.e., they neutralize over 70\% of panel strains), this result suggests that for an antibody to be broad, its escape mediating variants should either be rare  (i.e, small mutational target size) or  intrinsically costly  (i.e., incurring a high fitness cost), but it is not necessary to satisfy both of these requirements. For instance, we find that 10E8 has a relatively weak selection but has a small escape target size, while PGT151 has a larger escape target size but makes up for it by having mutants with unavoidably high fitness cost. 
	
The fitness-limited versus the mutation-limited  strategies have different implications for the design of combination cocktails. The small mutational target size of 10E8 makes it the best candidate antibody for mono-therapy among the antibodies we consider because because the escape variants against this antibody are less likely to circulate in a patient's serum prior to treatment. However, in combination, 10E8 appears less often in top ranked therapies than PGT151. PGT151 is unremarkable on its own because of a relatively large target size, but the high cost of escape makes it especially promising in combination therapies. Overall, fitness-limited bNAbs like PGT151 are more effective against high diversity viral populations,  while mutation-limited bNAbs such as 10E8 are more effective against low diversity viral populations. Indeed, the best ranked therapy, namely the combination of PG9, PGT151, and VRC01, combines antibodies that target different regions of the virus and also have both types of fitness- and  mutation-limited strategies for coverage against the full variability of viral diversities found in pre-treatment individuals (Fig.~\ref{Fig4}C) participating in the clinical trials.

\section{Discussion}
HIV therapy with passive bNAb infusion has become a promising alternative to anti-retroviral drugs for suppressing and preventing the disease in patients without a need for daily administration. The current obstacle is the frequent escape of the virus seen in  mono- and even combination bNAb therapy trials~\cite{Caskey:2015hm,Bar:2016hg,Caskey:2017el,bar-onSafetyAntiviralActivity2018}. The key is to identify bNAb cocktails that can target multiple vulnerable regions on the virus in order to reduce the likelihood for the rise of resistant variants with escape-mediating mutations  in all of these regions.  Identifying an optimal bNAb cocktail can be a combinatorially difficult problem, and designing patient trials for all the potential combinations is a costly pursuit.

Here, we have proposed a computational approach to predict the efficacy of a bNAb therapy trial based on population genetics of HIV escape, which we parametrize using high-throughput HIV sequence data collected from a separate cohort of bNAb-naive patients~\cite{Zanini:2015gg}. Specifically,  we infer the mutational target size for escape and the fitness cost associated with escape-mediating mutations in the absence of a given bNAb. These quantities together with the neutral diversity of HIV within a patient parametrize our stochastic model for HIV dynamics subject to bNAb infusion, based on which we can accurately predict  the distribution of rebound times for HIV in therapy trials with 10-1074, 3BNC117 and their combination.  Consistent with previous work on VRC01~\cite{Saha:2020fd}, we found that viral rebounds in  bNAb trials are primarily mediated by the escape variants present either in the patients' sera or their latent reservoirs prior to treatment, and that the escape is not {likely to be driven} by the emergence of spontaneous mutations that establish during the therapy.


One key measure of  success for a bNAb trial is the suppression of early viral rebound. Our model can accurately predict the rebound times of HIV subject to three distinct therapies~\cite{Caskey:2015hm,Caskey:2017el,bar-onSafetyAntiviralActivity2018},  based on the fitness and the mutational characteristics of escape variants inferred from high-throughput HIV sequence data. This approach enables us to characterize routes of HIV escape  from other bNAbs, for which therapy trials are not available,   and to design optimal therapies. We used deep mutational scanning data~\cite{Dingens:2019fd} to identify escape-mediating variants against 9 different bNAbs for HIV. Our genetic analysis shows that bNAbs gain breadth and limit viral escape either due to their small mutational target size for escape or because of the large intrinsic fitness cost  incurred by  escape mutations. bNAbs with mutation-limited strategy are more effective at preventing escape in patients with low viral genetic diversity, while bNAbs selection-limited strategy with more effective at high viral diversity. To suppress the chance of viral rebound  to below  $1\%$, we show that a combo-therapy with 3 bNAbs with a mixture of mutation- and selection-limited strategies that target different regions of the viral envelope is necessary. Such combination can counter the full variation of viral diversity observed in patients. We found that  PG9, PG151, and VRC01, which respectively target V2 loop, Interface, and CD4 binding site of HIV envelope, form an optimal combination for a 3-bNAb therapy to limit HIV-1 escape in patients infected with clade B of the virus. 


We rest our analysis primarily on the predictive power of the  observed variant frequencies in the untreated patients. Our model weighs these frequencies with respect to the viral diversity in a mathematically and biologically consistent way.  However, we ignore the dynamics of antibody concentration and IC50 neutralization during treatments, the details of T-cell dynamics during infection~\cite{Perelson:2002kw}, and also the evolutionary features of the genetic data, such as epistasis between loci~\cite{Bonhoeffer:2004cf,Zhang:2020bs}, genetic linkage~\cite{Zanini:2015gg}, and codon usage bias~\cite{Meintjes:2005ji}. The statistical fidelity of our model to the observed variant frequencies implies that many of the mechanistic details are at best secondary to the coarse-grained features  of  viral escape, and specifically the distribution of the rebound times. Nonetheless, our approach falls short of predicting the detailed characteristics of viremia traces in patients, especially at very short or very long times,  during which the dynamics of T-cell response or the decay of bNAbs could play a role~\cite{Lu:2016id,Reeves:2020ca,Saha:2020fd}. 


Our predictions are limited by our ability to identify the escape variants for each bNAb, either based on  trial data, patient surveillance, or {\em in-vitro} assays such as DMS experiments. DMS data for viral escape~\cite{Dingens:2017bk,Dingens:2019fd,Schommers:2020er} are generally of high quality but consider only a single HIV genetic background (e.g. in \cite{Dingens:2017bk} this background is BF520.W14M.C2), whereas clinical data will have diverse baseline viruses between and within individuals. In addition, DMS data can be noisy for variants that grow very poorly in the absence of a bNAb, since  growth without antibodies is the first stage  of these experiments, followed by   growth  in the presence of  bNAbs. This is likely to be the reason that the DMS experiments  fail to capture a clear escape signal against bNAbs such as VRC01 or 3BNC117 that target the CD4 binding site of HIV, where mutations can be extremely deleterious  in the absence of the bNAb. As such, we continue to need more data and more powerful statistical techniques for calling escape variants, using {\it in-vivo} approaches. For example, passive infusion of bNAb  in humanized mouse models~\cite{Gruell:2017}, which can, at least partially, reproduce the natural diversity of a human infection would be valuable. These efforts would take us a step closer to rational design for  bNAb therapy and a more model-guided clinical trials.

Our approach showcases that, when feasible,  combining high-throughput genetic data with ecological and population genetics models can have surprisingly broad  applicability, and their interpretability can shed light  into the complex dynamics of these populations.  Application of similar methods to therapy design to curb the escape of cancer tumors against immune- or chemo-therapy, the resistance in bacteria against antibiotics, or the escape of  seasonal influenza against vaccination is a promising avenue for future work. However, we expect that more  sophisticated methods for inferring fitness from evolutionary trajectories  may be necessary to capture the dynamical response of these populations.

\section*{Acknowledgements}
This work has been supported by the NSF CAREER award (grant No:~2045054), DFG grant (SFB1310) for Predictability in Evolution, and the MPRG funding through the Max Planck Society.

\bibliographystyle{plos2}
%\bibliography{Armita,Colin}
\begin{thebibliography}{10}
\providecommand{\url}[1]{\texttt{#1}}
\providecommand{\urlprefix}{URL }
\expandafter\ifx\csname urlstyle\endcsname\relax
  \providecommand{\doi}[1]{doi:\discretionary{}{}{}#1}\else
  \providecommand{\doi}{doi:\discretionary{}{}{}\begingroup
  \urlstyle{rm}\Url}\fi
\providecommand{\bibAnnoteFile}[1]{%
  \IfFileExists{#1}{\begin{quotation}\noindent\textsc{Key:} #1\\
  \textsc{Annotation:}\ \input{#1}\end{quotation}}{}}
\providecommand{\bibAnnote}[2]{%
  \begin{quotation}\noindent\textsc{Key:} #1\\
  \textsc{Annotation:}\ #2\end{quotation}}
\providecommand{\eprint}[2][]{\url{#2}}

\bibitem{Walker:2009cd}
Walker LM, et~al. (2009) {Broad and potent neutralizing antibodies from an
  African donor reveal a new HIV-1 vaccine target.}
\newblock Science 326: 285--289.
\bibAnnoteFile{Walker:2009cd}

\bibitem{Walker:2011ew}
Walker LM, et~al. (2011) {Broad neutralization coverage of HIV by multiple
  highly potent antibodies.}
\newblock Nature 477: 466--470.
\bibAnnoteFile{Walker:2011ew}

\bibitem{Liao:2013gs}
Liao HX, et~al. (2013) {Co-evolution of a broadly neutralizing HIV-1 antibody
  and founder virus.}
\newblock Nature 496: 469--476.
\bibAnnoteFile{Liao:2013gs}

\bibitem{Mouquet:2013he}
Mouquet H, Nussenzweig MC (2013) {HIV: Roadmaps to a vaccine.}
\newblock Nature 496: 441--442.
\bibAnnoteFile{Mouquet:2013he}

\bibitem{Klein:2013eb}
Klein F, et~al. (2013) {Antibodies in HIV-1 vaccine development and therapy.}
\newblock Science 341: 1199--1204.
\bibAnnoteFile{Klein:2013eb}

\bibitem{Kwong:2013ia}
Kwong PD, Mascola JR, Nabel GJ (2013) {Broadly neutralizing antibodies and the
  search for an HIV-1 vaccine: the end of the beginning.}
\newblock Nature Rev Immunol 13: 693--701.
\bibAnnoteFile{Kwong:2013ia}

\bibitem{Caskey:2015hm}
Caskey M, et~al. (2015) {Viraemia suppressed in HIV-1-infected humans by
  broadly neutralizing antibody 3BNC117.}
\newblock Nature 522: 487--491.
\bibAnnoteFile{Caskey:2015hm}

\bibitem{Caskey:2017el}
Caskey M, et~al. (2017) {Antibody 10-1074 suppresses viremia in HIV-1-infected
  individuals.}
\newblock Nat Med 23: 185--191.
\bibAnnoteFile{Caskey:2017el}

\bibitem{bar-onSafetyAntiviralActivity2018}
{Bar-On} Y, et~al. (2018) Safety and anti-viral activity of combination
  {{HIV}}-1 broadly neutralizing antibodies in viremic individuals.
\newblock Nature medicine 24: 1701--1707.
\bibAnnoteFile{bar-onSafetyAntiviralActivity2018}

\bibitem{sokRecentProgressBroadly2018}
Sok D, Burton DR (2018) Recent progress in broadly neutralizing antibodies to
  {{HIV}}.
\newblock Nature immunology 19: 1179--1188.
\bibAnnoteFile{sokRecentProgressBroadly2018}

\bibitem{zwickBroadlyNeutralizingAntibodies2001}
Zwick MB, et~al. (2001) Broadly neutralizing antibodies targeted to the
  membrane-proximal external region of human immunodeficiency virus type 1
  glycoprotein gp41.
\newblock Journal of virology 75: 10892--10905.
\bibAnnoteFile{zwickBroadlyNeutralizingAntibodies2001}

\bibitem{burtonBroadlyNeutralizingAntibodies2012}
Burton DR, Poignard P, Stanfield RL, Wilson IA (2012) Broadly neutralizing
  antibodies present new prospects to counter highly antigenically diverse
  viruses.
\newblock Science 337: 183--186.
\bibAnnoteFile{burtonBroadlyNeutralizingAntibodies2012}

\bibitem{Sparrow:2016ew}
Sparrow E, Friede M, Sheikh M, Torvaldsen S, Newall AT (2016) {Passive
  immunization for influenza through antibody therapies, a review of the
  pipeline, challenges and potential applications.}
\newblock Vaccine 34: 5442--5448.
\bibAnnoteFile{Sparrow:2016ew}

\bibitem{Ekiert:2012gr}
Ekiert DC, Wilson IA (2012) {Broadly neutralizing antibodies against influenza
  virus and prospects for universal therapies.}
\newblock Curr Opin Virol 2: 134--141.
\bibAnnoteFile{Ekiert:2012gr}

\bibitem{Durham:2019fq}
Durham ND, et~al. (2019) {Broadly neutralizing human antibodies against dengue
  virus identified by single B cell transcriptomics.}
\newblock eLife 8: 1113.
\bibAnnoteFile{Durham:2019fq}

\bibitem{Chen:2009ig}
Chen L, et~al. (2009) {Structural basis of immune evasion at the site of CD4
  attachment on HIV-1 gp120.}
\newblock Science 326: 1123--1127.
\bibAnnoteFile{Chen:2009ig}

\bibitem{Zhou:2010gx}
Zhou T, et~al. (2010) {Structural basis for broad and potent neutralization of
  HIV-1 by antibody VRC01.}
\newblock Science 329: 811--817.
\bibAnnoteFile{Zhou:2010gx}

\bibitem{West:2014jq}
West AP, et~al. (2014) {Structural insights on the role of antibodies in HIV-1
  vaccine and therapy.}
\newblock Cell 156: 633--648.
\bibAnnoteFile{West:2014jq}

\bibitem{Burton:2016ek}
Burton DR, Hangartner L (2016) {Broadly Neutralizing Antibodies to HIV and
  their role in vaccine design.}
\newblock Annu Rev Immunol 34: 635--659.
\bibAnnoteFile{Burton:2016ek}

\bibitem{Caskey:2016kk}
Caskey M, Klein F, Nussenzweig MC (2016) {Broadly Neutralizing Antibodies for
  HIV-1 Prevention or Immunotherapy}.
\newblock N Engl J Med 375: 2019--2021.
\bibAnnoteFile{Caskey:2016kk}

\bibitem{Gruell:2018ixa}
Gruell H, Klein F (2018) {Antibody-mediated prevention and treatment of HIV-1
  infection.}
\newblock Retrovirology 15: 73.
\bibAnnoteFile{Gruell:2018ixa}

\bibitem{Horwitz:2013gb}
Horwitz JA, et~al. (2013) {HIV-1 suppression and durable control by combining
  single broadly neutralizing antibodies and antiretroviral drugs in humanized
  mice.}
\newblock Proc Natl Acad Sci USA 110: 16538--16543.
\bibAnnoteFile{Horwitz:2013gb}

\bibitem{Shingai:2013kn}
Shingai M, et~al. (2013) {Antibody-mediated immunotherapy of macaques
  chronically infected with SHIV suppresses viraemia.}
\newblock Nature 503: 277--280.
\bibAnnoteFile{Shingai:2013kn}

\bibitem{Barouch:2013gn}
Barouch DH, et~al. (2013) {Therapeutic efficacy of potent neutralizing
  HIV-1-specific monoclonal antibodies in SHIV-infected rhesus monkeys.}
\newblock Nature 503: 224--228.
\bibAnnoteFile{Barouch:2013gn}

\bibitem{Julg:2017kg}
Julg B, et~al. (2017) {Virological Control by the CD4-Binding Site Antibody N6
  in Simian-Human Immunodeficiency Virus-Infected Rhesus Monkeys.}
\newblock J Virol 91: 1633.
\bibAnnoteFile{Julg:2017kg}

\bibitem{Bar:2016hg}
Bar KJ, et~al. (2016) {Effect of HIV Antibody VRC01 on Viral Rebound after
  Treatment Interruption.}
\newblock N Engl J Med 375: 2037--2050.
\bibAnnoteFile{Bar:2016hg}

\bibitem{Lienhardt:2012fl}
Lienhardt C, et~al. (2012) {New drugs for the treatment of tuberculosis: needs,
  challenges, promise, and prospects for the future.}
\newblock J Infect Dis 205: S241--9.
\bibAnnoteFile{Lienhardt:2012fl}

\bibitem{yuPredictingBroadlyNeutralizing}
Yu WH, et~al. Predicting the broadly neutralizing antibody susceptibility of
  the {{HIV}} reservoir.
\newblock JCI Insight 4: e130153.
\bibAnnoteFile{yuPredictingBroadlyNeutralizing}

\bibitem{waghOptimalCombinationsBroadly2016a}
Wagh K, et~al. (2016) Optimal {{Combinations}} of {{Broadly Neutralizing
  Antibodies}} for {{Prevention}} and {{Treatment}} of {{HIV}}-1 {{Clade C
  Infection}}.
\newblock PLOS Pathogens 12: e1005520.
\bibAnnoteFile{waghOptimalCombinationsBroadly2016a}

\bibitem{Lu:2016id}
Lu CL, et~al. (2016) {Enhanced clearance of HIV-1-infected cells by broadly
  neutralizing antibodies against HIV-1 in vivo.}
\newblock Science 352: 1001--1004.
\bibAnnoteFile{Lu:2016id}

\bibitem{Reeves:2020ca}
Reeves DB, et~al. (2020) {Mathematical modeling to reveal breakthrough
  mechanisms in the HIV Antibody Mediated Prevention (AMP) trials.}
\newblock PLoS Comput Biol 16: e1007626.
\bibAnnoteFile{Reeves:2020ca}

\bibitem{Saha:2020fd}
Saha A, Dixit NM (2020) {Pre-existing resistance in the latent reservoir can
  compromise VRC01 therapy during chronic HIV-1 infection.}
\newblock PLoS Comput Biol 16: e1008434.
\bibAnnoteFile{Saha:2020fd}

\bibitem{meijersPredictingVivoEscape2021}
Meijers M, Vanshylla K, Gruell H, Klein F, L{\"a}ssig M (2021) Predicting in
  vivo escape dynamics of {{HIV}}-1 from a broadly neutralizing antibody.
\newblock Proceedings of the National Academy of Sciences 118: e2104651118.
\bibAnnoteFile{meijersPredictingVivoEscape2021}

\bibitem{Lemey:2006wb}
Lemey P, Rambaut A, Pybus OG (2006) {HIV evolutionary dynamics within and among
  hosts}.
\newblock AIDS Rev 8: 125--140.
\bibAnnoteFile{Lemey:2006wb}

\bibitem{Zanini:2015gg}
Zanini F, et~al. (2015) {Population genomics of intrapatient HIV-1 evolution.}
\newblock eLife 4: e11282.
\bibAnnoteFile{Zanini:2015gg}

\bibitem{Feder:2016bc}
Feder AF, et~al. (2016) {More effective drugs lead to harder selective sweeps
  in the evolution of drug resistance in HIV-1.}
\newblock eLife 5: 1161.
\bibAnnoteFile{Feder:2016bc}

\bibitem{Nourmohammad:2019ij}
Nourmohammad A, Otwinowski J, Luksza M, Mora T, Walczak AM (2019) {Fierce
  Selection and Interference in B-Cell Repertoire Response to Chronic HIV-1.}
\newblock Mol Biol Evol 36: 2184--2194.
\bibAnnoteFile{Nourmohammad:2019ij}

\bibitem{Neher:2010dw}
Neher RA, Leitner T (2010) {Recombination rate and selection strength in HIV
  intra-patient evolution}.
\newblock PLoS Comput Biol 6: e1000660.
\bibAnnoteFile{Neher:2010dw}

\bibitem{Bonhoeffer:2004cf}
Bonhoeffer S, Chappey C, Parkin NT, Whitcomb JM, Petropoulos CJ (2004)
  {Evidence for positive epistasis in HIV-1.}
\newblock Science 306: 1547--1550.
\bibAnnoteFile{Bonhoeffer:2004cf}

\bibitem{Zhang:2020bs}
Zhang TH, et~al. (2020) {Predominance of positive epistasis among drug
  resistance-associated mutations in HIV-1 protease.}
\newblock PLoS Genet 16: e1009009.
\bibAnnoteFile{Zhang:2020bs}

\bibitem{Ferguson:2013kb}
Ferguson AL, et~al. (2013) {Translating HIV sequences into quantitative fitness
  landscapes predicts viral vulnerabilities for rational immunogen design}.
\newblock Immunity 38: 606--617.
\bibAnnoteFile{Ferguson:2013kb}

\bibitem{Dingens:2019fd}
Dingens AS, Arenz D, Weight H, Overbaugh J, Bloom JD (2019) {An Antigenic Atlas
  of HIV-1 Escape from Broadly Neutralizing Antibodies Distinguishes Functional
  and Structural Epitopes}.
\newblock Immunity 50: 520--532.e3.
\bibAnnoteFile{Dingens:2019fd}

\bibitem{Perelson:1996hv}
Perelson AS, Neumann AU, Markowitz M, Leonard JM, Ho DD (1996) {HIV-1 dynamics
  in vivo: virion clearance rate, infected cell life-span, and viral generation
  time.}
\newblock Science 271: 1582--1586.
\bibAnnoteFile{Perelson:1996hv}

\bibitem{lynchHIV1FitnessCost2015}
Lynch RM, et~al. (2015) {{HIV}}-1 {{Fitness Cost Associated}} with {{Escape}}
  from the {{VRC01 Class}} of {{CD4 Binding Site Neutralizing Antibodies}}.
\newblock Journal of Virology 89: 4201--4213.
\bibAnnoteFile{lynchHIV1FitnessCost2015}

\bibitem{scheidHIV1Antibody3BNC1172016}
Scheid JF, et~al. (2016) {{HIV}}-1 antibody {{3BNC117}} suppresses viral
  rebound in humans during treatment interruption.
\newblock Nature 535: 556--560.
\bibAnnoteFile{scheidHIV1Antibody3BNC1172016}

\bibitem{Pancera:2017dw}
Pancera M, Changela A, Kwong PD (2017) {How HIV-1 entry mechanism and broadly
  neutralizing antibodies guide structure-based vaccine design.}
\newblock Curr Opin HIV AIDS 12: 229--240.
\bibAnnoteFile{Pancera:2017dw}

\bibitem{Dingens:2017bk}
Dingens AS, Haddox HK, Overbaugh J, Bloom JD (2017) {Comprehensive Mapping of
  HIV-1 Escape from a Broadly Neutralizing Antibody.}
\newblock Cell Host Microbe 21: 777--787.e4.
\bibAnnoteFile{Dingens:2017bk}

\bibitem{Schommers:2020er}
Schommers P, et~al. (2020) {Restriction of HIV-1 Escape by a Highly Broad and
  Potent Neutralizing Antibody.}
\newblock Cell 180: 471--489.e22.
\bibAnnoteFile{Schommers:2020er}

\bibitem{Zanini:2017in}
Zanini F, Puller V, Brodin J, Albert J, Neher RA (2017) {In vivo mutation rates
  and the landscape of fitness costs of HIV-1.}
\newblock Virus Evol 3: vex003.
\bibAnnoteFile{Zanini:2017in}

\bibitem{Perelson:2002kw}
Perelson AS (2002) {Modelling viral and immune system dynamics.}
\newblock Nature Rev Immunol 2: 28--36.
\bibAnnoteFile{Perelson:2002kw}

\bibitem{Meintjes:2005ji}
Meintjes PL, Rodrigo AG (2005) {Evolution of relative synonymous codon usage in
  Human Immunodeficiency Virus type-1.}
\newblock J Bioinform Comput Biol 3: 157--168.
\bibAnnoteFile{Meintjes:2005ji}

\bibitem{Gruell:2017}
Gruell H, Klein F (2017) { Progress in HIV-1 antibody research using humanized
  mice.}
\newblock Curr Opin HIV AIDS 12: 285--293.
\bibAnnoteFile{Gruell:2017}

\end{thebibliography}

\end{document}




