
% \lipsum[1-3]



%% the main contents:


% \textbf{describe the basic conception importance of of RNA:}
% \textbf{describe the importance in the RNA structure and function prediction:}
% According to the basic dogma of the molecular biology, millions of gene sites are transcribed to produce Ribonucleic acid (RNA) transcripts (namely DNA → RNA), which are then translated to produce different types of protein sequences (namely RNA → protein) so as to maintain the normal functions of human bodies  \cite{chen2019computational,li2015statistics,dreyfuss1996transcript,mcknight1982transcriptional}.
% Such central dogma indicates that 
% \textcolor{blue}{Ash}
\forceindent RNA plays an important role in performing various types of biological functions, such as cell signaling, gene expression, and post-transcriptional regulations \cite{miao2017rna,caprara2000rna,atkins2011rna}. Determination of RNA structure or type is also an essential part of RNA-based therapeutics, including mRNA vaccines, RNA interference and CRISPR-based therapeutics \cite{li2020strategies, bora2012rna, pardi2018mrna}. Among all RNA transcripts, about 5\% served as mRNAs responsible for protein coding, while the substantial remaining portion is non-coding RNAs (ncRNAs) \cite{ chen2019computational, wang2011molecular}. Particularly, these ncRNAs must sustain specific structures to conduct corresponding biological functions. Different ncRNAs, such as small nuclear ribonucleoproteins (snRNPs), ribosomes, microRNAs, small nucleolar ribonucleoproteins (snoRNPs), long ncRNAs, and telomerase, also interact with
proteins to form stable RNA-protein complexes to perform specific functions \citep{cech2014noncoding, miao2017rna}.
Accurately modelling ncRNAs structures could help understand their performed functions and various biological processes. Despite a large number of ncRNA sequences, few of their structures and functions are known \citep{townshend2021geometric,yao2019cellular}. Traditionally, RNA three-dimensional (3D) structures assessed by experimental approaches, including nuclear magnetic resonance, X-ray crystallography and cryogenic electron microscopy, are expensive and time-consuming. Therefore, computational approaches are developed and applied to bridge the gap.

Regarding the RNA structure prediction, most existing approaches focus on the RNA secondary structure prediction, which could further be divided into three categories: thermodynamic methods, alignment-based methods, and deep learning (DL)-based methods. Foundational works can be traced back to the 1980s, when thermodynamic parameters-based methods were proposed \citep{zuker1984rna, waterman1978rna, rivas2013four} to predict RNA secondary structures. Nowadays, combined with dynamic programming, these approaches such as Vienna RNA \citep{ stadler2011viennarna,hofacker1994fast}, Mfold/UNAFold \citep{markham2008unafold,zuker2003mfold}, LinearFold \citep{huang2019linearfold}, and RNAstructure \citep{mathews1998updated,reuter2010rnastructure} are still widely used since they can incorporate a large volume of folding features (together with standard base pairs) to estimate the model parameters.
Thermodynamic-based methods are later improved by considering parameters of local structures, including experimental parameters (such as RNAstructure \cite{reuter2010rnastructure}, RNAfold \cite{lorenz2011viennarna}, RNAshapes \cite{janssen2015rna}) and machine learning parameters (such as ContextFold \cite{zakov2011rich}, CONTRAfold \cite{do2006contrafold}, CentroidFold \cite{sato2009centroidfold}). 
Though many currently used approaches fall into this category, their overall performance seems to hit the plateau. These methods usually do not consider all base pairs obtained from tertiary interactions \cite{nowakowski1997rna,rivas2013four,sato2009centroidfold}, which may miss essential information in their predictions.
%
%


To overcome the above problem, alignment-based methods built upon comparative sequence analysis are thus designed to determine the vital base pairs among homologous sequences \cite{singh2019rna}.
With the help of sufficient homologous sequences and their alignments, these alignment-based methods achieve excellent performance in predictions \cite{gutell2002accuracy,singh2019rna}. 
However, new issues arose from the limited number of known RNA families, given Rfam only contains several thousand RNA families. Since RNA is much less conserved, such property restricts the further improvement of alignment-based methods theoretically compared to the vast number of protein families.
On the other hand, with more RNA data available, several deep learning approaches have been recently developed in the community to improve the accuracy of RNA secondary structure prediction. For example, SPOT-RNA \cite{singh2019rna}, E2Efold \cite{chen2020rna}, MXfold2 \cite{sato2021rna}, and UFold \cite{fu2021ufold} are shown to be able to improve the prediction accuracy significantly on different datasets.
Nevertheless, the generalization capability of such DL-based methods still remains a problem, as the model architecture is explicitly designed for corresponding tasks and cannot generalize well to unknown RNA types \cite{chen2020rna}.
%
%

In contrast to secondary structure prediction, the modelling of RNA 3D structure is still under-explored due to the lack of 3D structure data.
In fact, computational optimization combined with deep learning methods may serve as an alternative way to solve the 3D problem. While there exist methods to optimize 3D structure with the minimum energy \citep{xiong2021pairing} given 2D information, deep learning methods could be utilized to solve the downgraded 2D problem (secondary structure, distance structure). 
For example, RNAcontact \cite{sun2021rna} could predict 3D closeness between base pairs by the deep residual neural networks. ARES, which consists of many processing layers, was proposed to score the predicted RNA structures \citep{townshend2021geometric}.
Despite the above attempts so far, there are no end-to-end DL-based methods that could generate RNA 3D structure directly. The lack of annotated 3D data is one major obstacle. 
% Other barriers, including designation or complexity, could be overcome gradually.

In addition to RNA structure, understanding its function is also vital. In particular, predicting the interaction between RNAs and proteins \cite{lam2019deep,sun2021predicting} could help to understand gene expression regulation. We could utilize either data from existing databases or data generated by biological experiments that classified RNAs into several functional groups for this application. With these extensive hand-crafted labels, DL-based approaches are then proposed to learn the underlying distribution of RNAs in different functional groups. Therefore, the corresponding functional group could be predicted given query RNA sequences. For instance, a deep discriminative neural network was developed to distinguish RNAs that can bind to specific RNA-binding proteins from the non-binding ones \cite{sun2021predicting}.
Furthermore, 1D CNN is designed to predict the protein expression levels regulated by human 5' UTRs from RNA sequence \cite{sample2019human}.


As the previous computational approaches rely heavily on the label information of RNA sequences, they all share the limitation of generalization. A model needs to be deliberately designed and tuned on specific tasks, yet challenging to transfer to other related studies. For instance, even E2Efold, a promising RNA secondary structure prediction model, performs well on dominant RNA types, including tRNAs and 5S rRNAs. Its performance on unknown RNA types degrades significantly. Also, models built upon RNA-protein interactions cannot gain accurate predictions on the UTR regulation. They are unfriendly to biological researchers because much effort will be needed to retrain specific deep learning based models. 
% This challenge causes internal friction beyond the community.

Notice that virtually every method omits the millions of unannotated RNA sequences and only utilizes the small annotated dataset with at most 30K sequences. 
To overcome the above generalization limitation, we suggest taking advantage of the enormous unannotated RNA sequence data, which contains the evolutionary information of RNA sequences. We propose a novel RNA foundation model (RNA-FM), as shown in Figure \ref{Fig.overview}, implying `ONE-FOR-ALL', with which various RNA-related downstream applications can be conducted via replacing its predicting layers only. The model is trained in a self-supervised manner that differs from all the previous RNA DL-based approaches. To the best of our knowledge, we are the first that attempt to extract meaningful RNA representations from unlabeled data and improve multiple downstream applications simultaneously. Precisely, the model is presented in two phases: pre-training and fine-tuning. Throughout the task-agnostic pre-training stage, the proposed RNA-FM is trained with 23 million ncRNA sequences from the RNAcentral database using self-supervised learning. Eventually, RNA-FM learns the sequential distribution and pattern that could potentially capture the underlying structural and functional information. Then, during the task-specific fine-tuning stage, the pre-trained RNA-FM model either generates sequence embeddings (features) that fit into downstream modules or is fine-tuned with a lightweight prediction layer.
With the powerful representation learned from unannotated ncRNA data, RNA-FM significantly improves the performance across a broad range of RNA structure/function-related prediction applications with minor modifications to the model architectures. In terms of RNA secondary structure prediction, RNA-FM outperforms LinearFold by up to 30\% and SPOT-RNA by up to 20\% in terms of F1 score on cross-dataset validation. Even on the low-redundant dataset, which is rather challenging for previous methods, RNA-FM still outperforms SPOT-RNA by up to 7.5\% and UFold by 4\% in terms of F1 score. As for 3D closeness prediction, a single model built upon RNA-FM can even exceed an ensemble method with 100 models by 30\% regarding the long-range top precision.
In addition to working well on the benchmark datasets, RNA-FM can model the regulatory elements in the SARS-CoV-2 genome and potentially illustrates the evolution of the virus variants.
Furthermore, the embedding from RNA-FM can achieve comparable performance with the \textit{in vivo} secondary structure feature on protein-RNA interaction prediction, even though RNA-FM is trained on unannotated RNA sequences only. 
All of these performance improvements suggest that the proposed large-scale pre-trained foundation model has implicitly captured both structural and functional information from RNA sequences alone.



\begin{figure}[!t]
\centering
\includegraphics[width=0.98\textwidth]{figs/overview1.png} 
\caption{
\textbf{Overview of RNA foundation model (RNA-FM) design and applications}. RNA-FM consists of 12 transformer layers. In the pre-training stage, the RNA-FM is trained with 23 million sequences from the RNAcentral database via self-supervised learning, \textit{i.e.}, reconstructing the masked tokens from the sequence alone. 
Thus, we could obtain effective feature representations of the query RNA sequences without using any labels.
In the fine-tuning stage, the representations from RNA-FM could significantly and consistently improve the performance on both structure-related and function-related applications with simple task-specific prediction modules.
} 
\label{Fig.overview}
\end{figure}



We summarize the main contributions of this work as follows:
\begin{itemize}
\item We propose the first RNA foundation model, RNA-FM, delivering rich representations for the ncRNA universe. In addition, we provide a web server such that the community can access the pre-trained model and its powerful representations. The server can be accessed with this link: \href{https://proj.cse.cuhk.edu.hk/rnafm/}{https://proj.cse.cuhk.edu.hk/rnafm/} The code and weights are available at \href{https://github.com/ml4bio/RNA-FM}{https://github.com/ml4bio/RNA-FM}. 

\item RNA-FM can produce interpretable RNA representations, which contain evolutionary information. Such embeddings can be used to infer the evolutionary trend of lncRNAs and SARS-CoV-2 variants.

\item The representations generated from RNA-FM can substantially improve the performance on multiple ncRNA structure/function prediction applications, with a desirable generalization property to the regulatory UTR regions of mRNAs and SARS-CoV-2.


\end{itemize}